diff -ruN wiringpi-rpi-6a17382a0ab40a9e36456e5a71bdce0a70421ffb.orig/devLib/Makefile wiringpi-rpi-6a17382a0ab40a9e36456e5a71bdce0a70421ffb/devLib/Makefile
--- wiringpi-rpi-6a17382a0ab40a9e36456e5a71bdce0a70421ffb.orig/devLib/Makefile	2021-06-19 18:07:42.000000000 +0200
+++ wiringpi-rpi-6a17382a0ab40a9e36456e5a71bdce0a70421ffb/devLib/Makefile	2021-06-22 22:47:42.595145625 +0200
@@ -37,9 +37,9 @@
 #DEBUG	= -g -O0
 DEBUG	= -O2
 CC	?= gcc
-INCLUDE	= -I.
+INCLUDE	= -I../wiringPi
 DEFS	= -D_GNU_SOURCE
-CFLAGS	= $(DEBUG) $(DEFS) -Wformat=2 -Wall -Winline $(INCLUDE) -pipe -fPIC $(EXTRA_CFLAGS)
+CFLAGS	+= $(DEBUG) $(DEFS) -Wformat=2 -Wall -Winline $(INCLUDE) -pipe -fPIC $(EXTRA_CFLAGS)
 
 LIBS    =
 
@@ -69,6 +69,7 @@
 $(DYNAMIC):	$(OBJ)
 	$Q echo "[Link (Dynamic)]"
 	$Q $(CC) -shared -Wl,-soname,libwiringPiDev.so$(WIRINGPI_SONAME_SUFFIX) -o libwiringPiDev.so.$(VERSION) -lpthread $(OBJ)
+	$Q ln -sf libwiringPiDev.so.$(VERSION) libwiringPiDev.so
 
 .c.o:
 	$Q echo [Compile] $<
@@ -93,7 +94,7 @@
 	$Q echo "[Install Dynamic Lib]"
 	$Q install -m 0755 -d						$(DESTDIR)$(PREFIX)/lib
 	$Q install -m 0755 libwiringPiDev.so.$(VERSION)			$(DESTDIR)$(PREFIX)/lib/libwiringPiDev.so.$(VERSION)
-	$Q ln -sf $(DESTDIR)$(PREFIX)/lib/libwiringPiDev.so.$(VERSION)	$(DESTDIR)/lib/libwiringPiDev.so
+	$Q ln -sf libwiringPiDev.so.$(VERSION)        $(DESTDIR)$(PREFIX)/lib/libwiringPiDev.so
 	$Q $(LDCONFIG)
 
 .PHONY:	install-static
diff -ruN wiringpi-rpi-6a17382a0ab40a9e36456e5a71bdce0a70421ffb.orig/gpio/Makefile wiringpi-rpi-6a17382a0ab40a9e36456e5a71bdce0a70421ffb/gpio/Makefile
--- wiringpi-rpi-6a17382a0ab40a9e36456e5a71bdce0a70421ffb.orig/gpio/Makefile	2021-06-19 18:07:42.000000000 +0200
+++ wiringpi-rpi-6a17382a0ab40a9e36456e5a71bdce0a70421ffb/gpio/Makefile	2021-06-22 22:56:03.145403795 +0200
@@ -33,10 +33,10 @@
 #DEBUG	= -g -O0
 DEBUG	= -O2
 CC	?= gcc
-INCLUDE	= -I$(DESTDIR)$(PREFIX)/include
-CFLAGS	= $(DEBUG) -Wall -Wextra $(INCLUDE) -Winline -pipe $(EXTRA_CFLAGS)
+INCLUDE	= -I../wiringPi -I../devLib
+CFLAGS	+= $(DEBUG) -Wall -Wextra $(INCLUDE) -Winline -pipe $(EXTRA_CFLAGS)
 
-LDFLAGS	= -L$(DESTDIR)$(PREFIX)/lib
+LDFLAGS	= -L../wiringPi -L../devLib
 LIBS    = -lwiringPi -lwiringPiDev -lpthread -lrt -lm -lcrypt
 
 # May not need to  alter anything below this line
@@ -52,9 +52,13 @@
 	$Q echo Need to run newVersion above.
 
 gpio:	$(OBJ)
-	$Q echo [Link]
+	$Q echo "[Link (Dynamic)]"
 	$Q $(CC) -o $@ $(OBJ) $(LDFLAGS) $(LIBS)
 
+gpio-static: $(OBJ)
+	$Q echo "[Link (Static)]"
+	$Q $(CC) -static -o gpio $(OBJ) ../wiringPi/libwiringPi.a ../devLib/libwiringPiDev.a -lpthread
+
 .c.o:
 	$Q echo [Compile] $<
 	$Q $(CC) -c $(CFLAGS) $< -o $@
diff -ruN wiringpi-rpi-6a17382a0ab40a9e36456e5a71bdce0a70421ffb.orig/wiringPi/Makefile wiringpi-rpi-6a17382a0ab40a9e36456e5a71bdce0a70421ffb/wiringPi/Makefile
--- wiringpi-rpi-6a17382a0ab40a9e36456e5a71bdce0a70421ffb.orig/wiringPi/Makefile	2021-06-19 18:07:42.000000000 +0200
+++ wiringpi-rpi-6a17382a0ab40a9e36456e5a71bdce0a70421ffb/wiringPi/Makefile	2021-06-22 23:06:27.758204125 +0200
@@ -39,7 +39,7 @@
 CC	?= gcc
 INCLUDE	= -I.
 DEFS	= -D_GNU_SOURCE
-CFLAGS	= $(DEBUG) $(DEFS) -Wformat=2 -Wall -Wextra -Winline $(INCLUDE) -pipe -fPIC $(EXTRA_CFLAGS)
+CFLAGS	+= $(DEBUG) $(DEFS) -Wformat=2 -Wall -Wextra -Winline $(INCLUDE) -pipe -fPIC $(EXTRA_CFLAGS)
 #CFLAGS	= $(DEBUG) $(DEFS) -Wformat=2 -Wall -Wextra -Wconversion -Winline $(INCLUDE) -pipe -fPIC
 
 LIBS    = -lm -lpthread -lrt -lcrypt
@@ -76,6 +76,7 @@
 $(DYNAMIC):	$(OBJ)
 	$Q echo "[Link (Dynamic)]"
 	$Q $(CC) -shared -Wl,-soname,libwiringPi.so$(WIRINGPI_SONAME_SUFFIX) -o libwiringPi.so.$(VERSION) $(OBJ) $(LIBS)
+	$Q ln -sf libwiringPi.so.$(VERSION) libwiringPi.so
 
 .c.o:
 	$Q echo [Compile] $<
@@ -101,7 +102,7 @@
 	$Q echo "[Install Dynamic Lib]"
 	$Q install -m 0755 -d						$(DESTDIR)$(PREFIX)/lib
 	$Q install -m 0755 libwiringPi.so.$(VERSION)			$(DESTDIR)$(PREFIX)/lib/libwiringPi.so.$(VERSION)
-	$Q ln -sf $(DESTDIR)$(PREFIX)/lib/libwiringPi.so.$(VERSION)	$(DESTDIR)/lib/libwiringPi.so
+	$Q ln -sf libwiringPi.so.$(VERSION)       $(DESTDIR)$(PREFIX)/lib/libwiringPi.so
 	$Q $(LDCONFIG)
 
 .PHONY: check-deb-destdir
