--- a/drivers/usb/storage/usb.c	2024-04-15 11:24:13.868827293 +0200
+++ b/drivers/usb/storage/usb.c	2024-04-15 11:24:22.324936125 +0200
@@ -71,7 +71,7 @@
 module_param(delay_use, uint, S_IRUGO | S_IWUSR);
 MODULE_PARM_DESC(delay_use, "seconds to delay before using a new device");
 
-static char quirks[128];
+static char quirks[256];
 module_param_string(quirks, quirks, sizeof(quirks), S_IRUGO | S_IWUSR);
 MODULE_PARM_DESC(quirks, "supplemental list of device IDs and their quirks");
 
