- integrated changes for building within buildroot env

Signed-off-by: Jens Maus <mail@jens-maus.de>

--- a/configure.ac.orig	2022-04-14 18:11:55.847643695 +0200
+++ b/configure.ac	2022-04-14 18:06:11.198482489 +0200
@@ -20,7 +20,7 @@
 AC_PROG_RANLIB
 
 # Checks for libraries.
-PKG_CHECK_MODULES([LIBCRYPT], [libcrypt])
+#PKG_CHECK_MODULES([LIBCRYPT], [libcrypt])
 
 # Checks for header files.
 AC_CHECK_HEADERS([arpa/inet.h fcntl.h malloc.h netdb.h netinet/in.h stdint.h stdlib.h string.h sys/ioctl.h sys/socket.h sys/time.h syslog.h termios.h unistd.h])
