--- buildroot/support/scripts/mkusers.orig	2016-12-14 11:50:14.723198182 +0100
+++ buildroot/support/scripts/mkusers	2016-12-14 11:51:42.993066846 +0100
@@ -303,7 +303,7 @@
 
     # Remove any previous instance of this user
     for _f in "${PASSWD}" "${SHADOW}"; do
-        sed -r -i -e '/^'"${username}"':.*/d;' "${_f}"
+        sed -r -i --follow-symlinks -e '/^'"${username}"':.*/d;' "${_f}"
     done
 
     _gid="$( get_gid "${group}" )"
