--- buildroot/toolchain/Config.in.orig
+++ buildroot/toolchain/Config.in
@@ -454,10 +454,15 @@
 	bool
 	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_5_2
 
+config BR2_TOOLCHAIN_HEADERS_AT_LEAST_5_4
+	bool
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_5_3
+
 # This order guarantees that the highest version is set, as kconfig
 # stops affecting a value on the first matching default.
 config BR2_TOOLCHAIN_HEADERS_AT_LEAST
 	string
+	default "5.4"  if BR2_TOOLCHAIN_HEADERS_AT_LEAST_5_4
 	default "5.3"  if BR2_TOOLCHAIN_HEADERS_AT_LEAST_5_3
 	default "5.2"  if BR2_TOOLCHAIN_HEADERS_AT_LEAST_5_2
 	default "5.1"  if BR2_TOOLCHAIN_HEADERS_AT_LEAST_5_1
--- buildroot/toolchain/toolchain-external/toolchain-external-custom/Config.in.options.orig
+++ buildroot/toolchain/toolchain-external/toolchain-external-custom/Config.in.options
@@ -109,6 +109,10 @@
 	    m = ( LINUX_VERSION_CODE >> 8  ) & 0xFF
 	    p = ( LINUX_VERSION_CODE >> 0  ) & 0xFF
 
+config BR2_TOOLCHAIN_EXTERNAL_HEADERS_5_4
+	bool "5.4.x"
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_5_4
+
 config BR2_TOOLCHAIN_EXTERNAL_HEADERS_5_3
 	bool "5.3.x"
 	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_5_3
