--- buildroot/package/linux-headers/Config.in.host.orig
+++ buildroot/package/linux-headers/Config.in.host
@@ -118,6 +118,10 @@
 	  This is used to hide/show some packages that have strict
 	  requirements on the version of kernel headers.
 
+config BR2_PACKAGE_HOST_LINUX_HEADERS_CUSTOM_5_4
+	bool "5.4.x"
+	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_5_4
+
 config BR2_PACKAGE_HOST_LINUX_HEADERS_CUSTOM_5_3
 	bool "5.3.x"
 	select BR2_TOOLCHAIN_HEADERS_AT_LEAST_5_3
