--- buildroot/package/wireguard-linux-compat/wireguard-linux-compat.hash.orig
+++ buildroot/package/wireguard-linux-compat/wireguard-linux-compat.hash
@@ -1,4 +1,4 @@
 # https://lists.zx2c4.com/pipermail/wireguard/2020-April/005296.html
-sha256  cf166348fbb67419528e73049ce001d29131aea367fa6aef9d3e223f7251e116  wireguard-linux-compat-1.0.20200413.tar.xz
+sha256  16e7ae4bef734b243428eea07f3b3c3d4721880c3ea8eb8f98628fd6ae5b77c3  wireguard-linux-compat-1.0.20200520.tar.xz
 # Locally calculated
 sha256  8177f97513213526df2cf6184d8ff986c675afb514d4e68a404010521b880643  COPYING
--- buildroot/package/wireguard-linux-compat/wireguard-linux-compat.mk.orig
+++ buildroot/package/wireguard-linux-compat/wireguard-linux-compat.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-WIREGUARD_LINUX_COMPAT_VERSION = 1.0.20200413
+WIREGUARD_LINUX_COMPAT_VERSION = 1.0.20200520
 WIREGUARD_LINUX_COMPAT_SITE = https://git.zx2c4.com/wireguard-linux-compat/snapshot
 WIREGUARD_LINUX_COMPAT_SOURCE = wireguard-linux-compat-$(WIREGUARD_LINUX_COMPAT_VERSION).tar.xz
 WIREGUARD_LINUX_COMPAT_LICENSE = GPL-2.0
