--- buildroot/package/lighttpd/lighttpd.hash.orig
+++ buildroot/package/lighttpd/lighttpd.hash
@@ -1,4 +1,4 @@
 # From http://www.lighttpd.net/
-sha256 0f8ad5aac7529d7b948b9d7e8cd0b4a9e177309d85d6bf6516e28e6e40d74f36  lighttpd-1.4.48.tar.xz
+sha256 29378312d8887cbc14ffe8a7fadef2d5a08c7e7e1be942795142346ad95629eb  lighttpd-1.4.50.tar.xz
 # Locally calculated
 sha256 5c98cad2fbaf5c5e2562bcbab401a7c557c1bb1bac9914ecc63730925052fb13  COPYING
--- buildroot/package/lighttpd/lighttpd.mk.orig
+++ buildroot/package/lighttpd/lighttpd.mk
@@ -5,7 +5,7 @@
 ################################################################################
 
 LIGHTTPD_VERSION_MAJOR = 1.4
-LIGHTTPD_VERSION = $(LIGHTTPD_VERSION_MAJOR).48
+LIGHTTPD_VERSION = $(LIGHTTPD_VERSION_MAJOR).50
 LIGHTTPD_SOURCE = lighttpd-$(LIGHTTPD_VERSION).tar.xz
 LIGHTTPD_SITE = http://download.lighttpd.net/lighttpd/releases-$(LIGHTTPD_VERSION_MAJOR).x
 LIGHTTPD_LICENSE = BSD-3-Clause
