--- buildroot/package/wiringpi/wiringpi.mk.orig
+++ buildroot/package/wiringpi/wiringpi.mk
@@ -4,8 +4,10 @@
 #
 ################################################################################
 
-WIRINGPI_VERSION = 2.46
-WIRINGPI_SITE = git://git.drogon.net/wiringPi
+WIRINGPI_VERSION = 50b7c5ed7d238a637db7d9c73978466eb022a8de
+WIRINGPI_SITE = git://github.com/wiringpi/wiringpi
+
+BR_NO_CHECK_HASH_FOR += $(WIRINGPI_SOURCE)
 
 WIRINGPI_LICENSE = LGPL-3.0+
 WIRINGPI_LICENSE_FILES = COPYING.LESSER
