--- buildroot/package/rpi-firmware/rpi-firmware.hash.orig
+++ buildroot/package/rpi-firmware/rpi-firmware.hash
@@ -1,2 +1,2 @@
 # Locally computed
-sha256 5edff641f216d2e09c75469dc2e9fc66aff290e212a1cd43ed31c499f99ea055 rpi-firmware-287af2a2be0787a5d45281d1d6183a2161c798d4.tar.gz
+sha256 b23af42de1db8940767d96a7a7eb86837d55a66d5d562576d975446f454874ae rpi-firmware-ebbecf992a466f9bffbd9e3f443f56f64994a277.tar.gz
--- buildroot/package/rpi-firmware/rpi-firmware.mk.orig
+++ buildroot/package/rpi-firmware/rpi-firmware.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-RPI_FIRMWARE_VERSION = 287af2a2be0787a5d45281d1d6183a2161c798d4
+RPI_FIRMWARE_VERSION = ebbecf992a466f9bffbd9e3f443f56f64994a277
 RPI_FIRMWARE_SITE = $(call github,raspberrypi,firmware,$(RPI_FIRMWARE_VERSION))
 RPI_FIRMWARE_LICENSE = BSD-3-Clause
 RPI_FIRMWARE_LICENSE_FILES = boot/LICENCE.broadcom
