--- buildroot/package/rpi-firmware/rpi-firmware.hash.orig
+++ buildroot/package/rpi-firmware/rpi-firmware.hash
@@ -1,2 +1,2 @@
 # Locally computed
-sha256	f1d631920ed4ae15f368ba7b8b3caa4ed604f5223372cc6debbd39a101eb8d74  rpi-firmware-81cca1a9380c828299e884dba5efd0d4acb39e8d.tar.gz
+sha256 eaa434671fadfff45d9073d9c9fa5d8f3aecd6bb5f32d080dca13abd5027b8fc  rpi-firmware-4508a68812b1e089d8e1de1d20aac79f5b93ed3a.tar.gz
--- buildroot/package/rpi-firmware/rpi-firmware.mk.orig
+++ buildroot/package/rpi-firmware/rpi-firmware.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-RPI_FIRMWARE_VERSION = 81cca1a9380c828299e884dba5efd0d4acb39e8d
+RPI_FIRMWARE_VERSION = 4508a68812b1e089d8e1de1d20aac79f5b93ed3a
 RPI_FIRMWARE_SITE = $(call github,raspberrypi,firmware,$(RPI_FIRMWARE_VERSION))
 RPI_FIRMWARE_LICENSE = BSD-3-Clause
 RPI_FIRMWARE_LICENSE_FILES = boot/LICENCE.broadcom
