--- buildroot/package/linux-firmware/Config.in.orig
+++ buildroot/package/linux-firmware/Config.in
@@ -8,6 +8,32 @@
 
 if BR2_PACKAGE_LINUX_FIRMWARE
 
+config BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS
+    bool "Compress firmware files"
+    help
+      Compress firmware files to save space in the root filesystem. This
+      is not as efficient as whole root filesystem compression, but it
+      is better performance-wise, because the firmwares are usually
+      accessed only when the driver is loaded.
+
+if BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS
+
+choice
+	prompt "Compression algorithm"
+	default BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS_XZ
+	help
+	  Select the compression algorithm to use for the firmware files.
+
+config BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS_XZ
+	bool "xz"
+
+config BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS_ZSTD
+	bool "zstd"
+
+endchoice
+
+endif
+
 menu "Audio firmware"
 
 config BR2_PACKAGE_LINUX_FIRMWARE_INTEL_SST_DSP
@@ -64,6 +90,16 @@
 	  Firmware files for Intel 3160/7260/7265/8260/8265 Wireless
 	  bluetooth support.
 
+config BR2_PACKAGE_LINUX_FIRMWARE_MT7921_BT
+	bool "MediaTek MT7921"
+	help
+	  Firmware files for MediaTek MT7921 bluetooth support
+
+config BR2_PACKAGE_LINUX_FIRMWARE_MT7922_BT
+	bool "MediaTek MT7922"
+	help
+	  Firmware files for MediaTek MT7922 bluetooth support
+
 config BR2_PACKAGE_LINUX_FIRMWARE_MEDIATEK_MT7650
 	bool "MediaTek MT7650"
 	help
@@ -400,6 +436,16 @@
 
 config BR2_PACKAGE_LINUX_FIRMWARE_MEDIATEK_MT7921
 	bool "MediaTek MT7921"
+	help
+	  MediaTek MT7921
+
+config BR2_PACKAGE_LINUX_FIRMWARE_MEDIATEK_MT7922
+	bool "MediaTek MT7922"
+	help
+	  MediaTek MT7922
+
+config BR2_PACKAGE_LINUX_FIRMWARE_MEDIATEK_MT7921
+	bool "MediaTek MT7921"
 	help
 	  MediaTek MT7921
 
--- buildroot/package/linux-firmware/linux-firmware.mk.orig
+++ buildroot/package/linux-firmware/linux-firmware.mk
@@ -9,6 +9,19 @@
 LINUX_FIRMWARE_SITE = $(BR2_KERNEL_MIRROR)/linux/kernel/firmware
 LINUX_FIRMWARE_INSTALL_IMAGES = YES
 
+LINUX_FIRMWARE_COMPRESSED_FILE_SUFFIX=
+ifeq ($(BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS_XZ),y)
+LINUX_FIRMWARE_DEPENDENCIES += host-xz
+LINUX_FIRMWARE_COMPRESS_CMD=$(HOST_DIR)/bin/xz -fv -C crc32 --lzma2=dict=2MiB
+LINUX_FIRMWARE_COMPRESSED_FILE_SUFFIX=.xz
+else
+ifeq ($(BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS_ZSTD),y)
+LINUX_FIRMWARE_DEPENDENCIES += host-zstd
+LINUX_FIRMWARE_COMPRESS_CMD=$(HOST_DIR)/bin/zstd -f -19 --rm
+LINUX_FIRMWARE_COMPRESSED_FILE_SUFFIX=.zst
+endif
+endif
+
 # Intel SST DSP
 ifeq ($(BR2_PACKAGE_LINUX_FIRMWARE_INTEL_SST_DSP),y)
 LINUX_FIRMWARE_FILES += intel/fw_sst_0f28.bin-48kHz_i2s_master
@@ -63,6 +76,18 @@
 LINUX_FIRMWARE_ALL_LICENSE_FILES += LICENCE.ibt_firmware
 endif
 
+# Mediatek MT7921 Bluetooth
+ifeq ($(BR2_PACKAGE_LINUX_FIRMWARE_MT7921_BT),y)
+LINUX_FIRMWARE_FILES += mediatek/BT_RAM_CODE_MT7961_1_2_hdr.bin
+LINUX_FIRMWARE_ALL_LICENSE_FILES += LICENCE.mediatek
+endif
+
+# Mediatek MT7922 Bluetooth
+ifeq ($(BR2_PACKAGE_LINUX_FIRMWARE_MT7922_BT),y)
+LINUX_FIRMWARE_FILES += mediatek/BT_RAM_CODE_MT7922_1_1_hdr.bin
+LINUX_FIRMWARE_ALL_LICENSE_FILES += LICENCE.mediatek
+endif
+
 # Qualcomm Atheros Rome 6174A Bluetooth
 ifeq ($(BR2_PACKAGE_LINUX_FIRMWARE_QUALCOMM_6174A_BT),y)
 LINUX_FIRMWARE_FILES += qca/rampatch_usb_00000302.bin qca/nvm_usb_00000302.bin
@@ -183,6 +208,13 @@
 LINUX_FIRMWARE_ALL_LICENSE_FILES += LICENCE.rtlwifi_firmware.txt
 endif
 
+# rtw89
+ifeq ($(BR2_PACKAGE_LINUX_FIRMWARE_RTL_RTW89),y)
+LINUX_FIRMWARE_FILES += \
+	rtw89/rtw*.bin
+LINUX_FIRMWARE_ALL_LICENSE_FILES += LICENCE.rtlwifi_firmware.txt
+endif
+
 # ar3011
 ifeq ($(BR2_PACKAGE_LINUX_FIRMWARE_AR3011),y)
 LINUX_FIRMWARE_FILES += ath3k-1.fw
@@ -413,6 +445,20 @@
 LINUX_FIRMWARE_ALL_LICENSE_FILES += LICENCE.mediatek
 endif
 
+# MT7921
+ifeq ($(BR2_PACKAGE_LINUX_FIRMWARE_MEDIATEK_MT7921),y)
+LINUX_FIRMWARE_FILES += mediatek/WIFI_MT7961_patch_mcu_1_2_hdr.bin \
+			mediatek/WIFI_RAM_CODE_MT7961_1.bin
+LINUX_FIRMWARE_ALL_LICENSE_FILES += LICENCE.mediatek
+endif
+
+# MT7922
+ifeq ($(BR2_PACKAGE_LINUX_FIRMWARE_MEDIATEK_MT7922),y)
+LINUX_FIRMWARE_FILES += mediatek/WIFI_MT7922_patch_mcu_1_1_hdr.bin \
+			mediatek/WIFI_RAM_CODE_MT7922_1.bin
+LINUX_FIRMWARE_ALL_LICENSE_FILES += LICENCE.mediatek
+endif
+
 # qca6174
 ifeq ($(BR2_PACKAGE_LINUX_FIRMWARE_QUALCOMM_6174),y)
 LINUX_FIRMWARE_FILES += ath10k/QCA6174
@@ -487,12 +533,22 @@
 endif
 
 ifeq ($(BR2_PACKAGE_LINUX_FIRMWARE_IWLWIFI_22000),y)
-LINUX_FIRMWARE_FILES += iwlwifi-QuZ-*.ucode iwlwifi-Qu-*.ucode
-LINUX_FIRMWARE_ALL_LICENSE_FILES += LICENCE.iwlwifi_firmware
-endif
-
-ifeq ($(BR2_PACKAGE_LINUX_FIRMWARE_IWLWIFI_22260),y)
-LINUX_FIRMWARE_FILES += iwlwifi-cc-a0-*.ucode
+LINUX_FIRMWARE_FILES += \
+	iwlwifi-Qu-b0-hr-b0-72.ucode \
+	iwlwifi-Qu-c0-hr-b0-72.ucode \
+	iwlwifi-Qu-b0-jf-b0-72.ucode \
+	iwlwifi-Qu-c0-jf-b0-72.ucode \
+	iwlwifi-QuZ-a0-hr-b0-72.ucode \
+	iwlwifi-QuZ-a0-jf-b0-72.ucode \
+	iwlwifi-cc-a0-72.ucode \
+	iwlwifi-so-a0-jf-b0-72.ucode \
+	iwlwifi-so-a0-hr-b0-72.ucode \
+	iwlwifi-so-a0-gf-a0-72.ucode \
+	iwlwifi-so-a0-gf-a0.pnvm \
+	iwlwifi-so-a0-gf4-a0-72.ucode \
+	iwlwifi-so-a0-gf4-a0.pnvm \
+	iwlwifi-ty-a0-gf-a0-72.ucode \
+	iwlwifi-ty-a0-gf-a0.pnvm
 LINUX_FIRMWARE_ALL_LICENSE_FILES += LICENCE.iwlwifi_firmware
 endif
 
@@ -887,15 +943,34 @@
 # sure we canonicalize the pointed-to file, to cover the symlinks of the form
 # a/foo -> ../b/foo  where a/ (the directory where to put the symlink) does
 # not yet exist.
+#
+# If BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS is enabled, modules are compressed
+# in this stage after being copied to the target directory. The compression
+# is not done on modules in the images directory, because they must not be
+# compressed when bundled via the EXTRA_FIRMWARE option.
+#
+# $1: target directory
+# $2: 'images' (uncompressed) or 'target' (compressed if enabled)
 define LINUX_FIRMWARE_INSTALL_FW
 	mkdir -p $(1)
 	$(TAR) xf $(@D)/br-firmware.tar -C $(1)
+	if [ "$(BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS)" = "y" ] && [ "$(2)" = "target" ]; then \
+		$(TAR) tf $(@D)/br-firmware.tar | while read f; do \
+			if [ -f $(1)/$$f ]; then \
+				$(LINUX_FIRMWARE_COMPRESS_CMD) $(1)/$$f; \
+			fi ; \
+		done ; \
+	fi
 	cd $(1) ; \
+	file_suffix="" ; \
+	if [ "$(BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS)" = "y" ] && [ "$(2)" = "target" ]; then \
+		file_suffix="$(LINUX_FIRMWARE_COMPRESSED_FILE_SUFFIX)" ; \
+	fi ; \
 	sed -r -e '/^Link: (.+) -> (.+)$$/!d; s//\1 \2/' $(@D)/WHENCE | \
 	while read f d; do \
-		if test -f $$(readlink -m $$(dirname "$$f")/$$d); then \
+		if test -f $$(readlink -m $$(dirname "$$f")/$${d}$${file_suffix}); then \
 			mkdir -p $$(dirname "$$f") || exit 1; \
-			ln -sf $$d "$$f" || exit 1; \
+			ln -sf $${d}$${file_suffix} "$${f}$${file_suffix}" || exit 1; \
 		fi ; \
 	done
 endef
@@ -903,11 +978,22 @@
 endif  # LINUX_FIRMWARE_FILES || LINUX_FIRMWARE_DIRS
 
 define LINUX_FIRMWARE_INSTALL_TARGET_CMDS
-	$(call LINUX_FIRMWARE_INSTALL_FW, $(TARGET_DIR)/lib/firmware)
+	$(call LINUX_FIRMWARE_INSTALL_FW,$(TARGET_DIR)/lib/firmware,target)
 endef
 
 define LINUX_FIRMWARE_INSTALL_IMAGES_CMDS
-	$(call LINUX_FIRMWARE_INSTALL_FW, $(BINARIES_DIR))
+	$(call LINUX_FIRMWARE_INSTALL_FW,$(BINARIES_DIR),images)
+endef
+
+define LINUX_FIRMWARE_LINUX_CONFIG_FIXUPS
+	$(if $(BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS_XZ),
+		$(call KCONFIG_ENABLE_OPT,CONFIG_FW_LOADER_COMPRESS)
+		$(call KCONFIG_ENABLE_OPT,CONFIG_FW_LOADER_COMPRESS_XZ)
+		$(call KCONFIG_DISABLE_OPT,CONFIG_FW_LOADER_COMPRESS_ZSTD))
+	$(if $(BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS_ZSTD),
+		$(call KCONFIG_ENABLE_OPT,CONFIG_FW_LOADER_COMPRESS)
+		$(call KCONFIG_DISABLE_OPT,CONFIG_FW_LOADER_COMPRESS_XZ)
+		$(call KCONFIG_ENABLE_OPT,CONFIG_FW_LOADER_COMPRESS_ZSTD))
 endef
 
 $(eval $(generic-package))
