--- buildroot/package/lighttpd/lighttpd.hash.orig	2017-01-06 02:02:36.703928134 +0100
+++ buildroot/package/lighttpd/lighttpd.hash	2017-01-06 02:03:05.707711270 +0100
@@ -1,2 +1,2 @@
 # From http://www.lighttpd.net/
-sha256 fe0c4a06dd2408a83ee7a2bfedc45e09597f3313cbda82485507573ae8fa948a  lighttpd-1.4.43.tar.xz
+sha256 adb66ca985651957feb209c91c55ebbf917d23630bfc3a216a2f70043c7b5422  lighttpd-1.4.44.tar.xz
--- buildroot/package/lighttpd/lighttpd.mk.orig	2017-01-06 02:02:44.379871078 +0100
+++ buildroot/package/lighttpd/lighttpd.mk	2017-01-06 02:03:15.711635667 +0100
@@ -5,7 +5,7 @@
 ################################################################################
 
 LIGHTTPD_VERSION_MAJOR = 1.4
-LIGHTTPD_VERSION = $(LIGHTTPD_VERSION_MAJOR).43
+LIGHTTPD_VERSION = $(LIGHTTPD_VERSION_MAJOR).44
 LIGHTTPD_SOURCE = lighttpd-$(LIGHTTPD_VERSION).tar.xz
 LIGHTTPD_SITE = http://download.lighttpd.net/lighttpd/releases-$(LIGHTTPD_VERSION_MAJOR).x
 LIGHTTPD_LICENSE = BSD-3c
