--- buildroot/package/nut/nut.mk.orig	2017-04-15 14:35:11.631620680 +0200
+++ buildroot/package/nut/nut.mk	2017-04-15 21:21:02.292137847 +0200
@@ -18,7 +18,9 @@
 # since the default location (/var/state/ups) maybe readonly.
 NUT_CONF_OPTS = \
 	--with-altpidpath=/var/run/upsd \
-	--without-hal
+	--without-hal \
+	--with-user=nobody \
+	--with-group=nogroup
 
 # For uClibc-based toolchains, nut forgets to link with -lm
 ifeq ($(BR2_TOOLCHAIN_USES_UCLIBC),y)
