--- buildroot/package/rpi-wifi-firmware/rpi-wifi-firmware.hash.orig
+++ buildroot/package/rpi-wifi-firmware/rpi-wifi-firmware.hash
@@ -1,2 +1,3 @@
 # Locally calculated
-sha256  872fde4f9942d9aba805880d6eaddfe050305626fd58ad955bfe77c04f6b75a5  brcmfmac43430-sdio.txt
+sha256 e4ce7ad5ce72406f958ed758539b65d958d1c0eb3e46e3f8e9f02b510d88e0b0 brcmfmac43430-sdio.txt
+sha256 15698c62457bcf25e60d063e6c666d6e1b7dacdf2b03e6d14ebbc619de6da6b7 brcmfmac43455-sdio.txt
--- buildroot/package/rpi-wifi-firmware/rpi-wifi-firmware.mk.orig
+++ buildroot/package/rpi-wifi-firmware/rpi-wifi-firmware.mk
@@ -4,20 +4,23 @@
 #
 ################################################################################
 
-RPI_WIFI_FIRMWARE_VERSION = 54bab3d6a6d43239c71d26464e6e10e5067ffea7
+RPI_WIFI_FIRMWARE_VERSION = 86e88fbf0345da49555d0ec34c80b4fbae7d0cd3
 # brcmfmac43430-sdio.bin comes from linux-firmware
-RPI_WIFI_FIRMWARE_SOURCE = brcmfmac43430-sdio.txt
+RPI_WIFI_FIRMWARE_SOURCE = brcmfmac43430-sdio.txt brcmfmac43455-sdio.txt
 # git repo contains a lot of unrelated files
-RPI_WIFI_FIRMWARE_SITE = https://raw.githubusercontent.com/RPi-Distro/firmware-nonfree/$(RPI_WIFI_FIRMWARE_VERSION)/brcm80211/brcm
+RPI_WIFI_FIRMWARE_SITE = https://raw.githubusercontent.com/RPi-Distro/firmware-nonfree/$(RPI_WIFI_FIRMWARE_VERSION)/brcm
 RPI_WIFI_FIRMWARE_LICENSE = PROPRIETARY
 
 define RPI_WIFI_FIRMWARE_EXTRACT_CMDS
-	cp $(DL_DIR)/$($(PKG)_SOURCE) $(@D)/
+	for file in $($(PKG)_SOURCE); do \
+		cp $(DL_DIR)/$${file} $(@D)/; \
+	done
 endef
 
 define RPI_WIFI_FIRMWARE_INSTALL_TARGET_CMDS
-	$(INSTALL) -D -m 0644 $(@D)/$(RPI_WIFI_FIRMWARE_SOURCE) \
-		$(TARGET_DIR)/lib/firmware/brcm/$(RPI_WIFI_FIRMWARE_SOURCE)
+	for file in $(RPI_WIFI_FIRMWARE_SOURCE); do \
+	  $(INSTALL) -D -m 0644 $(@D)/$${file} $(TARGET_DIR)/lib/firmware/brcm/${file}; \
+	done
 endef
 
 $(eval $(generic-package))
