--- buildroot/package/rpi-firmware/rpi-firmware.mk.orig	2017-08-27 08:23:11.249935624 +0200
+++ buildroot/package/rpi-firmware/rpi-firmware.mk	2017-08-27 08:21:43.616467076 +0200
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-RPI_FIRMWARE_VERSION = 685b3ceb0a6d6d6da7b028ee409850e83fb7ede7
+RPI_FIRMWARE_VERSION = 7f7df25d0f02eed1627e02a42d88c273c40bb5b0
 RPI_FIRMWARE_SITE = $(call github,raspberrypi,firmware,$(RPI_FIRMWARE_VERSION))
 RPI_FIRMWARE_LICENSE = BSD-3-Clause
 RPI_FIRMWARE_LICENSE_FILES = boot/LICENCE.broadcom
@@ -12,10 +12,9 @@
 
 ifeq ($(BR2_PACKAGE_RPI_FIRMWARE_INSTALL_DTBS),y)
 define RPI_FIRMWARE_INSTALL_DTB
-	$(INSTALL) -D -m 0644 $(@D)/boot/bcm2708-rpi-b.dtb $(BINARIES_DIR)/rpi-firmware/bcm2708-rpi-b.dtb
-	$(INSTALL) -D -m 0644 $(@D)/boot/bcm2708-rpi-b-plus.dtb $(BINARIES_DIR)/rpi-firmware/bcm2708-rpi-b-plus.dtb
-	$(INSTALL) -D -m 0644 $(@D)/boot/bcm2709-rpi-2-b.dtb $(BINARIES_DIR)/rpi-firmware/bcm2709-rpi-2-b.dtb
-	$(INSTALL) -D -m 0644 $(@D)/boot/bcm2710-rpi-3-b.dtb $(BINARIES_DIR)/rpi-firmware/bcm2710-rpi-3-b.dtb
+	for dtb in $(@D)/boot/*.dtb; do \
+		$(INSTALL) -D -m 0644 $${dtb} $(BINARIES_DIR)/rpi-firmware/$${dtb##*/} || exit 1; \
+	done
 endef
 endif
 
--- buildroot/package/rpi-firmware/rpi-firmware.hash.orig	2017-08-27 08:23:11.249935624 +0200
+++ buildroot/package/rpi-firmware/rpi-firmware.hash	2017-08-27 08:22:16.261014154 +0200
@@ -1,2 +1,2 @@
 # Locally computed
-sha256 42edea8317869f5a5ed87dfe009ceffce6a784ff40cb5da3e9e8a4fc61d79610 rpi-firmware-685b3ceb0a6d6d6da7b028ee409850e83fb7ede7.tar.gz
+sha256 ace6fe5dd2f6b01771bad6ae3d63787075f52caf5cf5b915f3a302a0e1bbe18a rpi-firmware-7f7df25d0f02eed1627e02a42d88c273c40bb5b0.tar.gz
