--- buildroot/package/rpi-userland/rpi-userland.hash.orig	2017-09-26 10:51:13.148087571 +0200
+++ buildroot/package/rpi-userland/rpi-userland.hash	2017-09-26 10:55:27.888339215 +0200
@@ -1,2 +1,2 @@
 # Locally computed
-sha256 476cde13d0d4bbf44ffc4d1d545847464d5c270bb72e4d940e43e5972c4de9da rpi-userland-29835184aa70c13d77a11d721e934a7298bdf46a.tar.gz
+sha256 46a37eaf0de598ce5804e48a5efc2120e07d753a57ed83463ebfa551562eefe6 rpi-userland-29835184aa70c13d77a11d721e934a7298bdf46a.tar.gz
