--- buildroot/package/psplash/psplash.mk.orig
+++ buildroot/package/psplash/psplash.mk
@@ -20,13 +20,13 @@
 endif
 
 ifeq ($(BR2_PACKAGE_PSPLASH_PROGRESS_BAR),y)
-PSPLASH_CONF_OPTS += --enable-progress-bar
+PSPLASH_CONF_OPTS +=
 else
 PSPLASH_CONF_OPTS += --disable-progress-bar
 endif
 
 ifeq ($(BR2_PACKAGE_PSPLASH_STARTUP_MSG),y)
-PSPLASH_CONF_OPTS += --enable-startup-msg
+PSPLASH_CONF_OPTS +=
 else
 PSPLASH_CONF_OPTS += --disable-startup-msg
 endif
