--- buildroot/package/rpi-firmware/Config.in.orig
+++ buildroot/package/rpi-firmware/Config.in
@@ -11,6 +11,10 @@
 
 if BR2_PACKAGE_RPI_FIRMWARE
 
+config BR2_PACKAGE_RPI_FIRMWARE_VERSION
+	string "Custom firmware version"
+	default "83dafbc92c0e63f76ca2ecdd42462d56489d1c77"
+
 config BR2_PACKAGE_RPI_FIRMWARE_BOOTCODE_BIN
 	bool "rpi 0/1/2/3 bootcode.bin"
 	help
--- buildroot/package/rpi-firmware/rpi-firmware.mk.orig
+++ buildroot/package/rpi-firmware/rpi-firmware.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-RPI_FIRMWARE_VERSION = 83dafbc92c0e63f76ca2ecdd42462d56489d1c77
+RPI_FIRMWARE_VERSION = $(call qstrip,$(BR2_PACKAGE_RPI_FIRMWARE_VERSION))
 RPI_FIRMWARE_SITE = $(call github,raspberrypi,firmware,$(RPI_FIRMWARE_VERSION))
 RPI_FIRMWARE_LICENSE = BSD-3-Clause
 RPI_FIRMWARE_LICENSE_FILES = boot/LICENCE.broadcom
