--- buildroot/package/ffmpeg/ffmpeg.mk.orig
+++ buildroot/package/ffmpeg/ffmpeg.mk
@@ -281,7 +281,7 @@
 FFMPEG_CONF_OPTS += --disable-vdpau
 endif
 
-ifeq ($(BR2_PACKAGE_RPI_USERLAND),y)
+ifeq ($(BR2_PACKAGE_RPI_USERLAND)$(BR2_arm),yy)
 FFMPEG_CONF_OPTS += --enable-mmal --enable-omx --enable-omx-rpi \
 	--extra-cflags=-I$(STAGING_DIR)/usr/include/IL
 FFMPEG_DEPENDENCIES += rpi-userland
--- buildroot/package/rpi-userland/0006-interface-vcos-pthreads-CMakeLists.txt-fix-build-wit.patch.orig
+++ buildroot/package/rpi-userland/0006-interface-vcos-pthreads-CMakeLists.txt-fix-build-wit.patch
@@ -1,4 +1,4 @@
-From 220c37b496f5c883bbfa955ee7fee471e41c34d4 Mon Sep 17 00:00:00 2001
+From c8270cea98bcce0d22ddc8e2f86eae2c9989efd4 Mon Sep 17 00:00:00 2001
 From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
 Date: Fri, 25 Feb 2022 19:21:06 +0100
 Subject: [PATCH] interface/vcos/pthreads/CMakeLists.txt: fix build with
@@ -20,7 +20,7 @@
  1 file changed, 4 insertions(+), 2 deletions(-)
 
 diff --git a/interface/vcos/pthreads/CMakeLists.txt b/interface/vcos/pthreads/CMakeLists.txt
-index 1d81ca3..43aca03 100644
+index 1d81ca3..100207f 100644
 --- a/interface/vcos/pthreads/CMakeLists.txt
 +++ b/interface/vcos/pthreads/CMakeLists.txt
 @@ -33,12 +33,14 @@ set (SOURCES
@@ -32,11 +32,11 @@
  if (VCOS_PTHREADS_BUILD_SHARED)
     add_library (vcos SHARED ${SOURCES})
 -   target_link_libraries (vcos pthread dl rt)
-+   target_link_libraries (vcos pthread dl rt ${EXECINFO_LIBRARY})
++   target_link_libraries (vcos pthread dl rt $<$<BOOL:${EXECINFO_LIBRARY}>:${EXECINFO_LIBRARY}>)
  else ()
     add_library (vcos ${SOURCES})
 -   target_link_libraries (vcos pthread rt)
-+   target_link_libraries (vcos pthread rt ${EXECINFO_LIBRARY})
++   target_link_libraries (vcos pthread rt $<$<BOOL:${EXECINFO_LIBRARY}>:${EXECINFO_LIBRARY}>)
  endif ()
  
  
--- buildroot/package/rpi-userland/Config.in.orig
+++ buildroot/package/rpi-userland/Config.in
@@ -1,13 +1,13 @@
 config BR2_PACKAGE_RPI_USERLAND
 	bool "rpi-userland"
-	depends on BR2_arm
+	depends on BR2_arm || BR2_aarch64
 	depends on BR2_INSTALL_LIBSTDCPP
 	depends on BR2_TOOLCHAIN_HAS_THREADS
 	depends on !BR2_STATIC_LIBS
-	select BR2_PACKAGE_HAS_LIBEGL
-	select BR2_PACKAGE_HAS_LIBGLES
-	select BR2_PACKAGE_HAS_LIBOPENVG
-	select BR2_PACKAGE_HAS_LIBOPENMAX
+	select BR2_PACKAGE_HAS_LIBEGL if !BR2_aarch64
+	select BR2_PACKAGE_HAS_LIBGLES if !BR2_aarch64
+	select BR2_PACKAGE_HAS_LIBOPENVG if !BR2_aarch64
+	select BR2_PACKAGE_HAS_LIBOPENMAX if !BR2_aarch64
 	help
 	  Raspberry Pi Userland contains the necessary library to use
 	  the VideoCore driver.
@@ -40,6 +40,6 @@
 endif
 
 comment "rpi-userland needs a toolchain w/ C++, threads, dynamic library"
-	depends on BR2_arm
+	depends on BR2_arm || BR2_aarch64
 	depends on !BR2_INSTALL_LIBSTDCPP || !BR2_TOOLCHAIN_HAS_THREADS || \
 		BR2_STATIC_LIBS
--- buildroot/package/rpi-userland/rpi-userland.mk.orig
+++ buildroot/package/rpi-userland/rpi-userland.mk
@@ -11,7 +11,9 @@
 RPI_USERLAND_INSTALL_STAGING = YES
 RPI_USERLAND_CONF_OPTS = -DVMCS_INSTALL_PREFIX=/usr
 
+ifeq ($(BR2_arm),y)
 RPI_USERLAND_PROVIDES = libegl libgles libopenmax libopenvg
+endif
 
 ifeq ($(BR2_PACKAGE_RPI_USERLAND_HELLO),y)
 
@@ -42,4 +44,8 @@
 endef
 RPI_USERLAND_POST_INSTALL_TARGET_HOOKS += RPI_USERLAND_POST_TARGET_CLEANUP
 
+ifeq ($(BR2_aarch64),y)
+RPI_USERLAND_CONF_OPTS += -DARM64=ON
+endif
+
 $(eval $(cmake-package))
