--- buildroot/package/linux-tools/S10hyperv.orig
+++ buildroot/package/linux-tools/S10hyperv
@@ -6,10 +6,13 @@
 # shellcheck source=/dev/null
 [ -r "/etc/default/hyperv" ] && . "/etc/default/hyperv"
 
+# only continue if we are in a HyperV platform
+[ -e "/sys/bus/vmbus" ] || exit 0
+
 start_one() {
 	printf 'Starting %s: ' "$1"
 	# shellcheck disable=SC2086 # we need the word splitting
-	start-stop-daemon -b -m -S -q -p "$PIDDIR/$1.pid" -x "/sbin/$1" -- -n
+	start-stop-daemon -b -m -S -q -p "$PIDDIR/$1.pid" -x "/usr/sbin/$1" -- -n
 	status=$?
 	if [ "$status" -eq 0 ]; then
 		echo "OK"
