--- buildroot/linux/Config.in.orig
+++ buildroot/linux/Config.in
@@ -434,6 +434,12 @@
 	  Choose this option to support Device Tree overlays
 	  on the target system.
 
+config BR2_LINUX_KERNEL_INSTALL_INTREE_OVERLAYS
+	bool "Deploy in-tree Device Tree overlays"
+	help
+	  Deploy in-tree device tree overlays stored in
+	  the architectures device tree directory overlays/.
+
 endif
 
 config BR2_LINUX_KERNEL_INSTALL_TARGET
--- buildroot/linux/linux.mk.orig
+++ buildroot/linux/linux.mk
@@ -459,6 +459,14 @@
 	)
 endef
 endif # BR2_LINUX_KERNEL_APPENDED_DTB
+ifeq ($(BR2_LINUX_KERNEL_INSTALL_INTREE_OVERLAYS),y)
+define LINUX_INSTALL_OVERLAYS
+	$(foreach ovldtb,$(wildcard $(LINUX_ARCH_PATH)/boot/dts/overlays/*.dtbo), \
+		$(INSTALL) -D -m 0644 $(ovldtb) $(1)/overlays/$(notdir $(ovldtb))
+	)
+	$(INSTALL) -D -m 0644 $(LINUX_ARCH_PATH)/boot/dts/overlays/overlay_map.dtb $(1)/overlays/
+endef
+endif # BR2_LINUX_KERNEL_INSTALL_INTREE_OVERLAYS
 endif # BR2_LINUX_KERNEL_DTB_IS_SELF_BUILT
 endif # BR2_LINUX_KERNEL_DTS_SUPPORT
 
@@ -529,6 +537,7 @@
 define LINUX_INSTALL_KERNEL_IMAGE_TO_TARGET
 	$(call LINUX_INSTALL_IMAGE,$(TARGET_DIR)/boot)
 	$(call LINUX_INSTALL_DTB,$(TARGET_DIR)/boot)
+	$(call LINUX_INSTALL_OVERLAYS,$(TARGET_DIR)/boot)
 endef
 endif
 
@@ -543,6 +552,7 @@
 define LINUX_INSTALL_IMAGES_CMDS
 	$(call LINUX_INSTALL_IMAGE,$(BINARIES_DIR))
 	$(call LINUX_INSTALL_DTB,$(BINARIES_DIR))
+	$(call LINUX_INSTALL_OVERLAYS,$(BINARIES_DIR))
 endef
 
 ifeq ($(BR2_STRIP_strip),y)
