--- buildroot/package/brcmfmac_sdio-firmware-rpi/brcmfmac_sdio-firmware-rpi.hash.orig
+++ buildroot/package/brcmfmac_sdio-firmware-rpi/brcmfmac_sdio-firmware-rpi.hash
@@ -1,3 +1,3 @@
 # Locally calculated
-sha256  2f0917b104739455dd488dd8f5af2ee4430801a7ac8fe8d9866e74bfbb185356  brcmfmac_sdio-firmware-rpi-26ff205b45dc109b498a70aaf182804ad9dbfea5.tar.gz
+sha256  02809dad206de1051226a58d1fb15f930bc2f19544bb2865893712250760067e  brcmfmac_sdio-firmware-rpi-4c1789eb8705336b956b3bf1a3e6e3359c0555e4.tar.gz
 sha256  ea8b7b7b6cfc6fd30587ed977100f6a542734ce53218b5b63de16180acddd599  LICENSE
--- buildroot/package/brcmfmac_sdio-firmware-rpi/brcmfmac_sdio-firmware-rpi.mk.orig
+++ buildroot/package/brcmfmac_sdio-firmware-rpi/brcmfmac_sdio-firmware-rpi.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-BRCMFMAC_SDIO_FIRMWARE_RPI_VERSION = 26ff205b45dc109b498a70aaf182804ad9dbfea5
+BRCMFMAC_SDIO_FIRMWARE_RPI_VERSION = 4c1789eb8705336b956b3bf1a3e6e3359c0555e4
 BRCMFMAC_SDIO_FIRMWARE_RPI_SITE = $(call github,LibreELEC,brcmfmac_sdio-firmware-rpi,$(BRCMFMAC_SDIO_FIRMWARE_RPI_VERSION))
 BRCMFMAC_SDIO_FIRMWARE_RPI_LICENSE = PROPRIETARY
 BRCMFMAC_SDIO_FIRMWARE_RPI_LICENSE_FILES = LICENSE
