--- buildroot/package/file/file.hash.orig
+++ buildroot/package/file/file.hash
@@ -1,6 +1,7 @@
 # Locally calculated after verifying signature
-# https://astron.com/pub/file/file-5.46.tar.gz.asc
+# https://astron.com/pub/file/file-5.45.tar.gz.asc
 # using key BE04995BA8F90ED0C0C176C471112AB16CB33B3A
-sha256  c9cc77c7c560c543135edc555af609d5619dbef011997e988ce40a3d75d86088  file-5.46.tar.gz
+sha256  fc97f51029bb0e2c9f4e3bffefdaf678f0e039ee872b9de5c002a6d09c784d82  file-5.45.tar.gz
 sha256  0bfa856a9930bddadbef95d1be1cf4e163c0be618e76ea3275caaf255283e274  COPYING
+sha256  4ccb60d623884ef637af4a5bc16b2cb350163e2135e967655837336019a64462  src/mygetopt.h
 sha256  2e14b6fa9a74f952ebfde3c96527158df281c953cfaf35609eb854d4da30131c  src/vasprintf.c
--- buildroot/package/file/file.mk.orig
+++ buildroot/package/file/file.mk
@@ -4,10 +4,10 @@
 #
 ################################################################################
 
-FILE_VERSION = 5.46
+FILE_VERSION = 5.45
 FILE_SITE = https://astron.com/pub/file
-FILE_LICENSE = BSD-2-Clause, BSD-3-Clause (vasprintf.c)
-FILE_LICENSE_FILES = COPYING src/vasprintf.c
+FILE_LICENSE = BSD-2-Clause, BSD-4-Clause (one file), BSD-3-Clause (one file)
+FILE_LICENSE_FILES = COPYING src/mygetopt.h src/vasprintf.c
 FILE_CPE_ID_VALID = YES
 
 FILE_AUTORECONF = YES
