--- a/www/rega/esp/datapointconfigurator.fn
+++ b/www/rega/esp/datapointconfigurator.fn
@@ -1381,7 +1381,7 @@
         Call("/esp/datapointconfigurator.fn::CreateUnknown()");
       }
     }
-    if ( (isKnownControl) && (bSV)  && (oEName.Find("HmIP-RF.") == -1) )
+    if ( (isKnownControl) && (bSV)  && (oEName.Find("HmIP-RF.") == -1) && (oEName.Find("CUxD.CUX") == -1) )
     {
       WriteLine( '</tr></table>');
       WriteLine( "<script>conInfo('SysVar found.');</script>" );
