--- a/usr/lib/tcl8.2/homematic/homematic.tcl
+++ b/usr/lib/tcl8.2/homematic/homematic.tcl
@@ -62,7 +62,7 @@
 namespace eval ::HomeMatic::Addon {
   global ::HomeMatic::Addon::_CONFIG_FILE_
   
-  namespace export GetAll AddConfigPage
+  namespace export GetAll AddConfigPage RemoveConfigPage
   
   set ::HomeMatic::Addon::_CONFIG_FILE_ "/etc/config/hm_addons.cfg"
 }
@@ -82,6 +82,19 @@
   ::HomeMatic::Util::SaveFile $filename [array get addons]
 }
 
+# removes an addon configuration page
+proc ::HomeMatic::Addon::RemoveConfigPage { id } {
+  global ::HomeMatic::Addon::_CONFIG_FILE_
+  set filename $::HomeMatic::Addon::_CONFIG_FILE_
+  
+  array set addons [::HomeMatic::Util::LoadFile $filename]
+  # remove addon config, if present
+  if {[info exists addons($id)]} { 
+    unset addons($id)
+  }  
+  ::HomeMatic::Util::SaveFile $filename [array get addons]
+}
+
 proc ::HomeMatic::Addon::GetAll { } {
   global ::HomeMatic::Addon::_CONFIG_FILE_
   set filename $::HomeMatic::Addon::_CONFIG_FILE_
