--- a/www/webui/webui.js
+++ b/www/webui/webui.js
@@ -14746,9 +14746,7 @@
       //channels         : this.filter(this.channels)
     });
 
-    if (! userIsNoExpert) {
-      jQuery(".j_expertChannel").show();
-    }
+    jQuery(".j_expertChannel").show();
 
     translateJSTemplate("#ChannelChooserDialog");
     translatePage(".j_rooms, .j_functions");
@@ -15234,9 +15232,7 @@
       channels         : this.filter(this.channels)
     });
 
-    if (! userIsNoExpert) {
-      jQuery(".j_expertChannel").show();
-    }
+    jQuery(".j_expertChannel").show();
 
     // Add extended channel description and hide the second user channel
     jQuery(".j_extChnDescr").each(function(index){
