--- a/www/config/cp_maintenance.cgi
+++ b/www/config/cp_maintenance.cgi
@@ -1260,6 +1260,7 @@
 }
 
 proc action_update_start {} {
+  puts ""
   catch { exec killall hss_lcd }
   catch { exec lcdtool {Saving   Data...  } }
   rega system.Save()
@@ -1276,6 +1277,7 @@
 }
 
 proc action_reboot {} {
+  puts ""
   catch { exec killall hss_lcd }
   catch { exec lcdtool {Saving   Data...  } }
   rega system.Save()
@@ -1284,6 +1286,7 @@
   exec /sbin/reboot
 }
 proc action_shutdown {} {
+  puts ""
   catch { exec killall hss_lcd }
   catch { exec lcdtool {Saving   Data...  } }
   rega system.Save()
--- a/www/config/cp_security.cgi
+++ b/www/config/cp_security.cgi
@@ -1563,6 +1563,7 @@
 }
 
 proc action_reboot {} {
+  puts ""
   exec /sbin/reboot
 }
 
