--- a/www/webui/webui.js
+++ b/www/webui/webui.js
@@ -8813,10 +8813,10 @@
     this.callback = callback;
     if (this.isPresent === false)
     {
-      //this.isPresent = true;
-      //new ConfigDataLoader(callback);
-      this.configDataLoader.showMessage();
-      this.showMessage = true;
+      this.isPresent = true;
+      new ConfigDataLoader(callback);
+      //this.configDataLoader.showMessage();
+      //this.showMessage = true;
     }
     else
     {
