diff -ruN monit-5.29.0.orig/configure.ac monit-5.29.0/configure.ac
--- monit-5.29.0.orig/configure.ac	2021-08-24 18:15:35.000000000 +0200
+++ monit-5.29.0/configure.ac	2021-11-20 10:34:31.074783265 +0100
@@ -8,7 +8,7 @@
 # Example: 5.0_beta2
 AC_INIT([monit], [5.29.0], [monit-general@nongnu.org])
 AC_CONFIG_AUX_DIR(config)
-AC_CONFIG_MACRO_DIR([m4])
+AC_CONFIG_MACRO_DIRS([m4 config])
 AM_INIT_AUTOMAKE
 AC_CONFIG_SRCDIR([src/monit.c])
 AC_CONFIG_SUBDIRS([libmonit])
diff -ruN monit-5.29.0.orig/libmonit/configure.ac monit-5.29.0/libmonit/configure.ac
--- monit-5.29.0.orig/libmonit/configure.ac	2021-08-24 18:15:35.000000000 +0200
+++ monit-5.29.0/libmonit/configure.ac	2021-11-20 10:34:54.123439163 +0100
@@ -4,7 +4,7 @@
 
 AC_INIT([libmonit], [1.0], [monit-dev@tildeslash.com])
 AC_CONFIG_AUX_DIR(config)
-AC_CONFIG_MACRO_DIR([m4])
+AC_CONFIG_MACRO_DIRS([m4 config])
 AM_INIT_AUTOMAKE
 AC_CONFIG_SRCDIR([src])
 
