diff -ruN monit-5.27.1.orig/configure.ac monit-5.27.1/configure.ac
--- monit-5.27.1.orig/configure.ac	2020-10-24 20:26:57.000000000 +0200
+++ monit-5.27.1/configure.ac	2020-11-19 17:31:30.656661653 +0100
@@ -9,6 +9,7 @@
 AC_INIT([monit], [5.27.1], [monit-general@nongnu.org])
 AC_CONFIG_AUX_DIR(config)
 AC_CONFIG_MACRO_DIR([m4])
+AC_CONFIG_MACRO_DIR([config])
 AM_INIT_AUTOMAKE
 AC_CONFIG_SRCDIR([src/monit.c])
 AC_CONFIG_SUBDIRS([libmonit])
diff -ruN monit-5.27.1.orig/libmonit/configure.ac monit-5.27.1/libmonit/configure.ac
--- monit-5.27.1.orig/libmonit/configure.ac	2020-10-24 20:26:57.000000000 +0200
+++ monit-5.27.1/libmonit/configure.ac	2020-11-19 17:31:44.220707976 +0100
@@ -5,6 +5,7 @@
 AC_INIT([libmonit], [1.0], [monit-dev@tildeslash.com])
 AC_CONFIG_AUX_DIR(config)
 AC_CONFIG_MACRO_DIR([m4])
+AC_CONFIG_MACRO_DIR([config])
 AM_INIT_AUTOMAKE
 AC_CONFIG_SRCDIR([src])
 
