--- a/detect_radio_module/radiomoduleconnector.cpp
+++ b/detect_radio_module/radiomoduleconnector.cpp
@@ -66,8 +66,8 @@ RadioModuleConnector::RadioModuleConnector(int fd) : _fd(fd)
     tty.c_cc[VTIME] = 1;
     tty.c_cc[VMIN] = 0;
 
-    cfsetispeed(&tty, 115200);
-    cfsetospeed(&tty, 115200);
+    cfsetispeed(&tty, B115200);
+    cfsetospeed(&tty, B115200);
     tcsetattr(_fd, TCSANOW, &tty);
   }
 }
