--- a/drivers/video/bcm2835.c.orig	2019-07-05 18:01:17.498092686 +0200
+++ b/drivers/video/bcm2835.c	2019-07-05 18:03:12.451445197 +0200
@@ -19,7 +19,7 @@
 
 	debug("bcm2835: Query resolution...\n");
 	ret = bcm2835_get_video_size(&w, &h);
-	if (ret)
+	if (ret || w == 0 || h == 0)
 		return -EIO;
 
 	debug("bcm2835: Setting up display for %d x %d\n", w, h);
@@ -27,6 +27,9 @@
 				       BCM2835_MBOX_ALPHA_MODE_IGNORED,
 				       &fb_base, &fb_size, &pitch);
 
+	if (ret)
+		return -EIO;
+
 	debug("bcm2835: Final resolution is %d x %d\n", w, h);
 
 	/* Enable dcache for the frame buffer */
