--- occu/arm-gnueabihf/packages/lighttpd/etc/lighttpd/conf.d/cgi.conf.orig
+++ occu/arm-gnueabihf/packages/lighttpd/etc/lighttpd/conf.d/cgi.conf
@@ -1,5 +1,7 @@
 #### CGI module
 cgi.assign = ( ".pl"  => "/usr/bin/perl",
-                ".cgi" => "/opt/hm/bin/tclsh",
+                ".cgi" => "/bin/tclsh",
                 ".ccc" => ""
              )
+cgi.x-sendfile = "enable"
+cgi.x-sendfile-docroot = ( "/usr/local/tmp" )
--- occu/arm-gnueabihf/packages/lighttpd/etc/lighttpd/conf.d/setenv.conf.orig
+++ occu/arm-gnueabihf/packages/lighttpd/etc/lighttpd/conf.d/setenv.conf
@@ -1,3 +1,3 @@
 setenv.add-environment = (
-        "TZ" => "CET-1CEST-2,M3.5.0/02:00:00,M10.5.0/03:00:00"
+        "EQ3_OS_PLATFORM" => "HM-RASPBERRYMATIC"
 )
--- occu/arm-gnueabihf/packages/lighttpd/etc/lighttpd/lighttpd.conf.orig
+++ occu/arm-gnueabihf/packages/lighttpd/etc/lighttpd/lighttpd.conf
@@ -14,10 +14,10 @@
 ## chroot example aswell.
 ##
 var.log_root    = "/var/log"
-var.server_root = "/opt/hm/www"
+var.server_root = "/www/"
 var.state_dir   = "/var/run"
 var.home_dir    = "/var/lib/lighttpd"
-var.conf_dir    = "/opt/hm/etc/lighttpd"
+var.conf_dir    = "/etc/lighttpd"
 
 ## 
 ## run the server chrooted.
@@ -87,6 +87,37 @@
 ##
 server.port = 80
 
+$SERVER["socket"] == ":8181" {
+	server.document-root = server_root
+}
+
+$SERVER["socket"] == ":443" {
+	server.document-root = server_root
+	ssl.engine = "enable"
+	ssl.pemfile = "/etc/config/server.pem"
+}
+
+$SERVER["socket"] == "[::]:80" {
+	server.document-root = server_root
+	server.use-ipv6 = "enable"
+}
+
+$SERVER["socket"] == "[::]:8181" {
+	server.document-root = server_root
+	server.use-ipv6 = "enable"
+}
+
+$SERVER["socket"] == "[::]:443" {
+	server.document-root = server_root
+	ssl.engine = "enable"
+	ssl.pemfile = "/etc/config/server.pem"
+	server.use-ipv6 = "enable"
+}
+
+#$HTTP["scheme"] == "https" {
+#	setenv.add-response-header  = ( "Strict-Transport-Security" => "max-age=31536000; includeSubdomains; preload")
+#}
+
 ##
 ## Use IPv6?
 ##
@@ -127,6 +158,11 @@
 server.pid-file = state_dir + "/lighttpd.pid"
 
 ##
+## number of worker processes to spawn.
+##
+server.max-worker = 4
+
+##
 #######################################################################
 
 #######################################################################
@@ -230,14 +266,14 @@
 ##
 ## Default: 5
 ##
-#server.max-keep-alive-idle = 5
+server.max-keep-alive-idle = 60
 
 ##
 ## How many keep-alive requests until closing the connection.
 ##
 ## Default: 16
 ##
-#server.max-keep-alive-requests = 16
+server.max-keep-alive-requests = 50
 
 ##
 ## Maximum size of a request in kilobytes.
@@ -248,18 +284,26 @@
 #server.max-request-size = 0
 
 ##
+## Maximum size of a request headers in kilobytes.
+## By default it is 8k (8192).
+##
+## Requests with a header larger than this will end in a 431 error.
+##
+server.max-request-field-size = 65536
+
+##
 ## Time to read from a socket before we consider it idle.
 ##
 ## Default: 60
 ##
-#server.max-read-idle = 60
+server.max-read-idle = 600
 
 ##
 ## Time to write to a socket before we consider it idle.
 ##
 ## Default: 360
 ##
-#server.max-write-idle = 360
+server.max-write-idle = 600
 
 ##
 ##  Traffic Shaping 
@@ -370,7 +414,12 @@
 ##
 ## defaults to /var/tmp as we assume it is a local harddisk
 ##
-server.upload-dirs = ( "/var/tmp" )
+server.upload-dirs = ( "/usr/local/tmp" )
+
+##
+## default size of temp files
+##
+server.upload-temp-file-size = 16777216
 
 ##
 #######################################################################
@@ -439,7 +488,6 @@
 ##
 ## custom includes like vhosts.
 ##
-#include "conf.d/config.conf"
-#include_shell "cat /etc/lighttpd/vhosts.d/*.conf"
+include "/etc/config/lighttpd/*.conf"
 ##
 #######################################################################
--- occu/arm-gnueabihf/packages/lighttpd/etc/lighttpd/modules.conf.orig
+++ occu/arm-gnueabihf/packages/lighttpd/etc/lighttpd/modules.conf
@@ -50,6 +50,7 @@
 #  "mod_usertrack",
   "mod_cgi",
   "mod_proxy",
+  "mod_openssl",
 )
 
 ##
--- occu/firmware/HM-MOD-UART/fwmap.orig
+++ occu/firmware/HM-MOD-UART/fwmap
@@ -1,7 +1,7 @@
 #Type           Filename                Version
 
 #Coprozessor
-CCU2                    coprocessor_update.eq3                         	1.4.1		# CoProzessor HM-MOD-UART
+#CCU2                    coprocessor_update.eq3                         	1.4.1		# CoProzessor HM-MOD-UART
 
-#CCU2			dualcopro_si1002_update_blhm.eq3		2.8.6		# Dual CoProzessor HM-MOD-UART
+CCU2			dualcopro_si1002_update_blhm.eq3		2.8.6		# Dual CoProzessor HM-MOD-UART
 
--- occu/WebUI/www/api/backup/version.cgi.orig
+++ occu/WebUI/www/api/backup/version.cgi
@@ -1 +1,11 @@
-### rfd-config##source common.tclproc main { } {	puts "Content-Type: text/plain"	puts ""	puts -nonewline [loadFile /boot/VERSION]}startup
\ No newline at end of file
+##
+# rfd-config
+##
+source common.tcl
+proc main { } {
+ puts "Content-Type: text/plain"
+ puts ""
+ puts -nonewline [loadFile /VERSION]
+}
+
+startup
--- occu/WebUI/www/api/homematic.cgi.orig
+++ occu/WebUI/www/api/homematic.cgi
@@ -22,15 +22,13 @@
 # um eine Notlösung. 
 ##
 
-set central true
+set central true 
 
 # Zentrale oder Konfigtool (/www/ = Zentrale)
 if {$env(DOCUMENT_ROOT) != "\/www\/"} then {
   set central false
 }
 
-set central true
-
 #####################
 # Verwendete Module #
 #####################
--- occu/WebUI/www/config/cp_maintenance.cgi.orig
+++ occu/WebUI/www/config/cp_maintenance.cgi
@@ -394,7 +394,7 @@
                     table_row {
                       table_data {
                         division {class="CLASS20908" style="display: none"} {id="btnFwDownload"} {} "onClick=\"window.location.href='$REMOTE_FIRMWARE_SCRIPT?cmd=download&version=$cur_version&serial=$serial&lang=de&product=HM-CCU2';\"" {}
-                        division {class="CLASS20908"}  "onClick=\"showCCULicense();\"" {puts "\${dialogSettingsCMBtnPerformSoftwareUpdateDownload}"}
+                        division {class="CLASS20908"}  "onClick=\"window.open('https://github.com/jens-maus/RaspberryMatic/releases','_blank');\"" {puts "\${dialogSettingsCMBtnPerformSoftwareUpdateDownload}"}
                       }
                     }
                   }
@@ -446,27 +446,7 @@
           }
         }
         table_data {align="left"} {class="CLASS20921"} {
-          puts "\${dialogSettingsCMHintSoftwareUpdate1}"
-          number_list {class="j_noForcedUpdate"} {
-            li {
-              ${dialogSettingsCMHintSoftwareUpdate2}            }
-            li {
-               ${dialogSettingsCMHintSoftwareUpdate3}
-            }
-            li {
-               ${dialogSettingsCMHintSoftwareUpdate3a}
-            }
-            set bat_level [get_bat_level]
-            if {$bat_level < 50} {
-              set msg " \${dialogSettingsCMHintSoftwareUpdate4a} $bat_level%. "
-              append msg  \${dialogSettingsCMHintSoftwareUpdate4b}
-              li $msg
-            }
-          }
-
-          division {class="j_forcedUpdate" style="padding:10px;"} {
-            puts "<br/>\${dialogSettingsCMHintSoftwareUpdate2}"
-          }
+          puts "\${dialogSettingsCMHintSoftwareUpdateRaspMatic}"
         }
       }
       table_row {class="CLASS20902 j_noForcedUpdate j_fwUpdateOnly"} {
--- occu/WebUI/www/config/cp_security.cgi.orig
+++ occu/WebUI/www/config/cp_security.cgi
@@ -258,7 +258,7 @@
         } else {
           exec touch /usr/local/.doFactoryReset
         }
-        exec kill -SIGQUIT 1
+        exec /sbin/reboot
     }]} {
 
       # TWIST-22
@@ -333,13 +333,13 @@
 
 proc action_backup_restore_check {} {
     global env
-    cd /tmp/
+    cd /usr/local/tmp/
     
     http_head
     set i 0
     if { [catch {
-        exec tar xf new_config.tar
-        file delete -force /tmp/new_config.tar
+        exec tar xf /usr/local/tmp/new_config.tar 2>/dev/null
+        file delete -force /usr/local/tmp/new_config.tar
     
     set config_version [read_version "firmware_version"]
     set ccu1_backup false
@@ -471,10 +471,10 @@
 
 proc action_backup_restore_go {} {
   global env
-  cd /tmp/
+  cd /usr/local/tmp/
   http_head
     
-  set system_version [read_version "/boot/VERSION"]
+  set system_version [read_version "/VERSION"]
   set config_version [read_version "firmware_version"]
   set ccu1_backup false
   if { [version_compare $config_version 2.0.0] < 0 } {
@@ -569,7 +569,7 @@
     }
     if { "CCU2_CCU2" == $migration_mode } {
       # CCU2 ==> CCU2
-      if { [catch {exec tar xzf /tmp/usr_local.tar.gz} errorMessage] } {
+      if { [catch {exec tar xzf /usr/local/tmp/usr_local.tar.gz} errorMessage] } {
           # set msg "Beim Einspielen des Systembackups ist ein Fehler aufgetreten. Bitte versuchen Sie es erneut. "
           # append msg "Falls dieser Fehler wiederholt Auftritt, wenden Sie sich bitte mit der folgenden Fehlermeldung an den Kundenservice:\n<br>"
           # append msg $errorMessage
@@ -583,31 +583,32 @@
     file delete -force /usr/local/eQ-3-Backup/restore
     file mkdir /usr/local/eQ-3-Backup/restore
     cd /usr/local/eQ-3-Backup/restore
-      if { [catch {exec tar xzf /tmp/usr_local.tar.gz --strip 2} errorMessage] } {
+      if { [catch {exec tar xzf /usr/local/tmp/usr_local.tar.gz --strip 2} errorMessage] } {
           # show message "Beim Einspielen des Systembackups ist ein Fehler aufgetreten. Bitte versuchen Sie es erneut. "
           put_message "\${dialogSettingsSecurityMessageSysBackupErrorTitle}" "\${dialogSettingsSecurityMessageSysBackupErrorContent} $errorMessage"
           set backuperror true
         } else {
           set backuperror false
+          exec touch /usr/local/.doBackupRestore
         }
     } elseif { "CCU2_CCU3" == $migration_mode } {
           # CCU2 ==> CCU3
-        # be sure /tmp/backup is empty
-        file delete -force /tmp/backup
+        # be sure /usr/local/tmp/backup is empty
+        file delete -force /usr/local/tmp/backup
         # be sure restore directory is empty
         file delete -force /usr/local/eQ-3-Backup/restore
-        file mkdir /tmp/backup
+        file mkdir /usr/local/tmp/backup
 
-        cd /tmp/backup
-        if { [catch {exec tar xzf /tmp/usr_local.tar.gz} errorMessage] } {
+        cd /usr/local/tmp/backup
+        if { [catch {exec tar xzf /usr/local/tmp/usr_local.tar.gz} errorMessage] } {
           put_message "\${dialogSettingsSecurityMessageSysBackupErrorTitle}" "\${dialogSettingsSecurityMessageSysBackupErrorContent} $errorMessage"
           set backuperror true
         } else {
           # only do this when the compatibility check says ok:
           #copy old files
           file mkdir /usr/local/eQ-3-Backup/restore/etc/config
-          if { [file exists /tmp/backup/usr/local/etc/config] } {
-            cd /tmp/backup/usr/local/etc/config
+          if { [file exists /usr/local/tmp/backup/usr/local/etc/config] } {
+            cd /usr/local/tmp/backup/usr/local/etc/config
           }
 
           if { [file exists InterfacesList.xml] } { file copy InterfacesList.xml /usr/local/eQ-3-Backup/restore/etc/config/InterfacesList.xml }
@@ -636,26 +637,31 @@
           if { [file exists time.conf] } { file copy time.conf /usr/local/eQ-3-Backup/restore/etc/config/time.conf }
           set backuperror false
           cd /
+          exec touch /usr/local/.doBackupRestore
         }
       }
     } else {  # backup for version < 2
+      catch {
+        exec killall ReGaHss
+        exec sleep 5
+      } 
       #delete existing files
-      file delete -force /tmp/backup
+      file delete -force /usr/local/tmp/backup
       file delete -force /etc/config/hs485d /etc/config/hs485types /etc/config/rfd /etc/config/userprofiles
       file delete -force /etc/config/homematic.regadom /etc/config/homematic.regadom.bak
       file delete -force /etc/config/ids /etc/config/ntpclient /etc/config/rega.conf /etc/config/syslog
       file delete -force /etc/config/tweaks /etc/config/TZ /etc/config/server.pem /etc/config/keys
       file delete -force /etc/config/time.conf
 
-      file mkdir /tmp/backup
-      cd /tmp/backup
-      if { [catch {exec tar xzf /tmp/usr_local.tar.gz} errorMessage] } {
+      file mkdir /usr/local/tmp/backup
+      cd /usr/local/tmp/backup
+      if { [catch {exec tar xzf /usr/local/tmp/usr_local.tar.gz} errorMessage] } {
         put_message "\${dialogSettingsSecurityMessageSysBackupErrorTitle}" "\${dialogSettingsSecurityMessageSysBackupErrorContent} $errorMessage"
         set backuperror true
       } else {
         #copy old files
-        if { [file exists /tmp/backup/usr/local/etc/config] } {
-          cd /tmp/backup/usr/local/etc/config
+        if { [file exists /usr/local/tmp/backup/usr/local/etc/config] } {
+          cd /usr/local/tmp/backup/usr/local/etc/config
         }
 
         if { [file exists hs485d] } { file copy hs485d /etc/config/hs485d }
@@ -709,6 +715,8 @@
         catch {exec eq3configcmd rfd-interface-copy rfd.conf /etc/config/rfd.conf}
 
         set backuperror false
+        exec /sbin/reboot
+        return
       }
     cd /
   }
@@ -729,8 +737,7 @@
   }
 
   if { "false" == $backuperror } {
-        exec mount -o remount,ro /usr/local
-        exec mount -o remount,rw /usr/local
+        exec sync
         division {class="popupTitle"} {
             puts "\${dialogSettingsSecurityMessageSysBackupRestartSystemTitle}"
         }
@@ -771,8 +778,6 @@
     cgi_javascript {
       puts "translatePage('#messagebox');"
     }
-
-    file delete -force /tmp/new_config.tar /tmp/firmware_version /tmp/signature /tmp/usr_local.tar.gz /tmp/backup
 }
 
 proc put_message {title msg args} {
@@ -1215,30 +1220,27 @@
 
 proc action_create_backup {} {
     set HOSTNAME [exec hostname]
+    set system_version [read_version "/VERSION"]
     set iso8601_date [exec date -Iseconds]
+    set tmpdir [exec mktemp -d -p /usr/local/tmp]
     regexp {^(\d+)-(\d+)-(\d+)T(\d+):(\d+):(\d+)([+-]\d+)$} $iso8601_date dummy year month day hour minute second zone
+    set backupfile [set HOSTNAME]-$system_version-$year-$month-$day-$hour$minute.sbk
     #save DOM
     rega system.Save()
     cd /
-    exec tar czf /tmp/usr_local.tar.gz usr/local
-    cd /tmp/
+    catch { exec tar --owner=root --group=root --exclude=usr/local/tmp --exclude=usr/local/lost+found --exclude-tag=.nobackup --one-file-system --ignore-failed-read -czf $tmpdir/usr_local.tar.gz usr/local }
+    cd $tmpdir/
     #sign the configuration with the current key
     exec crypttool -s -t 1 <usr_local.tar.gz >signature
     #store the current key index
     exec crypttool -g -t 1 >key_index
-    file copy -force /boot/VERSION firmware_version
-    set fd [open "|tar c usr_local.tar.gz signature firmware_version key_index"]
-    catch {fconfigure $fd -translation binary}
-    catch {fconfigure $fd -encoding binary}
-    puts "Content-Type:application/x-download"
-    puts "Content-Disposition:attachment;filename=[set HOSTNAME]-$year-$month-$day.sbk\n"
-    catch {fconfigure stdout -translation binary}
-    catch {fconfigure stdout -encoding binary}
-    while { ! [eof $fd]} {
-        puts -nonewline [read $fd 65536]
-    }
-    close $fd
-    file delete -force /tmp/usr_local.tar.gz /tmp/firmware_version /tmp/signature
+    file copy -force /VERSION firmware_version
+    catch { exec tar --owner=root --group=root -cf /usr/local/tmp/last_backup.sbk usr_local.tar.gz signature firmware_version key_index }
+    cd /
+    exec rm -rf $tmpdir
+    puts "X-Sendfile: /usr/local/tmp/last_backup.sbk"
+    puts "Content-Type: application/octet-stream"
+    puts "Content-Disposition: attachment; filename=\"$backupfile\"\n"
 }
 
 proc action_backup_upload {} {
--- occu/WebUI/www/config/cp_time.cgi.orig
+++ occu/WebUI/www/config/cp_time.cgi
@@ -645,7 +645,9 @@
     if { $fd <0 } { return 0 }
     puts $fd "$TIMEZONES($timezone)"
     close $fd
-    
+ 
+    catch {exec /bin/updateTZ.sh}
+   
     rega_script "var x=system.Longitude($lon);var y=system.Latitude($lat);var a=dom.ChangedTimeManually();"
     
     catch {exec /sbin/hwclock -wu}
--- occu/WebUI/www/config/help.cgi.orig
+++ occu/WebUI/www/config/help.cgi
@@ -3,8 +3,8 @@
 source common.tcl
 load tclrega.so
 
-set PRODUCT_NAME "Zentrale"
-set PRODUCT_ID "CCU[getProduct]"
+set PRODUCT_NAME "RaspberryMatic"
+set PRODUCT_ID "CCU"
 set HOMEMATIC_URL "http://www.homematic.com"
 set HOMEMATIC_IP_URL "https://www.homematic-ip.com"
 
@@ -18,8 +18,16 @@
 proc action_put_page {} {
   global PRODUCT_NAME PRODUCT_ID HOMEMATIC_URL HOMEMATIC_IP_URL HELP_URL HELP_IP_URL LANGUAGE CUR_YEAR env
 
-  set SERIAL [read_var /var/ids SerialNumber]
-  set VERSION [read_var /boot/VERSION VERSION]
+  if {[file exist /var/board_sgtin]} {
+    set SERIAL [exec cat /var/board_sgtin]
+  } elseif {[file exist /var/board_serial]} {
+    set SERIAL [exec cat /var/board_serial]
+  } elseif {[file exist /sys/module/plat_eq3ccu2/parameters/board_serial]} {
+    set SERIAL [exec cat /sys/module/plat_eq3ccu2/parameters/board_serial]
+  } else {
+    set SERIAL "n/a"
+  }
+  set VERSION [read_var /VERSION VERSION]
   set HELP_ARGUMENTS "lang=$LANGUAGE&vers=$VERSION&serial=$SERIAL"
 
 
@@ -57,6 +65,7 @@
                 puts "<li><h1 class='helpTitle'><u>\${menuHelpPage}</u></h1></li>"
                 puts "<li><a target='_blank' href=\'$HELP_URL\'>\${dialogHelpLinkOnlineHelpA}</a></li>"
                 puts "<li><a target='_blank' href=\'$HELP_IP_URL\'>\${dialogHelpLinkOnlineHelpB}</a></li>"
+                puts "<li><a target='_blank' href='https://raspberrymatic.de/#memo-documentation'>RaspberryMatic</a></li>"
                 puts "<li><a target='_blank' href='/licenseinfo.htm'>\${lblLicenseInformation}</a></li>"
               puts "</ul>"
       puts "</td>"
@@ -75,6 +84,7 @@
           # puts "<li>DOM [rega dom.BuildLabel()]</li>"
           puts "<li><a target='_blank' href=\"$HOMEMATIC_URL\">\${homepage} \${LabelHomeMatic}</a></li>"
           puts "<li><a target='_blank' href=\"$HOMEMATIC_IP_URL/start.html\">\${homepage} \${dialogHelpLinkOnlineHelpB}</a></li>"
+          puts "<li><a target='_blank' href=\"https://raspberrymatic.de/\">\${homepage} RaspberryMatic</a></li>"
           puts "<li style='padding-top:25px;'><img src='/ise/img/homematic_logo_small.png'/></li>"
         puts "</ul>"
 
--- occu/WebUI/www/config/ic_common.tcl.orig
+++ occu/WebUI/www/config/ic_common.tcl
@@ -23,10 +23,10 @@
 set GL_FLAG_RECEIVER_UNKNOWN     0x8
 
 set HTMLTITLE "HomeMatic Interne Konfiguration"
-set INTERFACES_FILE "/opt/hm/etc/config/InterfacesList.xml"
+set INTERFACES_FILE "/etc/config/InterfacesList.xml"
 
 if { ![info exists env(CONFIG_ROOT)] } {
-    set env(CONFIG_ROOT) "/opt/hm/etc/config"
+    set env(CONFIG_ROOT) "/etc/config"
 }
 
 set USERPROFILESPATH "$env(CONFIG_ROOT)/userprofiles"
--- occu/WebUI/www/webui/js/lang/de/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.js
@@ -1,8 +1,8 @@
 jQuery.extend(true,HMIdentifier , {
   "de" : {
-    "CCUFullNameHeader" : "Homematic CCU",
-    "CCUFullNameText" : "Homematic CCU",
-    "CCUShortName" : "CCU",
+    "CCUFullNameHeader" : "RaspberryMatic",
+    "CCUFullNameText" : "RaspberryMatic",
+    "CCUShortName" : "RaspMatic",
     "BidCosRF" : "BidCos-RF",
     "BidCosWired" : "BidCos-Wired",
     "VirtualDevices" : "VirtualDevices",
@@ -478,6 +478,7 @@
     "dialogSettingsCMHintSoftwareUpdate3" : "F%FChren Sie vor dem Update eine Datensicherung durch.",
     "dialogSettingsCMHintSoftwareUpdate4a" : "Der Ladezustand der Batterien betr%E4gt nur noch ",
     "dialogSettingsCMHintSoftwareUpdate4b" : "Um einem Datenverlust durch Ausfall der Stromversorgung vorzubeugen, empfehlen wir, die Batterien vor dem Update zu erneuern.",
+    "dialogSettingsCMHintSoftwareUpdateRaspMatic" : "RaspberryMatic ist ein gemeinschaftliches Projekt. Wenn Sie die Weiterentwicklung gerne in Form einer Geldspende unterst%FCtzen wollen k%F6nen Sie das gerne mittels einer <a href='https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL' target='_blank'>PayPal Spende</a> tun.<br><br><a href='https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL' target='_blank'><img alt='PayPal' border=0 src='https://www.paypalobjects.com/en_US/i/btn/btn_donateCC_LG.gif'></a>",
     "dialogSettingsCMHintRestart" : "Zentrale neu starten. Ver%E4nderte Einstellungen werden vorher gespeichert.",
     "dialogSettingsCMHintShutdown" : "Zentrale herunterfahren. Ver%E4nderte Einstellungen werden vorher gespeichert.",
     "dialogSettingsCMHintRestartSafeMode" : "HomeMatic Zentrale einmalig im abgesicherten Modus starten. Im abgesicherten Modus wird die installierte Zusatzsoftware nicht gestartet. Ver%E4nderte Einstellungen werden vorher gespeichert.",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.js
@@ -1,8 +1,8 @@
 jQuery.extend(true,HMIdentifier , {
   "en" : {
-    "CCUFullNameHeader" : "Homematic CCU",
-    "CCUFullNameText" : "Homematic CCU",
-    "CCUShortName" : "CCU",
+    "CCUFullNameHeader" : "RaspberryMatic",
+    "CCUFullNameText" : "RaspberryMatic",
+    "CCUShortName" : "RaspMatic",
     "BidCosRF" : "BidCos-RF",
     "BidCosWired" : "BidCos-Wired",
     "HmIPRF" : "HmIP-RF",
@@ -477,6 +477,7 @@
     "dialogSettingsCMHintSoftwareUpdate3" : "Please perform a data backup before starting the update.",
     "dialogSettingsCMHintSoftwareUpdate4a" : "The charging status the batteries is only at ",
     "dialogSettingsCMHintSoftwareUpdate4b" : "To prevent from data loss due to power failure, it is recommended to replace the batteries before starting the update.",
+    "dialogSettingsCMHintSoftwareUpdateRaspMatic" : "RaspberryMatic is a community driven project. If you want to support it, please consider a <a href='https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL' target='_blank'>PayPal donation</a> to support further development.<br><br><a href='https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL' target='_blank'><img alt='PayPal' border=0 src='https://www.paypalobjects.com/en_US/i/btn/btn_donateCC_LG.gif'></a>",
     "dialogSettingsCMHintRestart" : "Restart CCU. The changed settings will be saved before.",
     "dialogSettingsCMHintShutdown" : "Shutdown CCU. The changed settings will be saved before.",
     "dialogSettingsCMHintRestartSafeMode" : "Start HomeMatic Central Control Unit in safe mode once. In safe mode, the installed additional software will not be started. The changed settings will be saved before.",
--- occu/WebUI/www/webui/js/lang/tr/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/tr/translate.lang.js
@@ -1,8 +1,8 @@
 jQuery.extend(true,HMIdentifier , {
   "tr" : {
-    "CCUFullNameHeader" : "Homematic CCU",
-    "CCUFullNameText" : "Homematic CCU",
-    "CCUShortName" : "CCU",
+    "CCUFullNameHeader" : "RaspberryMatic",
+    "CCUFullNameText" : "RaspberryMatic",
+    "CCUShortName" : "RaspMatic",
     "BidCosRF" : "BidCos-RF",
     "BidCosWired" : "BidCos-Wired",
     "HmIPRF" : "HmIP-RF",
@@ -479,6 +479,7 @@
     "dialogSettingsCMHintSoftwareUpdate3" : "G%FCncellemeden %F6nce verileri emniyete alin.",
     "dialogSettingsCMHintSoftwareUpdate4a" : "Bataryalarin sarj durumu artik sadece ",
     "dialogSettingsCMHintSoftwareUpdate4b" : "Elektrik kesintisi nedeniyle veri kaybini %F6nlemek icin bataryalari g%FCncellemeyi baslatmadan %F6nce degistirmenizi %F6neriyoruz.",
+    "dialogSettingsCMHintSoftwareUpdateRaspMatic" : "RaspberryMatic is a community driven project. If you want to support it, please consider a <a href='https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL' target='_blank'>PayPal donation</a> to support further development.<br><br><a href='https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL' target='_blank'><img alt='PayPal' border=0 src='https://www.paypalobjects.com/en_US/i/btn/btn_donateCC_LG.gif'></a>",
     "dialogSettingsCMHintRestart" : "Merkezi yeniden baslat. Degistirilmis ayarlar %F6nceden kaydedilir.",
     "dialogSettingsCMHintShutdown" : "Shutdown CCU. The changed settings will be saved before.",
     "dialogSettingsCMHintRestartSafeMode" : "HomeMatic merkezini bir kere g%FCvenli modda baslatin. G%FCvenli modda y%FCklenen ek yazilim baslatilmaz. Degistirilmis ayarlar %F6nceden kaydedilir.",
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -19349,7 +19349,7 @@
   {
     this.m_ccuProduct = getProduct();
     this.preURL = (this.m_ccuProduct < 3) ? "" : "ccu3-";
-    this.m_product ="HM-CCU" + this.m_ccuProduct;
+    this.m_product = "HM-RASPBERRYMATIC";
     this.m_URLServer = (isHTTPS) ? "https://"+this.preURL+"update.homematic.com:8443" : "http://"+this.preURL+"update.homematic.com";
 
     var serial = homematic("CCU.getSerial");
@@ -19359,7 +19359,7 @@
     var script = document.createElement("script");
     script.id = "homematic_com_script";
     script.type = "text/javascript";
-    script.src = this.m_URLServer + "/firmware/download?cmd=js_check_version&version="+WEBUI_VERSION+"&product="+this.m_product+"&serial=" + serial;
+    script.src = "https://gitcdn.xyz/repo/jens-maus/RaspberryMatic/master/release/LATEST-VERSION.js?_version_=" + WEBUI_VERSION;
     $("body").appendChild(script);
   },
 
@@ -19411,7 +19411,7 @@
       var script = document.createElement("script");
       script.id = "homematic_com_script_" + index;
       script.type = "text/javascript";
-      script.src =  this.m_URLServer + "/firmware/download?cmd=js_check_version&product=" + product + "&serial=0";
+      script.src =  this.m_URLServer + "/firmware/download?cmd=js_check_version&product=" + product + "&serial=0" + "&ts=" + Date.now();
       $("body").appendChild(script);
       homematic.com.callback = callback;
   },
@@ -19421,7 +19421,7 @@
       var script = document.createElement("script");
       script.id = "homematic_com_script_fw";
       script.type = "text/javascript";
-      script.src =  this.m_URLServer + "/firmware/api/firmware/search/DEVICE";
+      script.src =  this.m_URLServer + "/firmware/api/firmware/search/DEVICE?ts=" + Date.now();
       $("body").appendChild(script);
       homematic.com.callback = callback;
   },
