--- occu/WebUI/www/api/methods/ccu/setsshpassword.tcl.orig
+++ occu/WebUI/www/api/methods/ccu/setsshpassword.tcl
@@ -8,7 +8,7 @@
 # Rückgabewert: "noError" oder sed-Fehlermeldung
 ##
 
-  set digest [exec mkpasswd -m sha512 "$args(passwd)"]
+  set digest [exec echo "$args(passwd)" | mkpasswd -m sha512]
   set command "s%^root:\\(\[^:\]*:\\)%root:$digest:%"
   set rc [catch {exec sed -i "$command" /etc/config/shadow} msg]
 
