--- occu/WebUI/www/config/cp_maintenance.cgi.orig
+++ occu/WebUI/www/config/cp_maintenance.cgi
@@ -470,7 +470,7 @@
           puts "\${dialogSettingsCMHintRestart}"
         }
       }
-set comment {
+#set comment {
       table_row {class="CLASS20902 j_noForcedUpdate j_fwUpdateOnly"} {
         table_data {class="CLASS20903"} $styleMaxWidth {
           puts "\${dialogSettingsCMTDCCUShutdown}"
@@ -486,7 +486,7 @@
           puts "\${dialogSettingsCMHintShutdown}"
         }
       }
-}
+#}
       # Abgesicherter Modus
       table_row {class="CLASS20902 j_noForcedUpdate j_fwUpdateOnly"} {
         table_data {class="CLASS20903"} $styleMaxWidth {
