--- occu/WebUI/www/config/cp_security.cgi.orig
+++ occu/WebUI/www/config/cp_security.cgi
@@ -930,6 +930,15 @@
               }
             }
             table_row {
+              td {width="20"} {}
+              table_data {align="left"} {
+                puts "\${dialogSettingsSecurityLblSecKeyChanges}"
+              }
+              table_data {align="right"} {
+                cgi_text keychanges=[get_current_key_index] {size="3"} {id="key_changes"} {type="text"} {disabled=""}
+              }
+            }
+            table_row {
               table_data {align="right"} {class="CLASS20812"} {colspan="3"} {
                 division {class="popupControls CLASS20811"} {
                   division {class="CLASS20813"} {onClick="OnChangeKey()"} {
--- occu/WebUI/www/webui/js/lang/de/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.js
@@ -488,6 +488,7 @@
     "dialogSettingsSecurityLblEnterSecKey" : "System-Sicherheitsschl%FCssel Eingabe",
     "dialogSettingsSecurityLblSecKey" : "Sicherheitsschl%FCssel",
     "dialogSettingsSecurityLblSecKeyRepeat" : "Sicherheitsschl%FCssel<br/>Wiederholung",
+    "dialogSettingsSecurityLblSecKeyChanges" : "Anzahl vorgenommener<br/>Schl%FCssel%E4nderungen",
     "dialogSettingsSecurityBtnSaveKey" : "Schl%FCssel %FCbernehmen",
     "dialogSettingsSecurityBtnBackupCreate" : "Backup erstellen",
     "dialogSettingsSecurityBtnBackupUpload" : "Backup einspielen",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.js
@@ -488,6 +488,7 @@
     "dialogSettingsSecurityLblEnterSecKey" : "Enter security key",
     "dialogSettingsSecurityLblSecKey" : "Security key",
     "dialogSettingsSecurityLblSecKeyRepeat" : "Repeat<br/>security key",
+    "dialogSettingsSecurityLblSecKeyChanges" : "Number of key changes",
     "dialogSettingsSecurityBtnSaveKey" : "Apply security key",
     "dialogSettingsSecurityBtnBackupCreate" : "Create backup",
     "dialogSettingsSecurityBtnBackupUpload" : "Import backup",
