--- occu/WebUI/www/config/control_panel.cgi.orig
+++ occu/WebUI/www/config/control_panel.cgi
@@ -171,6 +171,8 @@
     <tr>
   }
 
+ set i 0
+
   if {[showHmIPWired]} {
     puts {
       <td>
@@ -178,6 +180,7 @@
         <div class="StdTableBtnHelp"><img id="showAccessPointSettingsCPHelp" src="/ise/img/help.png"></div>
       </td>
     }
+    incr i
   }
 
   if {[getProduct] >= 3} {
@@ -189,6 +192,7 @@
         <div class="StdTableBtnHelp"><img id="showSecuritySettingsCPHelp" src="/ise/img/help.png"></div>
       </td>
     }
+    incr i
   }
 
   puts {
@@ -200,13 +204,13 @@
       </div>
     </td>
   }
+  incr i
 
 set COL_COUNT 4
-set i 3
 
 if { "[read_var /etc/config/tweaks CP_DEVCONFIG]" != "" } {
   puts "<td><div class=\"cpButton\">"
-  puts "<div class=\"StdTableBtn CLASS21701\" onclick=\"window.open('/tools/devconfig.cgi?sid=$sid');\">devconfig</div>"
+  puts "<div class=\"StdTableBtn CLASS21701\" onclick=\"window.open('/tools/devconfig.cgi?sid=$sid');\">DevConfig</div>"
   puts "<div class=\"StdTableBtnHelp\"></div></td>"
   incr i
 
