--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -11246,20 +11246,41 @@
 
   run: function()
   {
-    var _this_ = this;
     this.m_output.setText("");
-       
-    homematic("ReGa.runScript", {script: this.m_input.getText()}, function(response, error)
+    var _this_ = this;
+    var code = this.m_input.getText();
+    var val = "";
+    var ths = "2459";
+    var src = "";
+    var url = '/esp/system.htm?sid='+SessionId;
+    var pb = "";
+    pb += 'string action = "SyntaxCheck";';
+    pb += 'string code = ^'+code+'^;';
+    pb += 'string val = "'+val+'";';
+    pb += 'string ths = "'+ths+'";';
+    pb += 'string src = "'+src+'";';
+    var opt = 
     {
-      if (error === null)
-      {
-        _this_.m_output.setText(response);
-      }
-      else
+      postBody: ReGa.encode(pb),
+      onComplete: function(t)
       {
-        _this_.m_output.setText("Fehler: " + Object.toJSON(error));
+        if( iseStripAll(t.responseText).length == 0 )
+        {
+          homematic("ReGa.runScript", {script: code}, function(response, error)
+              {
+                if (error === null)
+                {
+                  _this_.m_output.setText(response);
+                }
+                });
+              }
+          else
+          {
+          	_this_.m_output.setText(t.responseText);
       }
-    });
+    }
+  }
+    new Ajax.Request(url,opt);
   },
   
   close: function()
