--- occu/WebUI/www/rega/esp/programs.fn.orig
+++ occu/WebUI/www/rega/esp/programs.fn
@@ -291,7 +291,7 @@
                       sValueUnit = "";
                     }
                     
-                    if((iCT1 != 5) &&  (iCT1 != 6) && (iCT1 != 8) && (iCT1 != 10) & (iCT1 != 11) & (iCT1 != 9) )
+                    if( (iCT1 != 1) && (iCT1 != 5) && (iCT1 != 6) && (iCT1 != 8) && (iCT1 != 9) && (iCT1 != 10) && (iCT1 != 11) && (iCT1 != 12) )
                     {
                       oSC.ConditionType(6);
                       iCT1 = 6;
@@ -308,6 +308,11 @@
                     if( iCT1 == 9 ) { sCT = "${ruleConditionLblGreaterOrEqualThan} "#iRV1.ToString(2)#sValueUnit; }
                     !if( iCT1 == 11 ) { sCT = "kleiner oder gleich "#iRV1.ToString(2)#sValueUnit; }
                     if( iCT1 == 11 ) { sCT = "${ruleConditionLblLessOrEqualThan} "#iRV1.ToString(2)#sValueUnit; }
+                    !if( iCT1 == 5 ) { sCT = "ist gleich "#iRV1.ToString(2)#sValueUnit; }
+                    if( iCT1 == 5 ) { sCT = "${ruleConditionLblEqualThan} "#iRV1.ToString(2)#sValueUnit; }
+                    !if( iCT1 == 12 ) { sCT = "ist ungleich "#iRV1.ToString(2)#sValueUnit; }
+                    if( iCT1 == 12 ) { sCT = "${ruleConditionLblNotEqualThan} "#iRV1.ToString(2)#sValueUnit; }
+
                     sRet = sRet#' '#sCT;
                   }
                   boolean bAction = ( oLV.ValueSubType() == istAction );
--- occu/WebUI/www/rega/esp/sico.inc.orig
+++ occu/WebUI/www/rega/esp/sico.inc
@@ -326,7 +326,7 @@
           sValueUnit = "";
         }
 
-        if( (iCT1 != 5) && (iCT1 != 6) && (iCT1 != 8) && (iCT1 != 10) & (iCT1 != 11) & (iCT1 != 9) )
+        if( (iCT1 != 5) && (iCT1 != 6) && (iCT1 != 8) && (iCT1 != 9) && (iCT1 != 10) && (iCT1 != 11) && (iCT1 != 12) )
         {
           oSC.ConditionType(6);
           iCT1 = 6;
@@ -343,6 +343,10 @@
         if( iCT1 == 9 ) { sCT = "${ruleConditionLblGreaterOrEqualThan} "#iRV1.ToString(2)#sValueUnit; }
         !if( iCT1 == 11 ) { sCT = "kleiner oder gleich "#iRV1.ToString(2)#sValueUnit; }
         if( iCT1 == 11 ) { sCT = "${ruleConditionLblLessOrEqualThan} "#iRV1.ToString(2)#sValueUnit; }
+        !if( iCT1 == 5 ) { sCT = "gleich "#iRV1.ToString(2)#sValueUnit; }
+        if( iCT1 == 5) { sCT = "${ruleConditionLblEqualThan} "#iRV1.ToString(2)#sValueUnit; }
+        !if( iCT1 == 12 ) { sCT = "ungleich "#iRV1.ToString(2)#sValueUnit; }
+        if( iCT1 == 12 ) { sCT = "${ruleConditionLblNotEqualThan} "#iRV1.ToString(2)#sValueUnit; }
 
         Write( ' <b class="CLASS02201" onclick="CreatePopup(ID_SET_VALUE_RANGE,'#oSC.ID()#');">'#sCT#'</b>' );
       }
--- occu/WebUI/www/rega/pages/msg/setValueRange.htm.orig
+++ occu/WebUI/www/rega/pages/msg/setValueRange.htm
@@ -61,7 +61,7 @@
     <td class="CLASS00802">
     <!-- for testing <div class="CLASS00803"><b>${lblValueRange}: <% Write(valMin # " - " # valMax # " - " # oLV.ValueType() # " - Unit: " # sValueUnit); %></b></div> -->
     <!-- <div class="CLASS00803"><b>${lblValueRange}: <% Write(valMin # " - " # valMax # " " # sValueUnit); %></b></div> -->
-    <div class="CLASS00803"><b>${lblValueRange}:</b></div>
+    <div class="CLASS00803"><b>${lblValueRange} / ${lblValue}:</b></div>
     <table>
     <tr><td style="height: 15px;"></td></tr>
     <tr>
@@ -85,7 +85,7 @@
       <td>
         <% string sChecked = ""; if(iCT==5) { sChecked="checked"; } %>
         <input id="Radio0" type="radio" <% Write(sChecked); %> onclick="SetCT(this);" name="valueRange"/>
-        <span class="CLASS00804">${ruleConditionLblIsEqual}</span>
+        <span class="CLASS00804">${ruleConditionLblEqualThan}</span>
       </td>
       <td>
         <input id="Text0" type="text" value="<% Write(iRV1); %>" class="CLASS00806"/>
@@ -93,7 +93,17 @@
       </td>
     </tr>
 
-
+    <tr>
+      <td>
+        <% string sChecked = ""; if(iCT==12) { sChecked="checked"; } %>
+        <input id="Radio6" type="radio" <% Write(sChecked); %> onclick="SetCT(this);" name="valueRange"/>
+        <span class="CLASS00804">${ruleConditionLblNotEqualThan}</span>
+      </td>
+      <td>
+        <input id="Text7" type="text" value="<% Write(iRV1); %>" class="CLASS00806"/>
+        <span class="CLASS00804"><% Write(sValueUnit); %></span>
+      </td>
+    </tr>
 
     <tr>
       <td>
@@ -177,6 +187,7 @@
     if( elem.id=="Radio3" ) ct = 10;
     if( elem.id=="Radio4" ) ct = 9;
     if( elem.id=="Radio5" ) ct = 11;
+    if( elem.id=="Radio6" ) ct = 12;
   }
   SaveValueRange = function()
   {
@@ -264,6 +275,18 @@
         } else {
           if ( ( (rv1/1) < (min/1) ) || ( (rv1/1) > (max/1) ) ) {
             go = 6;
+          }
+        }
+        break;
+      case 12:
+        rv1 = $("Text7").value;
+        if( vu == "100%" ) rv1 = rv1/100;
+        go = 0;
+        if (isNaN(rv1)) {
+          go = 5;
+        } else {
+          if ( ( (rv1/1) < (min/1) ) || ( (rv1/1) > (max/1) ) ) {
+            go = 6;
           }
         }
         break;
--- occu/WebUI/www/webui/js/lang/de/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.js
@@ -780,13 +780,15 @@
     "ruleConditionSelectDeviceList" : "Ger%E4teauswahl",
     "ruleConditionLblWhen" : "bei",
     "ruleConditionSelectSystemState" : "Systemzustand",
-    "ruleConditionLblWithinValueRange" : "im Wertebereich",
+    "ruleConditionLblWithinValueRange" : "im Wertebereich / mit Wert",
     "ruleConditionLblRangeFrom" : "von",
     "ruleConditionLblRangeLessThan" : "bis kleiner",
     "ruleConditionLblGreaterThan" : "gr%F6%DFer als",
     "ruleConditionLblLessThan" : "kleiner als",
     "ruleConditionLblGreaterOrEqualThan" : "gr%F6%DFer oder gleich",
     "ruleConditionLblLessOrEqualThan" : "kleiner oder gleich",
+    "ruleConditionLblEqualThan" : "gleich",
+    "ruleConditionLblNotEqualThan" : "ungleich",
     "ruleConditionSelectTimeControl" : "Zeitsteuerung",
     "ruleConditionLblTimeOnce" : "Einmalig",
     "ruleConditionLblTimePeriodically" : "Periodisch",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.js
@@ -218,7 +218,7 @@
     "lblChannelMappingYes" : "with",
     "lblChannelMappingNo" : "without",
     "lblValueDescription" : "Value description",
-    "lblValueRange" : "Value range",
+    "lblValueRange" : "Value range / with value",
     "lblSystemVariable" : "System variable",
     "lblLeft" : "Left",
     "lblRight" : "Right",
@@ -813,13 +813,15 @@
     "ruleConditionLblWhen" : "when",
 
     "ruleConditionSelectSystemState" : "System state",
-    "ruleConditionLblWithinValueRange" : "within value range",
+    "ruleConditionLblWithinValueRange" : "within value range / with value",
     "ruleConditionLblRangeFrom" : "from",
     "ruleConditionLblRangeLessThan" : "and less than",
     "ruleConditionLblGreaterThan" : "more than",
     "ruleConditionLblLessThan" : "less than",
     "ruleConditionLblGreaterOrEqualThan" : "more or equal",
     "ruleConditionLblLessOrEqualThan" : "less or equal",
+    "ruleConditionLblEqualThan" : "equal",
+    "ruleConditionLblNotEqualThan" : "not equal",
 
     "ruleConditionSelectTimeControl" : "Time control",
     "ruleConditionLblTimeOnce" : "Non-recurring",
