--- occu/WebUI/www/upnp/basic_dev.cgi.orig
+++ occu/WebUI/www/upnp/basic_dev.cgi
@@ -14,7 +14,16 @@
 }
 
 proc get_serial_number {} {
-    return [exec cat /sys/module/plat_eq3ccu2/parameters/board_serial]
+    if {[file exist /var/board_sgtin]} {
+      set serial [exec cat /var/board_sgtin]
+    } elseif {[file exist /var/board_serial]} {
+      set serial [exec cat /var/board_serial]
+    } elseif {[file exist /sys/module/plat_eq3ccu2/parameters/board_serial]} {
+      set serial [exec cat /sys/module/plat_eq3ccu2/parameters/board_serial]
+    } else {
+      set serial ""
+    }
+    return $serial
 }
 
 proc get_hostname {} {
@@ -22,10 +31,14 @@
 }
 
 set hostname "[get_hostname]"
-if { "$hostname" == "homematic-ccu2" } {
+if { "$hostname" == "homematic-ccu3" } {
+    set RESOURCE(TITLE) "HomeMatic Central CCU3"
+} elseif { "$hostname" == "homematic-raspi" } {
+    set RESOURCE(TITLE) "HomeMatic Central RaspberryMatic"
+} elseif { "$hostname" == "homematic-ccu2" } {
     set RESOURCE(TITLE) "HomeMatic Central CCU2"
 } else {
-    set RESOURCE(TITLE) "$hostname"
+    set RESOURCE(TITLE) "HomeMatic Central - $hostname"
 }
 
 set RESOURCE(MANUFACTURER) "EQ3"
