--- occu/WebUI/www/rega/esp/system.fn.orig
+++ occu/WebUI/www/rega/esp/system.fn
@@ -457,6 +457,9 @@
       ! Systemvariable vom Typ Zeichenkette
       if (oSV.ValueSubType() == istChar8859) {
         svValue = oSV.Value();
+        svValue = svValue.Replace("\r\n", "<br/>");
+        svValue = svValue.Replace("\r", "<br/>");
+        svValue = svValue.Replace("\n", "<br/>");
       }
       
       WriteLine("updateSysVar('" # svId # "', '" # svValue # "');");
@@ -1410,6 +1413,9 @@
   if ( bString )
   {
     sSysVarVal = cObjTmp.Value();
+    sSysVarVal = sSysVarVal.Replace("\r\n", "<br/>");
+    sSysVarVal = sSysVarVal.Replace("\r", "<br/>");
+    sSysVarVal = sSysVarVal.Replace("\n", "<br/>");
   }
 }
 
