--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -20759,7 +20759,7 @@
       case ID_EDIT_SCRIPT:
         this.setTitle( translateKey("dialogEditScript") /*"Skript bearbeiten"*/ );
         this.addToPostBody( 'string sdid = "'+this.type+'";' );
-        this.setWidth(800);
+        this.setWidth('auto');
         this.setFile( "/pages/msg/editScript.htm" );
         break;
       case ID_CONTROL_TEST:
