--- occu/WebUI/www/config/cp_maintenance.cgi.orig
+++ occu/WebUI/www/config/cp_maintenance.cgi
@@ -250,16 +250,6 @@
     <p class="CLASS20914">
     ${dialogSettingsCMDialogPerformSoftwareUpdateP1}
     </p>
-    <p class="CLASS20914">
-      ${dialogSettingsCMDialogPerformSoftwareUpdateP2}
-      <ol class="CLASS20915">
-        <li>${dialogSettingsCMDialogPerformSoftwareUpdateLi1}</li>
-        <li>
-        ${dialogSettingsCMDialogPerformSoftwareUpdateLi2}
-        </li>
-        <li>${dialogSettingsCMDialogPerformSoftwareUpdateLi3}</li>
-      </ol>
-    </p>
   } "_empty_"  
   puts ""
   cgi_javascript {
--- occu/WebUI/www/config/ic_common.tcl.orig
+++ occu/WebUI/www/config/ic_common.tcl
@@ -473,7 +473,6 @@
 
   html xmlns=\"http://www.w3.org/1999/xhtml\" xml:lang=\"de\" lang=\"de\" {
     head {
-      put_meta_nocache
       puts "<link rel=\"stylesheet\" type=\"text/css\" href=\"style.css\" />"
       title "$HTMLTITLE - Geräteparameter"
 
@@ -567,7 +566,6 @@
 
   html {
     head {
-      put_meta_nocache
     }
     body {
       if { ![catch { xmlrpc $url activateLinkParamset [list string $address] [list string $ps_id] [list bool $long_push] } ] } then {
@@ -692,7 +690,6 @@
 
     html {
       head {
-        put_meta_nocache
       }
       body {
         if {$ret == "1"} then {
@@ -710,7 +707,6 @@
 
     html {
       head {
-        put_meta_nocache
       }
       body {
         if {$ret == "1"} then {
@@ -731,7 +727,6 @@
 
   html {
     head {
-      put_meta_nocache
       puts "<link rel=\"stylesheet\" type=\"text/css\" href=\"style.css\">"
       title "$HTMLTITLE - Profil nicht gefunden"
     }
@@ -749,7 +744,6 @@
 
   html {
     head {
-      put_meta_nocache
       puts "<link rel=\"stylesheet\" type=\"text/css\" href=\"style.css\">"
       title "$HTMLTITLE - Profil nicht gefunden"
     }
@@ -1722,12 +1716,6 @@
   return [get_Pulse2 arr $id $ps($pname) $dev_address pulse $extraparam]
 }
 
-proc put_meta_nocache {} {
-  puts "<meta http-equiv=\"cache-control\" content=\"no-cache\" />"
-  puts "<meta http-equiv=\"pragma\"        content=\"no-cache\" />"
-  puts "<meta http-equiv=\"expires\"       content=\"0\" />"
-}
-
 proc get_InputElem {name id ps_arr pname {extraparam ""}} {
   
   upvar $ps_arr ps
--- occu/WebUI/www/config/ic_ifacecmd.cgi.orig
+++ occu/WebUI/www/config/ic_ifacecmd.cgi
@@ -718,7 +718,6 @@
 
     html {
       head {
-        put_meta_nocache
         puts "<title>response of request with command: $cmd</title>"
       }
       body {
--- occu/WebUI/www/rega/index.htm.orig
+++ occu/WebUI/www/rega/index.htm
@@ -5,9 +5,6 @@
     <title>HomeMatic WebUI</title>
     <meta http-equiv="Content-Language" content="de">
     <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
-    <meta http-equiv="cache-control" content="no-cache">
-    <meta http-equiv="pragma"        content="no-cache">
-    <meta http-equiv="expires"       content="0">
 
     <%
       string devtype = "pc";
--- occu/WebUI/www/rega/pages/index.htm.orig
+++ occu/WebUI/www/rega/pages/index.htm
@@ -6,9 +6,6 @@
     <% Call("/esp/system.fn::CheckUserFavorite()"); %>
     <meta http-equiv="Content-Language" content="de">
     <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
-    <meta http-equiv="cache-control" content="no-cache">
-    <meta http-equiv="pragma"        content="no-cache">
-    <meta http-equiv="expires"       content="0">
 
     <link rel="shortcut icon" href="/ise/img/favicon.ico" type="image/vnd.microsoft.icon"/>
     <link rel="icon"          href="/ise/img/favicon.ico" type="image/vnd.microsoft.icon" />
--- occu/WebUI/www/tools/designer.html.orig
+++ occu/WebUI/www/tools/designer.html
@@ -2,9 +2,6 @@
 <html>
 <head>
 <title>HomeMatic-Highlighting-Editor</title>
-<meta http-equiv="cache-control" content="no-cache" />
-<meta http-equiv="pragma"        content="no-cache" />
-<meta http-equiv="expires"       content="0" />
 <script type="text/javascript" src="/webui/js/extern/prototype.js"></script>
 <script type="text/javascript" src="/webui/js/extern/wz_jsgraphics.js"></script>
 <script type="text/javascript" src="/tools/js/ic_gd.js"></script>
--- occu/WebUI/www/tools/js/common/httploader.js.orig
+++ occu/WebUI/www/tools/js/common/httploader.js
@@ -12,21 +12,10 @@
   /*####################*/
   
   /**
-   * Hängt die Version der HomeMatic WebUI an eine URL an,
-   * um Probleme mit dem Browsercache zu umgehen.
-   **/
-  var addVersion = function(url)
-  {
-    if (0 <= url.indexOf("?")) { return url + "&_version_=" + WEBUI_VERSION; }
-    else                       { return url + "?_version_=" + WEBUI_VERSION; }
-  };
-  
-  /**
    * Lädt synchron Daten und gibt das XMLHttpRequest-Objekt zurück.
    **/
   var load = function(method, url, data)
   {
-    url = addVersion(url);
     var xhr = XMLHttpRequest_create();
     
     if (null !== xhr)
--- occu/WebUI/www/webui/js/lang/loadTextResource.js.orig
+++ occu/WebUI/www/webui/js/lang/loadTextResource.js
@@ -123,9 +123,7 @@
       url: url + res,
       type:  "GET",
       async: false,
-      contentType: "application/x-www-form-urlencoded;charset=ISO-8859-1",
-      dataType: "script",
-      cache: false
+      contentType: "application/x-www-form-urlencoded;charset=ISO-8859-1"
     });
 
     request.done(function(result) {
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -5850,21 +5850,10 @@
   /*####################*/
   
   /**
-   * Hängt die Version der HomeMatic WebUI an eine URL an,
-   * um Probleme mit dem Browsercache zu umgehen.
-   **/
-  var addVersion = function(url)
-  {
-    if (0 <= url.indexOf("?")) { return url + "&_version_=" + WEBUI_VERSION; }
-    else                       { return url + "?_version_=" + WEBUI_VERSION; }
-  };
-  
-  /**
    * Lädt synchron Daten und gibt das XMLHttpRequest-Objekt zurück.
    **/
   var load = function(method, url, data)
   {
-    url = addVersion(url);
     var xhr = XMLHttpRequest_create();
     
     if (null !== xhr)
@@ -18724,7 +18713,6 @@
     url: "/ise/htm/header.htm",
     dataType: "html",
     async: false,
-    cache: false,
     context: document.body
     }).done(function(response) {
       jQuery("#header").html(response);
@@ -23277,7 +23265,7 @@
       var script = document.createElement("script");
       script.id = "homematic_com_script_" + index;
       script.type = "text/javascript";
-      script.src =  this.m_URLServer + "/firmware/download?cmd=js_check_version&product=" + product + "&serial=0";
+      script.src =  this.m_URLServer + "/firmware/download?cmd=js_check_version&product=" + product + "&serial=0" + "&ts=" + Date.now();
       $("body").appendChild(script);
       homematic.com.callback = callback;
   },
@@ -23288,7 +23276,7 @@
       script.id = "homematic_com_script_fw";
       script.type = "text/javascript";
       // script.src =  this.m_URLServer + "/firmware/api/firmware/search/DEVICE";
-      script.src =  this.m_URLServer + "/firmware/api/firmware/search/DEVICE?product=HM-CCU"+getProduct()+"&version="+WEBUI_VERSION;
+      script.src =  this.m_URLServer + "/firmware/api/firmware/search/DEVICE?product=HM-CCU"+getProduct()+"&version="+WEBUI_VERSION+"&ts=" + Date.now();
       $("body").appendChild(script);
       homematic.com.callback = callback;
   },
