--- occu/WebUI/www/config/cp_maintenance.cgi.orig
+++ occu/WebUI/www/config/cp_maintenance.cgi
@@ -250,16 +250,6 @@
     <p class="CLASS20914">
     ${dialogSettingsCMDialogPerformSoftwareUpdateP1}
     </p>
-    <p class="CLASS20914">
-      ${dialogSettingsCMDialogPerformSoftwareUpdateP2}
-      <ol class="CLASS20915">
-        <li>${dialogSettingsCMDialogPerformSoftwareUpdateLi1}</li>
-        <li>
-        ${dialogSettingsCMDialogPerformSoftwareUpdateLi2}
-        </li>
-        <li>${dialogSettingsCMDialogPerformSoftwareUpdateLi3}</li>
-      </ol>
-    </p>
   } "_empty_"  
   puts ""
   cgi_javascript {
--- occu/WebUI/www/tools/js/common/httploader.js.orig
+++ occu/WebUI/www/tools/js/common/httploader.js
@@ -12,21 +12,10 @@
   /*####################*/
   
   /**
-   * Hängt die Version der HomeMatic WebUI an eine URL an,
-   * um Probleme mit dem Browsercache zu umgehen.
-   **/
-  var addVersion = function(url)
-  {
-    if (0 <= url.indexOf("?")) { return url + "&_version_=" + WEBUI_VERSION; }
-    else                       { return url + "?_version_=" + WEBUI_VERSION; }
-  };
-  
-  /**
    * Lädt synchron Daten und gibt das XMLHttpRequest-Objekt zurück.
    **/
   var load = function(method, url, data)
   {
-    url = addVersion(url);
     var xhr = XMLHttpRequest_create();
     
     if (null !== xhr)
--- occu/WebUI/www/webui/js/lang/loadTextResource.js.orig
+++ occu/WebUI/www/webui/js/lang/loadTextResource.js
@@ -123,9 +123,7 @@
       url: url + res,
       type:  "GET",
       async: false,
-      contentType: "application/x-www-form-urlencoded;charset=ISO-8859-1",
-      dataType: "script",
-      cache: false
+      contentType: "application/x-www-form-urlencoded;charset=ISO-8859-1"
     });
 
     request.done(function(result) {
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -5782,21 +5782,10 @@
   /*####################*/
   
   /**
-   * Hängt die Version der HomeMatic WebUI an eine URL an,
-   * um Probleme mit dem Browsercache zu umgehen.
-   **/
-  var addVersion = function(url)
-  {
-    if (0 <= url.indexOf("?")) { return url + "&_version_=" + WEBUI_VERSION; }
-    else                       { return url + "?_version_=" + WEBUI_VERSION; }
-  };
-  
-  /**
    * Lädt synchron Daten und gibt das XMLHttpRequest-Objekt zurück.
    **/
   var load = function(method, url, data)
   {
-    url = addVersion(url);
     var xhr = XMLHttpRequest_create();
     
     if (null !== xhr)
@@ -17643,7 +17632,6 @@
     url: "/ise/htm/header.htm",
     dataType: "html",
     async: false,
-    cache: false,
     context: document.body
     }).done(function(response) {
       jQuery("#header").html(response);
