--- datapointconfigurator.fn.orig	2018-02-26 22:56:12.000000000 +0100
+++ datapointconfigurator.fn	2018-02-26 22:55:47.000000000 +0100
@@ -748,7 +748,7 @@
       WriteLine( '</tr></table><table cellspacing="0" class="ctrlTbl CLASS02510" border="0" ><tr>' );
       Call("/esp/datapointconfigurator.fn::CreateUnknown()");
     }
-    if ( (isKnownControl) && (bSV)  && (oEName.Find("HmIP-RF.") == -1) )
+    if ( (isKnownControl) && (bSV)  && (oEName.Find("HmIP-RF.") == -1) && (oEName.Find("CUxD.CUX") == -1) )
     {
       WriteLine( '</tr></table>');
       WriteLine( "<script>conInfo('SysVar found.');</script>" );
