--- occu/WebUI/www/rega/esp/side.inc.orig
+++ occu/WebUI/www/rega/esp/side.inc
@@ -448,17 +448,17 @@
 
                 if (iV == "0") {
 
-                  if (oCH.HSSID() == "RGBW_COLOR") {
+                  if (oDP.HSSID().Find("USER_COLOR") != -1) {
                     ! deprecated iV = "0,200,0.5,0";
                     iV = "{'ACT_HSV_COLOR_VALUE_STORE':0,'ACT_BRIGHTNESS_STORE':200,'RAMP_TIME_STORE':0.5,'ON_TIME_STORE':0}";
                   }
 
-                  if (oCH.HSSID() == "RGBW_AUTOMATIC") {
+                  if (oDP.HSSID().Find("USER_PROGRAM") != -1) {
                     ! deprecated iV = "0,200,0.5,0,0,199";
                     iV = "{'ACT_COLOR_PROGRAM_STORE':0,'ACT_BRIGHTNESS_STORE':200,'RAMP_TIME_STORE':0.5,'ON_TIME_STORE':0, 'ACT_MIN_BORDER_STORE':0, 'ACT_MAX_BORDER_STORE':199}";
                   }
                 }
-                Write( '<input id="prgStringOptionValue'#oSD.ID()#'" onfocus="SetRGBWController('#oSD.ID()#',\''#oCH.HSSID()#'\');" type="text" class="SelectBox" size="10" value="'#iV#'" onchange="iseSingleDestination.SetValue('#oSD.ID()#',this.value,\'STRING\');" /> ' );
+                Write( '<input id="prgStringOptionValue'#oSD.ID()#'" onfocus="SetRGBWController('#oSD.ID()#',\''#oDP.HSSID()#'\');" type="text" class="SelectBox" size="10" value="'#iV#'" onchange="iseSingleDestination.SetValue('#oSD.ID()#',this.value,\'STRING\');" /> ' );
               }
 
               if ((jalousieActor != - 1) && (oDP.HSSID().Find("LEVEL_COMBINED") != -1)) {
@@ -518,7 +518,7 @@
                 Write( '<span id="setStatusDisplayEPaper'#oSD.ID()#'" class="CLASS02401" onclick="SetStatusDisplay('#oSD.ID()#',\'DIS-EP\');"><img src="/ise/img/notepad.png" /></span>' );
               }
               if (rgbw_controller != -1) {
-                Write( '<span id="setRGBWController'#oSD.ID()#'" class="CLASS02401" onclick="SetRGBWController('#oSD.ID()#',\''#oCH.HSSID()#'\');"><img src="/ise/img/notepad.png" /></span>' );
+                Write( '<span id="setRGBWController'#oSD.ID()#'" class="CLASS02401" onclick="SetRGBWController('#oSD.ID()#',\''#oDP.HSSID()#'\');"><img src="/ise/img/notepad.png" /></span>' );
               }
 
               if ((jalousieActor != - 1) && (oDP.HSSID().Find("LEVEL_COMBINED") != -1)) {
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -13155,7 +13155,7 @@
        jQuery("#RGBWControllerBrightness").show();
        break;
        */
-      case "RGBW_COLOR" :
+      case "USER_COLOR" :
         /*
          Array curValues
          0 Color val
@@ -13203,7 +13203,7 @@
         jQuery("#RGBWControllerColor").show();
         this.__activateColorPicker();
         break;
-      case "RGBW_AUTOMATIC":
+      case "USER_PROGRAM":
         /*
          Array curValues
          0 Program
@@ -13375,7 +13375,7 @@
       case "BRIGHTNESS" :
         this.configString = this.__getBrightness();
         break;
-      case "RGBW_COLOR" :
+      case "USER_COLOR" :
         var freeRampTimeHour = jQuery("#colorRGBEnterFreeRampTimeHour").val(),
           freeRampTimeMin = jQuery("#colorRGBEnterFreeRampTimeMin").val(),
           freeRampTimeSec = jQuery("#colorRGBEnterFreeRampTimeSec").val();
@@ -13398,7 +13398,7 @@
 
         this.configString = this.__getColor();
         break;
-      case "RGBW_AUTOMATIC":
+      case "USER_PROGRAM":
 
         var freeRampTimeHour = jQuery("#prgRGBEnterFreeRampTimeHour").val(),
           freeRampTimeMin = jQuery("#prgRGBEnterFreeRampTimeMin").val(),
