--- occu/WebUI/www/config/stringtable_de.txt.orig
+++ occu/WebUI/www/config/stringtable_de.txt
@@ -1081,6 +1081,7 @@
 SHUTTER_CONTACT|STATE=FALSE	${stringTableShutterContactStateFalse}
 SHUTTER_CONTACT|STATE=OPEN	${stringTableShutterContactStateTrue}
 SHUTTER_CONTACT|STATE=TRUE	${stringTableShutterContactStateTrue}
+SHUTTER_CONTACT|EVENT_DELAYTIME	${stringTableShutterContactEventDelayTime}
 SHUTTER_TRANSMITTER|LEVEL	${stringTableBlindLevel}
 SHUTTER_TRANSMITTER|LEVEL_2	${stringTableJalousieSlatsLevel}
 
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -4336,6 +4336,7 @@
 elvST['SHUTTER_CONTACT|STATE=FALSE'] = '${stringTableShutterContactStateFalse}';
 elvST['SHUTTER_CONTACT|STATE=OPEN'] = '${stringTableShutterContactStateTrue}';
 elvST['SHUTTER_CONTACT|STATE=TRUE'] = '${stringTableShutterContactStateTrue}';
+elvST['SHUTTER_CONTACT|EVENT_DELAYTIME'] = '${stringTableShutterContactEventDelayTime}';
 elvST['SHUTTER_TRANSMITTER|LEVEL'] = '${stringTableBlindLevel}';
 elvST['SHUTTER_TRANSMITTER|LEVEL_2'] = '${stringTableJalousieSlatsLevel}';
 elvST['SHUTTER_VIRTUAL_RECEIVER'] = '${stringTableShutterVirtualReceiverTitle}';
