--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -16400,6 +16400,7 @@
   m_gatewayTableKeyHeader: null,
   m_gatewayTableIPHeader: null,
   m_gatewayTableStateHeader: null,
+  m_gatewayTableDCStateHeader: null,
   m_gatewayTableActionHeader: null,
   m_gatewayTableBody: null,
   m_optionsButtonBar: null,
@@ -16512,7 +16513,14 @@
     //this.m_gatewayTableStateHeader.appendChild(document.createTextNode("Status"));
     this.m_gatewayTableStateHeader.appendChild(document.createTextNode(translateKey("thState")));
     this.m_gatewayTableHeadRow.appendChild(this.m_gatewayTableStateHeader);
-    
+
+    /* Spaltenüberschrift DutyCycle Status (Tabelle für BidCoS-RF Gateways */
+    this.m_gatewayTableDCStateHeader = document.createElement("th");
+    this.m_gatewayTableDCStateHeader.className = "bidcosrf_tableheader";
+    //this.m_gatewayTableStateHeader.appendChild(document.createTextNode("Status"));
+    this.m_gatewayTableDCStateHeader.appendChild(document.createTextNode("DutyCycle / Firmware"));
+    this.m_gatewayTableHeadRow.appendChild(this.m_gatewayTableDCStateHeader);
+
     /* Spaltenüberschrift Verbunden (Tabelle für BidCoS-RF Gateways */
     this.m_gatewayTableActionHeader = document.createElement("th");
     this.m_gatewayTableActionHeader.className = "bidcosrf_tableheader";
@@ -16832,6 +16840,15 @@
 				    	 lgw = rfGateways[j];
 				    	}
 				    }
+				    for (var loop = 0; loop < gatewayStatus.length; loop++) {
+				    	gatewaysn = gatewayStatus[loop].address;
+				    	if (lgwStatus.serial == gatewaysn) {
+				    		var textDC = "";
+				    		textDC += gatewayStatus[loop].dutyCycle + "% / ";
+				    		textDC += gatewayStatus[loop].fwVersion;
+				    		lgw.setDCState(textDC);
+				    	}
+				    }
 		        	if(lgw) 
 		        	{
 					  if(lgwStatus.connstat == "NO_ERROR") {
@@ -16855,6 +16872,7 @@
 				      textB = translateKey("lanGatewayLblNotActive");
 					}
 					lgw.setState(textB);
+					
 			  }//if lgwstatus
 			  else {
 			  	lgw.setState(translateKey("lanGatewayLblNotActive"));
@@ -16865,6 +16883,23 @@
 		  {
 			  var text = "";
 	   	      var status = rfGateways[i].determineState(gatewayStatus);
+	   	      var cfglan = rfGateways[i].getAddress(gatewayStatus);
+
+	   	      for (var loop = 0; loop < gatewayStatus.length; loop++) {
+	   	      	gatewaysn = gatewayStatus[loop].address;
+
+	   	      	if (cfglan == gatewaysn) {
+	   	      		textFW = gatewayStatus[loop].fwVersion;
+	   	      		if (textFW < "965") {
+	   	      			var textDC = gatewayStatus[loop].dutyCycle + "% / 0." + gatewayStatus[loop].fwVersion + " / Update 0.965";
+	   	      			rfGateways[i].setDCState(textDC);
+	   	      		} else {
+	   	      			textDC = gatewayStatus[loop].dutyCycle + "% / 0." + gatewayStatus[loop].fwVersion;
+	   	      			rfGateways[i].setDCState(textDC);
+	   	      		}
+	   	      	}
+	   	  	  }
+
 		      if (status)
 		      {
 		        text = (status.isConnected) ? translateKey("lanGatewayLblConnected") : translateKey("lanGatewayLblNotConnected");
@@ -17065,6 +17100,7 @@
   m_keyLabel: null,
   m_ipLabel: null,
   m_stateLabel: null,
+  m_stateDCLabel: null,
   m_actionCell: null,
   m_deleteButton: null,
   m_onDeleteHandler: null,
@@ -17117,6 +17153,11 @@
     this.m_stateLabel.className = "bidcosrf_tablecell";
     this.m_element.appendChild(this.m_stateLabel);
     
+    /* DutyCycle Status */
+    this.m_stateDCLabel = document.createElement("td");
+    this.m_stateDCLabel.className = "bidcosrf_tablecell";
+    this.m_element.appendChild(this.m_stateDCLabel);
+    
     /* Aktion */
     this.m_actionCell = document.createElement("td");
     this.m_actionCell.className = "bidcosrf_actioncell";
@@ -17256,6 +17297,12 @@
     return this;
   },
   
+  setDCState: function(state)
+  {
+    this.m_stateDCLabel.innerHTML = "";
+    this.m_stateDCLabel.appendChild(document.createTextNode(state));
+  },
+  
   getElement: function()
   {
     return this.m_element;
