--- occu/WebUI/www/tools/devconfig.cgi.orig
+++ occu/WebUI/www/tools/devconfig.cgi
@@ -198,13 +198,13 @@
                             puts ""
                             table_row { 
                                 table_data {
-                                    h3 [url "$dev_descr(TYPE) ($dev_descr(ADDRESS))" $env(SCRIPT_NAME)?cmd=config&device=$dev_descr(ADDRESS)&iface=$iface&$urlsid]
+                                    h3 [url "$dev_descr(ADDRESS)" $env(SCRIPT_NAME)?cmd=config&device=$dev_descr(ADDRESS)&iface=$iface&$urlsid]
                                 }
                                 table_data {
-                                    h3 [url "Delete Device" $env(SCRIPT_NAME)?cmd=delete&device=$dev_descr(PARENT)&iface=$iface&$urlsid]
+                                    h3 "$dev_descr(TYPE)"
                                 }
                                 table_data {
-                                    h3 [url "Delete And Reset Device" $env(SCRIPT_NAME)?cmd=delete&device=$dev_descr(PARENT)&reset=1&iface=$iface&$urlsid]
+                                    h4 [url "Device: $dev_descr(PARENT)" $env(SCRIPT_NAME)?cmd=list_devices&expand=$dev_descr(PARENT)&$urlsid]
                                 }
                             }
                         }
@@ -303,17 +303,55 @@
                             puts ""
                             table_row { 
                                 table_data colspan=2 {
-                                    h3 [url "$dev_descr(TYPE) ($dev_descr(ADDRESS))" $env(SCRIPT_NAME)?cmd=list_devices&expand=$dev_descr(ADDRESS)&$urlsid]
+                                    h3 [url "$dev_descr(ADDRESS)" $env(SCRIPT_NAME)?cmd=list_devices&expand=$dev_descr(ADDRESS)&$urlsid]
                                 }
                                 table_data {
-                                    h3 [url "Delete" $env(SCRIPT_NAME)?cmd=delete&device=$dev_descr(ADDRESS)&iface=$iface&$urlsid]
+                                    h3 "$dev_descr(TYPE)"
                                 }
                                 table_data {
-                                    h3 [url "Delete And Reset" $env(SCRIPT_NAME)?cmd=delete&device=$dev_descr(ADDRESS)&reset=1&iface=$iface&$urlsid]
+                                    h5 "|"
                                 }
                                 table_data {
-                                    if { [info exist dev_descr(AVAILABLE_FIRMWARE)] } {
-                                        h3 [url "Firmware Update ($dev_descr(FIRMWARE) to $dev_descr(AVAILABLE_FIRMWARE))" $env(SCRIPT_NAME)?cmd=firmware_update&device=$dev_descr(ADDRESS)&iface=$iface&$urlsid]
+                                    h5 [url "Delete" $env(SCRIPT_NAME)?cmd=delete&device=$dev_descr(ADDRESS)&iface=$iface&$urlsid]
+                                }
+                                table_data {
+                                    h5 "|"
+                                }
+                                table_data {
+                                    h5 [url "Delete+Reset" $env(SCRIPT_NAME)?cmd=delete&device=$dev_descr(ADDRESS)&reset=1&iface=$iface&$urlsid]
+                                }
+                                table_data {
+                                    h5 "|"
+                                }
+                                table_data {
+                                    h5 [url "Delete+Force" $env(SCRIPT_NAME)?cmd=delete&device=$dev_descr(ADDRESS)&force=1&iface=$iface&$urlsid]
+                                }
+                                table_data {
+                                    h5 "|"
+                                }
+                                table_data {
+                                    h5 [url "Delete+Defer" $env(SCRIPT_NAME)?cmd=delete&device=$dev_descr(ADDRESS)&defer=1&iface=$iface&$urlsid]
+                                }
+                                table_data {
+                                    h5 "|"
+                                }
+                                table_data {
+                                    h5 [url "Delete+Reset+Defer" $env(SCRIPT_NAME)?cmd=delete&device=$dev_descr(ADDRESS)&reset=1&defer=1&iface=$iface&$urlsid]
+                                }
+                                table_data {
+                                    h5 "|"
+                                }
+                                table_data {
+                                    h5 [url "Delete+Reset+Force" $env(SCRIPT_NAME)?cmd=delete&device=$dev_descr(ADDRESS)&reset=1&force=1&iface=$iface&$urlsid]
+                                }
+                                table_data {
+                                    h5 "|"
+                                }
+                                table_data {
+                                    if { [info exist dev_descr(AVAILABLE_FIRMWARE)] &&
+                                         $dev_descr(AVAILABLE_FIRMWARE) != "0.0.0" &&
+                                         $dev_descr(AVAILABLE_FIRMWARE) != $dev_descr(FIRMWARE) } {
+                                        h5 [url "FW-Upd ($dev_descr(FIRMWARE) to $dev_descr(AVAILABLE_FIRMWARE))" $env(SCRIPT_NAME)?cmd=firmware_update&device=$dev_descr(ADDRESS)&iface=$iface&$urlsid]
                                     }
                                 }
                             }
@@ -322,7 +360,10 @@
                             table_row { 
                                 table_data width=10 { }
                                 table_data {
-                                    h3 [url "$dev_descr(TYPE) ($dev_descr(ADDRESS))" $env(SCRIPT_NAME)?cmd=config&device=$dev_descr(ADDRESS)&iface=$iface&$urlsid]
+                                    h3 [url "$dev_descr(ADDRESS)" $env(SCRIPT_NAME)?cmd=config&device=$dev_descr(ADDRESS)&iface=$iface&$urlsid]
+                                }
+                                table_data {
+                                    h4 "$dev_descr(TYPE)"
                                 }
                                 table_data colspan=3 { }
                             }
@@ -357,7 +398,28 @@
                           xmlrpc $url system.methodHelp rssiInfo
                       } ] } { continue }
                       array_clear rssi_map
-                      set rssi_list [xmlrpc $url rssiInfo ]
+                      if { [catch {
+                        set rssi_list [xmlrpc $url rssiInfo ]
+                      } ] } {
+                        if { "$iface" == "HmIP-RF" } {
+                          set rssi_list ""
+                          set devlist [xmlrpc $url listDevices [list bool 0]]
+                          foreach dev $devlist {
+                            array_clear dev_descr
+                            array set dev_descr $dev
+                            if { "$dev_descr(PARENT)" == "" } {
+                              set rssi_device 65536
+                              set rssi_peer 65536
+                              catch { set rssi_device [xmlrpc $url getValue "$dev_descr(ADDRESS):0" "RSSI_DEVICE"] }
+                              catch { set rssi_peer [xmlrpc $url getValue "$dev_descr(ADDRESS):0" "RSSI_PEER"] }
+                              set name ""
+                              catch { set name $devnames($dev_descr(ADDRESS)) }
+                              set rssi_list "\"$name $dev_descr(ADDRESS)\" { $dev_descr(ADDRESS) { $rssi_device $rssi_peer }} $rssi_list"
+                            }
+                          }
+                        } else { continue }
+                      }
+                      table_data colspan=100 { puts "<hr/><h3><u>$iface:</u></h3>" }
                       array set rssi_map $rssi_list
                       foreach dev [lsort [array names rssi_map]] {
                           puts ""
@@ -522,7 +584,7 @@
     set green [rssi_color $RSSI_BAD $RSSI_MEDIUM $rssi]
     set color [format "#%02X%02X00" $red $green]
     table_data "bgcolor=\"$color\"" {
-        puts "[set rssi]dBm"
+        puts "[set rssi]&nbsp;dBm"
     }
 }
 
@@ -1045,8 +1107,16 @@
     global url iface env urlsid
     import device
     set reset 0
+    set force 0
+    set defer 0
     catch { import reset }
-    xmlrpc $url deleteDevice [list string $device] [list boolean $reset]
+    catch { import force }
+    catch { import defer }
+    set option 0
+    if { $reset == 1 } { set option [expr {$option + 1}] }
+    if { $force == 1 } { set option [expr {$option + 2}] }
+    if { $defer == 1 } { set option [expr {$option + 4}] }
+    xmlrpc $url deleteDevice [list string $device] [list int $option]
     redirect $env(SCRIPT_NAME)?cmd=list_devices&iface=$iface&$urlsid
 }
 
