--- occu/WebUI/www/config/cp_software.cgi.orig
+++ occu/WebUI/www/config/cp_software.cgi
@@ -117,7 +117,31 @@
     #    Nach der Installation wird die Zentrale automatisch neu gestartet. Sie k&ouml;nnen sich dann &uuml;ber die Schaltfl&auml;che unten neu anmelden.
     #} {"Neu anmelden" "window.location.href='/';"}
 
-    put_message "\${dialogSettingsExtraSoftwareHintPerformInstallationTitle}" "\${dialogSettingsExtraSoftwareHintPerformInstallationContent}" {"\${btnNewLogin}" "window.location.href='/';"}
+    if {[getProduct] == 3} {
+       set result 0
+       if { [catch {exec /bin/install_addon 2>/dev/null} results] } {
+         if { [lindex $::errorCode 0] == "CHILDSTATUS" } {
+           set result [lindex $::errorCode 2]
+         } else {
+           # Some kind of unexpected failure
+           set result 100
+         }
+       } else {
+         set result 0
+       }
+
+       if { $result == 0 } {
+         put_message "\${dialogSettingsExtraSoftwareHintPerformInstallationTitle}" "\${dialogSettingsExtraSoftwareHintPerformInstallationContentNoReboot}" {"\${btnOk}" "window.location.href='/';"}
+       } elseif { $result == 10 } {
+         put_message "\${dialogSettingsExtraSoftwareHintPerformInstallationTitle}" "\${dialogSettingsExtraSoftwareHintPerformInstallationContent}" {"\${btnNewLogin}" "window.location.href='/';"}
+         exec /sbin/reboot
+       } else {
+         put_message "Error ($result)" "\${dialogSettingsExtraSoftwareHintPerformInstallationFailure}"
+       }
+    } else {
+       put_message "\${dialogSettingsExtraSoftwareHintPerformInstallationTitle}" "\${dialogSettingsExtraSoftwareHintPerformInstallationContent}" {"\${btnNewLogin}" "window.location.href='/';"}
+       exec /sbin/reboot
+    }
 
     puts ""
     cgi_javascript {
@@ -510,10 +534,6 @@
 }
 
 proc action_install_start {} {
-    if {[getProduct] == 3} {
-       exec touch /usr/local/.doAddonInstall
-    }
-    exec /sbin/reboot
 }
 
 cgi_eval {
--- occu/WebUI/www/webui/js/lang/de/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.js
@@ -689,6 +689,8 @@
     "dialogSettingsExtraSoftwareInstallMessageBoxHintLowBat_b" : "Um einen Datenverlust durch Ausfall der Stromversorgung vorzubeugen, empfehlen wir, die Batterien vor dem Einspielen der Software zu erneuern.",
     "dialogSettingsExtraSoftwareHintPerformInstallationTitle" : "Software installieren",
     "dialogSettingsExtraSoftwareHintPerformInstallationContent" : "Die Software wird installiert. Nach der Installation wird die Zentrale automatisch neu gestartet. Sie k%F6nnen sich dann neu anmelden.",
+    "dialogSettingsExtraSoftwareHintPerformInstallationContentNoReboot" : "Die Installation des Addons war erfolgreich. Bitte stellen Sie sicher, dass das Addon ordnungsgem%E4&szlig; funktioniert oder starten Sie bei etwaigen Problemen die Zentrale neu.",
+    "dialogSettingsExtraSoftwareHintPerformInstallationFailure" : "Ein unerwarteter Fehler ist w%E4hrend der Addon-Installation aufgetreten. Bei der hochgeladenen Datei handelt es sich entweder nicht um ein Addon-Archiv, ist defekt oder ist f%FCr eine andere Hardware vorgesehen. Bitte versuchen Sie es erneut oder kontaktieren Sie den Autor des Addons.",
     "commentSettingsExtraSoftwareEnd" : "***********************************************",
     "commentDialogNewDevices" : "***********************************************",
     "dialogNewDevicesTitle" : "Ger%E4te anlernen",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.js
@@ -698,6 +698,8 @@
 
     "dialogSettingsExtraSoftwareHintPerformInstallationTitle" : "Install software",
     "dialogSettingsExtraSoftwareHintPerformInstallationContent" : "The software is being installed. After installation, the CCU will be restarted automatically. You can log in again afterwards.",
+    "dialogSettingsExtraSoftwareHintPerformInstallationContentNoReboot" : "The installation of the addon was successful. Please make sure the addon works as expected or perform a reboot if you identify any problems.",
+    "dialogSettingsExtraSoftwareHintPerformInstallationFailure" : "An unexpected error occurred while trying to install the addon. The uploaded file might be no valid addon archive, corrupt or you are trying to install the wrong hardware version of the addon. Please try again or contact the addon vendor.",
 
     "commentSettingsExtraSoftwareEnd" : "***********************************************",
     "commentDialogNewDevices" : "***********************************************",
