--- occu/WebUI/www/config/cp_software.cgi.orig
+++ occu/WebUI/www/config/cp_software.cgi
@@ -117,7 +117,31 @@
   #  Nach der Installation wird die Zentrale automatisch neu gestartet. Sie k&ouml;nnen sich dann &uuml;ber die Schaltfl&auml;che unten neu anmelden.
   #} {"Neu anmelden" "window.location.href='/';"}
 
-  put_message "\${dialogSettingsExtraSoftwareHintPerformInstallationTitle}" "\${dialogSettingsExtraSoftwareHintPerformInstallationContent}" {"\${btnNewLogin}" "window.location.href='/';"}
+  if {[getProduct] == 3} {
+    set result 0
+    if { [catch {exec /bin/install_addon 2>/dev/null} results] } {
+      if { [lindex $::errorCode 0] == "CHILDSTATUS" } {
+        set result [lindex $::errorCode 2]
+      } else {
+        # Some kind of unexpected failure
+        set result 100
+      }
+    } else {
+      set result 0
+    }
+
+    if { $result == 0 } {
+      put_message "\${dialogSettingsExtraSoftwareHintPerformInstallationTitle}" "\${dialogSettingsExtraSoftwareHintPerformInstallationContentNoReboot}" {"\${btnOk}" "window.location.href='/';"}
+    } elseif { $result == 10 } {
+      put_message "\${dialogSettingsExtraSoftwareHintPerformInstallationTitle}" "\${dialogSettingsExtraSoftwareHintPerformInstallationContent}" {"\${btnNewLogin}" "window.location.href='/';"}
+      exec /sbin/reboot -d 2 2>/dev/null >/dev/null &
+    } else {
+      put_message "Error ($result)" "\${dialogSettingsExtraSoftwareHintPerformInstallationFailure}"
+    }
+  } else {
+    put_message "\${dialogSettingsExtraSoftwareHintPerformInstallationTitle}" "\${dialogSettingsExtraSoftwareHintPerformInstallationContent}" {"\${btnNewLogin}" "window.location.href='/';"}
+    exec /sbin/reboot -d 2 2>/dev/null >/dev/null &
+  }
 
   puts ""
   cgi_javascript {
@@ -367,7 +391,7 @@
                   table {
                     table_row {
                       table_data {
-                        division {class="CLASS21412"} {onClick="installAddon();"} {
+                        division {class="CLASS21412"} {onClick="document.upload_form.submit();"} {
                           #puts "Installieren"
                           puts "\${dialogSettingsExtraSoftwareBtnInstallSoftware}"
                         }
@@ -523,14 +547,7 @@
 }
 
 proc action_install_start {} {
-  if {[getProduct] == 3} {
-     exec touch /usr/local/.doAddonInstall
-     exec /sbin/reboot
-  } else {
-    # CCU 2
-    exec /bin/kill -SIGQUIT 1
-  }
-  
+  puts ""
 }
 
 cgi_eval {
--- occu/WebUI/www/webui/js/lang/de/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.js
@@ -688,13 +688,15 @@
     "dialogSettingsExtraSoftwareBtnInstallSoftware" : "Installieren",
     "dialogSettingsExtraSoftwareBtnStartInstallSoftware" : "Installation starten",
     "dialogSettingsExtraSoftwareLblSelectExtraSoftware" : "Zusatzsoftware ausw%E4hlen:",
-    "dialogSettingsExtraSoftwareHintSelectExtraSoftware" : "<b>Hinweis:</b><br/>Vom Anwender installierte Zusatzsoftware kann zu unerw%FCnschten Ergebnissen bis hin zu Datenverlust und Systeminstabilit%E4ten f%FChren.<br/><br/><b><u>F%FCr vom Anwender installierte Zusatzsoftware %FCbernimmt die eQ-3 AG keine Haftung.</u></b><br/><br/>Zum Abschlu%DF der Installation wird die Zentrale automatisch neu gestartet.",
+    "dialogSettingsExtraSoftwareHintSelectExtraSoftware" : "<b>Hinweis:</b><br/>Vom Anwender installierte Zusatzsoftware kann zu unerw%FCnschten Ergebnissen bis hin zu Datenverlust und Systeminstabilit%E4ten f%FChren.<br/><br/><b><u>F%FCr vom Anwender installierte Zusatzsoftware %FCbernimmt die eQ-3 AG keine Haftung.</u></b><br/><br/>Zum Abschlu%DF der Installation muss die Zentrale mitunter neu gestartet werden.",
     "dialogSettingsExtraSoftwareInstallMessageBoxTitle" : "Firmware-Update - Best%E4tigung",
     "dialogSettingsExtraSoftwareInstallMessageBoxContent" : "Die Zusatzsoftware wurde hochgeladen. Sie k%F6nnen jetzt durch Klick auf die Schaltfl%E4che unten die Installation starten.",
     "dialogSettingsExtraSoftwareInstallMessageBoxHintLowBat_a" : "Achtung!<br/>Der Ladezustand der Batterien betr%E4gt nur noch ",
     "dialogSettingsExtraSoftwareInstallMessageBoxHintLowBat_b" : "Um einen Datenverlust durch Ausfall der Stromversorgung vorzubeugen, empfehlen wir, die Batterien vor dem Einspielen der Software zu erneuern.",
     "dialogSettingsExtraSoftwareHintPerformInstallationTitle" : "Software installieren",
     "dialogSettingsExtraSoftwareHintPerformInstallationContent" : "Die Software wird installiert. Nach der Installation wird die Zentrale automatisch neu gestartet. Sie k%F6nnen sich dann neu anmelden.",
+    "dialogSettingsExtraSoftwareHintPerformInstallationContentNoReboot" : "Die Installation des Addons war erfolgreich. Bitte stellen Sie sicher, dass das Addon ordnungsgem%E4&szlig; funktioniert oder starten Sie bei etwaigen Problemen die Zentrale neu.",
+    "dialogSettingsExtraSoftwareHintPerformInstallationFailure" : "Ein unerwarteter Fehler ist w%E4hrend der Addon-Installation aufgetreten. Bei der hochgeladenen Datei handelt es sich entweder nicht um ein Addon-Archiv, ist defekt oder ist f%FCr eine andere Hardware vorgesehen. Bitte versuchen Sie es erneut oder kontaktieren Sie den Autor des Addons.",
     "commentSettingsExtraSoftwareEnd" : "***********************************************",
     "commentDialogNewDevices" : "***********************************************",
     "dialogNewDevicesTitle" : "Ger%E4te anlernen",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.js
@@ -693,7 +693,7 @@
     "dialogSettingsExtraSoftwareBtnStartInstallSoftware" : "Start installation",
     "dialogSettingsExtraSoftwareLblSelectExtraSoftware" : "Select additional software:",
 
-    "dialogSettingsExtraSoftwareHintSelectExtraSoftware" : "<b>Please note:</b><br/>Additional software installed by the user can lead to unexpected results, data loss or even system instability.<br/><br/><b><u>eQ-3 AG does not assume any liability for additional software installed by the user.</u></b><br/><br/>To finish installation, the CCU will be restarted automatically.",
+    "dialogSettingsExtraSoftwareHintSelectExtraSoftware" : "<b>Please note:</b><br/>Additional software installed by the user can lead to unexpected results, data loss or even system instability.<br/><br/><b><u>eQ-3 AG does not assume any liability for additional software installed by the user.</u></b><br/><br/>To finish installation, the CCU might have to be restarted.",
 
     "dialogSettingsExtraSoftwareInstallMessageBoxTitle" : "Firmware update - confirmation",
     "dialogSettingsExtraSoftwareInstallMessageBoxContent" : "The additional software has been uploaded. You can now start the installation by click on the button below.",
@@ -704,6 +704,8 @@
 
     "dialogSettingsExtraSoftwareHintPerformInstallationTitle" : "Install software",
     "dialogSettingsExtraSoftwareHintPerformInstallationContent" : "The software is being installed. After installation, the CCU will be restarted automatically. You can log in again afterwards.",
+    "dialogSettingsExtraSoftwareHintPerformInstallationContentNoReboot" : "The installation of the addon was successful. Please make sure the addon works as expected or perform a reboot if you identify any problems.",
+    "dialogSettingsExtraSoftwareHintPerformInstallationFailure" : "An unexpected error occurred while trying to install the addon. The uploaded file might be no valid addon archive, corrupt or you are trying to install the wrong hardware version of the addon. Please try again or contact the addon vendor.",
 
     "commentSettingsExtraSoftwareEnd" : "***********************************************",
     "commentDialogNewDevices" : "***********************************************",
