--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -15828,10 +15828,11 @@
   ],
   INTERFACES:
   [
-    {id: "INTERFACE_BIDCOS_RF", name: translateKey("BidCosRF")}, // BidCos-RF
-    {id: "INTERFACE_BIDCOS_WIRED", name: translateKey("BidCosWired")}, // BidCos-Wired
-    {id: "INTERFACE_HMIP_RF", name: translateKey("HmIPRF")}, // HmIP-RF
-    {id: "INTERFACE_VIRTUAL_DEVICES", name: translateKey("VirtualDevices")},
+    {id: "INTERFACE_BIDCOS_RF", name: "BidCos-RF"}, // BidCos-RF
+    {id: "INTERFACE_BIDCOS_WIRED", name: "BidCos-Wired"}, // BidCos-Wired
+    {id: "INTERFACE_HMIP_RF", name: "HmIP-RF"}, // HmIP-RF
+    {id: "INTERFACE_VIRTUAL_DEVICES", name: "VirtualDevices"},
+    {id: "INTERFACE_CUXD", name: "CUxD"}, //
     {id: "INTERFACE_SYSTEM", name: "System"} //
   ],
   
