--- occu/WebUI/www/lang/js/de/translate.lang.js.orig
+++ occu/WebUI/www/lang/js/de/translate.lang.js
@@ -352,6 +352,7 @@
     "btnAddChannel" : "Kanal<br/>hinzuf%FCgen",
     "btnAddChannelWOLineBreak" : "Kanal hinzuf%FCgen",
     "btnAddSysVariable" : "Systemvariable hinzuf%FCgen",
+    "btnSortSysVariable" : "alphabetisch sortieren",
     "btnAddProgram" : "Programm hinzuf%FCgen",
     "btnAddSeparator" : "Trennzeile hinzuf%FCgen",
     "btnChooseChannel" : "Kanalauswahl",
--- occu/WebUI/www/lang/js/en/translate.lang.js.orig
+++ occu/WebUI/www/lang/js/en/translate.lang.js
@@ -351,6 +351,7 @@
     "btnAddChannel" : "Add<br/>channel",
     "btnAddChannelWOLineBreak" : "Add channel",
     "btnAddSysVariable" : "Add system variable",
+    "btnSortSysVariable" : "Sort system variables alphabetically",
     "btnAddProgram" : "Add program",
     "btnAddSeparator" : "Add separator",
     "btnChooseChannel" : "Channel selection",
--- occu/WebUI/www/rega/pages/msg/userAccountConfigAdmin.htm.orig
+++ occu/WebUI/www/rega/pages/msg/userAccountConfigAdmin.htm
@@ -180,6 +180,9 @@
             <td class="CLASS00709">
               <div class="FooterButton CLASS00716" onclick="addSysVar()">${btnAddSysVariable}</div>
             </td>
+            <td class="CLASS00709">
+              <div class="FooterButton CLASS00716" onclick="sortSysVar()">${btnSortSysVariable}</div>
+            </td>
           </tr>
         </table>
     </td>
@@ -288,6 +291,14 @@
     return arTmp;
   };
   
+  
+  sortSysVar = function()
+  {
+    if (iUserId != 0) {
+     iseUser.sortSysVarTable(iUserId);
+    }
+  };
+
   addSysVar = function()
   {
     dlgResult = "";
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -22166,7 +22166,23 @@
     };
     new Ajax.Updater("userSysVarTbl", url, opts);
   },
-  
+
+  sortSysVarTable: function(uid) {                                                                 
+    var url = "/esp/system.htm?sid="+SessionId;      
+    var pb = "((dom.GetObject(ID_USERS)).Get(" + uid + ")).UserSharedObjects().SortByName();";
+    pb += "string action = 'BuildUserSvTable';";                
+    pb += "integer userId = "+ uid +";";         
+    var opts = {                                                                               
+      postBody: ReGa.encode(pb),                                                           
+      onComplete: function(response) {               
+        if (200 == response.status) { 
+          translatePage("#userSysVarTbl");                       
+        }                                        
+      }                                                                                        
+    };                                                                                     
+    new Ajax.Updater("userSysVarTbl", url, opts);
+  },
+    
   addSysVarBuildTable: function(uid, newVarId) {
     //var url = "/esp/system.htm?sid="+SessionId+"&curDateTime="+new Date().getTime();
     var url = "/esp/system.htm?sid="+SessionId;
