--- occu/WebUI/www/config/easymodes/smoke_detector_ch_master.tcl.orig
+++ occu/WebUI/www/config/easymodes/smoke_detector_ch_master.tcl
@@ -1,6 +1,6 @@
 #!/bin/tclsh
 #Kanal-EasyMode!
-source $env(DOCUMENT_ROOT)/config/easymodes/em_common.tcl
+source [file join $env(DOCUMENT_ROOT) config/easymodes/em_common.tcl]
 
 # set PROFILE_PNAME(Team) "Zu welcher Gruppe geh&ouml;rt der Rauchmelder?"
 set PROFILE_PNAME(Team) "\${SD_QuestionGroup}"
--- occu/WebUI/www/config/ic_common.tcl.orig
+++ occu/WebUI/www/config/ic_common.tcl
@@ -7,8 +7,8 @@
 sourceOnce common.tcl
 sourceOnce ic_metadata.tcl
 
-source $env(DOCUMENT_ROOT)/config/easymodes/EnterFreeValue.tcl
-source $env(DOCUMENT_ROOT)/config/easymodes/etc/getStatusDisplayHelp.tcl
+source [file join $env(DOCUMENT_ROOT) config/easymodes/EnterFreeValue.tcl]
+source [file join $env(DOCUMENT_ROOT) config/easymodes/etc/getStatusDisplayHelp.tcl]
 
 
 #Flags für xmlrpc Aufruf "getLinks"
@@ -29,7 +29,7 @@
     set env(CONFIG_ROOT) "/etc/config"
 }
 
-set USERPROFILESPATH "$env(CONFIG_ROOT)/userprofiles"
+set USERPROFILESPATH [file join $env(CONFIG_ROOT) userprofiles]
 catch { file mkdir $USERPROFILESPATH }
 
 set wired "n.a"
@@ -42,7 +42,7 @@
 array set nav_td ""
 array set TYPE_MAP { "BOOL" "bool" "ENUM" "int" "INTEGER" "int" "FLOAT" "double" "STRING" "string"}
 
-set IC_SETTINGS_CONF_FILE "$env(CONFIG_ROOT)/ic_settings.dat"
+set IC_SETTINGS_CONF_FILE [file join $env(CONFIG_ROOT) ic_settings.dat]
 array set IC_SETTINGS_VALUES ""
 #Defaults (werden von der Settings-Datei überschrieben):
 set IC_SETTINGS_VALUES(EXPERTMODE) "off"
--- occu/WebUI/www/config/ic_deviceparameters.cgi.orig
+++ occu/WebUI/www/config/ic_deviceparameters.cgi
@@ -672,7 +672,7 @@
     # wichtig f. freie Werteingabe
     array set dev_descr_sender [array get ch_descr]
 
-    catch {source $env(DOCUMENT_ROOT)/config/easymodes/$ch_paramid\_intkey.tcl}
+    catch {source [file join $env(DOCUMENT_ROOT) config/easymodes/$ch_paramid\_intkey.tcl]}
 
     array set receiver_ps [xmlrpc $iface_url($iface) getParamset [list string $receiver] [list string $sender]]
     array set ps_descr_receiver [xmlrpc $iface_url($iface) getParamsetDescription [list string $receiver] [list string $sender]]
@@ -846,14 +846,14 @@
     }
     #=====
 
-    set sourcePath "$env(DOCUMENT_ROOT)config/easymodes/$ch_paramid.tcl"
+    set sourcePath [file join $env(DOCUMENT_ROOT) config/easymodes/$ch_paramid.tcl]
 
     if {[isHmIP] == "true" || [isHmIPGroup $ch_descr(PARENT_TYPE)] == "true" } {
-      if {[file exist $env(DOCUMENT_ROOT)config/easymodes/hmip/$ch_paramid.tcl]} {
-        set sourcePath "$env(DOCUMENT_ROOT)config/easymodes/hmip/$ch_paramid.tcl"
-      } elseif {[file exists $env(DOCUMENT_ROOT)config/easymodes/hmip/$ch_descr(TYPE).tcl]} {
+      if {[file exists [file join $env(DOCUMENT_ROOT) config/easymodes/hmip/$ch_paramid.tcl]]} {
+        set sourcePath [file join $env(DOCUMENT_ROOT) config/easymodes/hmip/$ch_paramid.tcl]
+      } elseif {[file exists [file join $env(DOCUMENT_ROOT) config/easymodes/hmip/$ch_descr(TYPE).tcl]]} {
         set ch_paramid "$ch_descr(TYPE)"
-        set sourcePath "$env(DOCUMENT_ROOT)config/easymodes/hmip/$ch_descr(TYPE).tcl"
+        set sourcePath [file join $env(DOCUMENT_ROOT) config/easymodes/hmip/$ch_descr(TYPE).tcl]
       }
     }
 
@@ -880,7 +880,7 @@
         set s "" 
         if {[isVirtual $ch_paramid] == "true"} {
           set file "$ch_paramid\Params.tcl"
-          catch {source $env(DOCUMENT_ROOT)/config/easymodes/$file}
+          catch {source [file join $env(DOCUMENT_ROOT) config/easymodes/$file]}
           # catch { set s [cmd_link_paramset $iface $ch_descr(ADDRESS) MASTER MASTER CHANNEL_$ch_descr(INDEX)] }
           catch { set_htmlParams $iface $ch_address ch_ps ch_ps_descr CHANNEL_$ch_descr(INDEX) "" }
           append s $HTML_PARAMS(separate_1)
@@ -950,14 +950,14 @@
                 if { [catch { array set ch_descr [xmlrpc $iface_url($iface) getDeviceDescription [list string $sender]] } ] } then { 
                   #continue
                 } 
-                catch {source $env(DOCUMENT_ROOT)/config/easymodes/$ch_paramid.tcl}
+                catch {source [file join $env(DOCUMENT_ROOT) config/easymodes/$ch_paramid.tcl]}
               } 
         
               if {[isVirtual $ch_paramid] == "true"} { 
                 if { [catch { array set ch_descr [xmlrpc $iface_url($iface) getDeviceDescription [list string $sender]] } ] } then { 
                   continue
                 }
-                catch {source $env(DOCUMENT_ROOT)/config/easymodes/$ch_paramid.tcl} 
+                catch {source [file join $env(DOCUMENT_ROOT) config/easymodes/$ch_paramid.tcl]}
               }
 
               array_clear receiver_ps
--- occu/WebUI/www/config/ic_setprofiles.cgi.orig
+++ occu/WebUI/www/config/ic_setprofiles.cgi
@@ -852,7 +852,7 @@
     set sender_paramid [getExistingParamId $paramids]
 
     if {($sender_paramid == "") && ([isHmIP] == "true")} {
-      if { [file exists $env(DOCUMENT_ROOT)config/easymodes/linkHmIP_$dev_descr_sender(TYPE).tcl] } {
+      if { [file exists [file join $env(DOCUMENT_ROOT) config/easymodes/linkHmIP_$dev_descr_sender(TYPE).tcl]] } {
         set sender_paramid linkHmIP_$dev_descr_sender(TYPE)
       }
     }
@@ -867,7 +867,7 @@
     set receiver_paramid [getExistingParamId $paramids]
 
     if {($receiver_paramid == "") && ([isHmIP] == "true")} {
-      if { [file exists $env(DOCUMENT_ROOT)config/easymodes/linkHmIP_$dev_descr_receiver(TYPE).tcl] } {
+      if { [file exists [file join $env(DOCUMENT_ROOT) config/easymodes/linkHmIP_$dev_descr_receiver(TYPE).tcl]] } {
         set receiver_paramid linkHmIP_$dev_descr_receiver(TYPE)
       }
     }
--- occu/WebUI/www/config/st_values.cgi.orig
+++ occu/WebUI/www/config/st_values.cgi
@@ -1,8 +1,8 @@
 #!/bin/tclsh
 source ../cgi.tcl
 
-set ST_FILENAME "$env(DOCUMENT_ROOT)/config/stringtable_de.txt"
-set JS_FILENAME "$env(DOCUMENT_ROOT)/config/st_values.js"
+set ST_FILENAME [file join $env(DOCUMENT_ROOT) config/stringtable_de.txt]
+set JS_FILENAME [file join $env(DOCUMENT_ROOT) config/st_values.js]
 	
 proc put_js_functions {} {
 
