--- occu/WebUI/www/config/easymodes/SWITCH_VIRTUAL_RECEIVER/KEY_TRANSCEIVER.tcl.orig
+++ occu/WebUI/www/config/easymodes/SWITCH_VIRTUAL_RECEIVER/KEY_TRANSCEIVER.tcl
@@ -69,10 +69,10 @@
 set PROFILE_2(LONG_CT_OFFDELAY)               0
 set PROFILE_2(LONG_CT_ON)                     0
 set PROFILE_2(LONG_CT_ONDELAY)                0
-set PROFILE_2(LONG_JT_OFF)                    {1 6}
+set PROFILE_2(LONG_JT_OFF)                    {6 1}
 set PROFILE_2(LONG_JT_OFFDELAY)               6
 set PROFILE_2(LONG_JT_ON)                     {4 6}
-set PROFILE_2(LONG_JT_ONDELAY)                {3 6}
+set PROFILE_2(LONG_JT_ONDELAY)                {6 3}
 set PROFILE_2(LONG_MULTIEXECUTE)              {0 6}
 set PROFILE_2(LONG_OFFDELAY_TIME_BASE)        {0 range 0 - 7}
 set PROFILE_2(LONG_OFFDELAY_TIME_FACTOR)      {0 range 0 - 31}
