--- occu/WebUI/www/api/methods/ccu/setuserackinstallwizard.tcl.orig
+++ occu/WebUI/www/api/methods/ccu/setuserackinstallwizard.tcl
@@ -8,8 +8,20 @@
 # Rückgabewert: immer true
 ##
 
-catch {exec touch /etc/config/userprofiles/userAckInstallWizard_[string map {{ } ;} $args(userName)]}
+set userName [string map {
+                " " ";"
+                "/" ""
+                "Ä" "AE"
+                "Ü" "UE"
+                "Ö" "OE"
+                "ä" "ae"
+                "ü" "ue"
+                "ö" "oe"
+                "ß" "ss"
+             } $args(userName)]
 
-jsonrpc_response true
+regsub -all {[^\u0020-\u007e]+} $userName "" userName
 
+catch {exec touch /etc/config/userprofiles/userAckInstallWizard_$userName}
 
+jsonrpc_response true
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -6568,8 +6568,22 @@
         }
 
         var usrPwd = homematic('User.getUserPWD', {'userID': userId});
+        var usrName = userName.replace(/[ \/\xC4\xD6\xDC\xE4\xF6\xFC\xDF]/g, function(m) {
+                        return {
+                          ' ': ';',
+                          '/': '',
+                          '\xC4': 'AE',
+                          '\xD6': 'OE'
+                          '\xDC': 'UE',
+                          '\xE4': 'ae',
+                          '\xF6': 'oe',
+                          '\xFC': 'ue',
+                          '\xDF': 'ss'
+                        }[m];
+                      });
+
         if ( usrPwd == "") {
-            var result = homematic('CCU.existsFile', {'file': "/etc/config/userprofiles/userAckInstallWizard_" + userName.replace(" ", ";")});
+            var result = homematic('CCU.existsFile', {'file': "/etc/config/userprofiles/userAckInstallWizard_" + usrName});
             if (!result) {
               var checkUpdateContentRunning = window.setInterval(function () {
                 if (!bUpdateContentRunning) {
@@ -6589,9 +6603,9 @@
 
           // User password set
           // The User will see a hint that new firewall settings are active
-          if ((getUPL() == UPL_USER) && (! homematic('CCU.existsFile', {'file': "/etc/config/userprofiles/userAckInstallWizard_" + userName.replace(" ", ";")}))) {
+          if ((getUPL() == UPL_USER) && (! homematic('CCU.existsFile', {'file': "/etc/config/userprofiles/userAckInstallWizard_" + usrName}))) {
             new MessageBox.show(translateKey("dglUserNewFwSettingsTitle"),translateKey("dglUserNewFwSettingsContent"));
-            homematic("CCU.setUserAckInstallWizard", {'userName' : userName});
+            homematic("CCU.setUserAckInstallWizard", {'userName' : usrName});
           }
         }
       }
