--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -27868,19 +27868,18 @@
       dcNotAvailable = -1,
       dcAlarm = 89;  // Attention when dc >= 90%
 
-    homematic("Interface.listBidcosInterfaces", {"interface": ifaceBidCosRF}, function (interfaceStatus) {
+    homematic("SysVar.getValueByName", {"name": "DutyCycle"}, function(dcVal) {
 
       var dutyCycleProgressElem = jQuery("#dutyCycleProgress"),
         dutyCycleProgressBarElm = jQuery("#dutyCycleProgressBar"),
         dutyCycleValElm = jQuery("#dutyCycleVal"),
         trDutyCycle = jQuery("[name='trDutyCycle']"),
         trPartingLineElm = jQuery("#partingLine1"),
-        dcVal,
         width, value;
 
-      if (interfaceStatus && (typeof interfaceStatus[0].dutyCycle != "undefined")) {
-        dcVal = parseInt(interfaceStatus[0].dutyCycle);
-        conInfo("dutyCycle - " + ifaceBidCosRF + ": " + dcVal + dcUnit);
+      if (typeof dcVal !== "undefined") {
+        dcVal = Math.floor(dcVal);
+        conInfo("dutyCycle - " + ifaceBidCosRF + ": " + dcVal + " " + dcUnit);
         arInterfaceDutyCycle[ifaceBidCosRF] = ((dcVal >= 0) && (dcVal <= 100)) ? dcVal : dcNotAvailable;
       } else {
         conInfo("No gateway status for the interface " + ifaceBidCosRF + " available!");
@@ -27888,7 +27887,7 @@
       }
 
       if (arInterfaceDutyCycle[ifaceBidCosRF] != dcNotAvailable) {
-        dutyCycleValElm.text(arInterfaceDutyCycle[ifaceBidCosRF] + dcUnit);
+        dutyCycleValElm.text(arInterfaceDutyCycle[ifaceBidCosRF] + " " + dcUnit);
 
         width = parseInt(dutyCycleProgressElem.css("width"));
         value = width - (width / 100 * arInterfaceDutyCycle[ifaceBidCosRF]);
