--- occu/WebUI/www/rega/esp/programs.fn.orig	2019-07-11 10:09:32.000000000 +0200
+++ occu/WebUI/www/rega/esp/programs.fn	2019-07-11 10:05:41.000000000 +0200
@@ -849,15 +849,17 @@
     object oOrig = dom.GetObject( system.GetVar("orig") );
     if( oCopy && oOrig )
     {
+      boolean isActive = true;
       if( system.GetVar("copy") != system.GetVar("orig") )
       {
+        isActive = oOrig.Active();
         oCopy.ProgramCopyTo( oOrig.ID() );
         dom.DeleteObject( oCopy.ID() );
         oOrig.ProgramCopyID( ID_ERROR );        
         Write( oOrig.Name() );
       }
       oOrig.ProgramUpdate();
-      oOrig.Active(true);
+      oOrig.Active(isActive);
     }
   }
   return;