--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -25971,8 +25971,9 @@
 logout = function() {
   regaMonitor.stop();
   InterfaceMonitor.stop();
-  // The second url-param has to be appended by a '?' instead of a '&'
-  location.href = "/logout.htm?sid=" + SessionId+"?lang="+getLang();
+  location.href = "/logout.htm?lang="+getLang();
+  homematic('Session.logout', {});
+  homematic('system.saveObjectModel', {});
 };
 
 
