--- occu/WebUI/www/rega/esp/sico.fn.orig
+++ occu/WebUI/www/rega/esp/sico.fn
@@ -13,6 +13,22 @@
   return;
 }
 
+function ::SetNegateCondition() {
+  if( system.IsVar("scid") && system.IsVar("nc") )
+  {
+    object oTmp = dom.GetObject( system.GetVar("scid") );
+    if( oTmp )
+    {
+      string sNC = system.GetVar("nc");
+      if (sNC == "false") {
+        oTmp.NegateCondition(false);
+      } else {
+        oTmp.NegateCondition(true);
+      }
+    }
+  }
+}
+
 function ::SetOperatorType()
 {
   string sCI = "SetOperatorType[SICO]: ";
--- occu/WebUI/www/rega/esp/sico.htm.orig
+++ occu/WebUI/www/rega/esp/sico.htm
@@ -9,4 +9,5 @@
   if( action == "SetLeftAndRightValue" ) { Call("sico.fn::SetLeftAndRightValue()"); }
   if( action == "SetValueRange" ) { Call("sico.fn::SetValueRange()"); }
   if( action == "ReloadSiCo" ) { Call("sico.fn::ReloadSiCo()"); }
+  if( action == "SetNegateCondition" ) { Call("sico.fn::SetNegateCondition()"); }
 %>
\ No newline at end of file
--- occu/WebUI/www/rega/esp/sico.inc.orig
+++ occu/WebUI/www/rega/esp/sico.inc
@@ -66,6 +66,10 @@
       boolean bSysVarGeneric = ( (iLVT == ivtSystemId) && (isGenericValueType)  && (oLV.ValueSubType() == istGeneric ) );
       boolean bSysVarString  = ( (iLVT == ivtSystemId) && (oLV.ValueType() == ivtString) && (oLV.ValueSubType() == istChar8859) );
 
+      string sChecked = "";
+      if( oSC.NegateCondition() ) { sChecked = "checked"; }
+      Write( '&nbsp;${ruleConditionLblNot}<input type="checkbox" '#sChecked#' class="SelectBox"  onchange="iseSingleCondition.SetNegateCondition('#oSC.ID()#',this.checked);" >' );
+      
       if (bSysVarString)
       {
         !Write("<i>bei</i>");
--- occu/WebUI/www/webui/js/lang/de/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.js
@@ -777,6 +777,7 @@
     "ruleCondition" : "Bedingung:",
     "ruleConditionIf" : "Wenn...",
     "ruleConditionSelectDeviceList" : "Ger%E4teauswahl",
+    "ruleConditionLblNot" : "nicht",
     "ruleConditionLblWhen" : "bei",
     "ruleConditionSelectSystemState" : "Systemzustand",
     "ruleConditionLblWithinValueRange" : "im Wertebereich",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.js
@@ -809,6 +809,7 @@
     "ruleConditionIf" : "If...",
 
     "ruleConditionSelectDeviceList" : "Device selection",
+    "ruleConditionLblNot" : "not",
     "ruleConditionLblWhen" : "when",
 
     "ruleConditionSelectSystemState" : "System state",
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -24823,6 +24823,24 @@
     new Ajax.Request(url,opts);        
   },
 
+  SetNegateCondition: function(scid,nc) {                                                          
+    var url = "/esp/sico.htm?sid="+SessionId;            
+    var pb = '';                          
+    pb += 'string action = "SetNegateCondition";';                                     
+    pb += 'string scid = "'+scid+'";';               
+    pb += 'string nc = "'+nc+'";';                                                                     
+    var opts =                                                                                     
+    {                                                
+      postBody: ReGa.encode(pb),      
+      onComplete:function(t)                                    
+      {                                          
+        if(dbg){alert( t.responseText );}                                                      
+        ReloadSingleCondition(scid);                                                       
+      }                                              
+    };                                
+    new Ajax.Request(url,opts);                                  
+  },
+  
   SetLeftValue: function(scid,lv)
   {
     var url = "/esp/sico.htm?sid="+SessionId;
