--- occu/WebUI/www/config/cp_software.cgi.orig
+++ occu/WebUI/www/config/cp_software.cgi
@@ -243,6 +243,7 @@
     puts "\${dialogSettingsExtraSoftwareTitle}"
   }
   division {class="CLASS21406"} {
+    puts "<div style='height:400px;width:100%;overflow:scroll;'>"
     table {class="popupTable"} {border="1"} {
       set scripts ""
       set loop -1
@@ -385,6 +386,7 @@
       }
     }
   }
+  puts "</div>"
   division {class="popupControls"} {
     table {
       table_row {
