--- occu/WebUI/www/config/cp_maintenance.cgi.orig
+++ occu/WebUI/www/config/cp_maintenance.cgi
@@ -1259,6 +1259,7 @@
 }
 
 proc action_update_start {} {
+  puts ""
   catch { exec killall hss_lcd }
   catch { exec lcdtool {Saving   Data...  } }
   rega system.Save()
@@ -1275,6 +1276,7 @@
 }
 
 proc action_reboot {} {
+  puts ""
   catch { exec killall hss_lcd }
   catch { exec lcdtool {Saving   Data...  } }
   rega system.Save()
@@ -1283,6 +1285,7 @@
   exec /sbin/reboot
 }
 proc action_shutdown {} {
+  puts ""
   catch { exec killall hss_lcd }
   catch { exec lcdtool {Saving   Data...  } }
   rega system.Save()
--- occu/WebUI/www/config/cp_security.cgi.orig
+++ occu/WebUI/www/config/cp_security.cgi
@@ -1563,6 +1563,7 @@
 }
 
 proc action_reboot {} {
+  puts ""
   exec /sbin/reboot
 }
 
