--- occu/WebUI/www/api/methods/ccu/getversion.tcl.orig
+++ occu/WebUI/www/api/methods/ccu/getversion.tcl
@@ -0,0 +1,27 @@
+##
+# CCU.getVersion
+# Liefert die Firmware-Version der HomeMatic Zentrale
+#
+# Parameter: <keine>
+#
+# Rückgabewert: [string]
+#  Firmware-Version der HomeMatic Zentrale
+##
+
+proc read_var { filename varname} {
+    set fd [open $filename r]
+    set var ""
+    if { $fd >=0 } {
+        while { [gets $fd buf] >=0 } {
+          if [regexp "^ *$varname *= *(.*)$" $buf dummy var] break
+        }
+      close $fd
+    }
+    return $var
+}
+
+proc get_version { } {
+    return [read_var /VERSION VERSION]
+}
+
+jsonrpc_response [json_toString [get_version]]
--- occu/WebUI/www/api/methods/interface/reportvalueusage.tcl.orig
+++ occu/WebUI/www/api/methods/interface/reportvalueusage.tcl
@@ -0,0 +1,21 @@
+##
+# Interface.reportValueUsage
+# Teilt der Schnittstelle mit, wie häufig die Logikschicht einen Wert verwendet
+#
+# Parameter:
+#   interface : [string] Bezeichnung der Schnittstelle
+#   address   : [string] Adresse des logischen Geräts
+#   valueId   : [string] Wert des logischen Geräts
+#   refCounter: [integer] Anzahl der verwendungen des Wertes
+#
+# Rückgabewert: [boolean]
+#   true
+##
+
+set address $args(address)
+set valueId $args(valueId)
+set refCounter $args(refCounter)
+
+checkXmlRpcStatus [catch {xmlrpc $interface(URL) reportValueUsage [list string $address] [list string $valueId] [list int $refCounter] }]
+
+jsonrpc_response true
