--- occu/WebUI/www/webui/js/lang/de/translate.lang.extension.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.extension.js
@@ -524,6 +524,7 @@
     "lblErrorCheck" : "Fehlerpr%FCfung",
     "lblNoError" : "Kein Fehler",
     "lblError" : "Fehler",
+    "lblAccessPointError" : "Die Nutzung eines HmIPW-DRAP oder HmIP-HAP Access Points setzt den Einsatz eines RPI-RF-MOD/HmIP-CCU3 kompatiblen Funkmodules voraus.",
     "lblInfoStatusControlEnergyCounter" : "Energie-Z%E4hler " + HMIdentifier.de.CCUShortName,
     "lblInfoStatusControlDevice" : "Energie-Z%E4hler Ger%E4t",
     "lblInfoStatusControlVoltage" : "Spannung",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.extension.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.extension.js
@@ -522,6 +522,7 @@
     "lblErrorCheck" : "Error check",
     "lblNoError" : "No error",
     "lblError" : "Error",
+    "lblAccessPointError" : "The use of a HmIPW-DRAP or HmIP-HAP access point requires the use of a RPI-RF-MOD/HmIP-CCU3 compatible rf module.",
     "lblInfoStatusControlEnergyCounter" : "Energy counter " + HMIdentifier.en.CCUShortName,
     "lblInfoStatusControlDevice" : "Energy counter device",
     "lblInfoStatusControlVoltage" : "Voltage",
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -41555,15 +41555,15 @@
   jQuery.each(arDrap2Update, function(index, data) {
     drapSGtin += data.address + ",";
   });
-  CreateCPPopup("/pages/jpages/hmip/AccessPoint/show", "draps="+drapSGtin.slice(0,-1) );
+  CreateCPPopup("/pages/jpages/hmip/AccessPoint/show", "draps="+drapSGtin.slice(0,-1), 'lblAccessPointError');
 };
 
 showAccessPoint = function() {
-  CreateCPPopup("/pages/jpages/hmip/AccessPoint/show");
+  CreateCPPopup("/pages/jpages/hmip/AccessPoint/show", '', 'lblAccessPointError');
 };
 
-CreateCPPopup = function(src, pb) {
-  dlgPopup = new cpMessageBox(src, pb);
+CreateCPPopup = function(src, pb, errorLbl) {
+  dlgPopup = new cpMessageBox(src, pb, errorLbl);
 
   PopupClose = function() {
     dlgPopup.close();
@@ -41599,7 +41599,7 @@
 
 cpMessageBox.prototype =
 {
-  initialize: function(src, pb)
+  initialize: function(src, pb, errorLbl)
   {
     if ( $('messagebox') ){
       $("messagebox").hide();
@@ -41611,7 +41611,7 @@
     this.setWidth(800);
 
     this.createMessagebox();
-    this.LoadFromFile(src, pb);
+    this.LoadFromFile(src, pb, errorLbl);
   },
 
   getViewPortDim : function()
@@ -41735,7 +41735,7 @@
     this.removeMessagebox();
   },
   
-  LoadFromFile: function(src, pb) {
+  LoadFromFile: function(src, pb, errorLbl) {
     var url = src;
     if(url.indexOf('?sid=') === -1)
     {                                                                                 
@@ -41751,6 +41751,15 @@
       onComplete: function(trans) {
         centerMessageBox();
         $('messagebox').style.display="";
+      },
+      onFailure: function(response) {
+        $('centerbox').hide();
+        $('trlayer').hide();
+        if(typeof errorLbl != 'undefined') {
+          MessageBox.show(translateKey('lblError'), translateKey(errorLbl), '', 480,120);
+        } else {
+          Ajax_failure(url, response.statusText);
+        }
       }
     };
     new Ajax.Updater('messagebox',url,opt);
