--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -49020,8 +49020,8 @@
   CreateCPPopup("/config/cp_maintenance.cgi", "downloadOnly=1");
 };
 
-CreateCPPopup = function(src, pb) {
-  dlgPopup = new cpMessageBox(src, pb);
+CreateCPPopup = function(src, pb, errorLbl) {
+  dlgPopup = new cpMessageBox(src, pb, errorLbl);
 
   PopupClose = function() {
     dlgPopup.close();
@@ -49061,7 +49061,7 @@
 
 cpMessageBox.prototype =
 {
-  initialize: function(src, pb)
+  initialize: function(src, pb, errorLbl)
   {
     if ( $('messagebox') ){
       $("messagebox").hide();
@@ -49073,7 +49073,7 @@
     this.setWidth(800);
 
     this.createMessagebox();
-    this.LoadFromFile(src, pb);
+    this.LoadFromFile(src, pb, errorLbl);
   },
 
   getViewPortDim : function()
@@ -49197,7 +49197,7 @@
     this.removeMessagebox();
   },
   
-  LoadFromFile: function(src, pb) {
+  LoadFromFile: function(src, pb, errorLbl) {
     var url = src;
     if(url.indexOf('?sid=') === -1)
     {                                                                                 
@@ -49213,6 +49213,15 @@
       onComplete: function(trans) {
         centerMessageBox();
         $('messagebox').style.display="";
+      },
+      onFailure: function(response) {
+        $('centerbox').hide();
+        $('trlayer').hide();
+        if(typeof errorLbl != 'undefined') {
+          MessageBox.show(translateKey('lblError'), translateKey(errorLbl), '', 480,120);
+        } else {
+          Ajax_failure(url, response.statusText);
+        }
       }
     };
     new Ajax.Updater('messagebox',url,opt);
