--- occu/WebUI/www/rega/esp/system.fn.orig
+++ occu/WebUI/www/rega/esp/system.fn
@@ -1317,13 +1317,14 @@
 ! [userId], [opt: sTmpVars]
 function ::BuildUserSvTable() {
   WriteLine('<table cellpadding="3" cellspacing="1" class="CLASS02105" width="100%">');
-  WriteLine('<colgroup> <col style="width:15%;" /> <col style="width:20%;" /> <col style="width:15%;" /> <col style="width:20%;" /> <col style="width:15%;" /> <col style="width:15%;" />');
+  WriteLine('<colgroup> <col style="width:15%;" /> <col style="width:20%;" /> <col style="width:15%;" /> <col style="width:20%;" /> <col style="width:15%;" /> <col style="width:11%;" /> <col style="width:4%;" />');
   WriteLine('</colgroup> <tbody> <tr> <th class="CLASS02106">${thName}</th>');
   WriteLine('<th class="CLASS02107">${thDescription}</th>');
   WriteLine('<th class="CLASS02107">${thTypeOfVariable}</th>');
   WriteLine('<th class="CLASS02107">${thValues}</th>');
   WriteLine('<th class="CLASS02107">${thUnit}</th>');
   WriteLine('<th class="CLASS02107">${thAction}</th>');
+  WriteLine('<th class="CLASS02107"></th>');                                                                  
   WriteLine('</tr> ');
   
   string enumTmp = "";
@@ -1389,6 +1390,13 @@
             WriteLine("onclick='delFromTmpVars("#tmp#");'>${btnRemove}</div>");
           }
         WriteLine("</td>");
+        WriteLine("<td class='WhiteBkg CLASS02110'>");
+          !WriteLine("<table class='CLASS02108'>");                                                                                        
+          WriteLine("<table>");                                                                                        
+            WriteLine("<tr><td class='tSysvarMoveUp' onclick='iseUser.MoveSysVar("#userId#", "#tmp#", -1);'></td></tr>");                   
+            WriteLine("<tr><td class='tSysvarMoveDown' onclick='iseUser.MoveSysVar("#userId#", "#tmp#", 1);'></td></tr>");                  
+          WriteLine("</table>");
+        WriteLine("</td>");  
       WriteLine("</tr>");
     }
   }
@@ -1416,6 +1424,25 @@
   return;
 }
 
+function ::UserMoveSysVarBuildTable() {                                  
+  integer uid = system.GetSessionVar("sessionUserID");                   
+  object oUser = dom.GetObject( uid );                                   
+  if( (oUser.UserLevel() != iulGuest) && (oUser.UserLevel() != iulNone) )                                            
+  {                                                                                                                   
+    integer targetUID = system.GetVar("userId");                         
+    if( (targetUID == uid) || (oUser.UserLevel() == iulAdmin) )          
+    {                                                                    
+      object user = dom.GetObject(targetUID);                            
+      if (user)                                                          
+      {                                                                  
+        user.UserSharedObjects().MoveObject(system.GetVar("svId"),system.GetVar("svMoveDirection"));
+        Call("system.fn::BuildUserSvTable()");                                                      
+      }                                                                                             
+    }                                                                                               
+  }                                                                                                 
+  return;                                                                                                        
+} 
+
 ! [alPC], [alPDA]
 function ::setAutoLogin() {
  object oUser = dom.GetObject(system.GetSessionVar("sessionUserID"));
--- occu/WebUI/www/rega/esp/system.htm.orig
+++ occu/WebUI/www/rega/esp/system.htm
@@ -29,6 +29,7 @@
   if( action == "BuildUserSvTable"           ) { Call( "system.fn::BuildUserSvTable()" );           }
   if( action == "AddSysVarAndBuildTable"     ) { Call( "system.fn::AddSysVarAndBuildTable()" );     }
   if( action == "UserDeleteSysVarBuildTable" ) { Call( "system.fn::UserDeleteSysVarBuildTable()" ); }
+  if( action == "UserMoveSysVarBuildTable"   ) { Call( "system.fn::UserMoveSysVarBuildTable()" );   }
   if( action == "setAutoLogin"               ) { Call( "system.fn::setAutoLogin()" );               }
   if( action == "getLastTimeOfChn"           ) { Call( "system.fn::getLastTimeOfChn()" );           }
   if( action == "checkName"                  ) { Call( "system.fn::checkName()");                   }
--- occu/WebUI/www/webui/style.css.orig
+++ occu/WebUI/www/webui/style.css
@@ -1389,6 +1389,30 @@
   cursor:pointer;
 }
 
+.tSysvarMoveUp
+{
+  background-image: url(/ise/img/up.png);
+  background-position: center;
+  height: 15px;
+  width: 20px;
+  background-size: contain;
+  background-repeat: round;
+  cursor: pointer;
+  mix-blend-mode: darken;
+}
+
+.tSysvarMoveDown
+{
+  background-image: url(/ise/img/down.png);
+  background-position: center;
+  height: 15px;
+  width: 20px;
+  background-size: contain;
+  background-repeat: round;
+  cursor: pointer;
+  mix-blend-mode: darken;
+}
+
 .oLine
 {
   background-color: $_(background); /* #183473 */
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -25920,6 +25920,27 @@
     new Ajax.Updater("userSysVarTbl", url, opts);
   },
   
+  MoveSysVar: function(uid, svId, dir) {
+    var url = "/esp/system.htm?sid=" + SessionId;
+    var pb = "string action = 'UserMoveSysVarBuildTable';";
+    pb += "integer userId = "+ uid +";";
+    pb += "integer svId = " + svId + ";";
+    pb += "integer svMoveDirection = " + dir + ";";
+    var opts = {
+      postBody: ReGa.encode(pb),
+      onSuccess: function(t) {
+        // funktioniert seltsamerweise nur durch den "Updater"-Aufruf nicht
+        $("userSysVarTbl").innerHTML = t.responseText;
+      },
+      onComplete: function(response) {
+        if (200 == response.status) {
+          translatePage("#userSysVarTbl");
+        }
+      }
+    };
+    new Ajax.Updater("userSysVarTbl", url, opts);
+  },
+    
   buildTmpSysVarTable: function(arSysVars) {
     var url = "/esp/system.htm?sid="+SessionId;
     var pb = "string action = 'BuildUserSvTable';";
