--- occu/WebUI/www/config/cp_software.cgi.orig
+++ occu/WebUI/www/config/cp_software.cgi
@@ -247,7 +247,15 @@
       set scripts ""
       set loop -1
       catch { set scripts [glob /etc/config/rc.d/*] }
-      foreach s $scripts {
+      array set scriptsArray ""                                           
+      foreach s $scripts {                                                                             
+        array set sw_info ""
+        get_info $s sw_info                      
+        if { ![info exists sw_info(Name)] } continue                             
+        set scriptsArray([string tolower $sw_info(Name)]) $s
+      }                                      
+      foreach name [lsort [array names scriptsArray]] {
+        set s $scriptsArray($name)  
         incr loop;
         catch {
           if { ! [file executable $s] } continue
