--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -37861,7 +37861,8 @@
 
 ProofAndSetValue = function(srcid, dstid, min, max, dstValueFactor, event)
 {
-  var valElm = $(srcid);
+  var srcElm = $(srcid);
+  var dstElm = $(dstid);
 
   // Falls das Tasten-Event nicht mit übergeben wurde ....
   var keyCode = 0,
@@ -37880,15 +37881,14 @@
   var value = $F(srcid);
 
   // Check if float is allowed
-  if (min.indexOf(".") == -1 && max.indexOf(".") == -1) {
+  if (min.toString().indexOf(".") == -1 && max.toString().indexOf(".") == -1) {
     min = parseInt(min);
     max = parseInt(max);
     value = (roundValue05(parseInt(value))).toString();
-    valElm.value = value;
   } else {
     min = parseFloat(min);
     max = parseFloat(max);
-    valElm.value = parseFloat(value);
+    srcElm.value = parseFloat(value);
   }
 
   //replace , by .
@@ -37901,7 +37901,7 @@
     value += '.';
     if (tokens[1]) value += tokens[1];
 
-    valElm.value = value;
+    srcElm.value = value;
   }
 
 
@@ -37940,24 +37940,23 @@
 
   if (ok)
   {
-    //valElm.style.backgroundColor = "white"; // problem with firefox 69.0.3 (64-bit)
-    valElm.style.backgroundColor = "#fffffe";
-    valElm.value = value * dstValueFactor;
-    valElm.setAttribute("valvalid", "true");
+    //srcElm.style.backgroundColor = "white"; // problem with firefox 69.0.3 (64-bit)
+    srcElm.style.backgroundColor = "#fffffe";
+    dstElm.value = value * dstValueFactor;
+    srcElm.setAttribute("valvalid", "true");
     // Cursortasten abfangen, ansonsten springt der Cursor im Texteingabefeld
     // beim IE (Version 8 u. 9) mit jedem Druck auf eine Cursortaste ans Ende des Wertes.
     // Man kann nicht mittels Cursor-Links nach links wandern, da der Cursor immer ans Ende springt.
     // [HM-1293]
     if ((keyCode) < 37 && (keyCode > 40) ) {    
-      valElm.value = value;
+      srcElm.value = value;
     }
   }
   else
   {
-    valElm.setAttribute("valvalid", "false");
-    valElm.style.backgroundColor = "red";
-    valElm.value = finalVal * dstValueFactor;
-    window.setTimeout(function(){valElm.style.backgroundColor = "white";},1000);
+    srcElm.setAttribute("valvalid", "false");
+    srcElm.style.backgroundColor = "red";
+    dstElm.value = finalVal * dstValueFactor;
   }
 };
 
