--- occu/WebUI/www/webui/js/lang/de/translate.lang.extension.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.extension.js
@@ -82,13 +82,21 @@
     "generalDeviceChannelConfigLblLogged" : "Protokolliert:",
     "generalDeviceChannelConfigLblServiceMsg": "Servicemeldungen:",
     "generalDeviceChannelConfigLblFuncTest" : "Funktionstest",
+    "generalDeviceChannelConfigLblRenameChannel" : "Zugeh%F6rige Kan%E4le umbenennen",
+    "generalDeviceChannelConfigLblRenameChannelIncludingOwn" : "inklusive eigener Namen:",
+    "generalDeviceChannelConfigLblSeparator": "Trennzeichen:",
     "generalChannelConfigLblRooms" : "R%E4ume",
     "generalChannelConfigLblFunctions" : "Gewerke",
     "generalChannelConfigLblSender" : "Sender (Sensor)",
     "generalChannelConfigLblReceiver" : "Empf%E4nger (Aktor)",
     "generalChannelConfigLblNone" : "nicht verkn%FCpfbar",
     "generalDeviceChannelConfigBtnFuncTest" : "Test starten",
+    "generalDeviceChannelConfigBtnRenameChannels" : "Kan%E4le umbenennen",
+    "generalDeviceRenameChannelHint" : "Benennt alle Kan%E4le des Ger%E4ts um.<br/><br/>Die Kanalnamen lauten anschlie%DFend: <i>Ger%E4tename%26lt;Trennzeichen%26gt;Kanalnummer</i>",
+    "generalDeviceRenameChannelHintTooltip" : "Benennt alle Kan%E4le des Ger%E4ts um. Die Kanalnamen lauten anschlie%DFend: 'Ger%E4tename<Trennzeichen>Kanalnummer'",
+    "generalDeviceRenameChannelIncludingOwnHint" : "Ist diese Option aktiviert, werden bereits vorher selbst vergebene Kanalnamen mit umbenannt.",
     "generalDeviceConfigHint" : "Im Rahmen des Funktionstests wird gepr%FCft, ob die Kommunikation mit dem Ger%E4t fehlerfrei funktioniert.<br /> Dazu werden an alle mit dem Ger%E4t verkn%FCpften Aktoren Schaltbefehle gesendet, die deren Zustand %E4ndern. Sensoren, wie z.B. Fernbedienungen, melden sich im Allgemeinen erst dann, wenn sie manuell bet%E4tigt werden. Der Test gilt als bestanden, sobald die erste R%FCckmeldung von dem Ger%E4t empfangen wurde.",
+    "generalDeviceConfigHintTooltip" : "Im Rahmen des Funktionstests wird gepr%FCft, ob die Kommunikation mit dem Ger%E4t fehlerfrei funktioniert. Dazu werden an alle mit dem Ger%E4t verkn%FCpften Aktoren Schaltbefehle gesendet, die deren Zustand %E4ndern. Sensoren, wie z.B. Fernbedienungen, melden sich im Allgemeinen erst dann, wenn sie manuell bet%E4tigt werden. Der Test gilt als bestanden, sobald die erste R%FCckmeldung von dem Ger%E4t empfangen wurde.",
     "generalChannelConfigHint" : "<p>Im Rahmen des Funktionstests wird gepr%FCft, ob die Kommunikation mit dem Kanal fehlerfrei funktioniert.</p>",
     "generalChannelConfigHintSender" : "<p>Bei Sensoren wartet die Homematic Zentrale, bis diese sich melden. Eine Fernbedienung meldet sich z.B. erst dann, wenn sie manuell bet%E4tigt wird.</p>",
     "generalChannelConfigHintReceiver" : "<p>Bei Aktoren wird dazu in der Regel ein Schaltbefehl ausgel%F6st.</p>",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.extension.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.extension.js
@@ -81,13 +81,21 @@
     "generalDeviceChannelConfigLblVisible": "Visible:",
     "generalDeviceChannelConfigLblLogged": "Logged:",
     "generalDeviceChannelConfigLblFuncTest": "Functional test",
+    "generalDeviceChannelConfigLblRenameChannel" : "Rename associated channels",
+    "generalDeviceChannelConfigLblRenameChannelIncludingOwn" : "including own names:",
+    "generalDeviceChannelConfigLblSeparator": "Separator:",
     "generalChannelConfigLblRooms": "Rooms",
     "generalChannelConfigLblFunctions": "Functions",
     "generalChannelConfigLblSender": "Transmitter (sensor)",
     "generalChannelConfigLblReceiver": "Receiver (actuator)",
     "generalChannelConfigLblNone": "no connection possible",
     "generalDeviceChannelConfigBtnFuncTest": "Start test",
+    "generalDeviceChannelConfigBtnRenameChannels" : "Rename channels",
+    "generalDeviceRenameChannelHint" : "Renames all channels of the device according to a given pattern.<br/><br/>The new channel names will afterwards look like: <i>device_name%26lt;separator%26gt;channel_number</i>",
+    "generalDeviceRenameChannelHintTooltip" : "Renames all channels of the device according to a given pattern. The new channel names will afterwards look like: 'device_name%26lt;separator%26gt;channel_number'",
+    "generalDeviceRenameChannelIncludingOwnHint" : "If this option is selected, all channels will be renamed including manually renamed ones.",
     "generalDeviceConfigHint": "During the functional test the error-free communication to the device is checked.<br /> Therefore, switching commands will be send to all actuators connected to the device. Sensors (e.g. remote controls) are usually sending signals only if they are operated manually. The test is passed as soon as the first feedback will be received by the device.",
+    "generalDeviceConfigHintTooltip": "During the functional test the error-free communication to the device is checked. Therefore, switching commands will be send to all actuators connected to the device. Sensors (e.g. remote controls) are usually sending signals only if they are operated manually. The test is passed as soon as the first feedback will be received by the device.",
     "generalChannelConfigHint": "<p>During the functional test the error-free communication to the channel is checked.</p>",
     "generalChannelConfigHintSender": "<p>In connection with sensors the Homematic Central Control Unit waits until the sensors are sending. A remote control is sending only if it is operated manually.</p>",
     "generalChannelConfigHintReceiver": "<p>In connection with actuators usually switching commands will be send.</p>",
--- occu/WebUI/www/webui/style.css.orig
+++ occu/WebUI/www/webui/style.css
@@ -3194,7 +3194,7 @@
   top: 50%;
   left: 50%;
   width: 800px;
-  height: 450px;
+  height: 470px;
   margin-left: -400px;
   margin-top: -225px;
   background-color: $_(black); /* black */
@@ -3220,7 +3220,7 @@
   top: 24px;
   left: 2px;
   width: 796px;
-  height: 371px;
+  height: 391px;
   background-color: $_(white); /* white */
   color: $_(textColorB); /* black */
 /*  overflow: scroll;   */
@@ -3229,7 +3229,7 @@
 
 #DeviceConfigDialogFooter {
   position: absolute;
-  top: 398px;
+  top: 418px;
   left: 2px;
   width: 796px;
   height: 50px;
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -257,12 +257,42 @@
         <table border=\"0\"  class=\"DeviceConfigDialogTable\" width=\"250px\">\n
           <tr>\n
             <td width=\"50%\"><div id=\"DeviceConfigDialogTestButton\" class=\"StdButton\" name=\"generalDeviceChannelConfigBtnFuncTest\" onclick=\"DeviceConfigDialog.startTest();\">Test starten<\/div><\/td>\n
-            <td width=\"50%\"><div id=\"DeviceConfigDialogTestResult\">--:--:--<\/div><\/td>\n
+            <td width=\"40%\"><div id=\"DeviceConfigDialogTestResult\">--:--:--<\/div><\/td>\n
+\n
+			<td width=\"40%\"><div id=\"DeviceConfigDialogTestHint\"><img id=\"DeviceConfigDialogTestHintImg\" src=\"/ise/img/help.png\" style=\"cursor: pointer; width:18px; height:18px; position:relative; top:2px\" onclick=\"showParamHelp('generalDeviceConfigHint', '400', '200')\"><\/div><\/td>\n
+\n
           <\/tr>\n
         <\/table>\n
-        <div class=\"CLASS10805\" name=\"generalDeviceConfigHint\">\n
-          Im Rahmen des Funktionstests wird geprüft, ob die Kommunikation mit dem Gerät fehlerfrei funktioniert. Der Test gilt als bestanden, sobald die erste Rückmeldung von dem Gerät empfangen wurde. <br \/> Dazu werden an alle Aktoren des Geräts Schaltbefehle gesendet, die deren Zustand ändern. Sensoren, wie z.B. Fernbedienungen, melden sich im Allgemeinen erst dann, wenn sie durch ein entsprechendes Ereignis ausgelöst wurden.\n
-        <\/div>\n
+      <\/div>\n
+    <\/div>\n
+	\n
+    <div id=\"deviceFunctionTestPanel\" class=\"DeviceConfigDialogSection\">\n
+      <div class=\"CLASS10804\" name=\"generalDeviceChannelConfigLblRenameChannel\">Funktionstest<\/div>\n
+      <hr \/>\n
+      <div>\n
+        <table border=\"0\"  class=\"DeviceConfigDialogTable\" width=\"250px\">\n
+          <tr>\n
+		    <td width=\"50%\"><div id=\"DeviceConfigDialogRenameChannelButton\" class=\"StdButton\" name=\"generalDeviceChannelConfigBtnRenameChannels\" onclick=\"DeviceConfigDialog.renameChannels();\">Kanäle umbenennen<\/div><\/td>\n
+\n
+			<td width=\"40%\"><\/td>\n
+\n
+			<td width=\"40%\"><div id=\"DeviceConfigDialogRenameChannelButtonHint\"><img id=\"DeviceConfigDialogRenameChannelButtonHintImg\" src=\"/ise/img/help.png\" style=\"cursor: pointer; width:18px; height:18px; position:relative; top:2px\" onclick=\"showParamHelp('generalDeviceRenameChannelHint', '400', '100')\"><\/div><\/td>\n
+\n
+          <\/tr>\n
+\n
+		  <tr>			\n
+		    <td width=\"80%\" name=\"generalDeviceChannelConfigLblSeparator\">Trennzeichen:<\/td><td width=\"20%\"><input id=\"DeviceConfigDialog_DeviceChannelsRenameSeparator\" size=\"2\" type=\"text\" value=\":\"\/><\/td>\n
+          <\/tr>\n
+\n
+		  <tr>\n
+\n
+		    <td width=\"50%\" name=\"generalDeviceChannelConfigLblRenameChannelIncludingOwn\">inkl. eigener Namen:<\/td><td><input id=\"DeviceConfigDialog_renameChannelIncludingOwn\" type=\"checkbox\" \/><\/td>\n
+\n
+			<td width=\"50%\"><div id=\"DeviceConfigDialogRenameIncludingOwnHint\"><img id=\"DeviceConfigDialogRenameIncludingOwnHintImg\" src=\"/ise/img/help.png\" style=\"cursor: pointer; width:18px; height:18px; position:relative; top:2px\" onclick=\"showParamHelp('generalDeviceRenameChannelIncludingOwnHint', '400', '100')\"><\/div><\/td>\n
+\n
+		  <\/tr>\n
+\n
+        <\/table>\n
       <\/div>\n
     <\/div>\n
 \n
@@ -13067,11 +13097,13 @@
 DeviceConfigDialog = Singleton.create({
   LAYER_ID: "DeviceConfigDialogLayer",
   NAME_ID: "DeviceConfigDialog_DeviceName",
+  SEPARATOR_ID: "DeviceConfigDialog_DeviceChannelsRenameSeparator",
   USABLE_ID: "DeviceConfigDialog_isUsable",
   VISIBLE_ID: "DeviceConfigDialog_isVisible",
   LOGGED_ID: "DeviceConfigDialog_isLogged",
   SERVICEMSG_ID: "DeviceConfigDialog_enabledServiceMsg",
   TEST_RESULT_ID: "DeviceConfigDialogTestResult",
+  RENAME_OWN_ID: "DeviceConfigDialog_renameChannelIncludingOwn",
   RESULT_OK: 1,
   RESULT_ABORT: 0,
   POLL_INTERVAL: 3,
@@ -13119,6 +13151,9 @@
       this.__hideLogging();
     }
     translateJSTemplate("#DeviceConfigDialog");
+	jQuery("#DeviceConfigDialogTestHintImg").attr('title', translateKey("generalDeviceConfigHintTooltip"));
+	jQuery("#DeviceConfigDialogRenameChannelButtonHintImg").attr('title', translateKey("generalDeviceRenameChannelHintTooltip"));
+	jQuery("#DeviceConfigDialogRenameIncludingOwnHintImg").attr('title', translateKey("generalDeviceRenameChannelIncludingOwnHint"));
   },
 
   __hideFunctionTest: function() {
@@ -13179,6 +13214,35 @@
   },
   
   /**
+  * Benennt alle Channel des Device um
+  **/
+
+  ready: function()
+  {
+    this.ok();
+  },
+  
+  renameChannels: function() {
+    if (confirm("Alle Kanäle umbenennen?")) {	 
+        var renameOwnChannelNames = $(this.RENAME_OWN_ID).checked;	
+		
+	    this.device.channels.each(function(channel) {
+	      var chNumber = channel.address.split(':')[1];
+		  var origChName = this.device.name+":"+chNumber;
+		  
+		  //Channel bereits umbenannt und Option "eigene Namen umbenennen" wurde deaktiviert
+		  if (channel.name !== origChName && renameOwnChannelNames === false) return true;
+			  
+          var chName = $(this.NAME_ID).value+ $(this.SEPARATOR_ID).value + chNumber;
+	      homematic("Channel.setName", {id: channel.id, name: chName});
+	      channel.setName(chName);
+	    }, this);
+	    DeviceListPage.updateView(true);
+	    DeviceList.reload(this);
+    }		
+  },
+  
+  /**
    * Startet den Funktionstest.
    **/
   startTest: function() {
