--- occu/WebUI/www/config/ic_selchannel.cgi.orig
+++ occu/WebUI/www/config/ic_selchannel.cgi
@@ -318,7 +318,8 @@
       #puts "descrElem.attr(\"value\",translateString(descrElem.val()));"
       puts "jQuery(descrElem).val(translateString(descrElem.val()));"
 
-      puts "groupDescrElem.attr(\"value\",translateString(groupDescrElem.val()));"
+      #puts "groupDescrElem.attr(\"value\",translateString(groupDescrElem.val()));"
+      puts "jQuery(groupDescrElem).val(translateString(groupDescrElem.val()));"
     puts "</script>"
 
     #Warnung vor dem Überschreiben von Verknüpfungen----------------------------------
