--- occu/WebUI/www/api/methods/system/describe.tcl.orig
+++ occu/WebUI/www/api/methods/system/describe.tcl
@@ -29,7 +29,7 @@
   append result "\{"
   append result "\"name\": [json_toString $name]," 
   append result "\"level\": [json_toString $method(LEVEL)]," 
-  append result "\"summary\": [json_toString $method(INFO)],"
+  append result "\"summary\": [json_toString $method(INFO)]"
   append result "\}"
 }
 
