--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -13636,7 +13636,7 @@
     firewallPolicy = this.m_portListBox.getSelectedItem().id,
     hmscriptAccess = this.m_hmscriptListBox.getSelectedItem().id,
     mediolaAccess = this.m_mediolaListBox.getSelectedItem().id,
-    ips = this.m_ipTextArea.getText(),
+    ips = this.m_ipTextArea.getText().trim(),
     userPorts = this.m_portTextArea.getText().replace(/\s+/g, '').split(";");
 
     // Remove a trailing ; at the end of ips
