--- occu/WebUI/www/rega/pages/tabs/control/function.fn.orig
+++ occu/WebUI/www/rega/pages/tabs/control/function.fn
@@ -29,7 +29,7 @@
   if ((chn.HssType() == "PASSAGE_DETECTOR_DIRECTION_TRANSMITTER") && (chn.Label() == "HmIP-SPDR")) {
     ! Show the device address instead of the channel address because channel 2 and 3 are summarized together in channel 2.
     ! Channel 3 is not visible.
-    Write(chn.Label()#"<br/>&nbsp;"#device.Address()#"&nbsp;");
+    Write(chn.Label()#"<br/>&nbsp;"#oDev.Address()#"&nbsp;");
     showExtDesc = true;
   }
 
