--- occu/WebUI/www/config/cp_add_device.cgi.orig
+++ occu/WebUI/www/config/cp_add_device.cgi
@@ -712,8 +712,12 @@
     }
     puts {
       OnInbox = function() {
-        rf_install_mode(false);
-        hmip_install_mode_stop();
+        if (jQuery("body").data("BidCos-RF")) {
+          rf_install_mode(false);
+        }
+        if (jQuery("body").data("HmIP-RF")) {
+          hmip_install_mode_stop();
+        }
         PopupClose();
         WebUI.enter(NewDeviceListPage, {"fromTeachIn":true});
       }
