--- occu/WebUI/www/rega/esp/rule.inc.orig
+++ occu/WebUI/www/rega/esp/rule.inc
@@ -41,7 +41,7 @@
                     {
                       if( sSingleIndex.ToInteger() > 0 )
                       {
-                        if( oSC.OperatorType() == 0 ) { Write( "<div><b>?</b></div>" ); }
+                        if( oSC.OperatorType() == 0 ) { Write( "<div><b>${ruleSelectOR}</b></div>" ); }
                         if( oSC.OperatorType() == 1 ) { Write( "<div><b>${ruleSelectAND}</b></div>" ); }
                         if( oSC.OperatorType() == 2 ) { Write( "<div><b>${ruleSelectOR}</b></div>" ); }
                       }
@@ -75,7 +75,7 @@
               if( sCondIndex.ToInteger() < (iRuleCondCount-1) )
               {
                 Write( '<div class="CLASS02300">' );
-                if( oC.CndOperatorType() == 0 ) { Write( '?' ); }
+                if( oC.CndOperatorType() == 0 ) { Write( '${ruleSelectOR}' ); }
                 if( oC.CndOperatorType() == 1 ) { Write( '${ruleSelectAND}' ); }
                 if( oC.CndOperatorType() == 2 ) { Write( '${ruleSelectOR}' ); }
                 Write( '</div>' );
