--- occu/WebUI/www/rega/esp/controls/rgbw.fn.orig
+++ occu/WebUI/www/rega/esp/controls/rgbw.fn
@@ -197,7 +197,10 @@
   if (oRGB) {
     rgbID = oRGB.ID();
     rgbValue = oRGB.Value();
-    rgbDefaultValue = oRGB.MetaData("DEFAULT").ToString();
+    var oRGBMetaData = oRGB.MetaData("DEFAULT");
+    if (oRGBMetaData) {
+      rgbDefaultValue = oRGBMetaData.ToString();
+    }
   }
 
   WriteLine("<tr>");
