--- occu/WebUI/www/api/methods.conf.orig
+++ occu/WebUI/www/api/methods.conf
@@ -1062,6 +1062,20 @@
   ARGUMENTS {_session_id_}
 }
 
+CCU.getSystemLanguage {
+  LEVEL ADMIN
+  SCRIPT_FILE ccu/getsystemlanguage.tcl
+  INFO {Gibt die eingestellte Systemsprache zurueck}
+  ARGUMENTS {_session_id_}
+}
+
+CCU.setSystemLanguage {
+  LEVEL ADMIN
+  SCRIPT_FILE ccu/setsystemlanguage.tcl
+  INFO {Setzt die Systemsprache in /etc/config/systemLanguage}
+  ARGUMENTS {_session_id_ lang}
+}
+
 CCU.restartSSHDaemon {
   LEVEL ADMIN
   SCRIPT_FILE ccu/restartsshdaemon.tcl
--- occu/WebUI/www/api/methods/ccu/getsystemlanguage.tcl.orig
+++ occu/WebUI/www/api/methods/ccu/getsystemlanguage.tcl
@@ -0,0 +1,21 @@
+##
+# CCU.getSystemLanguage
+# Gibt die momentan eingestellte Systemsprache der CCU aus der
+# /etc/config/systemLanguage Datei zurueck.
+#
+# Parameter:
+#  keine
+#
+# Rückgabewert: [string]
+#  Kuerzel der Systemsprache (de, en) oder none
+##
+
+set lang "none"
+if {[catch {set fp [open /etc/config/systemLanguage r]}] == 0} {
+  if { $fp >= 0 } {
+    set lang [string trim [read $fp]]
+    close $fp
+  }
+}
+
+jsonrpc_response [json_toString $lang]
--- occu/WebUI/www/api/methods/ccu/setsystemlanguage.tcl.orig
+++ occu/WebUI/www/api/methods/ccu/setsystemlanguage.tcl
@@ -0,0 +1,207 @@
+##
+# CCU.setSystemLanguage
+# Setzt die Systemsprache der CCU in /etc/config/systemLanguage
+# und benennt die Namen aller Gewerke, Raeume und Standard-Systemvariablen
+# in die jeweilige Sprache um.
+#
+# Parameter:
+#  lang         : [string] Die gewaehlte Systemsprache (de, en)
+#
+# Rückgabewert: immer true
+##
+
+proc resetSystemLanguage {a} {
+
+  array set args $a
+
+  # define rega script
+  set script {
+
+    string oID;
+    string metaDataFactoryName = "FACTORY_NAME";
+  
+    object funcs = dom.GetObject(ID_FUNCTIONS);
+    object rooms = dom.GetObject(ID_ROOMS);
+    object sv    = dom.GetObject(ID_SYSTEM_VARIABLES);
+  
+    foreach(oID, funcs.EnumIDs()) {
+      object oObj = dom.GetObject(oID);
+      var meta = oObj.MetaData(metaDataFactoryName);
+      if (meta) {
+        string newName = meta;
+        oObj.Name(newName);
+        oObj.RemoveMetaData(metaDataFactoryName);
+      }
+    }
+
+    foreach(oID, rooms.EnumIDs()) {
+      object oObj = dom.GetObject(oID);
+      var meta = oObj.MetaData(metaDataFactoryName);
+      if (meta) {
+        string newName = meta;
+        oObj.Name(newName);
+        oObj.RemoveMetaData(metaDataFactoryName);
+      }
+    }
+
+    foreach(oID, sv.EnumIDs()) {
+      object oObj = dom.GetObject(oID);
+      var meta = oObj.MetaData(metaDataFactoryName);
+      if (meta) {
+        string newName = meta;
+        oObj.Name(newName);
+        oObj.RemoveMetaData(metaDataFactoryName);
+      }
+  
+      var metaValueName0 = oObj.MetaData(metaDataFactoryName#"_VALUE_NAME_0");
+      if (metaValueName0) {
+        string newName = metaValueName0;
+        oObj.ValueName0(newName);
+        oObj.RemoveMetaData(metaDataFactoryName#"_VALUE_NAME_0");
+      }
+  
+      var metaValueName1 = oObj.MetaData(metaDataFactoryName#"_VALUE_NAME_1");
+      if (metaValueName1) {
+        string newName = metaValueName1;
+        oObj.ValueName1(newName);
+        oObj.RemoveMetaData(metaDataFactoryName#"_VALUE_NAME_1");
+      }
+    }
+    
+    Write("true");
+  }
+
+  set result [hmscript $script args]
+  return $result
+}
+
+proc setSystemLanguage {a} {
+
+  array set args $a
+
+  # define rega script
+  set script {
+    string funcDefault    = "funcLight;funcHeating;funcClimateControl;funcWeather;funcEnvironment;funcSecurity;funcLock;funcButton;funcCentral;funcEnergy";
+    string funcTranslated = "Licht;Heizung;Klima;Wetter;Umwelt;Sicherheit;Verschluss;Taster;Zentrale;Energiemanagement";
+
+    string roomDefault = "roomLivingRoom;roomKitchen;roomBedroom;roomChildrensRoom1;roomChildrensRoom2;roomOffice;roomBathroom;roomGarage;roomHWR;roomGarden;roomTerrace";
+    string roomTranslated = "Wohnzimmer;Küche;Schlafzimmer;Kinderzimmer 1;Kinderzimmer 2;Büro;Badezimmer;Garage;Hauswirtschaftsraum;Garten;Terrasse";
+	
+    string svDefault   = "sysVarAlarmZone1;sysVarAlarmZone1Msg;sysVarAlarmZone1Triggered;sysVarAlarmZone1NotTriggered;sysVarPresence;sysVarPresenceMsg;sysVarPresencePresent;sysVarPresenceNotPresent;sysVarAlarmMessages;sysVarAlarmMessagesCount;sysVarServiceMessages;sysVarServiceMessagesCount;sysVarRainToday;sysVarRainTodayDescr;sysVarRainYesterday;sysVarRainYesterdayDescr";
+    string svTranslated   = "Alarmzone 1;Alarmmeldung Alarmzone 1;ausgelöst;nicht ausgelöst;Anwesenheit;Anwesenheit;anwesend;nicht anwesend;Alarmmeldungen;Anzahl Alarmmeldungen;Servicemeldungen;Anzahl Servicemeldungen;Regen heute;Regenmenge aktueller Tag;Regen gestern;Regenmenge vorheriger Tag";
+
+    if (lang == "en") {
+      funcTranslated = "Light;Heating;Climatic conditions;Weather;Environment;Security;Lock;Button;Central;Energy management";
+      roomTranslated = "Living room;Kitchen;Bed room;Childrens room 1;Childrens room 2;Home office;Bathroom;Garage;Utility room;Garden;Terrace";
+      svTranslated   = "Alarm zone 1;Alarm message alarm zone 1;triggered;not triggered;Presence;Presence;present;absent;Alarm messages;Number alarm messages;Service messages;Number service messages;Rain today;Rainfall current day;Rainfall yesterday;Rainfall previous day";
+    }
+
+    string oID;
+    string metaDataFactoryName = "FACTORY_NAME";
+  
+    object funcs = dom.GetObject(ID_FUNCTIONS);
+    object rooms = dom.GetObject(ID_ROOMS);
+    object sv    = dom.GetObject(ID_SYSTEM_VARIABLES);
+  
+    ! Functions
+    foreach(oID, funcs.EnumIDs()) {
+      object oObj = dom.GetObject(oID);
+      string listEntry;
+      integer idx = 0;
+      foreach(listEntry, funcDefault.Split(";")) {
+        if ((oObj.Name() == listEntry) || (oObj.Name() == "${"#listEntry#"}")) {
+          var meta = oObj.MetaData(metaDataFactoryName);
+          if (!meta) {
+            string newName = funcTranslated.StrValueByIndex(";", idx);
+            oObj.RemoveMetaData(metaDataFactoryName);
+            oObj.AddMetaData(metaDataFactoryName, oObj.Name());
+            oObj.Name(newName);
+          }
+        }
+        idx = idx + 1;
+      }
+    }
+  
+    ! Rooms
+    foreach(oID, rooms.EnumIDs()) {
+      object oObj = dom.GetObject(oID);
+      string listEntry;
+      integer idx = 0;
+      foreach(listEntry, roomDefault.Split(";")) {
+        if ((oObj.Name() == listEntry) || (oObj.Name() == "${"#listEntry#"}")) {
+          var meta = oObj.MetaData(metaDataFactoryName);
+          if (!meta) {
+            string newName = roomTranslated.StrValueByIndex(";", idx);
+            oObj.RemoveMetaData(metaDataFactoryName);
+            oObj.AddMetaData(metaDataFactoryName, oObj.Name());
+            oObj.Name(newName);
+          }
+        }
+        idx = idx + 1;
+      }
+    }
+
+    !Sysvars
+    foreach(oID, sv.EnumIDs()) {
+      object oObj = dom.GetObject(oID);
+      string listEntry;
+      integer idx = 0;
+      foreach(listEntry, svDefault.Split(";")) {
+        if ((oObj.Name() == listEntry) || (oObj.Name() == "${"#listEntry#"}")) {
+          var meta = oObj.MetaData(metaDataFactoryName);
+          if (!meta) {
+            string newName = svTranslated.StrValueByIndex(";", idx);
+            oObj.RemoveMetaData(metaDataFactoryName);
+            oObj.AddMetaData(metaDataFactoryName, oObj.Name());
+            oObj.Name(newName);
+          }
+        }
+  
+        if ((oObj.ValueName0() == listEntry) || (oObj.ValueName0() == "${"#listEntry#"}")) {
+          var meta = oObj.MetaData(metaDataFactoryName#"_VALUE_NAME_0");
+          if (!meta) {
+            string newValueName0 = svTranslated.StrValueByIndex(";", idx);
+            oObj.RemoveMetaData(metaDataFactoryName#"_VALUE_NAME_0");
+            oObj.AddMetaData(metaDataFactoryName#"_VALUE_NAME_0", oObj.ValueName0());
+            oObj.ValueName0(newValueName0);
+          }
+        }
+  
+        if ((oObj.ValueName1() == listEntry) || (oObj.ValueName1() == "${"#listEntry#"}")) {
+          var meta = oObj.MetaData(metaDataFactoryName#"_VALUE_NAME_1");
+          if (!meta) {
+            string newValueName1 = svTranslated.StrValueByIndex(";", idx);
+            oObj.RemoveMetaData(metaDataFactoryName#"_VALUE_NAME_1");
+            oObj.AddMetaData(metaDataFactoryName#"_VALUE_NAME_1", oObj.ValueName1());
+            oObj.ValueName1(newValueName1);
+          }
+        }
+    
+       idx = idx + 1;
+      }
+    }
+
+    Write("true");
+  }
+
+  set result [hmscript $script args]
+  return $result
+}
+
+# reset the system language change
+set result [resetSystemLanguage [array get args]]
+
+# set the system language to lang
+if { $result == "true" } {
+  set result [setSystemLanguage [array get args]]
+
+  if { $result == "true" } {
+    # create /etc/config/systemLanguage with the
+    # language as a content
+    set fd [open /etc/config/systemLanguage w]
+    puts -nonewline $fd $args(lang)
+    close $fd
+  }
+}
+
+jsonrpc_response $result
--- occu/WebUI/www/rega/pages/msg/systemLanguageConfig.htm.orig
+++ occu/WebUI/www/rega/pages/msg/systemLanguageConfig.htm
@@ -0,0 +1,62 @@
+<div class="popupTitle" style='width: 98.5%;'>
+  <%
+  Write(system.GetVar("title"));
+  %>
+</div>
+
+<div class="CLASS01400">
+  <table class="popupTableScript CLASS01401" cellpadding="0" cellspacing="0">
+    <tr>
+    <th class="TblHeaderUserConfig">${thChooseSystemLanguage}</th>
+    <th class="TblHeaderUserConfig">${thChooseSystemLanguageNote}</th>
+    </tr>
+    
+    <tr>
+    
+    <td class="CLASS01402" style='text-align: center; vertical-align:middle;'>
+    <table style="margin:auto;">
+    <tr><td style='text-align: center; vertical-align:middle;'><div>
+      <select id="cbLang">
+        <option value="de">${selectLangDE}</option>
+        <option value="en">${selectLangUK}</option>
+       </select></div></td>
+       </tr>
+       </table>
+    </td>
+    
+    <td class="CLASS01403">
+
+    <span class="CLASS01404">${hintSetSystemLanguageNoteRow1}</span><br/>
+    <span class="CLASS01405">${hintSetSystemLanguageNoteRow2}</span>
+
+    </td>
+
+    </tr>
+
+    <tr class="popupTableRow">
+    <td class="CLASS01406" colspan="3">
+      <table>
+      <tr>
+       <td width="10%" style='text-align: center; vertical-align:middle;'><div class="FooterButton CLASS01408" onclick="doSystemLanguageConfig();">${footerBtnOk}</div></td>
+	   <td width="80%"></td>
+       <td width="10%" style='text-align: center; vertical-align:middle;'><div class="FooterButton CLASS01408" onclick="cancelSystemLanguageConfig();">${footerBtnCancel}</div></td>
+      </tr>
+      </table>
+    </td>
+    </tr>
+  </table>
+</div>
+<script type="text/javascript">
+  var lang = homematic('CCU.getSystemLanguage');
+  if(lang === 'de' || lang === 'en') {
+    $('cbLang').value = lang;
+  }
+  doSystemLanguageConfig = function() {
+    var selValue = $("cbLang")[$("cbLang").selectedIndex].value;
+    homematic("CCU.setSystemLanguage", {'lang': selValue});
+    PopupClose();
+  };
+  cancelSystemLanguageConfig = function() {
+    PopupClose();
+  };
+</script>
--- occu/WebUI/www/rega/pages/tabs/admin/userAdministration.htm.orig
+++ occu/WebUI/www/rega/pages/tabs/admin/userAdministration.htm
@@ -173,6 +173,7 @@
     s += "<td style='text-align: center; vertical-align: middle;'><div class='FooterButton' onclick='WebUI.goBack()'>${footerBtnPageBack}</div></td>";
     s += "<td style='text-align: center; vertical-align: middle;'><div class='FooterButton' onclick='newUser();'>${footerBtnNew}</div></td>";
     s += "<td  style='text-align:center; vertical-align: middle;'><div class='FooterButton CLASS05305' onclick='showAutoLogin()'>${footerBtnAutoLogin}</div></td>";
+    s += "<td  style='text-align:center; vertical-align: middle;'><div class='FooterButton CLASS05305' onclick='showSystemLanguage()'>${footerBtnSystemLanguage}</div></td>";
   s += "</tr>";
   s += "</table>";
 
@@ -210,6 +211,10 @@
     CreatePopup(ID_AUTO_LOGIN_CONFIG);
   }
 
+  showSystemLanguage = function() {
+    CreatePopup(ID_SYSTEM_LANGUAGE_CONFIG);
+  }
+  
   translatePage("#userAdmin");
   jQuery("#userAdmin").show();
 
--- occu/WebUI/www/webui/js/lang/de/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.js
@@ -90,6 +90,7 @@
     "footerBtnCloseTree" : "Baumstruktur schlie%DFen",
     "footerBtnEdit" : "Bearbeiten",
     "footerBtnAutoLogin" : "Automatisches<br/>Anmelden",
+    "footerBtnSystemLanguage" : "Systemsprache<br />festlegen",
     "footerBtnVirtualChannelsShow" : "Virtuelle Kan%E4le anzeigen",
     "footerBtnVirtualChannelsHide" : "Virtuelle Kan%E4le ausblenden",
     "footerBtnSaveNewFav" : "Auswahl als neue<br />Favoritenseite speichern",
@@ -159,6 +160,8 @@
     "thPhoneNumber" : "Telefonnummer",
     "thAutomaticLogin" : "Automatisches<br/>Anmelden",
     "thChooseUser" : "Benutzerauswahl",
+    "thChooseSystemLanguage" : "Systemsprache ausw%E4hlen",
+    "thChooseSystemLanguageNote" : "Hinweis",
     "thSafetyNote" : "Sicherheitshinweis",
     "thGateway" : "Gateway",
     "thAccessCode" : "Zugriffscode",
@@ -244,6 +247,8 @@
     "hintUsrAutoLoginRow2" : "Eine automatische Anmeldung erm%F6glicht die Systemanmeldung ohne Eingabe von Benutzernamen und Passwort.<br/>",
     "hintUsrAutoLoginRow3" : "Der ausgew%E4hlte Benutzer wird automatisch ohne weitere Pr%FCfung angemeldet. Das System ist somit nicht mehr gegen externen Zugriff gesch%FCtzt.<br />",
     "hintUsrAutoLoginRow4" : "Der Zugriff auf andere Benutzerkonten erfolgt %FCber 'Abmelden' und entsprechender Neuanmeldung.<br />",
+    "hintSetSystemLanguageNoteRow1" : "Bitte beachten:",
+    "hintSetSystemLanguageNoteRow2" : "Durch setzen der Systemsprache der CCU werden interne Standardbezeichnungen von R%E4umen, Gewerken, Standard-Systemvariablen, usw. auf die ausgew%E4hlte Sprache gesetzt. Bereits manuell angepasste Bezeichnungen werden von dieser %C4nderung jedoch nicht erfasst und behalten Ihre jetzigen Bezeichnungen. Eine sp%E4tere Anpassung der Systemsprache ist weiterhin jederzeit m%F6glich.",
     "profileSettings" : "Profileinstellung",
     "profileSettingsSender" : "Profileinstellung - Sender",
     "profileSettingsReceiver" : "Profileinstellung - Empf%E4nger",
@@ -420,6 +425,7 @@
     "dialogStateInfoDataTransferToDeviceSuccess" : "Statusinfo: Daten%FCbertragung erfolgreich",
     "dialogWarningDataTransferToDeviceNotPossible" : "Achtung! Daten%FCbertragung nicht m%F6glich",
     "dialogConfigAutomaticLogin" : "Konfiguration - Automatisches Anmelden",
+    "dialogConfigSystemLanguage" : "Konfiguration - Systemsprache",
     "dialogChooseBidCosInterface" : "Auswahl BidCos-Interface",
     "dialogChooseChannel" : "Kanalauswahl",
     "dialogRestartSafeModeTitle" : "Neustart",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.js
@@ -89,6 +89,7 @@
     "footerBtnCloseTree" : "Close tree structure",
     "footerBtnEdit" : "Edit",
     "footerBtnAutoLogin" : "Log in<br/>automatically",
+    "footerBtnSystemLanguage" : "Set system<br/>language",
     "footerBtnVirtualChannelsShow" : "Show virtual channels",
     "footerBtnVirtualChannelsHide" : "Hide virtual channels",
     "footerBtnSaveNewFav" : "Save selection as new<br/>favourites page",
@@ -158,6 +159,8 @@
     "thPhoneNumber" : "Phone number",
     "thAutomaticLogin" : "Log in<br/>automatically",
     "thChooseUser" : "User selection",
+    "thChooseSystemLanguage" : "Choose system language",
+    "thChooseSystemLanguageNote" : "Note",
     "thSafetyNote" : "Security alert",
     "thGateway" : "Gateway",
     "thAccessCode" : "Access code",
@@ -243,6 +246,8 @@
     "hintUsrAutoLoginRow2" : "If automatic login is activated you will be logged-in to the system without entering your user name or password.<br/>",
     "hintUsrAutoLoginRow3" : "The selected user will be logged-in automatically without further validation. The system is no longer protected against external influences.<br />",
     "hintUsrAutoLoginRow4" : "You can access other user accounts with click on \"Login\". Afterwards, please login with the user name and password again.",
+    "hintSetSystemLanguageNoteRow1" : "Please note:",
+    "hintSetSystemLanguageNoteRow2" : "By choosing the system language of the CCU, internal names of rooms, functions, standard system variables, etc. will be fixed to the selected language. If you have already manually changed names, they won't be modified by this selection. In addition, choosing a different system language is still possible later on.",
     "profileSettings" : "Profile setting",
     "profileSettingsSender" : "Profile settings - transmitter",
     "profileSettingsReceiver" : "Profile settings - receiver",
@@ -419,6 +424,7 @@
     "dialogStateInfoDataTransferToDeviceSuccess" : "Status info: Data transfer successful",
     "dialogWarningDataTransferToDeviceNotPossible" : "Attention! Unable to transfer data",
     "dialogConfigAutomaticLogin" : "Configuration - automatic login",
+    "dialogConfigSystemLanguage" : "Configuration - system language",
     "dialogChooseBidCosInterface" : "Selection BidCos-Interface",
     "dialogChooseChannel" : "Channel selection",
     "dialogRestartSafeModeTitle" : "Restart",
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -7012,6 +7012,11 @@
             homematic("CCU.setUserAckInstallWizard", {'userName': usrName});
           }
         }
+
+        // open the systemLanguage config dialog if no systemLanguage was selected yet
+        if (homematic('CCU.getSystemLanguage') === 'none') {
+          CreatePopup(ID_SYSTEM_LANGUAGE_CONFIG);
+        }		
       }
     } else {
       WebUI.serviceMessageCount = setServiceMessage();
@@ -26732,7 +26737,12 @@
         this.setWidth(800);
         this.setFile( "/pages/msg/autoLoginConfig.htm" );
         break;
-       case ID_CHOOSE_LED:
+      case ID_SYSTEM_LANGUAGE_CONFIG:
+        this.setTitle( translateKey("dialogConfigSystemLanguage") /*"Systemsprache waehlen - Konfiguration" */);
+        this.setWidth(800);
+        this.setFile( "/pages/msg/systemLanguageConfig.htm" );
+        break;
+        case ID_CHOOSE_LED:
         this.setTitle( translateKey("dialogEnterValue") /*"Wert eingeben" */);
         this.addToPostBody( 'string type = "'+this.type+'";' );
         this.setWidth(800);
@@ -27551,6 +27561,7 @@
 ID_SET_OUCFM_MODE        = 1023;
 ID_SET_STATUS_DISPLAY    = 1024;
 ID_BIDCOS_INTERFACE = 1025;
+ID_SYSTEM_LANGUAGE_CONFIG = 2001;
 
 nUA = navigator.userAgent;
 NAV_IE = false;
