--- occu/WebUI/www/api/methods.conf.orig
+++ occu/WebUI/www/api/methods.conf
@@ -1062,6 +1062,13 @@
   ARGUMENTS {_session_id_}
 }
 
+CCU.setSystemLanguageConfigured {
+  LEVEL ADMIN
+  SCRIPT_FILE ccu/setsystemlanguageconfigured.tcl
+  INFO {Legt die Datei /etc/config/systemLanguageConfigured an}
+  ARGUMENTS {_session_id_}
+}
+
 CCU.restartSSHDaemon {
   LEVEL ADMIN
   SCRIPT_FILE ccu/restartsshdaemon.tcl
--- occu/WebUI/www/rega/esp/system.fn.orig
+++ occu/WebUI/www/rega/esp/system.fn
@@ -1580,3 +1580,186 @@
 function ::keepAlive()
 {
 }
+
+function ::refactorDefaultNames()
+{
+  string lang = system.GetVar("refactorLanguage");
+  
+  string funcDefault    = "funcLight;funcHeating;funcClimateControl;funcWeather;funcEnvironment;funcSecurity;funcLock;funcButton;funcCentral;funcEnergy";
+  string funcTranslated = "Licht;Heizung;Klima;Wetter;Umwelt;Sicherheit;Verschluss;Taster;Zentrale;Energiemanagement";
+  
+  string roomDefault = "roomLivingRoom;roomKitchen;roomBedroom;roomChildrensRoom1;roomChildrensRoom2;roomOffice;roomBathroom;roomGarage;roomHWR;roomGarden;roomTerrace";
+  string roomTranslated = "Wohnzimmer;Küche;Schlafzimmer;Kinderzimmer 1;Kinderzimmer 2;Büro;Badezimmer;Garage;Hauswirtschaftsraum;Garten;Terrasse";
+  
+  string svDefault   = "sysVarAlarmZone1;sysVarAlarmZone1Msg;sysVarAlarmZone1Triggered;sysVarAlarmZone1NotTriggered;sysVarPresence;sysVarPresenceMsg;sysVarPresencePresent;sysVarPresenceNotPresent;sysVarAlarmMessages;sysVarAlarmMessagesCount;sysVarServiceMessages;sysVarServiceMessagesCount;sysVarRainToday;sysVarRainTodayDescr;sysVarRainYesterday;sysVarRainYesterdayDescr";
+  string svTranslated   = "Alarmzone 1;Alarmmeldung Alarmzone 1;ausgelöst;nicht ausgelöst;Anwesenheit;Anwesenheit;anwesend;nicht anwesend;Alarmmeldungen;Anzahl Alarmmeldungen;Servicemeldungen;Anzahl Servicemeldungen;Regen heute;Regenmenge aktueller Tag;Regen gestern;Regenmenge vorheriger Tag";
+
+  if (lang == "EN") {
+    funcTranslated = "Light;Heating;Climatic conditions;Weather;Environment;Security;Lock;Button;Central;Energy management";
+    roomTranslated = "Livin room;Kitchen;Bed room;Children's room 1;Children's room 2;Home office;Bathroom;Garage;Utility room;Garden;Terrace";
+    svTranslated   = "Alarm zone 1;Alarm message alarm zone 1;triggered;not triggered;Presence;Presence;present;absent;Alarm messages;Number alarm messages;Service messages;Number service messages;Rain today;Rainfall current day;Rainfall yesterday;Rainfall previous day";
+  }
+  
+  string oID;
+  string metaDataFactoryName = "FACTORY_NAME";
+  
+  var funcs = dom.GetObject(ID_FUNCTIONS);
+  var rooms = dom.GetObject(ID_ROOMS);
+  var sv    = dom.GetObject(ID_SYSTEM_VARIABLES);
+  
+  ! Functions
+  foreach(oID, funcs.EnumIDs()) {
+    var oObj = dom.GetObject(oID);
+    string listEntry;
+    integer idx = 0;
+    foreach(listEntry, funcDefault.Split(";")) {
+      if ((oObj.Name() == listEntry) || (oObj.Name() == "${"#listEntry#"}")) {
+        var meta = oObj.MetaData(metaDataFactoryName);
+        if (!meta) {
+          string newName = funcTranslated.StrValueByIndex(";", idx);
+          oObj.RemoveMetaData(metaDataFactoryName);
+          oObj.AddMetaData(metaDataFactoryName, oObj.Name());
+          oObj.Name(newName);
+          WriteLine("old = "#oObj.Name()#", new = "#newName);
+        } else {
+          WriteLine(oObj.Name()#" already renamed");
+        }
+      }
+      idx = idx + 1;
+    }
+  }
+  
+  !Rooms
+  foreach(oID, rooms.EnumIDs()) {
+    var oObj = dom.GetObject(oID);
+    string listEntry;
+    integer idx = 0;
+    foreach(listEntry, roomDefault.Split(";")) {
+      if ((oObj.Name() == listEntry) || (oObj.Name() == "${"#listEntry#"}")) {
+        var meta = oObj.MetaData(metaDataFactoryName);
+        if (!meta) {
+          string newName = roomTranslated.StrValueByIndex(";", idx);
+          oObj.RemoveMetaData(metaDataFactoryName);
+          oObj.AddMetaData(metaDataFactoryName, oObj.Name());
+          oObj.Name(newName);
+          WriteLine("old = "#oObj.Name()#", new = "#newName);
+        } else {
+          WriteLine(oObj.Name()#" already renamed");
+        }
+      }
+      idx = idx + 1;
+    }
+  }
+  
+  !Sysvars
+  foreach(oID, sv.EnumIDs()) {
+    var oObj = dom.GetObject(oID);
+    string listEntry;
+    integer idx = 0;
+    foreach(listEntry, svDefault.Split(";")) {
+      if ((oObj.Name() == listEntry) || (oObj.Name() == "${"#listEntry#"}")) {
+        var meta = oObj.MetaData(metaDataFactoryName);
+        if (!meta) {
+          string newName = svTranslated.StrValueByIndex(";", idx);
+          oObj.RemoveMetaData(metaDataFactoryName);
+          oObj.AddMetaData(metaDataFactoryName, oObj.Name());
+          oObj.Name(newName);
+          
+          WriteLine("old = "#oObj.Name()#", new = "#newName);
+        } else {
+          WriteLine(oObj.Name()#" already renamed");
+        }
+      }
+      
+      if ((oObj.ValueName0() == listEntry) || (oObj.ValueName0() == "${"#listEntry#"}")) {
+        var meta = oObj.MetaData(metaDataFactoryName#"_VALUE_NAME_0");
+        if (!meta) {
+          string newValueName0 = svTranslated.StrValueByIndex(";", idx);
+          oObj.RemoveMetaData(metaDataFactoryName#"_VALUE_NAME_0");
+          oObj.AddMetaData(metaDataFactoryName#"_VALUE_NAME_0", oObj.ValueName0());
+          oObj.ValueName0(newValueName0);
+          
+          WriteLine("old ValueName0 = "#oObj.ValueName0()#", new = "#newValueName0);
+        } else {
+          WriteLine(oObj.ValueName0()#" already renamed");
+        }
+      }
+      
+      if ((oObj.ValueName1() == listEntry) || (oObj.ValueName1() == "${"#listEntry#"}")) {
+        var meta = oObj.MetaData(metaDataFactoryName#"_VALUE_NAME_1");
+        if (!meta) {
+          string newValueName1 = svTranslated.StrValueByIndex(";", idx);
+          oObj.RemoveMetaData(metaDataFactoryName#"_VALUE_NAME_1");
+          oObj.AddMetaData(metaDataFactoryName#"_VALUE_NAME_1", oObj.ValueName1());
+          oObj.ValueName1(newValueName1);
+          
+          WriteLine("old ValueName1 = "#oObj.ValueName1()#", new = "#newValueName1);
+        } else {
+          WriteLine(oObj.ValueName1()#" already renamed");
+        }
+      }
+      
+      idx = idx + 1;
+    }
+  }
+
+}
+
+function ::refactorDefaultNamesRollback()
+{
+  string oID;
+  string metaDataFactoryName = "FACTORY_NAME";
+  
+  var funcs = dom.GetObject(ID_FUNCTIONS);
+  var rooms = dom.GetObject(ID_ROOMS);
+  var sv    = dom.GetObject(ID_SYSTEM_VARIABLES);
+  
+  foreach(oID, funcs.EnumIDs()) {
+    var oObj = dom.GetObject(oID);
+    var meta = oObj.MetaData(metaDataFactoryName);
+    if (meta) {
+      string newName = meta;
+      WriteLine("rollback "#oObj.Name()#" to "#newName);
+      oObj.Name(newName);
+      oObj.RemoveMetaData(metaDataFactoryName);
+    }
+  }
+  
+  foreach(oID, rooms.EnumIDs()) {
+    var oObj = dom.GetObject(oID);
+    var meta = oObj.MetaData(metaDataFactoryName);
+    if (meta) {
+      string newName = meta;
+      WriteLine("rollback "#oObj.Name()#" to "#newName);
+      oObj.Name(newName);
+      oObj.RemoveMetaData(metaDataFactoryName);
+    }
+  }
+  
+  foreach(oID, sv.EnumIDs()) {
+    var oObj = dom.GetObject(oID);
+    var meta = oObj.MetaData(metaDataFactoryName);
+    if (meta) {
+      string newName = meta;
+      WriteLine("rollback "#oObj.Name()#" to "#newName);
+      oObj.Name(newName);
+      oObj.RemoveMetaData(metaDataFactoryName);
+    }
+    
+    var metaValueName0 = oObj.MetaData(metaDataFactoryName#"_VALUE_NAME_0");
+    if (metaValueName0) {
+      string newName = metaValueName0;
+      WriteLine("rollback "#oObj.ValueName0()#" to "#newName);
+      oObj.ValueName0(newName);
+      oObj.RemoveMetaData(metaDataFactoryName#"_VALUE_NAME_0");
+    }
+    
+    var metaValueName1 = oObj.MetaData(metaDataFactoryName#"_VALUE_NAME_1");
+    if (metaValueName1) {
+      string newName = metaValueName1;
+      WriteLine("rollback "#oObj.ValueName1()#" to "#newName);
+      oObj.ValueName1(newName);
+      oObj.RemoveMetaData(metaDataFactoryName#"_VALUE_NAME_1");
+    }
+  }
+}
\ No newline at end of file
--- occu/WebUI/www/rega/esp/system.htm.orig
+++ occu/WebUI/www/rega/esp/system.htm
@@ -34,4 +34,6 @@
   if( action == "getLastTimeOfChn"           ) { Call( "system.fn::getLastTimeOfChn()" );           }
   if( action == "checkName"                  ) { Call( "system.fn::checkName()");                   }
   if( action == "keepAlive"                  ) { Call( "system.fn::keepAlive()");                   }
+  if( action == "refactorDefaultNames"       ) { Call( "system.fn::refactorDefaultNames()");        }
+  if( action == "refactorDefaultNamesRollback") { Call( "system.fn::refactorDefaultNamesRollback()");}
 %>
--- occu/WebUI/www/webui/js/lang/de/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.js
@@ -91,6 +91,7 @@
     "footerBtnCloseTree" : "Baumstruktur schlie%DFen",
     "footerBtnEdit" : "Bearbeiten",
     "footerBtnAutoLogin" : "Automatisches<br/>Anmelden",
+    "footerBtnRefactorDefaultNames" : "Defaults<br />umbenennen",
     "footerBtnVirtualChannelsShow" : "Virtuelle Kan%E4le anzeigen",
     "footerBtnVirtualChannelsHide" : "Virtuelle Kan%E4le ausblenden",
     "footerBtnSaveNewFav" : "Auswahl als neue<br />Favoritenseite speichern",
@@ -161,6 +162,7 @@
     "thPhoneNumber" : "Telefonnummer",
     "thAutomaticLogin" : "Automatisches<br/>Anmelden",
     "thChooseUser" : "Benutzerauswahl",
+    "thChooseLanguage" : "Sprachauswahl",
     "thSafetyNote" : "Sicherheitshinweis",
     "thGateway" : "Gateway",
     "thAccessCode" : "Zugriffscode",
@@ -255,6 +257,7 @@
     "easyModeOptionUnknownProfile" : "Unbekanntes Profil not in use",
     "easyModeHintUseExpertMode" : "Bitte verwenden Sie hier den Expertenmodus not in use",
     "btnDirectDeviceLinks" : "Direkte Ger%E4teverkn%FCpfungen",
+    "btnRefactorRollback" : "R%FCckg%E4ngig",
     "lblDirectDeviceLinks1" : "%DCbersicht aller bestehenden direkten Ger%E4te-/ Kanalverkn%FCpfungen",
     "lblDirectDeviceLinks2" : "Direkte Ger%E4te-/ Kanalverkn%FCpfungen bearbeiten",
     "lblDirectDeviceLinks3" : "Direkte Ger%E4te-/ Kanalverkn%FCpfungen l%F6schen",
@@ -425,6 +428,7 @@
     "dialogStateInfoDataTransferToDeviceSuccess" : "Statusinfo: Daten%FCbertragung erfolgreich",
     "dialogWarningDataTransferToDeviceNotPossible" : "Achtung! Daten%FCbertragung nicht m%F6glich",
     "dialogConfigAutomaticLogin" : "Konfiguration - Automatisches Anmelden",
+    "dialogRefactorDefaultNames" : "Sprache der Werks- R%E4ume/Gewerke/Systemvariablen",
     "dialogChooseBidCosInterface" : "Auswahl BidCos-Interface",
     "dialogChooseChannel" : "Kanalauswahl",
     "dialogRestartSafeModeTitle" : "Neustart",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.js
@@ -90,6 +90,7 @@
     "footerBtnCloseTree" : "Close tree structure",
     "footerBtnEdit" : "Edit",
     "footerBtnAutoLogin" : "Log in<br/>automatically",
+    "footerBtnRefactorDefaultNames" : "Assign language values<br/>to internal ...",
     "footerBtnVirtualChannelsShow" : "Show virtual channels",
     "footerBtnVirtualChannelsHide" : "Hide virtual channels",
     "footerBtnSaveNewFav" : "Save selection as new<br/>favourites page",
@@ -160,6 +161,7 @@
     "thPhoneNumber" : "Phone number",
     "thAutomaticLogin" : "Log in<br/>automatically",
     "thChooseUser" : "User selection",
+    "thChooseLanguage" : "Language selection",
     "thSafetyNote" : "Security alert",
     "thGateway" : "Gateway",
     "thAccessCode" : "Access code",
@@ -245,6 +247,8 @@
     "hintUsrAutoLoginRow2" : "If automatic login is activated you will be logged-in to the system without entering your user name or password.<br/>",
     "hintUsrAutoLoginRow3" : "The selected user will be logged-in automatically without further validation. The system is no longer protected against external influences.<br />",
     "hintUsrAutoLoginRow4" : "You can access other user accounts with click on \"Login\". Afterwards, please login with the user name and password again.",
+    "hintRefactorDefaultNamesRow1" : "Attention!",
+    "hintRefactorDefaultNamesRow1" : "jp112sdl: not translated yet<br/>",
     "profileSettings" : "Profile setting",
     "profileSettingsSender" : "Profile settings - transmitter",
     "profileSettingsReceiver" : "Profile settings - receiver",
@@ -254,6 +258,7 @@
     "easyModeOptionUnknownProfile" : "Unknown profile",
     "easyModeHintUseExpertMode" : "Please use the expert mode here",
     "btnDirectDeviceLinks" : "Direct device connections",
+    "btnRefactorRollback" : "Rollback",
     "lblDirectDeviceLinks1" : "Overview of all direct connections to devices or channels",
     "lblDirectDeviceLinks2" : "Edit direct device / channel connections",
     "lblDirectDeviceLinks3" : "Delete direct device / channel connections",
@@ -424,6 +429,7 @@
     "dialogStateInfoDataTransferToDeviceSuccess" : "Status info: Data transfer successful",
     "dialogWarningDataTransferToDeviceNotPossible" : "Attention! Unable to transfer data",
     "dialogConfigAutomaticLogin" : "Configuration - automatic login",
+    "dialogRefactorDefaultNames" : "Rename pre-defined object names (rooms, functions, system variables)",
     "dialogChooseBidCosInterface" : "Selection BidCos-Interface",
     "dialogChooseChannel" : "Channel selection",
     "dialogRestartSafeModeTitle" : "Restart",
@@ -1101,7 +1107,7 @@
     "optionNoModification" : "No change",
     "optionNone" : "None",
     "optionNotActive" : "Not active",
-
+    
     "HelpTitle" : "Help"
   }
 });
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -7012,6 +7012,11 @@
             homematic("CCU.setUserAckInstallWizard", {'userName': usrName});
           }
         }
+        if (!homematic('CCU.existsFile', {'file': "/etc/config/systemLanguageConfigured"})) {
+          console.log("lang");
+          CreatePopup(ID_REFACTOR_DEFAULT_NAMES);
+          //homematic("CCU.setSystemLanguageConfigured");
+        }		
       }
     } else {
       WebUI.serviceMessageCount = setServiceMessage();
@@ -25416,6 +25421,37 @@
     new Ajax.Request(url, {postBody: ReGa.encode(pb)});  
   },
   
+  refactorDefaultNames: function(lng) {
+    var url = "/esp/system.htm?sid="+SessionId;
+    var pb = "string action = 'refactorDefaultNames';";
+    pb += "string refactorLanguage = '" + lng + "';";   
+    var opts =
+    {
+      postBody: ReGa.encode(pb),
+      onComplete:function(t)
+      {
+        PopupClose();
+        reloadPage();
+      }
+    };
+    new Ajax.Request(url,opts);     
+   },
+   
+   refactorDefaultNamesRollback: function() {
+    var url = "/esp/system.htm?sid="+SessionId;
+    var pb = "string action = 'refactorDefaultNamesRollback';";
+    var opts =
+    {
+      postBody: ReGa.encode(pb),
+      onComplete:function(t)
+      {
+        PopupClose();
+        reloadPage();
+      }
+    };
+    new Ajax.Request(url,opts);     
+   },
+    
   ClearHistoryData: function()
   {
     var url = "/esp/system.htm?sid="+SessionId;
@@ -26732,7 +26768,12 @@
         this.setWidth(800);
         this.setFile( "/pages/msg/autoLoginConfig.htm" );
         break;
-       case ID_CHOOSE_LED:
+      case ID_REFACTOR_DEFAULT_NAMES:
+        this.setTitle( translateKey("dialogRefactorDefaultNames") /*"Umbenennen der Standard-Objekte" */);
+        this.setWidth(400);
+        this.setFile( "/pages/msg/refactorDefaultNames.htm" );
+        break;
+        case ID_CHOOSE_LED:
         this.setTitle( translateKey("dialogEnterValue") /*"Wert eingeben" */);
         this.addToPostBody( 'string type = "'+this.type+'";' );
         this.setWidth(800);
@@ -27551,6 +27592,7 @@
 ID_SET_OUCFM_MODE        = 1023;
 ID_SET_STATUS_DISPLAY    = 1024;
 ID_BIDCOS_INTERFACE = 1025;
+ID_REFACTOR_DEFAULT_NAMES = 2001;
 
 nUA = navigator.userAgent;
 NAV_IE = false;
