--- occu/WebUI/www/api/methods.conf.orig
+++ occu/WebUI/www/api/methods.conf
@@ -1062,6 +1062,13 @@
   ARGUMENTS {_session_id_}
 }
 
+CCU.setSystemLanguage {
+  LEVEL ADMIN
+  SCRIPT_FILE ccu/setsystemlanguage.tcl
+  INFO {Setzt die Systemsprache in /etc/config/systemLanguage}
+  ARGUMENTS {_session_id_ lang}
+}
+
 CCU.restartSSHDaemon {
   LEVEL ADMIN
   SCRIPT_FILE ccu/restartsshdaemon.tcl
--- occu/WebUI/www/api/methods/ccu/setsystemlanguage.tcl.orig
+++ occu/WebUI/www/api/methods/ccu/setsystemlanguage.tcl
@@ -0,0 +1,13 @@
+##
+# CCU.systemLanguage
+# Setzt die Systemsprache der CCU in /etc/config/systemLanguage
+#
+# Parameter:
+#  lang         : [string] Die gewaehlte Systemsprache (de, en)
+#
+# Rückgabewert: immer true
+##
+
+catch {exec echo $args(lang) > /etc/config/systemLanguage}
+
+jsonrpc_response true
--- occu/WebUI/www/rega/esp/system.fn.orig
+++ occu/WebUI/www/rega/esp/system.fn
@@ -1580,3 +1580,188 @@
 function ::keepAlive()
 {
 }
+
+function ::setSystemLanguage()
+{
+  string lang = system.GetVar("systemLanguage");
+  
+  Call("system.fn::resetSystemLanguage()");
+
+  string funcDefault    = "funcLight;funcHeating;funcClimateControl;funcWeather;funcEnvironment;funcSecurity;funcLock;funcButton;funcCentral;funcEnergy";
+  string funcTranslated = "Licht;Heizung;Klima;Wetter;Umwelt;Sicherheit;Verschluss;Taster;Zentrale;Energiemanagement";
+  
+  string roomDefault = "roomLivingRoom;roomKitchen;roomBedroom;roomChildrensRoom1;roomChildrensRoom2;roomOffice;roomBathroom;roomGarage;roomHWR;roomGarden;roomTerrace";
+  string roomTranslated = "Wohnzimmer;Küche;Schlafzimmer;Kinderzimmer 1;Kinderzimmer 2;Büro;Badezimmer;Garage;Hauswirtschaftsraum;Garten;Terrasse";
+  
+  string svDefault   = "sysVarAlarmZone1;sysVarAlarmZone1Msg;sysVarAlarmZone1Triggered;sysVarAlarmZone1NotTriggered;sysVarPresence;sysVarPresenceMsg;sysVarPresencePresent;sysVarPresenceNotPresent;sysVarAlarmMessages;sysVarAlarmMessagesCount;sysVarServiceMessages;sysVarServiceMessagesCount;sysVarRainToday;sysVarRainTodayDescr;sysVarRainYesterday;sysVarRainYesterdayDescr";
+  string svTranslated   = "Alarmzone 1;Alarmmeldung Alarmzone 1;ausgelöst;nicht ausgelöst;Anwesenheit;Anwesenheit;anwesend;nicht anwesend;Alarmmeldungen;Anzahl Alarmmeldungen;Servicemeldungen;Anzahl Servicemeldungen;Regen heute;Regenmenge aktueller Tag;Regen gestern;Regenmenge vorheriger Tag";
+
+  if (lang == "en") {
+    funcTranslated = "Light;Heating;Climatic conditions;Weather;Environment;Security;Lock;Button;Central;Energy management";
+    roomTranslated = "Living room;Kitchen;Bed room;Childrens room 1;Childrens room 2;Home office;Bathroom;Garage;Utility room;Garden;Terrace";
+    svTranslated   = "Alarm zone 1;Alarm message alarm zone 1;triggered;not triggered;Presence;Presence;present;absent;Alarm messages;Number alarm messages;Service messages;Number service messages;Rain today;Rainfall current day;Rainfall yesterday;Rainfall previous day";
+  }
+  
+  string oID;
+  string metaDataFactoryName = "FACTORY_NAME";
+  
+  var funcs = dom.GetObject(ID_FUNCTIONS);
+  var rooms = dom.GetObject(ID_ROOMS);
+  var sv    = dom.GetObject(ID_SYSTEM_VARIABLES);
+  
+  ! Functions
+  foreach(oID, funcs.EnumIDs()) {
+    var oObj = dom.GetObject(oID);
+    string listEntry;
+    integer idx = 0;
+    foreach(listEntry, funcDefault.Split(";")) {
+      if ((oObj.Name() == listEntry) || (oObj.Name() == "${"#listEntry#"}")) {
+        var meta = oObj.MetaData(metaDataFactoryName);
+        if (!meta) {
+          string newName = funcTranslated.StrValueByIndex(";", idx);
+          oObj.RemoveMetaData(metaDataFactoryName);
+          oObj.AddMetaData(metaDataFactoryName, oObj.Name());
+          oObj.Name(newName);
+          WriteLine("old = "#oObj.Name()#", new = "#newName);
+        } else {
+          WriteLine(oObj.Name()#" already renamed");
+        }
+      }
+      idx = idx + 1;
+    }
+  }
+  
+  ! Rooms
+  foreach(oID, rooms.EnumIDs()) {
+    var oObj = dom.GetObject(oID);
+    string listEntry;
+    integer idx = 0;
+    foreach(listEntry, roomDefault.Split(";")) {
+      if ((oObj.Name() == listEntry) || (oObj.Name() == "${"#listEntry#"}")) {
+        var meta = oObj.MetaData(metaDataFactoryName);
+        if (!meta) {
+          string newName = roomTranslated.StrValueByIndex(";", idx);
+          oObj.RemoveMetaData(metaDataFactoryName);
+          oObj.AddMetaData(metaDataFactoryName, oObj.Name());
+          oObj.Name(newName);
+          WriteLine("old = "#oObj.Name()#", new = "#newName);
+        } else {
+          WriteLine(oObj.Name()#" already renamed");
+        }
+      }
+      idx = idx + 1;
+    }
+  }
+  
+  !Sysvars
+  foreach(oID, sv.EnumIDs()) {
+    var oObj = dom.GetObject(oID);
+    string listEntry;
+    integer idx = 0;
+    foreach(listEntry, svDefault.Split(";")) {
+      if ((oObj.Name() == listEntry) || (oObj.Name() == "${"#listEntry#"}")) {
+        var meta = oObj.MetaData(metaDataFactoryName);
+        if (!meta) {
+          string newName = svTranslated.StrValueByIndex(";", idx);
+          oObj.RemoveMetaData(metaDataFactoryName);
+          oObj.AddMetaData(metaDataFactoryName, oObj.Name());
+          oObj.Name(newName);
+          
+          WriteLine("old = "#oObj.Name()#", new = "#newName);
+        } else {
+          WriteLine(oObj.Name()#" already renamed");
+        }
+      }
+      
+      if ((oObj.ValueName0() == listEntry) || (oObj.ValueName0() == "${"#listEntry#"}")) {
+        var meta = oObj.MetaData(metaDataFactoryName#"_VALUE_NAME_0");
+        if (!meta) {
+          string newValueName0 = svTranslated.StrValueByIndex(";", idx);
+          oObj.RemoveMetaData(metaDataFactoryName#"_VALUE_NAME_0");
+          oObj.AddMetaData(metaDataFactoryName#"_VALUE_NAME_0", oObj.ValueName0());
+          oObj.ValueName0(newValueName0);
+          
+          WriteLine("old ValueName0 = "#oObj.ValueName0()#", new = "#newValueName0);
+        } else {
+          WriteLine(oObj.ValueName0()#" already renamed");
+        }
+      }
+      
+      if ((oObj.ValueName1() == listEntry) || (oObj.ValueName1() == "${"#listEntry#"}")) {
+        var meta = oObj.MetaData(metaDataFactoryName#"_VALUE_NAME_1");
+        if (!meta) {
+          string newValueName1 = svTranslated.StrValueByIndex(";", idx);
+          oObj.RemoveMetaData(metaDataFactoryName#"_VALUE_NAME_1");
+          oObj.AddMetaData(metaDataFactoryName#"_VALUE_NAME_1", oObj.ValueName1());
+          oObj.ValueName1(newValueName1);
+          
+          WriteLine("old ValueName1 = "#oObj.ValueName1()#", new = "#newValueName1);
+        } else {
+          WriteLine(oObj.ValueName1()#" already renamed");
+        }
+      }
+      
+      idx = idx + 1;
+    }
+  }
+
+}
+
+function ::resetSystemLanguage()
+{
+  string oID;
+  string metaDataFactoryName = "FACTORY_NAME";
+  
+  var funcs = dom.GetObject(ID_FUNCTIONS);
+  var rooms = dom.GetObject(ID_ROOMS);
+  var sv    = dom.GetObject(ID_SYSTEM_VARIABLES);
+  
+  foreach(oID, funcs.EnumIDs()) {
+    var oObj = dom.GetObject(oID);
+    var meta = oObj.MetaData(metaDataFactoryName);
+    if (meta) {
+      string newName = meta;
+      WriteLine("rollback "#oObj.Name()#" to "#newName);
+      oObj.Name(newName);
+      oObj.RemoveMetaData(metaDataFactoryName);
+    }
+  }
+  
+  foreach(oID, rooms.EnumIDs()) {
+    var oObj = dom.GetObject(oID);
+    var meta = oObj.MetaData(metaDataFactoryName);
+    if (meta) {
+      string newName = meta;
+      WriteLine("rollback "#oObj.Name()#" to "#newName);
+      oObj.Name(newName);
+      oObj.RemoveMetaData(metaDataFactoryName);
+    }
+  }
+  
+  foreach(oID, sv.EnumIDs()) {
+    var oObj = dom.GetObject(oID);
+    var meta = oObj.MetaData(metaDataFactoryName);
+    if (meta) {
+      string newName = meta;
+      WriteLine("rollback "#oObj.Name()#" to "#newName);
+      oObj.Name(newName);
+      oObj.RemoveMetaData(metaDataFactoryName);
+    }
+    
+    var metaValueName0 = oObj.MetaData(metaDataFactoryName#"_VALUE_NAME_0");
+    if (metaValueName0) {
+      string newName = metaValueName0;
+      WriteLine("rollback "#oObj.ValueName0()#" to "#newName);
+      oObj.ValueName0(newName);
+      oObj.RemoveMetaData(metaDataFactoryName#"_VALUE_NAME_0");
+    }
+    
+    var metaValueName1 = oObj.MetaData(metaDataFactoryName#"_VALUE_NAME_1");
+    if (metaValueName1) {
+      string newName = metaValueName1;
+      WriteLine("rollback "#oObj.ValueName1()#" to "#newName);
+      oObj.ValueName1(newName);
+      oObj.RemoveMetaData(metaDataFactoryName#"_VALUE_NAME_1");
+    }
+  }
+}
--- occu/WebUI/www/rega/esp/system.htm.orig
+++ occu/WebUI/www/rega/esp/system.htm
@@ -34,4 +34,5 @@
   if( action == "getLastTimeOfChn"           ) { Call( "system.fn::getLastTimeOfChn()" );           }
   if( action == "checkName"                  ) { Call( "system.fn::checkName()");                   }
   if( action == "keepAlive"                  ) { Call( "system.fn::keepAlive()");                   }
+  if( action == "setSystemLanguage"          ) { Call( "system.fn::setSystemLanguage()");           }
 %>
--- occu/WebUI/www/rega/pages/msg/systemLanguageConfig.htm.orig
+++ occu/WebUI/www/rega/pages/msg/systemLanguageConfig.htm
@@ -0,0 +1,50 @@
+<div class="popupTitle" style='width: 98.5%;'>
+  <%
+  Write(system.GetVar("title"));
+  %>
+</div>
+
+<div class="CLASS01400">
+  <table class="popupTableScript CLASS01401" cellpadding="0" cellspacing="0">
+    <tr>
+    <th class="TblHeaderUserConfig">${thChooseSystemLanguage}</th>
+    </tr>
+    
+    <tr>
+    
+    <td class="CLASS01402" style='text-align: center; vertical-align:middle;'>
+    <table style="margin:auto;">
+    <tr><td style='text-align: center; vertical-align:middle;'><div>
+      <select id="cbLang">
+        <option value="de">${selectLangDE}</option>
+        <option value="en">${selectLangUK}</option>
+       </select></div></td>
+       </tr>
+       </table>
+    </td>
+    
+    </tr>
+
+    <tr class="popupTableRow">
+    <td class="CLASS01406" colspan="3">
+      <table>
+      <tr>
+       <td width="10%" style='text-align: center; vertical-align:middle;'><div class="FooterButton CLASS01408" onclick="cancelSystemLanguageConfig();">${footerBtnCancel}</div></td>
+	   <td width="80%"></td>
+       <td width="10%" style='text-align: center; vertical-align:middle;'><div class="FooterButton CLASS01408" onclick="doSystemLanguageConfig();">${footerBtnOk}</div></td>
+      </tr>
+      </table>
+    </td>
+    </tr>
+  </table>
+</div>
+<script type="text/javascript">
+  doSystemLanguageConfig = function() {
+    var selValue = $("cbLang")[$("cbLang").selectedIndex].value;
+    iseSystem.setSystemLanguage(selValue);
+    homematic("CCU.setSystemLanguage", {'lang': selValue});
+  };
+  cancelSystemLanguageConfig = function() {
+    PopupClose();
+  };
+</script>
--- occu/WebUI/www/rega/pages/tabs/admin/userAdministration.htm.orig
+++ occu/WebUI/www/rega/pages/tabs/admin/userAdministration.htm
@@ -173,6 +173,7 @@
     s += "<td style='text-align: center; vertical-align: middle;'><div class='FooterButton' onclick='WebUI.goBack()'>${footerBtnPageBack}</div></td>";
     s += "<td style='text-align: center; vertical-align: middle;'><div class='FooterButton' onclick='newUser();'>${footerBtnNew}</div></td>";
     s += "<td  style='text-align:center; vertical-align: middle;'><div class='FooterButton CLASS05305' onclick='showAutoLogin()'>${footerBtnAutoLogin}</div></td>";
+    s += "<td  style='text-align:center; vertical-align: middle;'><div class='FooterButton CLASS05305' onclick='showSystemLanguage()'>${footerBtnSystemLanguage}</div></td>";
   s += "</tr>";
   s += "</table>";
 
@@ -210,6 +211,10 @@
     CreatePopup(ID_AUTO_LOGIN_CONFIG);
   }
 
+  showSystemLanguage = function() {
+    CreatePopup(ID_SYSTEM_LANGUAGE_CONFIG);
+  }
+  
   translatePage("#userAdmin");
   jQuery("#userAdmin").show();
 
--- occu/WebUI/www/webui/js/lang/de/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.js
@@ -91,6 +91,7 @@
     "footerBtnCloseTree" : "Baumstruktur schlie%DFen",
     "footerBtnEdit" : "Bearbeiten",
     "footerBtnAutoLogin" : "Automatisches<br/>Anmelden",
+    "footerBtnSystemLanguage" : "Systemsprache<br />festlegen",
     "footerBtnVirtualChannelsShow" : "Virtuelle Kan%E4le anzeigen",
     "footerBtnVirtualChannelsHide" : "Virtuelle Kan%E4le ausblenden",
     "footerBtnSaveNewFav" : "Auswahl als neue<br />Favoritenseite speichern",
@@ -161,6 +162,7 @@
     "thPhoneNumber" : "Telefonnummer",
     "thAutomaticLogin" : "Automatisches<br/>Anmelden",
     "thChooseUser" : "Benutzerauswahl",
+    "thChooseSystemLanguage" : "Systemsprache ausw%E4hlen",
     "thSafetyNote" : "Sicherheitshinweis",
     "thGateway" : "Gateway",
     "thAccessCode" : "Zugriffscode",
@@ -425,6 +427,7 @@
     "dialogStateInfoDataTransferToDeviceSuccess" : "Statusinfo: Daten%FCbertragung erfolgreich",
     "dialogWarningDataTransferToDeviceNotPossible" : "Achtung! Daten%FCbertragung nicht m%F6glich",
     "dialogConfigAutomaticLogin" : "Konfiguration - Automatisches Anmelden",
+    "dialogConfigSystemLanguage" : "Konfiguration - Systemsprache",
     "dialogChooseBidCosInterface" : "Auswahl BidCos-Interface",
     "dialogChooseChannel" : "Kanalauswahl",
     "dialogRestartSafeModeTitle" : "Neustart",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.js
@@ -90,6 +90,7 @@
     "footerBtnCloseTree" : "Close tree structure",
     "footerBtnEdit" : "Edit",
     "footerBtnAutoLogin" : "Log in<br/>automatically",
+    "footerBtnSystemLanguage" : "Set system<br/>language",
     "footerBtnVirtualChannelsShow" : "Show virtual channels",
     "footerBtnVirtualChannelsHide" : "Hide virtual channels",
     "footerBtnSaveNewFav" : "Save selection as new<br/>favourites page",
@@ -160,6 +161,7 @@
     "thPhoneNumber" : "Phone number",
     "thAutomaticLogin" : "Log in<br/>automatically",
     "thChooseUser" : "User selection",
+    "thChooseSystemLanguage" : "Choose system language",
     "thSafetyNote" : "Security alert",
     "thGateway" : "Gateway",
     "thAccessCode" : "Access code",
@@ -424,6 +426,7 @@
     "dialogStateInfoDataTransferToDeviceSuccess" : "Status info: Data transfer successful",
     "dialogWarningDataTransferToDeviceNotPossible" : "Attention! Unable to transfer data",
     "dialogConfigAutomaticLogin" : "Configuration - automatic login",
+    "dialogConfigSystemLanguage" : "Configuration - system language",
     "dialogChooseBidCosInterface" : "Selection BidCos-Interface",
     "dialogChooseChannel" : "Channel selection",
     "dialogRestartSafeModeTitle" : "Restart",
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -7012,6 +7012,9 @@
             homematic("CCU.setUserAckInstallWizard", {'userName': usrName});
           }
         }
+        if (!homematic('CCU.existsFile', {'file': "/etc/config/systemLanguage"})) {
+          CreatePopup(ID_SYSTEM_LANGUAGE_CONFIG);
+        }		
       }
     } else {
       WebUI.serviceMessageCount = setServiceMessage();
@@ -25416,6 +25419,22 @@
     new Ajax.Request(url, {postBody: ReGa.encode(pb)});  
   },
   
+  setSystemLanguage: function(lang) {
+    var url = "/esp/system.htm?sid="+SessionId;
+    var pb = "string action = 'setSystemLanguage';";
+    pb += "string systemLanguage = '" + lang + "';";
+    var opts =
+    {
+      postBody: ReGa.encode(pb),
+      onComplete:function(t)
+      {
+        PopupClose();
+        reloadPage();
+      }
+    };
+    new Ajax.Request(url,opts);     
+  },
+   
   ClearHistoryData: function()
   {
     var url = "/esp/system.htm?sid="+SessionId;
@@ -26732,7 +26751,12 @@
         this.setWidth(800);
         this.setFile( "/pages/msg/autoLoginConfig.htm" );
         break;
-       case ID_CHOOSE_LED:
+      case ID_SYSTEM_LANGUAGE_CONFIG:
+        this.setTitle( translateKey("dialogConfigSystemLanguage") /*"Systemsprache waehlen - Konfiguration" */);
+        this.setWidth(400);
+        this.setFile( "/pages/msg/systemLanguageConfig.htm" );
+        break;
+        case ID_CHOOSE_LED:
         this.setTitle( translateKey("dialogEnterValue") /*"Wert eingeben" */);
         this.addToPostBody( 'string type = "'+this.type+'";' );
         this.setWidth(800);
@@ -27551,6 +27575,7 @@
 ID_SET_OUCFM_MODE        = 1023;
 ID_SET_STATUS_DISPLAY    = 1024;
 ID_BIDCOS_INTERFACE = 1025;
+ID_SYSTEM_LANGUAGE_CONFIG = 2001;
 
 nUA = navigator.userAgent;
 NAV_IE = false;
