--- occu/WebUI/www/rega/pages/index.htm.orig
+++ occu/WebUI/www/rega/pages/index.htm
@@ -15,6 +15,7 @@
     <link rel="stylesheet" type="text/css" href="/webui/css/extern/jquery.jqplot.css" />
     <link rel="stylesheet" type="text/css" href="/webui/css/extern/msdropdown/dd.css" />
     <link rel="stylesheet" type="text/css" href="/webui/css/extern/spectrum.css" />
+    <link rel="stylesheet" type="text/css" href="/webui/css/extern/hmBootstrap.css" />
     <link id="idCss" rel="stylesheet" type="text/css" href="/webui/style.cgi" />
 
     <link rel="apple-touch-icon" href="/ise/img/rm-touch-icon-iphone.png" />
@@ -24,6 +25,7 @@
 
 
     <script type="text/javascript" src="/webui/js/extern/jquery.js"></script>
+    <script type="text/javascript" src="/webui/js/extern/bootstrap.min.js"></script>
     <script type="text/javascript" src="/webui/js/extern/jqueryURLPlugin.js"></script>
     <script type="text/javascript" src="/webui/js/extern/jquery.powertip.js"></script>
     <script type="text/javascript" src="/webui/js/extern/jquery.blockUI.js"></script>
--- occu/WebUI/www/rega/pages/tabs/startpage.htm.orig
+++ occu/WebUI/www/rega/pages/tabs/startpage.htm
@@ -255,35 +255,101 @@
 
             <!-- DutyCycle -->
             <tr name="trDutyCycle0" style="display: none">
-              <td id="dutyCycleAddr0" align="left"</td>
-              <td><div id="dutyCycleProgress0" align="right"><div id="dutyCycleProgressBar0"></div></div></td>
-              <td style="white-space: nowrap"><div id="dutyCycleVal0" style="width:75%; text-align:left"></div></td>
+              <td id="dutyCycleAddr0" style="vertical-align: top;text-align: left;"</td>
+              <td id="hmBootstrap">
+                <div class="progress"><div name="dutyCycleBar0" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" style="width:0%"><span id="dutyCycleVal0"></span></div>
+              </td>
             </tr>
             <tr name="trDutyCycle1" style="display: none">
-              <td id="dutyCycleAddr1" align="left"</td>
-              <td><div id="dutyCycleProgress1" align="right"><div id="dutyCycleProgressBar1"></div></div></td>
-              <td style="white-space: nowrap"><div id="dutyCycleVal1" style="width:75%; text-align:left"></div></td>
+              <td id="dutyCycleAddr1" style="vertical-align: top;text-align: left;"</td>
+              <td id="hmBootstrap">
+                <div class="progress"><div name="dutyCycleBar1" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" style="width:0%"><span id="dutyCycleVal1"></span></div>
+              </td>
             </tr>
             <tr name="trDutyCycle2" style="display: none">
-              <td id="dutyCycleAddr2" align="left"</td>
-              <td><div id="dutyCycleProgress2" align="right"><div id="dutyCycleProgressBar2"></div></div></td>
-              <td style="white-space: nowrap"><div id="dutyCycleVal2" style="width:75%; text-align:left"></div></td>
+              <td id="dutyCycleAddr2" style="vertical-align: top;text-align: left;"</td>
+              <td id="hmBootstrap">
+                <div class="progress"><div name="dutyCycleBar2" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" style="width:0%"><span id="dutyCycleVal2"></span></div>
+              </td>
             </tr>
             <tr name="trDutyCycle3" style="display: none">
-              <td id="dutyCycleAddr3" align="left"</td>
-              <td><div id="dutyCycleProgress3" align="right"><div id="dutyCycleProgressBar3"></div></div></td>
-              <td style="white-space: nowrap"><div id="dutyCycleVal3" style="width:75%; text-align:left"></div></td>
-            </tr>
+              <td id="dutyCycleAddr3" style="vertical-align: top;text-align: left;"</td>
+              <td id="hmBootstrap">
+                <div class="progress"><div name="dutyCycleBar3" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" style="width:0%"><span id="dutyCycleVal3"></span></div>
+              </td>
+            </tr>            
             <tr name="trDutyCycle4" style="display: none">
-              <td id="dutyCycleAddr4" align="left"</td>
-              <td><div id="dutyCycleProgress4" align="right"><div id="dutyCycleProgressBar4"></div></div></td>
-              <td style="white-space: nowrap"><div id="dutyCycleVal4" style="width:75%; text-align:left"></div></td>
-            </tr>
+              <td id="dutyCycleAddr4" style="vertical-align: top;text-align: left;"</td>
+              <td id="hmBootstrap">
+                <div class="progress"><div name="dutyCycleBar4" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" style="width:0%"><span id="dutyCycleVal4"></span></div>
+              </td>
+            </tr>            
             <tr name="trDutyCycle5" style="display: none">
-              <td id="dutyCycleAddr5" align="left"</td>
-              <td><div id="dutyCycleProgress5" align="right"><div id="dutyCycleProgressBar5"></div></div></td>
-              <td style="white-space: nowrap"><div id="dutyCycleVal5" style="width:75%; text-align:left"></div></td>
-            </tr>
+              <td id="dutyCycleAddr5" style="vertical-align: top;text-align: left;"</td>
+              <td id="hmBootstrap">
+                <div class="progress"><div name="dutyCycleBar5" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" style="width:0%"><span id="dutyCycleVal5"></span></div>
+              </td>
+            </tr>
+            <tr name="trDutyCycle6" style="display: none">
+              <td id="dutyCycleAddr6" style="vertical-align: top;text-align: left;"</td>
+              <td id="hmBootstrap">
+                <div class="progress"><div name="dutyCycleBar6" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" style="width:0%"><span id="dutyCycleVal6"></span></div>
+              </td>
+            </tr>
+            <tr name="trDutyCycle7" style="display: none">
+              <td id="dutyCycleAddr7" style="vertical-align: top;text-align: left;"</td>
+              <td id="hmBootstrap">
+                <div class="progress"><div name="dutyCycleBar7" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" style="width:0%"><span id="dutyCycleVal7"></span></div>
+              </td>
+            </tr>
+            <tr name="trDutyCycle8" style="display: none">
+              <td id="dutyCycleAddr8" style="vertical-align: top;text-align: left;"</td>
+              <td id="hmBootstrap">
+                <div class="progress"><div name="dutyCycleBar8" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" style="width:0%"><span id="dutyCycleVal8"></span></div>
+              </td>
+            </tr>            
+            <tr name="trDutyCycle9" style="display: none">
+              <td id="dutyCycleAddr9" style="vertical-align: top;text-align: left;"</td>
+              <td id="hmBootstrap">
+                <div class="progress"><div name="dutyCycleBar9" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" style="width:0%"><span id="dutyCycleVal9"></span></div>
+              </td>
+            </tr>  
+            <tr name="trDutyCycle10" style="display: none">
+              <td id="dutyCycleAddr10" style="vertical-align: top;text-align: left;"</td>
+              <td id="hmBootstrap">
+                <div class="progress"><div name="dutyCycleBar10" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" style="width:0%"><span id="dutyCycleVal10"></span></div>
+              </td>
+            </tr>
+            <tr name="trDutyCycle11" style="display: none">
+              <td id="dutyCycleAddr11" style="vertical-align: top;text-align: left;"</td>
+              <td id="hmBootstrap">
+                <div class="progress"><div name="dutyCycleBar11" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" style="width:0%"><span id="dutyCycleVal11"></span></div>
+              </td>
+            </tr>
+            <tr name="trDutyCycle12" style="display: none">
+              <td id="dutyCycleAddr12" style="vertical-align: top;text-align: left;"</td>
+              <td id="hmBootstrap">
+                <div class="progress"><div name="dutyCycleBar12" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" style="width:0%"><span id="dutyCycleVal12"></span></div>
+              </td>
+            </tr>
+            <tr name="trDutyCycle13" style="display: none">
+              <td id="dutyCycleAddr13" style="vertical-align: top;text-align: left;"</td>
+              <td id="hmBootstrap">
+                <div class="progress"><div name="dutyCycleBar13" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" style="width:0%"><span id="dutyCycleVal13"></span></div>
+              </td>
+            </tr>            
+            <tr name="trDutyCycle14" style="display: none">
+              <td id="dutyCycleAddr14" style="vertical-align: top;text-align: left;"</td>
+              <td id="hmBootstrap">
+                <div class="progress"><div name="dutyCycleBar14" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" style="width:0%"><span id="dutyCycleVal14"></span></div>
+              </td>
+            </tr>  
+            <tr name="trDutyCycle15" style="display: none">
+              <td id="dutyCycleAddr15" style="vertical-align: top;text-align: left;"</td>
+              <td id="hmBootstrap">
+                <div class="progress"><div name="dutyCycleBar15" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" style="width:0%"><span id="dutyCycleVal15"></span></div>
+              </td>
+            </tr> 
 
           </table>
           </td>
--- occu/WebUI/www/webui/style.css.orig
+++ occu/WebUI/www/webui/style.css
@@ -2321,27 +2321,6 @@
 progress::-moz-progress-bar { background: #3678c9; }
 progress::-webkit-progress-value { background: #3678c9; }
 
-#dutyCycleProgress0, #dutyCycleProgress1, #dutyCycleProgress2, #dutyCycleProgress3, #dutyCycleProgress4, #dutyCycleProgress5 {
-  width: 75%;
-
-  background: -moz-linear-gradient(0deg, rgba(13,255,0,1) 0%, rgba(255,255,0,1) 63%, rgba(255,0,0,1) 100%); /* ff3.6+ */
-  background: -webkit-gradient(linear, left top, right top, color-stop(0%, rgba(13,255,0,1)), color-stop(63%, rgba(255,255,0,1)), color-stop(100%, rgba(255,0,0,1))); /* safari4+,chrome */
-  background: -webkit-linear-gradient(0deg, rgba(13,255,0,1) 0%, rgba(255,255,0,1) 63%, rgba(255,0,0,1) 100%); /* safari5.1+,chrome10+ */
-  background: -o-linear-gradient(0deg, rgba(13,255,0,1) 0%, rgba(255,255,0,1) 63%, rgba(255,0,0,1) 100%); /* opera 11.10+ */
-  background: -ms-linear-gradient(0deg, rgba(13,255,0,1) 0%, rgba(255,255,0,1) 63%, rgba(255,0,0,1) 100%); /* ie10+ */
-  background: linear-gradient(90deg, rgba(13,255,0,1) 0%, rgba(255,255,0,1) 63%, rgba(255,0,0,1) 100%); /* w3c */
-  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#0DFF00', endColorstr='#ff0000',GradientType=1 ); /* ie6-9 */
-
-
-}
-
-#dutyCycleProgressBar0, #dutyCycleProgressBar1, #dutyCycleProgressBar2, #dutyCycleProgressBar3, #dutyCycleProgressBar4, #dutyCycleProgressBar5 {
-  width: 1%;
-  height: 12px;
-  background-color: #d9d9d9;
-}
-
-
 .Filter {
   margin:0;
   border:1px solid $_(black); /* black */
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -30136,73 +30136,72 @@
 };
 
 
-showDutyCycle = function() {
-  if (jQuery("#PagePathSpan").text() == translateKey("startPage")) {
-    var ifaceBidCosRF = "BidCos-RF",
-      ifaceHmIPRF = "HmIP-RF",
-      arInterfaceDutyCycle = {},
-      showPartingLine = false,
-      dcUnit = "%",
-      dcNotAvailable = -1,
-      dcAlarm = 89;  // Attention when dc >= 90%
-
-      homematic("Interface.getDutyCycle", {}, function(dcArray) {
-        if(jQuery.isArray(dcArray)) {
-          jQuery.each(dcArray, function(index, iface) {
-            var dutyCycleProgressElem = jQuery("#dutyCycleProgress"+index),
-              dutyCycleProgressBarElm = jQuery("#dutyCycleProgressBar"+index),
-              dutyCycleValElm = jQuery("#dutyCycleVal"+index),
-              dutyCycleAddrElm = jQuery("#dutyCycleAddr"+index),
-              trDutyCycle = jQuery("[name='trDutyCycle"+index+"']"),
-              trPartingLineElm = jQuery("#partingLine1"),
-              dcVal,
-              width, value;
-
-            if (typeof iface.dutyCycle !== "undefined") {
-              dcVal = Math.floor(iface.dutyCycle);
-              conInfo("dutyCycle - " + ifaceBidCosRF + ": " + dcVal + " " + dcUnit);
-              arInterfaceDutyCycle[ifaceBidCosRF] = ((dcVal >= 0) && (dcVal <= 100)) ? dcVal : dcNotAvailable;
-            } else {
-              conInfo("No gateway status for the interface " + ifaceBidCosRF + " available!");
-              arInterfaceDutyCycle[ifaceBidCosRF] = dcNotAvailable;
-            }
-
-            if (arInterfaceDutyCycle[ifaceBidCosRF] != dcNotAvailable) {
-              dutyCycleValElm.text(arInterfaceDutyCycle[ifaceBidCosRF] + " " + dcUnit);
-              if (iface.type === "CCU2") {
-                dutyCycleAddrElm.text("DutyCycle CCU:");
-              } else if(iface.name !== "") {
-                dutyCycleAddrElm.text("DutyCycle LGW ("+iface.name+"):");
-              } else {
-                dutyCycleAddrElm.text("DutyCycle LGW ("+iface.address+"):");
-              }
-
-              width = parseInt(dutyCycleProgressElem.css("width"));
-              value = width - (width / 100 * arInterfaceDutyCycle[ifaceBidCosRF]);
-
-              window.setTimeout(function () {
-                dutyCycleProgressBarElm.css("width", value + "px");
-              }, 25);
-
-              if (arInterfaceDutyCycle[ifaceBidCosRF] > dcAlarm) {
-                trDutyCycle.addClass("attention");
-              } else {
-                trDutyCycle.removeClass("attention");
-              }
-              trPartingLineElm.show();
-              showPartingLine = true;
-              trDutyCycle.css("display", "table-row");
-            } else {
-              trDutyCycle.css("display", "none");
-            }
-
-            if (!showPartingLine) {
-              trPartingLineElm.hide();
-            }
-          });
-        }
-      });
-  }
+showDutyCycle = function() {                                                                                                                                                                                                                        
+  if (jQuery("#PagePathSpan").text() == translateKey("startPage")) {                                                                                                                                                                                
+    var ifaceBidCosRF = "BidCos-RF",                                                                                                                                                                                                                
+      ifaceHmIPRF = "HmIP-RF",                                                                                                                                                                                                                      
+      arInterfaceDutyCycle = {},                                                                                                                                                                                                                    
+      showPartingLine = false,                                                                                                                                                                                                                      
+      dcUnit = "%",                                                                                                                                                                                                                                 
+      dcNotAvailable = -1,                                                                                                                                                                                                                          
+      dcWarn  = 69;  // Warning when dc >= 70%                                                                                                                                                                                                      
+      dcAlarm = 89;  // Attention when dc >= 90%                                                              
+                                                                                                              
+      homematic("Interface.getDutyCycle", {}, function(dcArray) {                                             
+        if(jQuery.isArray(dcArray)) {                                                                         
+          jQuery.each(dcArray, function(index, iface) {                                                       
+              var dutyCycleValElm = jQuery("#dutyCycleVal"+index),                                            
+              dutyCycleAddrElm = jQuery("#dutyCycleAddr"+index),                                              
+              trDutyCycle = jQuery("[name='trDutyCycle"+index+"']"),                                          
+              dutyCycleBar = jQuery("[name='dutyCycleBar"+index+"']"),                                        
+              trPartingLineElm = jQuery("#partingLine1"),                                                     
+              dcVal;                                                                                          
+                                                                                                              
+            if (typeof iface.dutyCycle !== "undefined") {                                                     
+              dcVal = Math.floor(iface.dutyCycle);                                                            
+              conInfo("dutyCycle - " + ifaceBidCosRF + ": " + dcVal + " " + dcUnit);                                            
+              arInterfaceDutyCycle[ifaceBidCosRF] = ((dcVal >= 0) && (dcVal <= 100)) ? dcVal : dcNotAvailable;       
+            } else {                                                                                                 
+              conInfo("No gateway status for the interface " + ifaceBidCosRF + " available!");                       
+              arInterfaceDutyCycle[ifaceBidCosRF] = dcNotAvailable;                                                  
+            }                                                                                                 
+                                                                                                              
+            if (arInterfaceDutyCycle[ifaceBidCosRF] != dcNotAvailable) {                                      
+              dutyCycleValElm.text(arInterfaceDutyCycle[ifaceBidCosRF] + " " + dcUnit);                       
+              if (iface.type === "CCU2") {                                                                    
+                dutyCycleAddrElm.text("DutyCycle CCU:");                                                      
+              } else if(iface.name !== "") {                                                                  
+                dutyCycleAddrElm.text("DutyCycle LGW ("+iface.name+"):");                                     
+              } else {                                                                                        
+                dutyCycleAddrElm.text("DutyCycle LGW ("+iface.address+"):");                                  
+              }                                                                                               
+                                                                                                              
+              dutyCycleBar.css("width",arInterfaceDutyCycle[ifaceBidCosRF]+"%");                              
+              dutyCycleBar.removeClass("progress-bar-success");                                               
+              dutyCycleBar.removeClass("progress-bar-warning");                                               
+              dutyCycleBar.removeClass("progress-bar-danger");                                                
+              if (arInterfaceDutyCycle[ifaceBidCosRF] > dcAlarm) {                                            
+                dutyCycleBar.addClass("progress-bar-danger");                                                 
+              } else if (arInterfaceDutyCycle[ifaceBidCosRF] > dcWarn) {                                      
+                dutyCycleBar.addClass("progress-bar-warning");                                                
+              } else {                                                                                        
+                dutyCycleBar.addClass("progress-bar-success");                                                
+              }                                                                                               
+                                                                                                              
+              trPartingLineElm.show();                                                                        
+              showPartingLine = true;                                                                         
+              trDutyCycle.css("display", "table-row");                                                        
+            } else {                                                                                          
+              trDutyCycle.css("display", "none");                                                             
+            }                                                                                                 
+                                                                                                              
+            if (!showPartingLine) {                                                                                                            
+              trPartingLineElm.hide();                                                                        
+            }                                                                                                                 
+          });                                                                                                     
+        }                                                                                                   
+      });                                                                              
+  }                                                                                            
 };
 
 // Show the duty cycle of all relevant interfaces
