--- occu/WebUI/www/config/easymodes/etc/hmipChannelConfigDialogs.tcl.orig
+++ occu/WebUI/www/config/easymodes/etc/hmipChannelConfigDialogs.tcl
@@ -1976,10 +1976,10 @@
         incr prn
         append html "<tr><td>\${ecoCoolingTemperature}</td>"
         # append html  "<td>[_getTextField $CHANNEL $param $ps($param) $prn]&nbsp;[_getUnit $param]&nbsp;[_getMinMaxValueDescr $param]<input id=\"comfortOld\" type=\"hidden\" value=\"$ps($param)\"></td>"
-        append html "<td>[getTextField $param $ps($param) $chn $prn]&nbsp;[getMinMaxValueDescr $param]<input id=\"comfortOld\" type=\"hidden\" value=\"$ps($param)\"</td>"
+        append html "<td>[getTextField $param $ps($param) $chn $prn]&nbsp;[getMinMaxValueDescr $param]<input id=\"comfortOld\" type=\"hidden\" value=\"$ps($param)\"/></td>"
 
         append html "<script type=\"text/javascript\">"
-          append html "jQuery(\"#separate_$CHANNEL\_$prn\").bind(\"blur\",function() {ProofAndSetValue(this.id, this.id, [getMinValue $param], [getMaxValue $param], 1);isEcoLTComfort(this.name);});"
+          append html "jQuery(\"#separate_$CHANNEL\_$prn\").bind(\"blur\",function() {ProofAndSetValue(this.id, this.id, \"[getMinValue $param]\", \"[getMaxValue $param]\", 1);isEcoLTComfort(this.name);});"
         append html "</script>"
 
         # right
@@ -1987,10 +1987,10 @@
         set param TEMPERATURE_LOWERING
         append html "<td>\${ecoHeatingTemperature}</td>"
         # append html  "<td>[_getTextField $CHANNEL $param $ps($param) $prn]&nbsp;[_getUnit $param]&nbsp;[_getMinMaxValueDescr $param]<input id=\"ecoOld\" type=\"hidden\" value=\"$ps($param)\"></td>"
-        append html "<td>[getTextField $param $ps($param) $chn $prn]&nbsp;[getMinMaxValueDescr $param]</td>"
+        append html "<td>[getTextField $param $ps($param) $chn $prn]&nbsp;[getMinMaxValueDescr $param]<input id=\"ecoOld\" type=\"hidden\" value=\"$ps($param)\"/></td>"
 
         append html "<script type=\"text/javascript\">"
-          append html "jQuery(\"#separate_$CHANNEL\_$prn\").bind(\"blur\",function() {ProofAndSetValue(this.id, this.id, [getMinValue $param], [getMaxValue $param], 1);isEcoLTComfort(this.name);});"
+          append html "jQuery(\"#separate_$CHANNEL\_$prn\").bind(\"blur\",function() {ProofAndSetValue(this.id, this.id, \"[getMinValue $param]\", \"[getMaxValue $param]\", 1);isEcoLTComfort(this.name);});"
         append html "</script>"
         append html "</tr>"
 
@@ -2095,7 +2095,7 @@
         append html "<td>[getTextField $param $ps($param) $chn $prn]&nbsp;[getMinMaxValueDescr $param]<input id=\"comfortOld\" type=\"hidden\" value=\"$ps($param)\"</td>"
 
         append html "<script type=\"text/javascript\">"
-          append html "jQuery(\"#separate_$CHANNEL\_$prn\").bind(\"blur\",function() {ProofAndSetValue(this.id, this.id, [getMinValue $param], [getMaxValue $param], 1);isEcoLTComfort(this.name);});"
+          append html "jQuery(\"#separate_$CHANNEL\_$prn\").bind(\"blur\",function() {ProofAndSetValue(this.id, this.id, \"[getMinValue $param]\", \"[getMaxValue $param]\", 1);isEcoLTComfort(this.name);});"
         append html "</script>"
       append html "</tr>"
     }
