--- occu/WebUI/www/config/easymodes/CC.js.orig
+++ occu/WebUI/www/config/easymodes/CC.js
@@ -295,14 +295,14 @@
 isEcoLTComfort = function(elmName) {
   // elmName should be e. g. TEMPERATURE_COMFORT
   var arName = elmName.split("_"),
-  elmType = arName[1], // e.g.COMFORT
-  isComfort = (elmType == "COMFORT") ? true : false,
-  comfElm = jQuery("input[name='"+arName[0]+"_COMFORT']"),
+  elmType = elmName.replace(`${arName[0]}_`, ""), // e.g.LOWERING_COOLING
+  isComfort = (elmType == "LOWERING_COOLING") ? true : false,
+  comfElm = jQuery("input[name='"+arName[0]+"_LOWERING_COOLING']"),
   comfValIsValid = (comfElm.attr("valvalid") == "true") ? true : false,
   ecoElm = jQuery("input[name='"+arName[0]+"_LOWERING']"),
   ecoValIsValid = (ecoElm.attr("valvalid") == "true") ? true : false,
-  comfVal = parseInt(jQuery(comfElm[0]).val()),
-  ecoVal = parseInt(jQuery(ecoElm[0]).val()),
+  comfVal = parseFloat(jQuery(comfElm[0]).val()),
+  ecoVal = parseFloat(jQuery(ecoElm[0]).val()),
   comfOldElm = jQuery("#comfortOld"),
   ecoOldElm = jQuery("#ecoOld"),
   errorRow = jQuery("#errorRow"),
@@ -314,37 +314,37 @@
   //console.log("comfValIsValid: " + comfValIsValid + " - ecoValIsValid: " + ecoValIsValid);
 
   switch (elmType) {
-    case "COMFORT":
+    case "LOWERING_COOLING":
       if (comfValIsValid) {
         if (comfVal < ecoVal) {
-          jQuery(comfElm[0]).val(parseFloat(jQuery(comfOldElm).val()).toFixed(2));
+          comfVal = parseFloat(jQuery(comfOldElm).val());
           errorRow.show();
           errorComfElm.html(errorMsg).show();
           errorComfElm.fadeOut(fadeOutTime);
         }
       } else {
-        jQuery(comfElm[0]).val(parseFloat(jQuery(comfOldElm).val()).toFixed(2));
+        comfVal = parseFloat(jQuery(comfOldElm).val());
       }
       break;
     case "LOWERING":
       if (ecoValIsValid) {
         if (ecoValIsValid && (ecoVal > comfVal)) {
-          jQuery(ecoElm[0]).val(parseFloat(jQuery(ecoOldElm).val()).toFixed(2));
+          ecoVal = parseFloat(jQuery(ecoOldElm).val());
           errorRow.show();
           errorEcoElm.html(errorMsg).show();
           errorEcoElm.fadeOut(fadeOutTime);
         }
       } else {
-        jQuery(ecoElm[0]).val(parseFloat(jQuery(ecoOldElm).val()).toFixed(2));
+        ecoVal = parseFloat(jQuery(ecoOldElm).val());
       }
       break;
   }
-  comfVal = parseFloat(jQuery(comfElm).val()).toFixed(2);
-  ecoVal = parseFloat(jQuery(ecoElm).val()).toFixed(2);
+  comfVal = comfVal.toFixed(2);
+  ecoVal = ecoVal.toFixed(2);
   jQuery(comfElm).val(comfVal);
   jQuery(ecoElm).val(ecoVal);
 
-  jQuery(comfOldElm).val(jQuery(comfElm[0]).val());
-  jQuery(ecoOldElm).val(jQuery(ecoElm[0]).val());
+  jQuery(comfOldElm).val(comfVal);
+  jQuery(ecoOldElm).val(ecoVal);
 
 };
