--- occu/WebUI/www/config/easymodes/cc_rt_dev_master.tcl.orig
+++ occu/WebUI/www/config/easymodes/cc_rt_dev_master.tcl
@@ -21,7 +21,12 @@
   upvar psDescr descr
   array_clear param_descr
   array set param_descr $descr($param)
-  set min [format {%1.1f} $param_descr(MIN)]
+  set min $param_descr(MIN)
+
+  # Limit float to 1 decimal places
+  if {[llength [split $min "."]] == 2} {
+    set min [format {%1.1f} $min]
+  }
   return "$min"
 }
 
@@ -30,16 +35,21 @@
   upvar psDescr descr
   array_clear param_descr
   array set param_descr $descr($param)
-  set max [format {%1.1f} $param_descr(MAX)]
+  set max $param_descr(MAX)
+
+  # Limit float to 1 decimal places
+  if {[llength [split $max "."]] == 2} {
+    set max [format {%1.1f} $max]
+  }
   return "$max"
 }
 
 proc getTextField {type param value prn} {
   global psDescr
   set elemId 'separate_$type\_$prn'
-  # Limit float to 2 decimal places
+  # Limit float to 1 decimal places
   if {[llength [split $value "."]] == 2} {
-    set value [format {%1.2f} $value]
+    set value [format {%1.1f} $value]
   }
   set s "<input id=$elemId type=\"text\" size=\"5\" value=$value name=\"$param\" onblur=\"ProofAndSetValue(this.id, this.id, '[getMinValue $param]', '[getMaxValue $param]', 1)\"/>"
 
@@ -73,10 +83,10 @@
   set min $param_descr(MIN)
   set max $param_descr(MAX)
 
-  # Limit float to 2 decimal places
+  # Limit float to 1 decimal places
   if {[llength [split $min "."]] == 2} {
-    set min [format {%1.2f} $min]
-    set max [format {%1.2f} $max]
+    set min [format {%1.1f} $min]
+    set max [format {%1.1f} $max]
   }
   return "($min - $max)"
 }
--- occu/WebUI/www/config/ic_common.tcl.orig
+++ occu/WebUI/www/config/ic_common.tcl
@@ -1383,7 +1383,7 @@
 
           append s "<input type=\"hidden\" name=\"$param_id\"   value=\"$value_orig\" $id                 $access style=\"visibility:hidden;display:none;\" />"
           append s "<input type=\"text\"   name=\"__$param_id\" value=\"$value\"       id=\"$input_idval\" $access $hidden"
-          append s "  onblur=\" ProofAndSetValue('$input_idval', '${idval}', parseFloat($min), parseFloat($max), parseFloat([expr 1 / $factor]));\" /></td>"
+          append s "  onblur=\" ProofAndSetValue('$input_idval', '${idval}', '$min', '$max', parseFloat([expr 1 / $factor]));\" /></td>"
           append s "<td><div id=\"${input_idval}_unit\" $hidden>$unit ($min-$max)</div></td>"
       }
       "ENUM" {
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -39027,55 +39027,70 @@
     if (tokens[1]) value += tokens[1];
     srcElm.value = value;
   }
+  
+  var parsedValue;
+  var parsedMin = parseFloat(min);
+  var parsedMax = parseFloat(max);
 
+  var minSplit = min.toString().split(".");
+  var maxSplit = max.toString().split(".");
+  var digits = 0;
+
+  if(minSplit.length === 2) {
+    digits = Math.max(digits, minSplit[1].length);
+  }
+  if(maxSplit.length === 2) {
+    digits = Math.max(digits, maxSplit[1].length);
+  }
   // Check if float is allowed
   try {
     if (min.toString().indexOf(".") == -1 && max.toString().indexOf(".") == -1) {
       min = parseInt(min);
       max = parseInt(max);
-      value = (roundValue05(parseInt(value)));
+      value = Math.round(parseFloat(value));
     } else {
-      min = parseFloat(min).toFixed(1);
-      max = parseFloat(max).toFixed(1);
-      value = parseFloat(value);
-      srcElm.value = parseFloat(value);
+      min = parseFloat(min).toFixed(digits);
+      max = parseFloat(max).toFixed(digits);
+      var roundFactor = Math.pow(10, digits);
+      value = Math.round(parseFloat(value) * roundFactor) / roundFactor;
     }
+    parsedValue = parseFloat(value);
   } catch(e) {conInfo(e);}
 
   if (typeof value == "undefined")
   {
-    finalVal = min;
+    finalVal = parsedMin;
     ok = false;
   }
   else if (isNaN(value))
   {
-    finalVal = min;
+    finalVal = parsedMin;
     ok = false;
   }
-  else if (value < min)
+  else if (parsedValue < parsedMin)
   {
-    finalVal = min;
+    finalVal = parsedMin;
     ok = false;
   }
-  else if (value > max)
+  else if (parsedValue > parsedMax)
   {
-    finalVal = max;
+    finalVal = parsedMax;
     ok = false;
   }
 
   if (ok)
   {
     srcElm.style.backgroundColor = "#fffffe";
-    dstElm.value = value * dstValueFactor;
-    srcElm.value = dstElm.value;
+    dstElm.value = (parsedValue.toFixed(digits) * dstValueFactor);
+    srcElm.value = parsedValue.toFixed(digits);
     srcElm.setAttribute("valvalid", "true");
   }
   else
   {
     srcElm.setAttribute("valvalid", "false");
     srcElm.style.backgroundColor = "red";
-    dstElm.value = finalVal * dstValueFactor;
-    srcElm.value = dstElm.value;
+    dstElm.value = (finalVal.toFixed(digits) * dstValueFactor);
+    srcElm.value = finalVal.toFixed(digits);
     window.setTimeout(function(){srcElm.style.backgroundColor = "white";},1000);
   }
 };
