--- occu/WebUI/www/rega/esp/system.fn.orig
+++ occu/WebUI/www/rega/esp/system.fn
@@ -1274,7 +1274,13 @@
   Write("<td>");
   string sClass = "ControlBtnOff";
   if( objProg.Active() ) { sClass = "ControlBtnOn"; }
-    Write("<table class='"#sClass#" CLASS02102' id='"#pid#"Act' >"); 
+  integer uid = system.GetSessionVar("sessionUserID");
+  object oUser = dom.GetObject(uid);
+  if(oUser.UserLevel() == iulAdmin) {
+    Write("<table class='"#sClass#" CLASS02102' id='"#pid#"Act' onclick='isePrograms.SetActive("#pid#",!" #objProg.Active() # ")'>");
+  } else {
+    Write("<table class='"#sClass#" CLASS02102' style='cursor: default; border-top: solid 1px black; border-right: solid 1px black;' id='"#pid#"Act' >");
+  }
     Write("<tr>");
     Write("<td  class='CLASS02103'>");
         Write("<div align='center'>");
@@ -1283,7 +1289,11 @@
       Write("</td>");
     Write("</tr>");
     !Write("<tr><td>Aktiv</td></tr>");
-    Write("<tr><td>${actionStatusControlLblActive}</td></tr>");
+    if( objProg.Active() ) {
+      Write("<tr><td>${actionStatusControlLblActive}</td></tr>");
+    } else {
+      Write("<tr><td>${actionStatusControlLblNotActive}</td></tr>");
+    }
     Write("</table>");
   Write("</td>");
   if (system.GetVar("iStatusOnly") == 0)
@@ -1292,15 +1302,17 @@
     !Write("<div id='"#pid#"Start' class='ControlBtnOff CLASS02104'>Start</div>");
     Write("<div id='"#pid#"Start' class='ControlBtnOff CLASS02104'>${actionStatusControlLblStart}</div>");
     Write("<script type='text/javascript'>");
-    Write("new iseButtonProg("#pid#", "#objProg.Active()#");");
+    Write("new iseButtonProg("#pid#", "#objProg.Active()#", false);");
     Write("</script>");
     Write("</td>");
+    Write("<td><div id='"#pid#"StartThenOnly' class='ControlBtnOff CLASS02104'>${actionStatusControlLblStartThenOnly}</div>");
+    Write("<script type='text/javascript'> new iseButtonProg("#pid#", "#objProg.Active()#", true);</script></td>");
   }
   else
   {
-    Write("<td></td>");
+    Write("<td></td><td></td>");
   }
-  Write("<td></td><td></td><td></td>");
+  Write("<td></td><td></td>");
   
   Write("</tr>");
   Write("</table>");
--- occu/WebUI/www/rega/pages/tabs/control/programs.htm.orig
+++ occu/WebUI/www/rega/pages/tabs/control/programs.htm
@@ -51,7 +51,7 @@
         <th class="tHeaderCell CLASS03303" >${thDescription}</th>
         <th class="tHeaderCell CLASS03303" >${thActivity}</th>
         <th class="tHeaderCell CLASS03303" >${thLastExecution}</th>
-        <th class="tHeaderCell CLASS03303" >${thAction}</th>
+        <th class="tHeaderCell CLASS03303" ><div><span>${thAction}</span><img src="/ise/img/help.png" style="cursor: pointer; width:24px; height:24px; vertical-align:middle;" onclick="MessageBox.show(translateKey('HelpTitle'), translateKey('actionStatusControlLblStartHelp'), '', 400, 250);"></div></th>
       </tr>
     </thead>
       <tbody>
--- occu/WebUI/www/webui/js/lang/de/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.js
@@ -764,8 +764,11 @@
     "dialogBackspacePrevent" : "Bitte verwenden Sie zum Navigieren die Bedienelemente der Weboberfl%E4che.",
     "commentDialogNewHelpEnd" : "***********************************************",
     "commentActionStatusControl" : "***********************************************",
-    "actionStatusControlLblActive" : "Aktiv",
-    "actionStatusControlLblStart" : "Start",
+    "actionStatusControlLblActive" : "aktiv",
+    "actionStatusControlLblNotActive" : "nicht aktiv",
+    "actionStatusControlLblStart" : "Ausf%FChren",
+    "actionStatusControlLblStartThenOnly" : "Ausf%FChren<br/>(nur 1. DANN)",
+    "actionStatusControlLblStartHelp" : "<b>Ausf%FChren:</b><br/><br/>Führt das Programm inkl. aller Bedingungspr%FCfungen aus. Es werden die Aktionen des &quot;DANN&quot; Teiles des Programmes ausgef%FChrt dessen Bedingungsblock &quot;WAHR&quot; ergibt. Andernfalls werden die Aktionen des &quot;SONST&quot; Teiles ausgef%FChrt.<br/><br/><b>Ausf%FChren (nur 1. DANN):</b><br/><br/>F%FChrt lediglich die Aktionen innerhalb des ersten &quot;DANN&quot; Teiles des Programmes - ohne jegliche Bedingungspr%FCfungen - aus.",
     "actionStatusControlLblOn" : "Ein",
     "actionStatusControlLblOff" : "Aus",
     "actionStatusControlLblShortKeyPress" : "Kurzer<br/>Tastendruck",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.js
@@ -791,8 +791,11 @@
 
     "commentActionStatusControl" : "***********************************************",
 
-    "actionStatusControlLblActive" : "Active",
-    "actionStatusControlLblStart" : "Start",
+    "actionStatusControlLblActive" : "active",
+    "actionStatusControlLblNotActive" : "not active",
+    "actionStatusControlLblStart" : "Execute",
+    "actionStatusControlLblStartThenOnly" : "Execute<br/>(only 1st THEN)",
+    "actionStatusControlLblStartHelp" : "<b>Execute:</b><br/><br/>Executes the program including all condition checks. The actions within the &quot;THEN&quot; part of the program will be executed if a condition block becomes &quot;TRUE&quot;. Otherwise the actions within the &quot;ELSE&quot; part will be executed accordingly.<br/><br/><b>Execute (only 1st THEN):</b><br/><br/>Executes only the actions of the first &quot;THEN&quot; part of the program without considering any condition check whatsoever.",
     "actionStatusControlLblOn" : "On",
     "actionStatusControlLblOff" : "Off",
     "actionStatusControlLblShortKeyPress" : "Short<br/>button press",
--- occu/WebUI/www/webui/style.css.orig
+++ occu/WebUI/www/webui/style.css
@@ -5015,9 +5015,11 @@
 
 .CLASS02104 {
 	cursor:pointer;
-	line-height: 100px;
 	height:100px;
 	width:100px;
+	display: flex;
+	align-items: center;
+	justify-content: center;
 }
 
 .CLASS02105 {
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -28969,14 +28969,14 @@
   return arTmp[0]+':'+arTmp[1];
 };
 
-ExecuteProgram = function(dpid)
+ExecuteProgram = function(dpid, thenOnly)
 {
   var url = "/esp/exec.htm?sid=" + SessionId;
   var pb = "";
   pb += "object o = dom.GetObject( "+dpid+" );";
   pb += "if( o )";
   pb += "{";
-  pb += "  o.ProgramExecute();";
+  pb += (thenOnly === true) ? "  o.ProgramExecute();" : "  o.State(1);";
   pb += "}";
   var opts = {postBody: ReGa.encode(pb)};
   if(dbg)alert(pb);
@@ -31741,13 +31741,15 @@
  **/
 iseButtonProg = Class.create();
 iseButtonProg.prototype = {
-  initialize: function(id, progActive) {
+  initialize: function(id, progActive, thenOnly) {
     this.id = id;
     this.progActive = progActive;
     
-    this.startBtn = $(id + "Start");
+    this.startBtn = (thenOnly === true) ? $(id + "StartThenOnly") : $(id + "Start");
     this.actBtn = $(id + "Act");
     
+    this.thenOnly = thenOnly;
+    
     if ( progActive ) { ControlBtn.on(this.actBtn); }
     
     // Add event handlers
@@ -31758,7 +31760,7 @@
   onClickStart: function() 
   {
     ControlBtn.pushed(this.startBtn);
-    ExecuteProgram(this.id);
+    ExecuteProgram(this.id, this.thenOnly);
     var t = this;
     new PeriodicalExecuter(function(pe)
     {
