--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -20568,6 +20568,15 @@
               }
               conInfo("Check firmware version of all devices - number of devices: " + self.devList.length);
               if (self.devList.length > 0) {
+								function sortorder_devList(a , b){
+									if(a.type == b.type){
+										return (a.address < b.address) ? -1 : 1;
+									}else{
+										return (a.type < b.type) ? -1 : 1;
+									}
+								}
+								
+								self.devList.sort(sortorder_devList);
                 self.fetchAndSetDeviceVersion();
               }
             });
