--- occu/WebUI/www/webui/js/lang/en/translate.lang.extension.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.extension.js
@@ -973,7 +973,7 @@
     "thAvailableVersion": "Available version",
     "thInstalledVersion": "Installed version",
     "btnReload": "Reload",
-    "btnAccessPoints": "Update Access Points with</br>incompatible FW",
+    "btnAccessPoints": "Update Access Points</br>with incompatible FW",
     "apSearchMigrationDescription": "The password of the Homematic IP Access Point and an active Internet connection are required for the update. The password (PW) can be found on the sticker on the device.",
     "apSearchDescription": "The password of the Homematic IP Access Point is required for the update. The password (PW) can be found on the sticker on the device.",
     "password": "Password",
