--- occu/WebUI/www/rega/esp/functions.fn.orig
+++ occu/WebUI/www/rega/esp/functions.fn
@@ -871,11 +871,11 @@
 
 function ::getSpecialTranslationPrgDest() {
   ! ASIR - Adds the string "Alarmsignal: " to the original value.
-  if (oDP.HSSID() == "ACOUSTIC_ALARM_SELECTION") {
+  if (oCH.HssType() == "ACOUSTIC_ALARM_SELECTION") {
     sValue = "${acousticAlarm}: "#sValue;
   }
   ! ASIR - Adds the string "Alarmsignal: " to the original value.
-  if (oDP.HSSID() == "OPTICAL_ALARM_SELECTION") {
+  if (oCH.HssType() == "OPTICAL_ALARM_SELECTION") {
     sValue = "${opticalAlarm}: "#sValue;
   }
 
