--- occu/WebUI/www/config/easymodes/EnterFreeValue.tcl.orig
+++ occu/WebUI/www/config/easymodes/EnterFreeValue.tcl
@@ -7,7 +7,7 @@
 set unit_sec  "s"
 set unit_day "d"
 set unit_perc  "%"
-set unit_temp   " &deg;C"
+set unit_temp   "&nbsp;&deg;C"
 set unit_cf    "0"
 set free_value  "\${enterValue}"
 
@@ -27,9 +27,9 @@
   catch {
     switch $dev_ps(DISPLAY_TEMPERATUR_UNIT) {
 
-        0  {set unit_temp  " &deg;C"; return 0}
-        1  {set unit_temp  " &deg;F"; return 1}
-      default {set unit_temp  " *"; return -1}
+        0  {set unit_temp  "&nbsp;&deg;C"; return 0}
+        1  {set unit_temp  "&nbsp;&deg;F"; return 1}
+      default {set unit_temp  "&nbsp;*"; return -1}
     }
   }
   return 0
--- occu/WebUI/www/config/easymodes/cc_tc_ch_master.tcl.orig
+++ occu/WebUI/www/config/easymodes/cc_tc_ch_master.tcl
@@ -137,7 +137,7 @@
 	# Heizungsregler die gesendeten Werte in Grad Celsius erwartet. Die Werte werden durch CC_conv_CF errechnet
 	append HTML_PARAMS(separate_1) "<input type=\"hidden\" id=\"separate_${special_input_id}_6\" name=\"TEMPERATUR_LOWERING_VALUE\">"
 	#Das C in <span>C</span> der naechsten Zeile wird in der Funktion /js/ic_setprofiles/CC_setUnit() evtl. in F(ahrenheit) gewandelt
-	append HTML_PARAMS(separate_1) "&deg;<span class=\"CF\">C</span></td></tr>"
+	append HTML_PARAMS(separate_1) "&nbsp;&deg;<span class=\"CF\">C</span></td></tr>"
 
 #	Komforttemperatur
 	append HTML_PARAMS(separate_1) "<tr><td>$PROFILE_PNAME(TEMPERATUR_COMFORT_VALUE)&nbsp;&nbsp;<img src=\"/ise/img/cc_sun.png\"></td><td>"
@@ -146,7 +146,7 @@
 	# Heizungsregler die gesendeten Werte in Grad Celsius erwartet. Die Werte werden durch CC_conv_CF errechnet
 	append HTML_PARAMS(separate_1) "<input type=\"hidden\" id=\"separate_${special_input_id}_5\" name=\"TEMPERATUR_COMFORT_VALUE\">"
 	#Das C in <span>C</span> der naechsten Zeile wird in der Funktion /js/ic_setprofiles/CC_setUnit() evtl. in F(ahrenheit) gewandelt
-	append HTML_PARAMS(separate_1) "&deg;<span class=\"CF\">C</span></td></tr>"
+	append HTML_PARAMS(separate_1) "&nbsp;&deg;<span class=\"CF\">C</span></td></tr>"
 	
 #	append HTML_PARAMS(separate_1) "<td colspan=\"2\"><hr class=\"CLASS20600\"/>"
 	append HTML_PARAMS(separate_1) "<td colspan=\"2\"><hr>"
@@ -193,7 +193,7 @@
 	# Heizungsregler die gesendeten Werte in Grad Celsius erwartet. Die Werte werden durch CC_conv_CF errechnet
 	append HTML_PARAMS(separate_1) "<input type=\"hidden\" id=\"separate_${special_input_id}_8\" name=\"TEMPERATUR_PARTY_VALUE\">"
 	#Das C in <span>C</span> der naechsten Zeile wird in der Funktion /js/ic_setprofiles/CC_setUnit() evtl. in F(ahrenheit) gewandelt
-	append HTML_PARAMS(separate_1) "&deg;<span class=\"CF\">C</span></td></tr></table>"
+	append HTML_PARAMS(separate_1) "&nbsp;&deg;<span class=\"CF\">C</span></td></tr></table>"
 
 #	rechte Zelle endet
 	append HTML_PARAMS(separate_1) "</td></tr></table>"
--- occu/WebUI/www/config/easymodes/cc_tc_ch_master_le_v1_9.tcl.orig
+++ occu/WebUI/www/config/easymodes/cc_tc_ch_master_le_v1_9.tcl
@@ -136,7 +136,7 @@
 	# Heizungsregler die gesendeten Werte in Grad Celsius erwartet. Die Werte werden durch CC_conv_CF errechnet
 	append HTML_PARAMS(separate_1) "<input type=\"hidden\" id=\"separate_${special_input_id}_6\" name=\"TEMPERATUR_LOWERING_VALUE\">"
 	#Das C in <span>C</span> der naechsten Zeile wird in der Funktion /js/ic_setprofiles/CC_setUnit() evtl. in F(ahrenheit) gewandelt
-	append HTML_PARAMS(separate_1) "&deg;<span class=\"CF\">C</span></td></tr>"
+	append HTML_PARAMS(separate_1) "&nbsp;&deg;<span class=\"CF\">C</span></td></tr>"
 
 #	Komforttemperatur
 	append HTML_PARAMS(separate_1) "<tr><td>$PROFILE_PNAME(TEMPERATUR_COMFORT_VALUE)&nbsp;&nbsp;<img src=\"/ise/img/cc_sun.png\"></td><td>"
@@ -145,7 +145,7 @@
 	# Heizungsregler die gesendeten Werte in Grad Celsius erwartet. Die Werte werden durch CC_conv_CF errechnet
 	append HTML_PARAMS(separate_1) "<input type=\"hidden\" id=\"separate_${special_input_id}_5\" name=\"TEMPERATUR_COMFORT_VALUE\">"
 	#Das C in <span>C</span> der naechsten Zeile wird in der Funktion /js/ic_setprofiles/CC_setUnit() evtl. in F(ahrenheit) gewandelt
-	append HTML_PARAMS(separate_1) "&deg;<span class=\"CF\">C</span></td></tr>"
+	append HTML_PARAMS(separate_1) "&nbsp;&deg;<span class=\"CF\">C</span></td></tr>"
 	
 #	append HTML_PARAMS(separate_1) "<td colspan=\"2\"><hr class=\"CLASS20600\"/>"
 	append HTML_PARAMS(separate_1) "<td colspan=\"2\"><hr>"
@@ -160,7 +160,7 @@
 	# Heizungsregler die gesendeten Werte in Grad Celsius erwartet. Die Werte werden durch CC_conv_CF errechnet
 	append HTML_PARAMS(separate_1) "<input type=\"hidden\" id=\"separate_${special_input_id}_8\" name=\"TEMPERATUR_PARTY_VALUE\">"
 	#Das C in <span>C</span> der naechsten Zeile wird in der Funktion /js/ic_setprofiles/CC_setUnit() evtl. in F(ahrenheit) gewandelt
-	append HTML_PARAMS(separate_1) "&deg;<span class=\"CF\">C</span></td></tr></table>"
+	append HTML_PARAMS(separate_1) "&nbsp;&deg;<span class=\"CF\">C</span></td></tr></table>"
 
 #	rechte Zelle endet
 	append HTML_PARAMS(separate_1) "</td></tr></table>"
--- occu/WebUI/www/pda/eq3/controls/generic.tcl.orig
+++ occu/WebUI/www/pda/eq3/controls/generic.tcl
@@ -105,7 +105,7 @@
 		
 		
 		! Ausgabe
-		Write(value # " " # unit);
+		Write(value # "&nbsp;" # unit);
 	}
 	
 #	puts "<pre>$script</pre>"
@@ -113,4 +113,4 @@
 	if {"" == $value } { set value "Unbekannt" }
 	
 	return $value
-}
\ No newline at end of file
+}
--- occu/WebUI/www/pda/eq3/controls/temp.tcl.orig
+++ occu/WebUI/www/pda/eq3/controls/temp.tcl
@@ -24,19 +24,19 @@
 	set status [rega_exec $script]
 	
 	switch -exact -- $cmd {
-		temp_up_5   { set correctionValue "(+5 &deg;C)" }
-		temp_up_2   { set correctionValue "(+2 &deg;C)" }
-		temp_up_1   { set correctionValue "(+1 &deg;C)" }
-		temp_down_1 { set correctionValue "(-1 &deg;C)" }
-		temp_down_2 { set correctionValue "(-2 &deg;C") }
-		temp_down_5 { set correctionValue "(-5 &deg;C)" }
+		temp_up_5   { set correctionValue "(+5&nbsp;&deg;C)" }
+		temp_up_2   { set correctionValue "(+2&nbsp;&deg;C)" }
+		temp_up_1   { set correctionValue "(+1&nbsp;&deg;C)" }
+		temp_down_1 { set correctionValue "(-1&nbsp;&deg;C)" }
+		temp_down_2 { set correctionValue "(-2&nbsp;&deg;C") }
+		temp_down_5 { set correctionValue "(-5&nbsp;&deg;C)" }
 		default     { set correctionValue "" }
 	}
 	
 	set    result [status_separator]
 	append result {<tr><td class="status">}
 	append result {<div class="caption">Soll-Temperatur</div>}
-	append result "<div class=\"center\">$status &deg;C $correctionValue</div>"
+	append result "<div class=\"center\">$status&nbsp;&deg;C $correctionValue</div>"
 	append result {</td></tr>}
 	
 	if { "" != $correctionValue } {
@@ -53,12 +53,12 @@
 	global sid favListId favId UI_URLBASE
 	set result ""
 	
-	append result [button "+5 &deg;" "$UI_URLBASE/fav.cgi?sid=$sid&uiStyle=[getUIStyle]&favListId=$favListId&favId=$favId&cmd=temp_up_5"]
-	append result [button "+2 &deg;" "$UI_URLBASE/fav.cgi?sid=$sid&uiStyle=[getUIStyle]&favListId=$favListId&favId=$favId&cmd=temp_up_2"]
-	append result [button "+1 &deg;" "$UI_URLBASE/fav.cgi?sid=$sid&uiStyle=[getUIStyle]&favListId=$favListId&favId=$favId&cmd=temp_up_1"]
-	append result [button "-1 &deg;" "$UI_URLBASE/fav.cgi?sid=$sid&uiStyle=[getUIStyle]&favListId=$favListId&favId=$favId&cmd=temp_down_1"]
-	append result [button "-2 &deg;" "$UI_URLBASE/fav.cgi?sid=$sid&uiStyle=[getUIStyle]&favListId=$favListId&favId=$favId&cmd=temp_down_2"]
-	append result [button "-5 &deg;" "$UI_URLBASE/fav.cgi?sid=$sid&uiStyle=[getUIStyle]&favListId=$favListId&favId=$favId&cmd=temp_down_5"]
+	append result [button "+5&nbsp;&deg;" "$UI_URLBASE/fav.cgi?sid=$sid&uiStyle=[getUIStyle]&favListId=$favListId&favId=$favId&cmd=temp_up_5"]
+	append result [button "+2&nbsp;&deg;" "$UI_URLBASE/fav.cgi?sid=$sid&uiStyle=[getUIStyle]&favListId=$favListId&favId=$favId&cmd=temp_up_2"]
+	append result [button "+1&nbsp;&deg;" "$UI_URLBASE/fav.cgi?sid=$sid&uiStyle=[getUIStyle]&favListId=$favListId&favId=$favId&cmd=temp_up_1"]
+	append result [button "-1&nbsp;&deg;" "$UI_URLBASE/fav.cgi?sid=$sid&uiStyle=[getUIStyle]&favListId=$favListId&favId=$favId&cmd=temp_down_1"]
+	append result [button "-2&nbsp;&deg;" "$UI_URLBASE/fav.cgi?sid=$sid&uiStyle=[getUIStyle]&favListId=$favListId&favId=$favId&cmd=temp_down_2"]
+	append result [button "-5&nbsp;&deg;" "$UI_URLBASE/fav.cgi?sid=$sid&uiStyle=[getUIStyle]&favListId=$favListId&favId=$favId&cmd=temp_down_5"]
 
 	return $result
 }
--- occu/WebUI/www/rega/esp/controls/heating_control.fn.orig
+++ occu/WebUI/www/rega/esp/controls/heating_control.fn
@@ -3,7 +3,7 @@
   real rState = 0;
   real rMin = 4.5;
   real rMax = 30.5;
-  string sUnit = "&deg;<i>C</i>";
+  string sUnit = "&deg;C";
   string bUnit = "V";
   string vUnit = "%";
   integer iReadOnlyState = 0;
@@ -41,7 +41,7 @@
     sUnit = oLevel.ValueUnit();
     if (sUnit == 'degree')
     {
-      sUnit = "&deg;<i>C</i>";
+      sUnit = "&deg;C";
     }
     ! rMin = oLevel.MetaData("MIN").ToFloat();
     ! rMax = oLevel.MetaData("MAX").ToFloat();
@@ -62,10 +62,10 @@
       WriteLine("<table id='"#chnId#"tblShowTemp' class='ControlBtnInfo hidden' style='width:200px'>");
         WriteLine("<tr>");
           WriteLine("<td>${stringTableWeatherTemperature}</td>");
-          WriteLine("<td><span id='"#chnId#"actTemp'></span><span><b>"#sUnit#"</b></span></td>");
+          WriteLine("<td><span id='"#chnId#"actTemp'></span><span>&nbsp;<b>"#sUnit#"</b></span></td>");
         WriteLine("</tr>");
       WriteLine("</table>");
-      !WriteLine("<div class='ControlBtnInfo _CLASS02550' style='line-height:15px; min-width:200px' id='"#chnId#"actualTemp' ><span>${stringTableWeatherTemperature}</span><span id='actTemp'"#sUnit#"></span></div>");
+      !WriteLine("<div class='ControlBtnInfo _CLASS02550' style='line-height:15px; min-width:200px' id='"#chnId#"actualTemp' ><span>${stringTableWeatherTemperature}</span><span id='actTemp'>&nbsp;"#sUnit#"</span></div>");
     WriteLine("</td>");
   WriteLine("</tr>");
  
@@ -76,7 +76,7 @@
        WriteLine("<table id='"#chnId#"tblShowBatteryState' class='ControlBtnInfo hidden' style='width:200px"#cssDisplay#"'>");
          WriteLine("<tr>");
            WriteLine("<td>${stringTableClimateControlRTTransceiverBatteryState}</td>");
-           WriteLine("<td><span id='"#chnId#"actBatteryState'></span><span><b>"#bUnit#"</b></span></td>");
+           WriteLine("<td><span id='"#chnId#"actBatteryState'></span><span>&nbsp;<b>"#bUnit#"</b></span></td>");
         WriteLine("</tr>");
        WriteLine("</table>");
      WriteLine("</td>");
@@ -90,7 +90,7 @@
        WriteLine("<table id='"#chnId#"tblShowValveState' class='ControlBtnInfo hidden' style='width:200px"#cssDisplay#"'>");
          WriteLine("<tr>");
            WriteLine("<td>${stringTableHeatingLevel}</td>");
-           WriteLine("<td><span id='"#chnId#"actValveState'></span><span><b>"#vUnit#"</b></span></td>");
+           WriteLine("<td><span id='"#chnId#"actValveState'></span><span>&nbsp;<b>"#vUnit#"</b></span></td>");
          WriteLine("</tr>");
        WriteLine("</table>");
      WriteLine("</td>");
@@ -103,7 +103,7 @@
     WriteLine("  <div id='button"#chnId#"' class='PercBtn' >");
     WriteLine("  <table width='100%' height='100%' class='CLASS02530'>");
     WriteLine("    <tr class='CLASS02531'>");
-    WriteLine("      <td colspan='2' class='CLASS02546'><input id='"#chnId#"Deg' type='text' value='"#rState.ToString(2)#"' size='4' class='CLASS02542' style='text-align:center' "#sReadOnly#" /><span id='"#chnId#"unitDegree'><b>"#sUnit#"</b></span></td></tr>");
+    WriteLine("      <td colspan='2' class='CLASS02546'><input id='"#chnId#"Deg' type='text' value='"#rState.ToString(2)#"' size='4' class='CLASS02542' style='text-align:center' "#sReadOnly#" /><span id='"#chnId#"unitDegree'>&nbsp;<b>"#sUnit#"</b></span></td></tr>");
     WriteLine("      <input id='"#chnId#"Perc' type='hidden' value='"#rState.ToString(2)#"' />");
     WriteLine("      <td style='text-align:center;' class='CLASS02548'>");
     WriteLine("      <table cellspacing='0' cellpadding='0'><tr><td><img id='"#chnId#"Up' height='12px' width='40px' src='/ise/img/up.png' class='CLASS02543' /></td></tr><tr><td><img id='"#chnId#"Down' height='12px' width='40px' src='/ise/img/down.png' class='CLASS02543' /></td></tr></table>");
@@ -242,7 +242,7 @@
   integer chInterfaceID = ch.Interface();
   object chInterface = dom.GetObject(chInterfaceID);
 
-  string tUnit = "&deg;<i>C</i>";
+  string tUnit = "&deg;C";
   string hUnit = "&#37;";
   string bUnit = "V";
   string vUnit = "%";
@@ -277,7 +277,7 @@
       WriteLine("<table class='ControlBtnInfo' style='width:200px'>");
         WriteLine("<tr>");
           WriteLine("<td  id='"#chnId#"ACTUAL_TEMPERATURE'>${stringTableWeatherTemperature}</td>");
-          WriteLine("<td><span id='"#chnId#"valACTUAL_TEMPERATURE'></span><span><b>"#tUnit#"</b></span></td>");
+          WriteLine("<td><span id='"#chnId#"valACTUAL_TEMPERATURE'></span><span>&nbsp;<b>"#tUnit#"</b></span></td>");
         WriteLine("</tr>");
       WriteLine("</table>");
     WriteLine("</td>");
@@ -287,7 +287,7 @@
         WriteLine("<table class='ControlBtnInfo' style='width:200px'>");
           WriteLine("<tr>");
             WriteLine("<td>${stringTableWeatherHumidity}</td>");
-            WriteLine("<td><span id='"#chnId#"valHUMIDITY'></span><span><b>"#hUnit#"</b></span></td>");
+            WriteLine("<td><span id='"#chnId#"valHUMIDITY'></span><span>&nbsp;<b>"#hUnit#"</b></span></td>");
           WriteLine("</tr>");
         WriteLine("</table>");
       WriteLine("</td>");
@@ -311,7 +311,7 @@
         WriteLine("<table class='ControlBtnInfo' style='width:200px'>");
           WriteLine("<tr>");
             WriteLine("<td id='"#chnId#"LEVEL'>${stringTableHeatingLevel}</td>");
-            WriteLine("<td><span id='"#chnId#"valLEVEL'></span><span><b>"#vUnit#"</b></span></td>");
+            WriteLine("<td><span id='"#chnId#"valLEVEL'></span><span>&nbsp;<b>"#vUnit#"</b></span></td>");
           WriteLine("</tr>");
         WriteLine("</table>");
       WriteLine("</td>");
@@ -324,7 +324,7 @@
         WriteLine("<table class='ControlBtnInfo' style='width:200px'>");
           WriteLine("<tr>");
             WriteLine("<td>${stringTableClimateControlRTTransceiverBatteryState}</td>");
-            WriteLine("<td><span id='"#chnId#"valBATTERY_STATE'></span><span><b>"#bUnit#"</b></span></td>");
+            WriteLine("<td><span id='"#chnId#"valBATTERY_STATE'></span><span>&nbsp;<b>"#bUnit#"</b></span></td>");
           WriteLine("</tr>");
         WriteLine("</table>");
       WriteLine("</td>");
@@ -342,7 +342,7 @@
     WriteLine("  <div id='button"#chnId#"' class='PercBtn' >");
     WriteLine("  <table width='100%' height='100%' class='CLASS02530'>");
     WriteLine("    <tr class='CLASS02531'>");
-    WriteLine("      <td colspan='2' class='CLASS02546'><input id='"#chnId#"Deg' type='text' value='"#rState.ToString(2)#"' size='4' class='CLASS02542' style='text-align:center' "#sReadOnly#" /><span id='"#chnId#"unitDegree'><b>"#tUnit#"</b></span></td></tr>");
+    WriteLine("      <td colspan='2' class='CLASS02546'><input id='"#chnId#"Deg' type='text' value='"#rState.ToString(2)#"' size='4' class='CLASS02542' style='text-align:center' "#sReadOnly#" /><span id='"#chnId#"unitDegree'>&nbsp;<b>"#tUnit#"</b></span></td></tr>");
     WriteLine("      <input id='"#chnId#"Perc' type='hidden' value='"#rState.ToString(2)#"' />");
     WriteLine("      <td style='text-align:center;' class='CLASS02548'>");
     WriteLine("      <table cellspacing='0' cellpadding='0'><tr><td><img id='"#chnId#"Up' height='12px' width='40px' src='/ise/img/up.png' class='CLASS02543' /></td></tr><tr><td><img id='"#chnId#"Down' height='12px' width='40px' src='/ise/img/down.png' class='CLASS02543' /></td></tr></table>");
@@ -525,7 +525,7 @@
 
 ! ALPHA-IP-RBGa
 function ::CreateHeatingControlHmIPa() {
-  string tUnit = "&deg;<i>C</i>";
+  string tUnit = "&deg;C";
   string hUnit = "&#37;";
   real actTemp = -1.0;
   real setPointValue = -1.0;
@@ -570,7 +570,7 @@
     WriteLine("<td>");
       WriteLine("<table class='ControlBtnInfo' style='width:200px'>");
         WriteLine("<tr>");
-          WriteLine("<td>${stringTableWeatherTemperature}:&nbsp;"#actTemp.ToString(2)#"<span><b>"#tUnit#"</b></span></td>");
+          WriteLine("<td>${stringTableWeatherTemperature}:&nbsp;"#actTemp.ToString(2)#"<span>&nbsp;<b>"#tUnit#"</b></span></td>");
         WriteLine("</tr>");
       WriteLine("</table>");
     WriteLine("</td>");
@@ -588,7 +588,7 @@
     WriteLine("<td>");
       WriteLine("<table class='ControlBtnInfo' style='width:200px'>");
         WriteLine("<tr>");
-          WriteLine("<td>${stringTableSetPointTemperature}: "#setPointValue.ToString(2)#"<span><b>"#tUnit#"</b></span></td>");
+          WriteLine("<td>${stringTableSetPointTemperature}: "#setPointValue.ToString(2)#"<span>&nbsp;<b>"#tUnit#"</b></span></td>");
         WriteLine("</tr>");
       WriteLine("</table>");
     WriteLine("</td>");
@@ -630,4 +630,4 @@
     WriteLine("</td>");
   WriteLine("</tr>");
 
-}
\ No newline at end of file
+}
--- occu/WebUI/www/rega/esp/controls/thermostat.fn.orig
+++ occu/WebUI/www/rega/esp/controls/thermostat.fn
@@ -4,7 +4,7 @@
   real rState = 0;
   real rMin = 6;
   real rMax = 30;
-  string sUnit = "&deg;<i>C</i>";
+  string sUnit = "&deg;C";
   integer iReadOnlyState = 0;
 
   object oLevel = cObj.DPByControl("TEMP.SETPOINT");
@@ -15,7 +15,7 @@
     sUnit = oLevel.ValueUnit();
     if (sUnit == 'degree')
     {
-      sUnit = "&deg;<i>C</i>";
+      sUnit = "&deg;C";
     }
     rMin = oLevel.MetaData("MIN").ToFloat();
     rMax = oLevel.MetaData("MAX").ToFloat();
@@ -34,7 +34,7 @@
   WriteLine("  <div id='button"#chnId#"' class='PercBtn' >");
   WriteLine("  <table width='100%' height='100%' class='CLASS02530'>");
   WriteLine("    <tr class='CLASS02531'>");
-  WriteLine("      <td colspan='2' class='CLASS02546'><input id='"#chnId#"Deg' type='text' value='"#rState.ToString(2)#"' size='3' class='CLASS02542' "#sReadOnly#" /><b>"#sUnit#"</b></td></tr>");
+  WriteLine("      <td colspan='2' class='CLASS02546'><input id='"#chnId#"Deg' type='text' value='"#rState.ToString(2)#"' size='3' class='CLASS02542' "#sReadOnly#" />&nbsp;<b>"#sUnit#"</b></td></tr>");
   WriteLine("      <input id='"#chnId#"Perc' type='hidden' value='"#rState.ToString(2)#"' />");
   WriteLine("      <td style='text-align: center' class='CLASS02548'>");
   WriteLine("      <table cellspacing='0' cellpadding='0'><tr><td><img id='"#chnId#"Up' height='12px' width='40px' src='/ise/img/up.png' class='CLASS02543' /></td></tr><tr><td><img id='"#chnId#"Down' height='12px' width='40px' src='/ise/img/down.png' class='CLASS02543' /></td></tr></table>");
@@ -50,4 +50,4 @@
   WriteLine("</script>");
 
   return;
-}
\ No newline at end of file
+}
--- occu/WebUI/www/rega/esp/datapointconfigurator.fn.orig
+++ occu/WebUI/www/rega/esp/datapointconfigurator.fn
@@ -1316,13 +1316,12 @@
             }
 
             string sValueUnit = oDP.ValueUnit();
-            string sSpace = " ";
-            if( sValueUnit == "degree" ) { sValueUnit = "&deg;"; sSpace = ""; }
+            string sSpace = "&nbsp;";
+            if( sValueUnit == "degree" ) { sValueUnit = "&deg;"; }
             if( sValueUnit == "100%" )
             {
               value = value * 100;
               sValueUnit = "%";
-              sSpace = "";
             }
 
             string sValue = "?";
--- occu/WebUI/www/rega/esp/programs.fn.orig
+++ occu/WebUI/www/rega/esp/programs.fn
@@ -614,7 +614,7 @@
                         
                         if( unit.Length() && sValue == "" )
                         {
-                          sRet=sRet#" "#unit;
+                          sRet=sRet#"&nbsp;"#unit;
                         }
                       }
                     }
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -14132,7 +14132,7 @@
     // Adds a selectbox for the temperature to the party mode dialog
     var selBox = "<select>";
     for (var loop = 5; loop <= 30; loop++) {
-      selBox += "<option value='"+loop+"'>"+loop+"°C</option>";
+      selBox += "<option value='"+loop+"'>"+loop+"&nbsp;&deg;C</option>";
     }
     selBox += "</select>";
 
@@ -26006,7 +26006,7 @@
       value = min;
     }
 
-    if ((unit == "°C") && (value != min) && (value != max)) {
+    if ((unit == "°C" || unit == "&deg;C") && (value != min) && (value != max)) {
       value = roundValue05(value);
     }
 
@@ -32223,7 +32223,7 @@
     this.min = min;
     this.max = max;
     this.factor = 100/(max-min);
-    this.unit = " &deg;C";
+    this.unit = "&nbsp;&deg;C";
     this.statusOFF = "OFF";
     this.statusON = "ON";
     this.opts = opts;
@@ -32685,7 +32685,7 @@
     this.max = 30;
     this.off = 4.5;
     this.on = 30.5;
-    this.unit = "°C";
+    this.unit = "&nbsp;&deg;C";
     this.factor = 100/(this.max-this.min);
     this.iViewOnly = false;
     this.bSliderPosFlag = false;
@@ -33166,7 +33166,7 @@
     this.min = min;
     this.max = max;
     this.factor = 100/(max-min);
-    this.unit = " mHz";
+    this.unit = "&nbsp;mHz";
     if(bSliderPosFlag) 
     {
         this.bSliderPosFlag = bSliderPosFlag;
@@ -34854,12 +34854,19 @@
   var isFarVal = false;
   if (!checkDecimal(tmpInput)) {
     // Zeichen für Temperaturangaben rausfiltern
+    if (tmpInput.indexOf('&deg;C') != -1) {
+      tmpInput = tmpInput.substr(0, tmpInput.indexOf('&deg;C'));
+    }
     if (tmpInput.indexOf('°C') != -1) {
       tmpInput = tmpInput.substr(0, tmpInput.indexOf('°C'));
     }
     if (tmpInput.indexOf('C') != -1) {
       tmpInput = tmpInput.substr(0, tmpInput.indexOf('C'));
     }
+    if (tmpInput.indexOf('&deg;F') != -1) {
+      tmpInput = tmpInput.substr(0, tmpInput.indexOf('&deg;F'));
+      isFarVal = true;      
+    }
     if (tmpInput.indexOf('°F') != -1) {
       tmpInput = tmpInput.substr(0, tmpInput.indexOf('°F'));
       isFarVal = true;      
@@ -35352,7 +35359,7 @@
   },
 
   changeToKilo: function() {
-    return (this.summedUpEnergy/1000).toFixed(3);
+    return (this.summedUpEnergy/1000).toFixed(2);
   },
 
   setEnergyCounterPanel: function() {
@@ -35427,8 +35434,8 @@
           energyCostCCU = ((energyConsumptionCCU * this.EnergyPrice) / unitFactorCCU).toFixed(2) + " " + this.energyConfig.currency,
           energyCostDevice = ((energyConsumptionDevice * this.EnergyPrice) / unitFactorDevice).toFixed(2) + " " + this.energyConfig.currency;
         if (this.measurementType == this.sensorTypeID.gas) {
-          j_energyCostCCU.text(energyConsumptionCCU.toFixed(1) + kWh + " = " + energyCostCCU).show();
-          j_energyCostDevice.text(energyConsumptionDevice.toFixed(1) + kWh + " = " + energyCostDevice).show();
+          j_energyCostCCU.text(energyConsumptionCCU.toFixed(2) + kWh + " = " + energyCostCCU).show();
+          j_energyCostDevice.text(energyConsumptionDevice.toFixed(2) + kWh + " = " + energyCostDevice).show();
         } else {
           j_energyCostCCU.text(energyCostCCU).show();
           j_energyCostDevice.text(energyCostDevice).show();
@@ -41777,7 +41784,7 @@
       //msg += "<td><input style=\"text-align: right;\" maxLength=\"5\" size=\"7\" id=\"starttime_"   +dayidx +"_"+i+"\" disabled=\"disabled\" type=\"text\" value=\""+ this.tom_toTime( endtime) +"\">Uhr</td>";
       msg += "<td><input style=\"text-align: right;\" maxLength=\"5\" size=\"7\" id=\""+prgNr+"starttime_" + dayidx + "_" + i + "\" disabled=\"disabled\" type=\"text\" value=\"" + this.tom_toTime(endtime) + "\">" + translateKey('lblProgramTimeExtension') + "</td>";
       msg += "<td><input style=\"text-align: right;\" maxLength=\"5\" size=\"7\" id=\""+prgNr+"endtime_" + dayidx + "_" + i + "\" " + (i == timeouts.length - 1 ? 'disabled=\"disabled\"' : '') + " type=\"text\" value=\"" + this.tom_toTime(timeouts[i][tom_endtime]) + "\" onblur=\""+prgNr+"tom.tom_checkAndSetTime ('" + day + "', this, " + i + "); document.getElementById('"+prgNr+"starttime_" + dayidx + "_" + (i + 1) + "').value=this.value; \">" + translateKey('lblProgramTimeExtension') + "</td>";
-      msg += "<td><input style=\"text-align: right;\" maxLength=\"4\" size=\"3\" name=\""+prgNr+"temp_tmp\" id=\""+prgNr+"temperature_" + dayidx + "_" + i + "_tmp\" value=\"" + timeouts[i][tom_temperature].toFixed(1) + "\" onblur=\""+prgNr+"tom.tom_setDirty ('" + day + "', this, " + i + ");\">&deg;<span class=\"CF\">C</span></td>";
+      msg += "<td><input style=\"text-align: right;\" maxLength=\"4\" size=\"3\" name=\""+prgNr+"temp_tmp\" id=\""+prgNr+"temperature_" + dayidx + "_" + i + "_tmp\" value=\"" + timeouts[i][tom_temperature].toFixed(1) + "\" onblur=\""+prgNr+"tom.tom_setDirty ('" + day + "', this, " + i + ");\">&nbsp;&deg;<span class=\"CF\">C</span></td>";
       // im naechsten Input-Feld stehen versteckt die zum Aktor zu uebertragenden Werte. Da der Aktor Werte im Grad Celsius erwartet,
       // muss der im Kanalparameter evtl. in F angezeigte Wert immer in Celsius umgerechntet und zwischengespeichert werden.
       msg += "<td><input style=\"display:none\" size=\"3\" name=\""+prgNr+"temp\" id=\""+prgNr+ "temperature_" + dayidx + "_" + i + "\" value=\"" + timeouts[i][tom_temperature].toFixed(1) + "\"></td>";
