--- occu/WebUI/www/config/cp_network.cgi.orig
+++ occu/WebUI/www/config/cp_network.cgi
@@ -304,8 +304,9 @@
   set dhcp ""
   set dns1 ""
   set dns2 ""
+  set vpn ""
   
-  read_config dhcp hostname ip mask gw dns1 dns2
+  read_config dhcp hostname ip mask gw dns1 dns2 vpn
   
   http_head
   
@@ -433,6 +434,42 @@
 
         }
       }
+      table_row {class="CLASS21115"} {
+        table_data {class="CLASS21116"} {
+          puts "\${dialogSettingsNetworkTDVPNSettings}"
+        }
+        table_data {align=left} {
+          table {
+            table_row {
+              set checked ""
+              if {$vpn == "1"} { set checked "checked=true" }
+              table_data {class="CLASS21112"} {colspan="2"} {
+                cgi_checkbox mode=vpn {id="check_vpn"} $checked
+                puts "\${dialogSettingsNetworkVPNLblActivate}"
+              }
+            }
+            table_row {
+              table_data {width="10"} { }
+              table_data {class="CLASS21112"} {
+                division {class="popupControls CLASS21107"} {
+                  if {$vpn == "1"} {
+                    division {class="CLASS21117"} {onClick="window.open('/tailscale/', '_blank').focus();"} {
+                      puts "\${dialogSettingsNetworkVPNLblAuth}"
+                    }
+                  } else {
+                    division {class="CLASS21117"} {style="color:grey"} {
+                      puts "\${dialogSettingsNetworkVPNLblAuth}"
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+        table_data {class="CLASS21113"} {align="center"} {
+          puts "<img src=\"/ise/img/help.png\" onclick=\"MessageBox.show(translateKey('dialogSettingsNetworkVPNLblActivate')+' - '+translateKey('HelpTitle'), translateKey('dialogSettingsNetworkVPNHelpP1')+'<br/><br/>'+translateKey('dialogSettingsNetworkVPNHelpP2'), '', 450, 320) ;\">"
+        }
+      }
       table_row {class="CLASS21119"} {
         table_data {class="CLASS21116"} {
           #puts "Zertifikat<br />erstellen"
@@ -632,6 +669,7 @@
         pb += "&dns2=0.0.0.0";
         }
         
+        pb += "&vpn="+(document.getElementById("check_vpn").checked?"1":"0");
         var opts = {
           postBody: pb,
           sendXML: false,
@@ -796,9 +834,9 @@
     import dhcp
     import dns1
     import dns2
-    if { [write_config $dhcp $hostname $ip $mask $gw $dns1 $dns2] } {
+    import vpn
+    if { [write_config $dhcp $hostname $ip $mask $gw $dns1 $dns2 $vpn] } {
       #activate the new settings
-    catch {exec "/etc/init.d/S40network" "restart" 2> &1 > /dev/null}
       puts "Success"
       # puts "\${dialogSettingsNetworkMessageSaveSettingsSucceed}"
       return
@@ -832,8 +870,8 @@
   return 0
 }
 
-proc get_current_config {dhcp_var hostname_var ip_var mask_var gw_var dns1_var dns2_var} {
-  upvar $dhcp_var dhcp $hostname_var hostname $ip_var ip $mask_var mask $gw_var gw $dns1_var dns1 $dns2_var dns2
+proc get_current_config {dhcp_var hostname_var ip_var mask_var gw_var dns1_var dns2_var vpn_var} {
+  upvar $dhcp_var dhcp $hostname_var hostname $ip_var ip $mask_var mask $gw_var gw $dns1_var dns1 $dns2_var dns2 $vpn_var vpn
   set ifconfig_result [exec /sbin/ifconfig eth0]
   if {! [regexp -line {inet addr:([\d.]+).*Mask:([\d.]+)[^\d.]*$} $ifconfig_result dummy ip mask]} {return 0}
   set fd -1
@@ -857,8 +895,8 @@
   return 1
 }
 
-proc read_config {dhcp_var hostname_var ip_var mask_var gw_var dns1_var dns2_var} {
-  upvar $dhcp_var dhcp $hostname_var hostname $ip_var ip $mask_var mask $gw_var gw $dns1_var dns1 $dns2_var dns2
+proc read_config {dhcp_var hostname_var ip_var mask_var gw_var dns1_var dns2_var vpn_var} {
+  upvar $dhcp_var dhcp $hostname_var hostname $ip_var ip $mask_var mask $gw_var gw $dns1_var dns1 $dns2_var dns2 $vpn_var vpn
   set fd -1
   catch {set fd [open "/etc/config/netconfig" r]}
   if { $fd <0 } { return 0 }
@@ -872,10 +910,14 @@
   if {! [get_property $netconfig "GATEWAY" gw] } {return 0}
   get_property $netconfig "NAMESERVER1" dns1
   get_property $netconfig "NAMESERVER2" dns2
+
+  # check for vpn enable status
+  set vpn [file exists "/etc/config/tailscaleEnabled"]
+  
   return 1
 }
 
-proc write_config {dhcp hostname ip mask gw dns1 dns2} {
+proc write_config {dhcp hostname ip mask gw dns1 dns2 vpn} {
   set fd -1
   catch {set fd [open "/etc/config/netconfig" r]}
   if { $fd <0 } { return 0 }
@@ -902,6 +944,18 @@
   puts -nonewline $fd $netconfig
   close $fd
 
+  # set vpn status
+  set cur_vpn [file exists "/etc/config/tailscaleEnabled"]
+  if { $vpn != $cur_vpn } {
+    if { $vpn == 1 } {
+      exec touch /etc/config/tailscaleEnabled
+      exec /etc/init.d/S46tailscaled restart >/dev/null 2>/dev/null
+    } else {
+      exec /etc/init.d/S46tailscaled stop >/dev/null 2>/dev/null
+      exec rm -f /etc/config/tailscaleEnabled
+    }
+  }
+
   return 1
 }
 
--- occu/WebUI/www/webui/js/lang/de/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.js
@@ -627,6 +627,7 @@
     "commentSettingsDialogNetwork" : "***********************************************",
     "dialogSettingsNetworkTitle" : HMIdentifier.de.CCUShortName + " - Netzwerkeinstellungen",
     "dialogSettingsNetworkTDIPSettings" : "IP-Einstellungen",
+    "dialogSettingsNetworkTDVPNSettings" : "VPN-Einstellungen",
     "dialogSettingsNetworkTDCreateCertificate" : "Zertifikat<br/>erstellen",
     "dialogSettingsNetworkIPSettingsLblHostname" : "Hostname: ",
     "dialogSettingsNetworkIPSettingsLblUseIPAdd" : "Folgende IP-Adresse verwenden: ",
@@ -639,6 +640,10 @@
     "dialogSettingsNetworkHintIPSettingsP1" : "Legen Sie die grundlegenden IP-Einstellungen Ihrer " + HMIdentifier.de.CCUFullNameText + " fest.",
     "dialogSettingsNetworkHintIPSettingsP2" : "%DCber den Hostnamen k%F6nnen Sie Ihrer " + HMIdentifier.de.CCUFullNameText + " einen Namen geben.",
     "dialogSettingsNetworkHintIPSettingsP3" : "Die " + HMIdentifier.de.CCUFullNameText + " bezieht ihre IP-Adresse und alle damit verbundenen Einstellungen von einem DHCP-Server. Falls sich in Ihrem Netzwerk kein solcher Server befindet, oder Sie aus anderen Gr%FCnden eigene Einstellungen verwenden m%F6chten, k%F6nnen Sie hier manuell eine IP-Adresse konfigurieren.",
+    "dialogSettingsNetworkVPNLblActivate" : "Tailscale VPN-Funktion",
+    "dialogSettingsNetworkVPNLblAuth" : "Authentifizierung",
+    "dialogSettingsNetworkVPNHelpP1" : "Die " + HMIdentifier.de.CCUFullNameText + " verf%FCgt %FCber eine interne VPN-Funktion, welche erlaubt die Zentrale %FCber einen sicheren Zugang f%FCr Zugriffe vom Internet aus (z.B. via Smartphone) verf%FCgbar zu machen. Hierf%FCr wird <a href='https://tailscale.com/'>Tailscale</a> als sichere VPN-L%F6sung eingesetzt, bei welcher es sich um ein Zeroconfig-VPN handelt das intern Wireguard einsetzt um somit ein dezentrales, privates und kostenfreies VPN zwischen Nutzerger%E4ten aufzubauen. Tailscale ist OpenSource, sicherheitsgepr%FCft und nutzt direkte Punkt-zu-Punkt Verbindungen, soda%DF gew%E4hrleistet ist, das keinerlei Nutzer-Traffic %FCber zentrale Server geleitet wird.",
+    "dialogSettingsNetworkVPNHelpP2" : "F%FCr die Aufnahme der Zentrale in ein Tailscale-VPN Netzwerk, muss nach Aktivierung dieser Funktion im Anschluss der 'Authentifizierung'-Knopf bet%E4tigt werden, damit mit ggf. bereits bestehenden Zugangsdaten die Zentrale dann in das pers%F6nliche VPN-Netzwerk integriert werden kann und alle darin befindlichen Ger%E4te unmittelbar Zugriff erhalten.",
     "dialogSettingsNetworkCertificateLblStep1" : "Schritt 1: Daten eingeben",
     "dialogSettingsNetworkCertificateLblHostname" : "Hostname",
     "dialogSettingsNetworkCertificateLblEMail" : "E-Mail Adresse",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.js
@@ -627,6 +627,7 @@
     "commentSettingsDialogNetwork" : "***********************************************",
     "dialogSettingsNetworkTitle" : HMIdentifier.en.CCUShortName + " network settings",
     "dialogSettingsNetworkTDIPSettings" : "IP settings",
+    "dialogSettingsNetworkTDVPNSettings" : "VPN settings",
     "dialogSettingsNetworkTDCreateCertificate" : "Create certificate",
     "dialogSettingsNetworkIPSettingsLblHostname" : "Host name: ",
     "dialogSettingsNetworkIPSettingsLblUseIPAdd" : "Use the following IP address: ",
@@ -639,6 +640,10 @@
     "dialogSettingsNetworkHintIPSettingsP1" : "Please define the basic IP settings of your " + HMIdentifier.en.CCUFullNameText + ".",
     "dialogSettingsNetworkHintIPSettingsP2" : "You can assign a name to your " + HMIdentifier.en.CCUFullNameText + " via the host name.",
     "dialogSettingsNetworkHintIPSettingsP3" : "The " + HMIdentifier.en.CCUFullNameText + " gets the IP address and all corresponding settings from a DHCP server. If your network does not have a DHCP server or if you would like to use your own settings, you can manually configure an own IP address here.",
+    "dialogSettingsNetworkVPNLblActivate" : "Tailscale VPN functionality",
+    "dialogSettingsNetworkVPNLblAuth" : "Authentification",
+    "dialogSettingsNetworkVPNHelpP1" : "The " + HMIdentifier.en.CCUFullNameText + " comes with internal VPN functionality which allows to make it accessible from the internet (e.g. via Smartphone) using a secure VPN solution fully under your control. For this functionality, the secure <a href='https://tailscale.com/'>Tailscale</a> VPN solution is used which provides a zeroconfig solution to easily integrate your devices in a wireguard-based, decentralized and cost-free user VPN. It is fully based on open source software, security proof and initiates point-to-point connections between your devices, thus no user traffic is routed over centralized server.",
+    "dialogSettingsNetworkVPNHelpP2" : "For making your device available, you will have to switch on the Tailscale VPN functionality and use the 'Authentication' button to access the local authentication page. From there you can then login with your existing Tailscale login credentials and thus add this central to your private Tailscale VPN network so that any device in that network can directly access it.",
     "dialogSettingsNetworkCertificateLblStep1" : "Step 1: Enter data",
     "dialogSettingsNetworkCertificateLblHostname" : "Host name",
     "dialogSettingsNetworkCertificateLblEMail" : "Email address",
