--- occu/WebUI/www/rega/esp/datapointconfigurator.fn.orig
+++ occu/WebUI/www/rega/esp/datapointconfigurator.fn
@@ -1612,14 +1612,14 @@
 }
 
 function ::HideCertainParameters() {
-  string tmp;
+  string tmpHCP;
   string param2Exclude = "IDENTIFY_DURATION,IDENTIFICATION_MODE_KEY_VISUAL,IDENTIFICATION_MODE_LCD_BACKLIGHT,IDENTIFY_TARGET_LEVEL,STATUS_FLAG_ERROR,STATUS_FLAG_LOW_BAT,STATUS_FLAG_PLAYING_FILE_ACTIVE,STATUS_FLAG_PLAYLIST_ACTIVE,ERROR_CODE_STATUS";
 
   showParam = true;
   ! Check if datapoint is no sysvar
   if (! oDP.IsTypeOf( OT_VARDP )) {
-    foreach (tmp, param2Exclude.Split(",")) {
-      if (oDP.HSSID().Find(tmp) != -1) {
+    foreach (tmpHCP, param2Exclude.Split(",")) {
+      if (oDP.HSSID().Find(tmpHCP) != -1) {
         showParam = false;
         return;
       }
