--- occu/WebUI/www/config/easymodes/MASTER_LANG/HmIP-ParamHelp.js.orig
+++ occu/WebUI/www/config/easymodes/MASTER_LANG/HmIP-ParamHelp.js
@@ -260,10 +260,10 @@
     "CALIBRATION_PPM" :
       "Bei der automatischen Kalibrierung des CO2-Sensors wird der niedrigste Messwert innerhalb vordefinierter Kalibrierintervalle als neue Referenz angenommen. " +
       "Frische Au%DFenluft hat einen typischen Wert von zirka 400 ppm. Bei normaler Innenraumnutzung wird angenommen, dass die Innenraumluft innerhalb eines Zeitraums von einer Woche mindestens einmal das Niveau von frischer Au%DFenluft erreicht. " +
-      "<br/><br/>Die Kalibrierung ben%F6tig mindestens drei Kalibrierintervalle, um den Endwert sicher zu bestimmen." +
+      "<br/><br/>Die Kalibrierung ben%F6tigt mindestens drei Kalibrierintervalle, um den Endwert sicher zu bestimmen." +
       "<br/><br/>F%FCr R%E4ume mit Dauerbelegung 24/7 muss diese Funktion deaktiviert werden. Eine Kalibrierung sollte in dem Fall alle 1-2 Jahre manuell durchgef%FChrt werden.",
 
-    "MOUNTING_ORIENTATION" : "Dieser Parameter erm%FCglicht es, den Displayinhalt um 180 Grad zu drehen.",
+    "MOUNTING_ORIENTATION" : "Dieser Parameter erm%F6glicht es, den Displayinhalt um 180 Grad zu drehen.",
     "MOUNTING_ORIENTATION_A" : "Dieser Parameter erm%F6glicht es, den Aktor im gew%FCnschten Winkel einzusetzen. ",
 
     "POWERUP_ONTIME_UNIT" :
--- occu/WebUI/www/config/ic_deviceFirmwareOverview.cgi.orig
+++ occu/WebUI/www/config/ic_deviceFirmwareOverview.cgi
@@ -67,8 +67,8 @@
   append html "<div class=\"FilterBodyWrapper\" id=\"filter_$param\" style=\"display:none;\">"
     append html "<div class=\"FilterBody\">"
       append html "<input class=\"FilterText\" id=\"searchText_$param\"  name=\"DeviceListPage.NameFilterText\" value=\"\" type=\"text\" onkeypress=\"checkEnterEsc(this.id, event.keyCode);\">"
-      append html "<div class=\"FilterButton\" name=\"filterSet\" onclick=\"setFilter('filter_$param');\">Setzen</div>"
-      append html "<div class=\"FilterButton\" name=\"filterClose\" onclick=\"closeFilter('filter_$param');\">Schliessen</div>"
+      append html "<div class=\"FilterButton\" name=\"filterSet\" onclick=\"setFilter('filter_$param');\">\${filterSet}</div>"
+      append html "<div class=\"FilterButton\" name=\"filterClose\" onclick=\"closeFilter('filter_$param');\">\${filterClose}</div>"
     append html "</div>"
   append html "</div>"
   return $html
--- occu/WebUI/www/config/stringtable_de.txt.orig
+++ occu/WebUI/www/config/stringtable_de.txt
@@ -1396,7 +1396,9 @@
 PRESS_LONG	${stringTableKeyPressLong}
 PRESS_LONG=TRUE	${stringTableKeyPressLongTrue}
 PRESS_LONG_RELEASE	${stringTableKeyPressLongRelease}
+PRESS_LONG_RELEASE=TRUE	${stringTableKeyPressLongRelease}
 PRESS_LONG_START	${stringTableKeyPressLongStart}
+PRESS_LONG_START=TRUE	${stringTableKeyPressLongStart}
 PRESS_SHORT	${stringTableKeyPressShort}
 PRESS_SHORT=TRUE	${stringTableKeyPressShortTrue}
 PROCESS=NOT_STABLE	${stringTableProcessNotStableGeneric}
--- occu/WebUI/www/webui/js/lang/de/translate.lang.extension.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.extension.js
@@ -77,7 +77,7 @@
     "dialogCreateLinkErrorContent3" : "das Ger%E4t innerhalb der Funkreichweite befindet und aktiv ist,",
     "dialogCreateLinkErrorContent4" : "das Ger%E4t im Anlernmodus befindet.",
     "dialogCreateLinkErrorContent5" : "\"Ignorieren\", wenn die Zentrale die %DCbertragung zum Ger%E4t bei n%E4chster Gelegenheit selbstst%E4ndig durchf%FChren soll. Bis dahin ist dieser Konfigurationsvorgang als offene Servicemeldung sichtbar.",
-    "dialogCreateLinkErrorContent6" : "<u>Achtung</u><br/> Es besteht die M%FCglichkeit, dass die maximale Anzahl der erlaubten Verkn%FCpfungen erreicht wurde.",
+    "dialogCreateLinkErrorContent6" : "<u>Achtung</u><br/> Es besteht die M%F6glichkeit, dass die maximale Anzahl der erlaubten Verkn%FCpfungen erreicht wurde.",
     "dialogCreateLinkErrorUnknownDevice" : "%DCbertragung nicht erfolgt, weil das Ger%E4t unbekannt ist.",
     "dialogCreateLinkHintLinkExists" : "Achtung! Die Verkn%FCpfung existiert bereits und wird %FCberschrieben.",
     "dialogCreateLinkMsgLinkExistsA" : "Sie sind dabei ",
--- occu/WebUI/www/webui/js/lang/de/translate.lang.label.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.label.js
@@ -117,8 +117,8 @@
     "lblInfoLevelReal" : "Realer Wert",
     "lblInfoLevelRealBrightness" : "Reale Helligkeit",
     "lblInfoStatusControlCurrent" : "Strom",
-    "lblInfoStatusControlDevice" : "Energie-Z%E4hler Ger%E4t",
-    "lblInfoStatusControlEnergyCounter" : "Energie-Z%E4hler " + HMIdentifier.de.CCUShortName,
+    "lblInfoStatusControlDevice" : "Energie-Z%E4hler - Ger%E4t",
+    "lblInfoStatusControlEnergyCounter" : "Energie-Z%E4hler - Zentrale",
     "lblInfoStatusControlFrequency" : "Netzfrequenz",
     "lblInfoStatusControlGas" : "Verbrauch",
     "lblInfoStatusControlPower" : "Leistung",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.label.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.label.js
@@ -117,8 +117,8 @@
     "lblInfoLevelReal" : "Actual value",
     "lblInfoLevelRealBrightness" : "Actual brightness",
     "lblInfoStatusControlCurrent" : "Current",
-    "lblInfoStatusControlDevice" : "Energy counter device",
-    "lblInfoStatusControlEnergyCounter" : "Energy counter " + HMIdentifier.en.CCUShortName,
+    "lblInfoStatusControlDevice" : "Energy counter - Device",
+    "lblInfoStatusControlEnergyCounter" : "Energy counter - Central",
     "lblInfoStatusControlFrequency" : "Frequency",
     "lblInfoStatusControlGas" : "Consumption",
     "lblInfoStatusControlPower" : "Power",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.stringtable.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.stringtable.js
@@ -86,7 +86,7 @@
     "stringTableCapacitiveFillingSensorCaseDesign" : "Case design",
     "stringTableCapacitiveFillingSensorCaseDesignHor" : "horizontal cylinder",
     "stringTableCapacitiveFillingSensorCaseDesignVer" : "vertical cylinder",
-    "stringTableCapacitiveFillingSensorCaseHeight" : "Case hight",
+    "stringTableCapacitiveFillingSensorCaseHeight" : "Case height",
     "stringTableCapacitiveFillingSensorCaseLength" : "Case length",
     "stringTableCapacitiveFillingSensorCaseWidth" : "Case width",
     "stringTableCapacitiveFillingSensorCustomCal" : "Custom calibration",
@@ -871,7 +871,7 @@
     "stringTableValveStateErrorPosition" : "Valve in error position",
     "stringTableValveStateFalse" : "Heating valve closed",
     "stringTableValveStateNotAvailable" : "Valve status unknown",
-    "stringTableValveStateRunToStart" : "Valve reverses to the initial position (vale is opened entirely)",
+    "stringTableValveStateRunToStart" : "Valve reverses to the initial position (valve is opened entirely)",
     "stringTableValveStateToTight" : "Valve sluggish/blocked",
     "stringTableValveStateTrue" : "Heating valve open",
     "stringTableValveStateWaitForAdaption" : "Waiting for adaption run",
