--- occu/WebUI/www/webui/js/lang/de/translate.lang.label.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.label.js
@@ -149,8 +149,8 @@
     "lblInfoLevelReal" : "Realer Wert",
     "lblInfoLevelRealBrightness" : "Reale Helligkeit",
     "lblInfoStatusControlCurrent" : "Strom",
-    "lblInfoStatusControlDevice" : "Energie-Z%E4hler Ger%E4t",
-    "lblInfoStatusControlEnergyCounter" : "Energie-Z%E4hler " + HMIdentifier.de.CCUShortName,
+    "lblInfoStatusControlDevice" : "Energie-Z%E4hler - Ger%E4t",
+    "lblInfoStatusControlEnergyCounter" : "Energie-Z%E4hler - Zentrale",
     "lblInfoStatusControlFrequency" : "Netzfrequenz",
     "lblInfoStatusControlGas" : "Verbrauch",
     "lblInfoStatusControlGasFlow" : "Gasdurchfluss",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.label.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.label.js
@@ -149,8 +149,8 @@
     "lblInfoLevelReal" : "Actual value",
     "lblInfoLevelRealBrightness" : "Actual brightness",
     "lblInfoStatusControlCurrent" : "Current",
-    "lblInfoStatusControlDevice" : "Energy counter device",
-    "lblInfoStatusControlEnergyCounter" : "Energy counter " + HMIdentifier.en.CCUShortName,
+    "lblInfoStatusControlDevice" : "Energy counter - Device",
+    "lblInfoStatusControlEnergyCounter" : "Energy counter - Central",
     "lblInfoStatusControlFrequency" : "Frequency",
     "lblInfoStatusControlGas" : "Consumption",
     "lblInfoStatusControlGasFlow" : "Gas flow",
