--- occu/WebUI/www/config/cp_software.cgi.orig
+++ occu/WebUI/www/config/cp_software.cgi
@@ -453,11 +453,11 @@
           sendXML: false,
           onSuccess: function(transport) {
             if (!transport.responseText.match(/^Success/g)){
-              alert(translateString(op_name) + translateKey('btnSysConfAdditionalSoftRemoveFailure') + transport.responseText);
+              alert(translateString(op_name) + translateKey('btnSysConfAdditionalSoftRemoveFailure') + transport.responseText.replace(/^Failure\n/, ''));
             }else{
               alert(translateString(op_name) + translateKey('btnSysConfAdditionalSoftRemoveSuccess'));
-              showSoftwareCP();
             }
+            showSoftwareCP();
           }
         };
         if ("uninstall" == op) 
@@ -523,13 +523,17 @@
   import script
   import op
   
-  if {[catch {exec $script $op}]} {
-    puts "Failure"
+  if {[catch {exec $script $op} result]} {
+    set errorfile /var/log/addon-uninstall-error.log
+    exec echo $result >>$errorfile
+    set result "Failure\nPlease see $errorfile on the central for more details."
+  } else {
+    set result "Success"
   }
   if { "$op" == "uninstall" } {
     exec rm -rf $script
   }
-  puts "Success"
+  puts $result
 }
 
 proc action_image_upload {} {
--- occu/WebUI/www/webui/js/lang/de/translate.lang.extension.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.extension.js
@@ -339,7 +339,7 @@
     "dialogSetSecKeyLoadBackuplblB" : "Die Zentrale befindet sich im Auslieferungszustand und es wurde noch kein Sicherheitsschl&uuml;ssel gesetzt.",
     "dialogSetSecKeyLoadBackupLblC" : "Sicherheitsschl&uuml;ssel:",
     "btnSysConfAdditionalSoftRemoveSuccess" : " wurde erfolgreich ausgef%FChrt.",
-    "btnSysConfAdditionalSoftRemoveFailure" : " fehlgeschlagen:\n",
+    "btnSysConfAdditionalSoftRemoveFailure" : " wurde mit Fehlern abgeschlossen:\n\n",
     "dialogChooseLEDTitle" : "LEDs ausw%E4hlen",
     "infoStatusControlLblDanger" : "Gefahr",
     "actionStatusControlUp" : "Auf",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.extension.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.extension.js
@@ -339,7 +339,7 @@
     "dialogSetSecKeyLoadBackuplblB" : "The CCU is in delivery status and a security key has not yet been set.",
     "dialogSetSecKeyLoadBackupLblC" : "Security key:",
     "btnSysConfAdditionalSoftRemoveSuccess" : " was successfully executed.",
-    "btnSysConfAdditionalSoftRemoveFailure" : " failed:\n",
+    "btnSysConfAdditionalSoftRemoveFailure" : " was executed with failures:\n\n",
     "dialogChooseLEDTitle" : "Select LEDs",
     "infoStatusControlLblDanger" : "Hazard",
     "actionStatusControlUp" : "Up",
