--- occu/WebUI/www/api/methods/safemode/enter.tcl.orig
+++ occu/WebUI/www/api/methods/safemode/enter.tcl
@@ -18,6 +18,6 @@
 
 # Neustart
 rega system.Save()
-exec /sbin/reboot
+exec /sbin/reboot -d 2 2>/dev/null >/dev/null &
 
-jsonrpc_response true
+jsonrpc_response true
--- occu/WebUI/www/config/cp_maintenance.cgi.orig
+++ occu/WebUI/www/config/cp_maintenance.cgi
@@ -1263,8 +1263,7 @@
     exec /bin/kill -SIGQUIT 1
   } else {
     exec touch /usr/local/.recoveryMode
-    exec sleep 5
-    exec /sbin/reboot
+    exec /sbin/reboot -d 2 2>/dev/null >/dev/null &
   }
 
 }
@@ -1275,8 +1274,7 @@
   catch { exec lcdtool {Saving   Data...  } }
   rega system.Save()
   catch { exec lcdtool {Reboot...       } }
-  exec sleep 5
-  exec /sbin/reboot
+  exec /sbin/reboot -d 2 2>/dev/null >/dev/null &
 }
 proc action_shutdown {} {
   puts ""
@@ -1284,9 +1282,8 @@
   catch { exec lcdtool {Saving   Data...  } }
   rega system.Save()
   catch { exec lcdtool {Shutdown...       } }
-  exec sleep 5
   catch { exec touch /tmp/shutdown }
-  exec /sbin/poweroff
+  exec /sbin/poweroff -d 2 2>/dev/null >/dev/null &
 }
 
 proc get_logserver {} {
--- occu/WebUI/www/config/cp_network.cgi.orig
+++ occu/WebUI/www/config/cp_network.cgi
@@ -142,7 +142,7 @@
 
 proc action_reboot {} {
   puts ""
-  exec /sbin/reboot
+  exec /sbin/reboot -d 2 2>/dev/null >/dev/null &
 }
 
 proc action_cert_update_go {} {
--- occu/WebUI/www/config/cp_security.cgi.orig
+++ occu/WebUI/www/config/cp_security.cgi
@@ -1558,7 +1558,7 @@
 
 proc action_reboot {} {
   puts ""
-  exec /sbin/reboot
+  exec /sbin/reboot -d 2 2>/dev/null >/dev/null &
 }
 
 proc _version_compare { v1 v2 } {
