--- occu/WebUI/www/webui/style.css.orig
+++ occu/WebUI/www/webui/style.css
@@ -8705,6 +8705,17 @@
   margin: 10px;
 }
 
+#configMainTable {
+  margin: 10px;
+  max-width: 1500px;
+}
+
+.cpCell {
+  width: 270px;
+  height: 70px;
+  float: left;  
+}
+
 .CLASS21700 {
 }
 
--- occu/WebUI/www/config/control_panel.cgi.orig
+++ occu/WebUI/www/config/control_panel.cgi
@@ -54,95 +54,83 @@
     setFooter("");
   </script>
 
-  <table id="mainTable" cellspacing="0" cellpadding="0" style="display:none">
-    <colgroup>
-      <col />
-      <col />
-      <col />
-      <col />
-    </colgroup>
-    <tr class="CLASS21700">
+  <div id="configMainTable" style="display:none">
     
+    
+    
     <!-- Zentralen-Wartung -->
-    <td>
+    <div class="cpCell">
       <div class="cpButton">
         <div class="StdTableBtn CLASS21701" onclick="showMaintenanceCP()">${btnSysConfCentralMaintenace}</div>
         <div class="StdTableBtnHelp"><img id="showMaintenanceCPHelp" src="/ise/img/help.png"></div>
       </div>
-    </td>
+    </div>
 
     <!-- Sicherheit -->
-    <td>
+    <div class="cpCell">
       <div class="cpButton">
         <div class="StdTableBtn CLASS21701" onclick="showSecurityCP()">${btnSysConfSecurity}</div>
         <div class="StdTableBtnHelp"><img id="showSecurityCPHelp" src="/ise/img/help.png"></div>
       </div>
-    </td>
+    </div>
 
     <!-- Zeit- und Positionseinstellung -->
-    <td>
+    <div class="cpCell">
       <div class="cpButton">
         <div  class="StdTableBtn CLASS21701" onclick="showTimeCP()">${btnSysConfTimePosSettings}</div>
         <div class="StdTableBtnHelp"><img id="showTimeCPHelp" src="/ise/img/help.png"></div>
       </div>
-    </td>
+    </div>
 }
 if {[isOldCCU]} {
   puts {
       <!-- Zenralen-Display-Einstellungen -->
-      <td>
+      <div class="cpCell">
         <div  class="StdTableBtn CLASS21703" onclick="showDisplayCP()">${btnSysConfDisplayConfig}</div>
-     </td>
+     </div>
 
-     <td class="_CLASS21702"></td>
+     <div class="cpCell _CLASS21702"></div>
 
-     </tr>
-     <tr>
+     
 
   }
 }
     
 puts { 
     <!-- Netzwerkeinstellungen -->
-    <td>
+    <div class="cpCell">
       <div class="cpButton">
         <div class="StdTableBtn CLASS21701" onclick="showNetworkCP()">${btnSysConfNetworkConfig}</div>
         <div class="StdTableBtnHelp"><img id="showNetworkCPHelp" src="/ise/img/help.png"></div>
       </div>
-    </td>
+    </div>
 }
 
-if {![isOldCCU]} {
-  puts {
-    </tr>
-    <tr>
-  }
-}
 
-
 if {[getProduct] >= 3} {
   puts {
-    <td>
+    <div class="cpCell">
       <div class="cpButton">
         <div class="StdTableBtn CLASS21701" onclick="new FirewallConfigDialog();">${btnSysConfFirewallConfig}</div>
         <div class="StdTableBtnHelp"><img id="newFirewallConfigDialogHelp" src="/ise/img/help.png"></div>
       </div>
-    </td>
+    </div>
   }
 } else {
   puts {
-    <td>
+    <div class="cpCell">
       <div class="cpButton">
         <div class="StdTableBtn CLASS21701" onclick="new FirewallConfigDialog_CCU2();">${btnSysConfFirewallConfig}</div>
         <div class="StdTableBtnHelp"><img id="newFirewallConfigDialogHelp" src="/ise/img/help.png"></div>
       </div>
-    </td>
+    </div>
   }
 }
 
 if {[isOldCCU]} {
   puts {
-    <td>
+    <div class="cpCell">
+	 <div class="cpButton">
       <div class="StdTableBtn CLASS21701" onclick="ConfigData.check(function() { WebUI.enter(BidcosRfPage); });">${btnSysConfBidCosConfig}</div>
       <ul>
         <li>${lblSysConfBidCosConfig1}</li> <!-- interne Antenne konfigurieren -->
@@ -151,7 +139,7 @@
   }
 } else {
   puts {
-    <td>
+    <div class="cpCell">
       <div class="cpButton">
         <div class="StdTableBtn CLASS21701" onclick="ConfigData.check(function() { WebUI.enter(BidcosRfPage); });">${btnSysConfLANGateway}</div>
   }
@@ -160,97 +148,72 @@
   puts {
         <div class="StdTableBtnHelp"><img id="showBidCosConfigHelp" src="/ise/img/help.png"></div>
       </div>
-    </td>
+    </div>
 
     <!-- Zusatzsoftware -->
-    <td>
+    <div class="cpCell">
       <div class="cpButton">
         <div  class="StdTableBtn CLASS21701" onclick="showSoftwareCP();">${btnSysConfAdditionalSoft}</div>
         <div class="StdTableBtnHelp"><img id="showSoftwareCPHelp" src="/ise/img/help.png"></div>
       </div>
-    </td>
+    </div>
 
     <!-- Storage Settings -->
-    <td>
+    <div class="cpCell">
       <div class="cpButton">
         <div class="StdTableBtn CLASS21701" onclick="showGeneralSettingsCP();">${btnSysConfGeneralSettings}</div>
         <div class="StdTableBtnHelp"><img id="showGeneralSettingsCPHelp" src="/ise/img/help.png"></div>
       </div>
-    </td>
+    </div>
 
-    </tr>
-    
-    <tr>
+  
   }
 
-  set i 0
-
-  if {[getProduct] >= 3} {
+    if {[getProduct] >= 3} {
     puts {
       <!-- Button Security Wizard - slimmed down InstallWizard without the admin password dialog -->
 
-      <td>
+      <div class="cpCell">
         <div  class="StdTableBtn CLASS21701" onclick="new DialogChooseSecuritySettings();">${btnSecuritySettings}</div>
         <div class="StdTableBtnHelp"><img id="showSecuritySettingsCPHelp" src="/ise/img/help.png"></div>
-      </td>
+      </div>
     }
-    incr i
+  
   }
 
   puts {
     <!-- Kopplungen -->
-    <td>
+    <div class="cpCell">
       <div class="cpButton">
         <div class="StdTableBtn CLASS21701" onclick="showCouplingCP();">${btnSysConfCoupling}</div>
         <div class="StdTableBtnHelp"><img id="showCouplingCPHelp" src="/ise/img/help.png"></div>
       </div>
-    </td>
+    </div>
   }
-  incr i
+  
 
   if { [getProduct] >= 3} {
-    puts "<td><div class=\"cpButton\">"
+    puts "<div class=\"cpCell\"><div class=\"cpButton\">"
     puts "<div class=\"StdTableBtn CLASS21701\" onclick=\"showAccessPoint();\">\${btnAccessPoints}</div>"
     puts "<div class=\"StdTableBtnHelp\"><img id=\"showAccessPointCPHelp\" src=\"/ise/img/help.png\"></div>"
-    incr i
+	puts "</div></div>"
+  
   }
 
   puts {
-    <td>
+    <div class="cpCell">
       <div class="cpButton">
         <div class="StdTableBtn CLASS21701" onclick="showAdvancedSettingsCP()">${btnSysConfAdvancedSettings}</div>
         <div class="StdTableBtnHelp"><img id="showAdvancedSettingsCPHelp" src="/ise/img/help.png"></div>
       </div>
-    </td>
+    </div>
    }
-   incr i
+  
 
-set COL_COUNT 4
-
-if { $i == $COL_COUNT } {
-  puts {
-    <td class="_CLASS21702"></td>
-    </tr>
-    <tr>
-  }
-  set i 0
-}
-
 if { "[read_var /etc/config/tweaks CP_DEVCONFIG]" != "" } {
-  puts "<td><div class=\"cpButton\">"
+  puts "<div class=\"cpCell\"><div class=\"cpButton\">"
   puts "<div class=\"StdTableBtn CLASS21701\" onclick=\"window.open('/tools/devconfig.cgi?sid=$sid');\">DevConfig</div>"
-  puts "<div class=\"StdTableBtnHelp\"></div></td>"
-  incr i
-
-  if { $i == $COL_COUNT } {
-    puts {
-      <td class="_CLASS21702"></td>
-      </tr>
-      <tr>
-    }
-    set i 0
-  }
-
+  puts "<div class=\"StdTableBtnHelp\"></div></div></div>"
 }
 
 
@@ -261,7 +224,7 @@
 
     array set arrDESCR $addon(CONFIG_DESCRIPTION)
 
-    puts "<td>"
+    puts "<div class=\"cpCell\">"
       puts "<div class='cpButton'>"
         puts "<div class=\"StdTableBtn CLASS21701\" onclick=\"window.open('$addon(CONFIG_URL)?sid=$sid');\">$addon(CONFIG_NAME)</div>"
         puts "<div id=\"btnAddOn_$addonId\" class=\"StdTableBtnHelp j_addOn\"><img id=\"showAddonInfo_$addonId\" src=\"/ise/img/help.png\"></div>"
@@ -269,7 +232,7 @@
         #puts "<ul id=\"description_$addonId\" style=\"display:none\">$addon(CONFIG_DESCRIPTION)</ul>"
         puts "<ul id=\"description_$addonId\" style=\"display:none\"></ul>"
       puts "</div>"
-    puts "</td>"
+    puts "</div>"
 
      puts "<script type=\"text/javascript\">"
 
@@ -291,34 +254,13 @@
      puts "</script>"
 
 
-    incr i
-    if { $i == $COL_COUNT } {
-      puts {
-        <td class="_CLASS21702"></td>
-        </tr>
-        <tr>
-      }
-      set i 0
-    }
-
     array_clear addon
   }
 }
 
-if { $i != 0 } {
-  while { $i < $COL_COUNT } {
-    puts {<td class="_CLASS21702"></td>}
-    incr i
-  }
-  puts {
-    <td class="_CLASS21702"></td>
-    </tr>
-  }
-}
 
-
 puts {
-  </table>
+  </div>
 }
 
 
@@ -389,6 +331,6 @@
       createAddOnTooltips();
     } catch(e) {};
 
-    jQuery("#mainTable").show();
+    jQuery("#configMainTable").show();
   </script>
 }
