--- occu/WebUI/www/config/ic_deviceparameters.cgi.orig
+++ occu/WebUI/www/config/ic_deviceparameters.cgi
@@ -1133,6 +1133,29 @@
   puts "</script>"
 }
 
+proc get_user_easylinkmode {} {
+  global sid
+
+  set session [string trim $sid "@" ]
+  
+  set isecmd ""
+  append isecmd "var user=system.GetSessionVarStr('$session');"
+  array set user [rega_script $isecmd]
+  
+  set userid [lindex [split $user(user) ";"] 0]
+  
+  set isecmd ""
+  append isecmd "var easylinkmode = true;"
+  append isecmd "object user = dom.GetObject($userid);"
+  append isecmd "if (user)"
+  append isecmd "{"
+  append isecmd "easylinkmode = user.UserEasyLinkMode();"
+  append isecmd "}"
+  array set result [rega_script $isecmd]  
+  
+  return $result(easylinkmode);
+}
+
 proc put_Header {} {
 
   #cgi_debug -on
@@ -1144,6 +1167,7 @@
 
   set HmIPIdentifier "HmIP-RF"
   set HmIPWIdentifier "HmIP-Wired"
+  set HMRFIdentifier "BidCos-RF"
   set type $dev_descr(TYPE)
 
   array set SENTRY ""
@@ -1159,7 +1183,7 @@
       set SENTRY(NAME) "$iface"
       append SENTRY(NAME) ".$address"
   }
-
+  set EASYLINKMODE [get_user_easylinkmode]
   set SENTRY(IMAGE) "<div id=\"picDiv_thumb\" class=\"CLASS22005\" onmouseover=\"picDivShow(jg_250, '$dev_descr(TYPE)', 250, '-1', this);\" onmouseout=\"picDivHide(jg_250);\"></div>"
   append SENTRY(IMAGE) "<script type=\"text/javascript\">"
   append SENTRY(IMAGE) "var jg_thumb = new jsGraphics(\"picDiv_thumb\");"
@@ -1170,6 +1194,9 @@
   set SENTRY(FIRMWARE) "<table id=\"id_firmware_table_$address\" class=\"id_firmware_table\" cellspacing=\"0\">"
   #append SENTRY(FIRMWARE) "<tr><td>Version:</td><td class=\"CLASS22006\">$dev_descr(FIRMWARE)</td></tr>"
   append SENTRY(FIRMWARE) "<tr><td>\${lblFirmwareVersion}</td><td id=\"curFwVersion\" class=\"CLASS22006\">$dev_descr(FIRMWARE)</td></tr>"
+  if {$EASYLINKMODE == "false"} {
+    set SENTRY(RESTOREDEVPARAMS) "<table><tr><td class=\"CLASS22012\" style='border: none !important;' onclick=\"RestoreDeviceParameters();\"><div class=\"CLASS21000\">Restore<br/>Config</div></td></tr></table>"
+  }
   if {$MODE == "DEVICEPARAMETERS"} then {
     set fw_update_rows ""
     if {($iface != $HmIPIdentifier) && ($iface != $HmIPWIdentifier)} {
@@ -1270,8 +1297,15 @@
   puts "  <col />"
   #Interface
   puts "  <col />"
-  #Firmware
-  puts "  <col />"
+  if {$EASYLINKMODE == "false"} {
+    #Firmware
+    puts "  <col />"
+    #Restore Config
+    puts "  <col />"
+  } else {
+    #Firmware
+    puts "  <col />"
+  }
   puts "</colgroup>"
   
   puts "<THEAD>"
@@ -1284,6 +1318,9 @@
   puts "<TD>\${thSerialNumber}</TD>"
   puts "<TD>\${thInterface}</TD>"
   puts "<TD>\${thFirmware}</TD>"
+  if {$EASYLINKMODE == "false"} {
+    puts "<TD>\${thActions}</TD>"
+  }
   puts "</TR>"
   
   puts "</THEAD>"
@@ -1298,6 +1335,9 @@
   puts "<td>$SENTRY(ADDRESS)</td>"
   puts "<td>$SENTRY(IFACE)</td>"
   puts "<td>$SENTRY(FIRMWARE)</td>"
+  if {$EASYLINKMODE == "false"} {
+    puts "<td>$SENTRY(RESTOREDEVPARAMS)</td>"
+  }
   puts "</tr>"
 
   puts "</TBODY>"
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -40186,6 +40186,21 @@
   return false;
 };
 
+RestoreDeviceParameters = function()
+{
+  ProgressBar = new ProgressBarMsgBox(translateKey("transferConfigData"), 0);
+  
+  ProgressBar.show();
+  ProgressBar.StartKnightRiderLight();
+  
+  homematic("Interface.restoreConfigToDevice", {"interface": $F('global_iface'), "address": $F('global_address')});
+  
+  ProgressBar.hide();
+  ProgressBar.StopKnightRiderLight();
+  
+  CheckConfigPending($F('global_iface'), $F('global_address'), '', $F('global_redirect_url'), true);
+}
+
 SaveDeviceParameters = function()
 {
   var actions = 0,
