--- occu/WebUI/www/ise/htm/header.htm.orig
+++ occu/WebUI/www/ise/htm/header.htm
@@ -7,7 +7,10 @@
 </colgroup>
 <tr>
   <!-- Logo -->
-          <td class="CLASS11100"><img id="headerLogo" src="/ise/img/rm-logo_small.png" onclick="LogoClick();" class="SmallLogo" alt="" /></td>
+  <td class="CLASS11100">
+    <img id="headerLogo" src="/ise/img/rm-logo_small.png" onclick="LogoClick();" class="SmallLogo" alt="" />
+    <img id="headerLogoRed" src="/ise/img/rm-logo_small_red.png" onclick="LogoClick();" class="SmallLogo hidden" alt="" />
+  </td>
   <!-- Username und Pfad-->
   <td class="CLASS11101">
     <table class="CLASS11102">
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -27682,37 +27682,31 @@
 };
 
 setAlarmMessageCount = function (count) {
-
   if ($("msgAlarms")) {
-    //$("msgAlarms").innerHTML = "${Alarmmeldungen} (" + count + ")";
-    $("msgAlarms").innerHTML = translateKey('alarmMsg') + " (" + count + ")";
-
-  }
-  if (count === 0) {
-    if ($("imgAlarms")) {
-      $("imgAlarms").src = "/ise/img/dot/green.png";
+    const newMsg = translateKey('alarmMsg') + " (" + count + ")";
+    if ($("msgAlarms").innerHTML !== newMsg) {
+      $("msgAlarms").innerHTML = newMsg;
     }
   }
-  else {
-    if ($("imgAlarms")) {
-      $("imgAlarms").src = "/ise/img/dot/red.png";
+  if ($("imgAlarms")) {
+    const newImg = (count === 0 ? '/ise/img/dot/green.png' : '/ise/img/dot/red.png');
+    if ($("imgAlarms").getAttribute('src') !== newImg) {
+      $("imgAlarms").src = newImg;
     }
   }
 };
 
 setServiceMessageCount = function (count) {
   if ($("msgServices")) {
-    $("msgServices").innerHTML = translateKey('serviceMsg') + " (" + count + ")";
-
-  }
-  if (count === 0) {
-    if ($("imgServices")) {
-      $("imgServices").src = "/ise/img/dot/green.png";
+    const newMsg = translateKey('serviceMsg') + " (" + count + ")";
+    if ($("msgServices").innerHTML !== newMsg) {
+      $("msgServices").innerHTML = newMsg;
     }
   }
-  else {
-    if ($("imgServices")) {
-      $("imgServices").src = "/ise/img/dot/yellow.png";
+  if ($("imgServices")) {
+    const newImg = (count === 0 ? '/ise/img/dot/green.png' : '/ise/img/dot/yellow.png');
+    if ($("imgServices").getAttribute('src') !== newImg) {
+      $("imgServices").src = newImg;
     }
   }
 };
@@ -29772,12 +29766,14 @@
 
 SwitchOnFlashLight = function()
 {
-  if ($("headerLogo")) { $("headerLogo").src = "/ise/img/rm-logo_small_red.png"; }
+  if ($("headerLogo")) { $("headerLogo").classList.add('hidden'); }
+  if ($("headerLogoRed")) { $("headerLogoRed").classList.remove('hidden'); }
 };
 
 SwitchOffFlashLight = function()
 {
-  if ($("headerLogo")) { $("headerLogo").src = "/ise/img/rm-logo_small.png"; }
+  if ($("headerLogo")) { $("headerLogo").classList.remove('hidden'); }
+  if ($("headerLogoRed")) { $("headerLogoRed").classList.add('hidden'); }
 };
 
 getAjaxLoadElem = function() {
