--- occu/WebUI/www/rega/login.htm.orig
+++ occu/WebUI/www/rega/login.htm
@@ -40,34 +40,13 @@
           FormSubmit();
         }
       }
-
-      recalcDivs = function()
-      {
-        var wHeight = 0;
-        if (window.innerHeight)
-        {
-          wHeight = window.innerHeight;
-        }
-        else if (document.documentElement.offsetHeight)
-        {
-          wHeight = document.documentElement.offsetHeight;
-        }
-
-        var wHeaderHeight = 72;
-        var wFooterHeight = 32;
-        var wContentHeight = wHeight - wHeaderHeight - wFooterHeight - 27;
-
-        $("navbar").setStyle({height: wHeaderHeight + "px"});
-        $("content").setStyle({height: wContentHeight + "px"});
-        $("footer").setStyle({height: wFooterHeight + "px"});
-      }
     </script>
   </head>
   <body>
-    <table style="width:100%; height:100%;" cellpadding="0" cellspacing="0" id="maintable">
+    <table id="maintable">
       <tr>
         <td id="navbar">
-          <table cellpadding="0" cellspacing="0">
+          <table>
             <tr>
               <td style="text-align: center;"><div style="width: 180px;"><img src="/ise/img/homematic_logo_small.png" alt="HomeMatic Logo" /></div></td>
               <td><span class="HeaderTitle" style="cursor: pointer;" onclick="gotoLoginPage();">${lblUsrLogin}</span></td>
@@ -80,7 +59,7 @@
       </tr>
       <tr>
         <td>
-          <div id="content" style="background-repeat:no-repeat;background-position:480px center;">
+          <div id="content">
             <table id="login_content" style="width: 200px;">
               <tr>
                 <!-- Liste der Benutzer -->
@@ -168,9 +147,6 @@
 
 
     <script type="text/javascript">
-      Event.observe(window, "resize", function(event) { recalcDivs(); });
-      recalcDivs();
-
       gotoLoginPage = function() {
         location.href = "/login.htm";
       };
--- occu/WebUI/www/rega/logout.htm.orig
+++ occu/WebUI/www/rega/logout.htm
@@ -9,33 +9,12 @@
     <script type="text/javascript" src="/webui/js/extern/prototype.js"></script>
     <script type="text/javascript" src="/webui/js/lang/loadTextResource.js"></script>
     <script type="text/javascript" src="/webui/js/lang/translate.js"></script>
-
-    <script type="text/javascript">
-      recalcDivs = function()
-      {
-        var wHeight = 0;
-        if (window.innerHeight) {
-          wHeight = window.innerHeight;
-        }
-        else if (document.documentElement.offsetHeight) {
-          wHeight = document.documentElement.offsetHeight;
-        }
-
-        var wHeaderHeight = 72;
-        var wFooterHeight = 32;
-        var wContentHeight = wHeight - wHeaderHeight - wFooterHeight - 27;
-
-        $("navbar").setStyle({height: wHeaderHeight + "px"});
-        $("content").setStyle({height: wContentHeight + "px"});
-        $("footer").setStyle({height: wFooterHeight + "px"});
-      };
-    </script>
   </head>
   <body>
-    <table style="width: 100%; height: 100%;" cellpadding="0" cellspacing="0" id="maintable" class="j_translate">
+    <table id="maintable" class="j_translate">
       <tr>
         <td id="navbar">
-          <table cellpadding="0" cellspacing="0">
+          <table>
           <tr>
             <td style="text-align: center;"><div style="width: 180px;" ><img src="/ise/img/homematic_logo_small.png" alt="HomeMatic Logo" /></div></td>
             <td><span class="HeaderTitle" style="cursor: pointer;" onclick="gotoLoginPage();">${lblUsrLogin}</span></td>
@@ -47,7 +26,7 @@
       </tr>
       <tr>
         <td>
-          <div id="content" style="background-repeat:no-repeat;background-position:480px center;">
+          <div id="content">
             <table id="logout_content" cellspacing="8">
               <tr>
                 <td style="vertical-align: top;">
@@ -66,9 +45,6 @@
     </table>
 
     <script type="text/javascript">
-      Event.observe(window, "resize", function(event) { recalcDivs(); });
-      recalcDivs();
-
       gotoLoginPage = function() {
         location.href = "/login.htm";
       };
--- occu/WebUI/www/webui/style.css.orig
+++ occu/WebUI/www/webui/style.css
@@ -100,6 +100,8 @@
 
 html,body
 {
+  width: 100vw; /* FIX: resize */
+  height: 100vh; /* FIX: resize */
   margin:0;
   padding:0;
   color: $_(textColorB); /* black */
@@ -134,9 +136,28 @@
  * * *   Element IDs                     * * *
  * * * * * * * * * * * * * * * * * * * * * * */
 
+/* Login & Logout page */
+
+#maintable {
+  width: 100vw;
+  height: 100vh;
+  border-collapse: collapse;  
+}
+
+#maintable tbody > tr > td {
+  padding: 0;
+}
+
+#maintable #content {
+  height: 100%;
+  background-repeat: no-repeat;
+  background-position: 480px center;
+}
+
 /* Navigation-Leiste */
 #navbar 
 {
+  height: 72px;
   margin: 0px;
   background-color: $_(background); /* #183473 */
   width:100%;
@@ -144,6 +165,10 @@
 }
  
 /* Haupt-Fenster */
+.Layer0 {
+  height: 100%; /* FIX: resize */
+}
+
 #_content
 {
   height: 70%;
@@ -177,6 +202,8 @@
     align-items: flex-start;
     /* Prevent chrome from outlining this by focus */
     outline: none;
+    /* Stretch to max available space */
+    height: calc(100% - 168px); /* FIX: header + menubar + footer: 72px + 41px + 55px = 168px */
 }
 
 #prgcontent
@@ -196,7 +223,7 @@
 
 /* Kopfzeile */
 #header {
-  height: 15%;
+  height: 72px; /* FIX: resize */
   background-color: $_(background);  /* #183473 */
   /* background-color: #dcdcdc; */
   color: $_(textColorB); /* black */
@@ -206,7 +233,7 @@
 
 /* Fusszeile */
 #footer {
-  height:15%;
+  height: 50px; /* FIX: resize */
   background-color: $_(background); /* #183473 */
   /* background-color: #dcdcdc; */
   color: $_(textColorB); /* black */
@@ -254,12 +281,14 @@
 /* Linker Bereich der Hauptansicht (falls in zwei Spalten getrennt) */
 #contentLeft 
 {
+  height: 100%; /* FIX: resize */
   vertical-align: top;
 }
 
 /* Rechter Bereich der Hauptansicht (falls in zwei Spalten getrennt) */
 #contentRight 
 { 
+  height: 100%; /* FIX: resize */
   border-left: solid 5px $_(navBarSeparator); /* #ffffff */ /* Trennlinie zwischen Navigation und Hauptbereich */
   /* padding-top: 10px; */
   vertical-align: top;
@@ -2550,8 +2579,8 @@
   cursor: pointer;
 }
 #menubar {
-  height: 34px;
-  width: 100%;
+  height: 34px; /* FIX: resize */
+  /* width: 100%; */
   border-bottom:solid 5px $_(navBarSeparator); /* #ffffff */
   background-color: $_(background); /* #183473 */
   z-index: 189;
@@ -6274,6 +6303,7 @@
 }
 
 .CLASS03901 {
+  height: 100%; /* FIX: resize */
 	width:100%;
 }
 
@@ -6335,6 +6365,7 @@
 }
 
 .CLASS04001 {
+  height: 100%; /* FIX: resize */
 	width:100%;
 }
 
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -6942,6 +6942,8 @@
    */
   onResize: function()
   {
+    /**
+     * FIX: resize - not required any more, since we moved to css styles
     var height       = WebUI.getHeight();
     var width        = WebUI.getWidth();
     var bodyOverflow = "hidden";
@@ -6955,8 +6957,9 @@
     if ($("menubar")) { Element.setStyle("menubar", {"height": this.MENUBAR_HEIGHT + "px", "width": width + "px"}); }
     if ($("content")) { Element.setStyle("content", {"height": contentHeight       + "px", "width": width + "px"}); }
     if ($("footer"))  { Element.setStyle("footer" , {"height": this.FOOTER_HEIGHT  + "px", "width": width + "px"}); }
+    */
 
-    if (this.currentPage) { this.currentPage.resize(); }
+    if (this.currentPage) { this.currentPage.resize(); } /* FIX: resize - ToDo */
 
     if(typeof dcTimeout == "undefined") {
       dcTimeout = window.setTimeout(function () {
@@ -20235,6 +20238,9 @@
    **/
   resizeUser: function()
   {
+    /**
+     * FIX: resize - partialy done
+     *
     var contentHeight     = $("content").getHeight();
     var contentWidth      = $("content").getWidth();
     var width = parseInt(contentWidth / 2);
@@ -20244,10 +20250,10 @@
       //Element.setStyle("contentLeft", {"height": contentHeight + "px", "width": width + "px"});
       Element.setStyle("contentLeft", {"height": contentHeight + "px", "width": "55%"});
     }
-    if ($("contentLeft"))
+    if ($("contentRight"))
     {
       //Element.setStyle("contentRight", {"height": contentHeight + "px", "width": (width - 10) + "px"});
-      Element.setStyle("contentRight", {"height": contentHeight + "px", "width": "44%"});
+      Element.setStyle("contentRight", {"height": contentHeight + "px"});
     }
     
     if ($("favSelector"))
@@ -20257,6 +20263,7 @@
       var favViewHeight = (contentHeight - FavSelectorHeight - FAV_MARGIN_HEIGHT) * 0.95;
       $("favView").setStyle({"height": favViewHeight});
     }
+    */
   },
 
   getPageMeasurement: function() {
