--- occu/WebUI/www/rega/login.htm.orig
+++ occu/WebUI/www/rega/login.htm
@@ -40,34 +40,13 @@
           FormSubmit();
         }
       }
-
-      recalcDivs = function()
-      {
-        var wHeight = 0;
-        if (window.innerHeight)
-        {
-          wHeight = window.innerHeight;
-        }
-        else if (document.documentElement.offsetHeight)
-        {
-          wHeight = document.documentElement.offsetHeight;
-        }
-
-        var wHeaderHeight = 72;
-        var wFooterHeight = 32;
-        var wContentHeight = wHeight - wHeaderHeight - wFooterHeight - 27;
-
-        $("navbar").setStyle({height: wHeaderHeight + "px"});
-        $("content").setStyle({height: wContentHeight + "px"});
-        $("footer").setStyle({height: wFooterHeight + "px"});
-      }
     </script>
   </head>
   <body>
-    <table style="width:100%; height:100%;" cellpadding="0" cellspacing="0" id="maintable">
+    <table id="maintable">
       <tr>
         <td id="navbar">
-          <table cellpadding="0" cellspacing="0">
+          <table>
             <tr>
               <td style="text-align: center;"><div style="width: 180px;"><img src="/ise/img/homematic_logo_small.png" alt="HomeMatic Logo" /></div></td>
               <td><span class="HeaderTitle" style="cursor: pointer;" onclick="gotoLoginPage();">${lblUsrLogin}</span></td>
@@ -80,7 +59,7 @@
       </tr>
       <tr>
         <td>
-          <div id="content" style="background-repeat:no-repeat;background-position:480px center;">
+          <div id="content">
             <table id="login_content" style="width: 200px;">
               <tr>
                 <!-- Liste der Benutzer -->
@@ -168,9 +147,6 @@
 
 
     <script type="text/javascript">
-      Event.observe(window, "resize", function(event) { recalcDivs(); });
-      recalcDivs();
-
       gotoLoginPage = function() {
         location.href = "/login.htm";
       };
--- occu/WebUI/www/rega/logout.htm.orig
+++ occu/WebUI/www/rega/logout.htm
@@ -9,33 +9,12 @@
     <script type="text/javascript" src="/webui/js/extern/prototype.js"></script>
     <script type="text/javascript" src="/webui/js/lang/loadTextResource.js"></script>
     <script type="text/javascript" src="/webui/js/lang/translate.js"></script>
-
-    <script type="text/javascript">
-      recalcDivs = function()
-      {
-        var wHeight = 0;
-        if (window.innerHeight) {
-          wHeight = window.innerHeight;
-        }
-        else if (document.documentElement.offsetHeight) {
-          wHeight = document.documentElement.offsetHeight;
-        }
-
-        var wHeaderHeight = 72;
-        var wFooterHeight = 32;
-        var wContentHeight = wHeight - wHeaderHeight - wFooterHeight - 27;
-
-        $("navbar").setStyle({height: wHeaderHeight + "px"});
-        $("content").setStyle({height: wContentHeight + "px"});
-        $("footer").setStyle({height: wFooterHeight + "px"});
-      };
-    </script>
   </head>
   <body>
-    <table style="width: 100%; height: 100%;" cellpadding="0" cellspacing="0" id="maintable" class="j_translate">
+    <table id="maintable" class="j_translate">
       <tr>
         <td id="navbar">
-          <table cellpadding="0" cellspacing="0">
+          <table>
           <tr>
             <td style="text-align: center;"><div style="width: 180px;" ><img src="/ise/img/homematic_logo_small.png" alt="HomeMatic Logo" /></div></td>
             <td><span class="HeaderTitle" style="cursor: pointer;" onclick="gotoLoginPage();">${lblUsrLogin}</span></td>
@@ -47,7 +26,7 @@
       </tr>
       <tr>
         <td>
-          <div id="content" style="background-repeat:no-repeat;background-position:480px center;">
+          <div id="content">
             <table id="logout_content" cellspacing="8">
               <tr>
                 <td style="vertical-align: top;">
@@ -66,9 +45,6 @@
     </table>
 
     <script type="text/javascript">
-      Event.observe(window, "resize", function(event) { recalcDivs(); });
-      recalcDivs();
-
       gotoLoginPage = function() {
         location.href = "/login.htm";
       };
--- occu/WebUI/www/rega/pages/tabs/startpage.htm.orig
+++ occu/WebUI/www/rega/pages/tabs/startpage.htm
@@ -275,7 +275,6 @@
 </div>
 
 <script type="text/javascript">
-  WebUI.resize();
   StartPage.showUpdate();
   if (forceUpdate) {
       StartPage.showCurrentFirmware();
@@ -286,6 +285,7 @@
       StartPage.showUpdate();
       StartPage.showCurrentFirmware();
       StartPage.showAllDeviceFirmware();
+      StartPage.showDutyCycle();
       jQuery("#forceUpdate").remove();
   }
   jQuery("#iFaceShowAll").text(translateKey("lblDutyCycleA") + getProduct()+ ":");
--- occu/WebUI/www/webui/style.css.orig
+++ occu/WebUI/www/webui/style.css
@@ -100,6 +100,8 @@
 
 html,body
 {
+  width: 100vw;
+  height: 100vh;
   margin:0;
   padding:0;
   color: $_(textColorB); /* black */
@@ -134,9 +136,28 @@
  * * *   Element IDs                     * * *
  * * * * * * * * * * * * * * * * * * * * * * */
 
+/* Login & Logout page */
+
+#maintable {
+  width: 100vw;
+  height: 100vh;
+  border-collapse: collapse;  
+}
+
+#maintable tbody > tr > td {
+  padding: 0;
+}
+
+#maintable #content {
+  height: 100%;
+  background-repeat: no-repeat;
+  background-position: 480px center;
+}
+
 /* Navigation-Leiste */
 #navbar 
 {
+  height: 72px;
   margin: 0px;
   background-color: $_(background); /* #183473 */
   width:100%;
@@ -144,6 +165,11 @@
 }
  
 /* Haupt-Fenster */
+.Layer0 {
+  height: 100%;
+  width: -moz-available; /* otherwise firefox does not resize to max browser width */
+}
+
 #_content
 {
   height: 70%;
@@ -177,6 +203,8 @@
     align-items: flex-start;
     /* Prevent chrome from outlining this by focus */
     outline: none;
+    /* Stretch to max available space */
+    height: calc(100% - 168px); /* FIX: header + menubar + footer: 72px + 41px + 55px = 168px */
 }
 
 #prgcontent
@@ -196,7 +224,7 @@
 
 /* Kopfzeile */
 #header {
-  height: 15%;
+  height: 72px;
   background-color: $_(background);  /* #183473 */
   /* background-color: #dcdcdc; */
   color: $_(textColorB); /* black */
@@ -206,7 +234,7 @@
 
 /* Fusszeile */
 #footer {
-  height:15%;
+  height: 50px;
   background-color: $_(background); /* #183473 */
   /* background-color: #dcdcdc; */
   color: $_(textColorB); /* black */
@@ -254,12 +282,14 @@
 /* Linker Bereich der Hauptansicht (falls in zwei Spalten getrennt) */
 #contentLeft 
 {
+  height: 100%;
   vertical-align: top;
 }
 
 /* Rechter Bereich der Hauptansicht (falls in zwei Spalten getrennt) */
 #contentRight 
 { 
+  height: 100%;
   border-left: solid 5px $_(navBarSeparator); /* #ffffff */ /* Trennlinie zwischen Navigation und Hauptbereich */
   /* padding-top: 10px; */
   vertical-align: top;
@@ -2551,7 +2581,7 @@
 }
 #menubar {
   height: 34px;
-  width: 100%;
+  /* width: 100%; */
   border-bottom:solid 5px $_(navBarSeparator); /* #ffffff */
   background-color: $_(background); /* #183473 */
   z-index: 189;
@@ -6274,6 +6304,7 @@
 }
 
 .CLASS03901 {
+  height: 100%;
 	width:100%;
 }
 
@@ -6335,6 +6366,7 @@
 }
 
 .CLASS04001 {
+  height: 100%;
 	width:100%;
 }
 
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -6695,7 +6695,6 @@
          1 - user selected YES
          */
         if (action == 1) {
-          WebUI.resize();
           jQuery(".Layer0").show();
           homematic("CCU.setSecurityHint");
           if (getProduct() >= 3) {
@@ -6930,42 +6929,8 @@
     this.currentPageOptions  = null;    // Argumente der aktuellen Seite
     this.previousPage        = null;    // Vorherige Seite
     this.previousPageOptions = null;    // Argumente der vorherigen Seite
-    this.onResizeHandler     = null;    // EventHandler: onResize
-
-    this.onResizeHandler = this.onResize.bindAsEventListener(this);
-    Event.observe(window, "resize", this.onResizeHandler);
   },  
   
-  /**
-   * Ereignis. Wird bei der Änderung der Fenstergröße aufgerufen.
-   * Passt die Steuerelemente entsprechend der Änderung an.
-   */
-  onResize: function()
-  {
-    var height       = WebUI.getHeight();
-    var width        = WebUI.getWidth();
-    var bodyOverflow = "hidden";
-
-    if (width  < this.MIN_WIDTH)  { width  = this.MIN_WIDTH;  bodyOverflow = "auto"; }
-    if (height < this.MIN_HEIGHT) { height = this.MIN_HEIGHT; bodyOverflow = "auto"; }
-    var contentHeight = height - this.STATIC_HEIGHT;
-
-    if ($("body"))    { Element.setStyle("body", {"overflow": bodyOverflow, "width": width  + "px", "height": height + "px"}); }
-    if ($("header"))  { Element.setStyle("header" , {"height": this.HEADER_HEIGHT  + "px", "width": width + "px"}); }
-    if ($("menubar")) { Element.setStyle("menubar", {"height": this.MENUBAR_HEIGHT + "px", "width": width + "px"}); }
-    if ($("content")) { Element.setStyle("content", {"height": contentHeight       + "px", "width": width + "px"}); }
-    if ($("footer"))  { Element.setStyle("footer" , {"height": this.FOOTER_HEIGHT  + "px", "width": width + "px"}); }
-
-    if (this.currentPage) { this.currentPage.resize(); }
-
-    if(typeof dcTimeout == "undefined") {
-      dcTimeout = window.setTimeout(function () {
-        showDutyCycle();
-        delete dcTimeout;
-      }, 10);
-    }
-  },
-
   /*########################*/
   /*# Öffentliche Elemente #*/
   /*########################*/
@@ -7163,8 +7128,6 @@
 
 
     if (result) {
-      WebUI.resize();
-
       if (!isFF) {
         jQuery(".Layer0").show();
         WebUI.enter(StartPage);
@@ -7331,13 +7294,6 @@
       throw new Error("content not defined");
     }
   },
-  /**
-   * Ruft das onResize-Ereignis auf.
-   **/
-  resize: function()
-  {
-    this.onResize();
-  },
   
   /**
    * Lädt eine Seite.
@@ -20127,11 +20083,6 @@
    **/
   leave: function() {},
   
-  /**
-   * Callback beim Verändern der Bildschirmgröße
-   **/
-  resize: function() {}
-  
 });
 /**
  * page.js
@@ -20159,13 +20110,6 @@
   {
   },
   
-  /**
-   * Callback. Wird beim Verändern des Bildschirms aufgerufen.
-   **/
-  resize: function()
-  {
-  }
-  
 });
 /**
  * StartPage.js
@@ -20190,19 +20134,6 @@
   initialize: function ()
   {
     this.deviceList = [];
-
-    switch (getUPL())
-    {
-      case UPL_ADMIN:
-      case UPL_USER:
-        this.resize = this.resizeUser;
-        break;
-      case UPL_GUEST:
-        this.rezise = this.resizeGuest;
-        break;
-      default:
-        break;
-    }
   },
 
   /**
@@ -20213,7 +20144,6 @@
     iseInitUpdateArrays();
     setPath("");
     setFooter("");
-    WebUI.resize();
   
     switch (getUPL())
     {
@@ -20229,35 +20159,6 @@
         break;
     }
   },
-  
-  /**
-   * Berechnet die Größe der normalen Startseite (Admin und Benutzer)
-   **/
-  resizeUser: function()
-  {
-    var contentHeight     = $("content").getHeight();
-    var contentWidth      = $("content").getWidth();
-    var width = parseInt(contentWidth / 2);
-    
-    if ($("contentLeft"))
-    {
-      //Element.setStyle("contentLeft", {"height": contentHeight + "px", "width": width + "px"});
-      Element.setStyle("contentLeft", {"height": contentHeight + "px", "width": "55%"});
-    }
-    if ($("contentLeft"))
-    {
-      //Element.setStyle("contentRight", {"height": contentHeight + "px", "width": (width - 10) + "px"});
-      Element.setStyle("contentRight", {"height": contentHeight + "px", "width": "44%"});
-    }
-    
-    if ($("favSelector"))
-    {   
-      var FavSelectorHeight = $("favSelector").getHeight();
-      var FAV_MARGIN_HEIGHT = 8;          
-      var favViewHeight = (contentHeight - FavSelectorHeight - FAV_MARGIN_HEIGHT) * 0.95;
-      $("favView").setStyle({"height": favViewHeight});
-    }
-  },
 
   getPageMeasurement: function() {
     var pageElements = ["body","#header", "#menubar", "#favSelector", "#favView", "#footer"],
@@ -20493,11 +20394,9 @@
     }
   },
   
-  /**
-   * Berechnet die Größe der Startseite für den Gast
-   **/
-  resizeGuest: function() { }   
-  
+  showDutyCycle: function() {
+    showDutyCycle();
+  }
   
 });
 ControlPage = new function()
@@ -20514,10 +20413,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 ControlRoomsPage = new function()
 {
@@ -20533,10 +20428,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 ControlFunctionsPage = new function()
 {
@@ -20552,10 +20443,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 ControlFavoritesPage = new function()
 {
@@ -20572,10 +20459,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 ControlFavoritePage = new function()
 {
@@ -20591,10 +20474,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 ControlDevicesPage = new function()
 {
@@ -20610,10 +20489,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 ControlProgramsPage = new function()
 {
@@ -20629,10 +20504,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 ControlVariablesPage = new function()
 {
@@ -20648,10 +20519,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 ControlProtocolPage = new function()
 {
@@ -20667,10 +20534,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 LinksAndProgramsPage = new function()
 {
@@ -20686,10 +20549,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 LinkListPage = new function()
 {
@@ -20714,10 +20573,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 ProgramListPage = new function()
 {
@@ -20733,10 +20588,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 LinkEditProfilePage = new function()
 {
@@ -20757,10 +20608,6 @@
   {
   };
   
-  this.resize = function()
-  {
-  };
-  
 };AlarmMessagesPage = new function()
 {
   
@@ -20776,10 +20623,6 @@
     if ($("msgAlarms")) { $("msgAlarms").removeClassName("Messages_Selected"); }
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 ServiceMessagesPage = new function()
 {
@@ -20796,10 +20639,6 @@
     if ($("msgServices")) { $("msgServices").removeClassName("Messages_Selected"); }
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 HelpPage = new function()
 {
@@ -20816,10 +20655,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 adminFirstStartup = new function()
 {
@@ -20833,10 +20668,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 userFirstStartup = new function()
 {
@@ -20850,10 +20681,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 /**
  * devielistpage.js
@@ -21587,10 +21414,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-
  }();
 
 }
@@ -21608,10 +21431,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 FunctionListPage = new function()
 {
@@ -21626,10 +21445,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 NewDeviceListPage = new function()
 {
@@ -21653,10 +21468,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 RoomListPage = new function()
 {
@@ -21672,10 +21483,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 SystemConfigPageUser = new function()
 {
@@ -21691,10 +21498,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 SystemConfigPage = new function()
 {
@@ -21710,10 +21513,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 UserAdminPageUser = new function()
 {
@@ -21729,10 +21528,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 UserAdminPageAdmin = new function()
 {
@@ -21748,10 +21543,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 SystemControlPage = new function()
 {
@@ -21767,10 +21558,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 CreateDiagramPage = new function()
 {
@@ -21809,10 +21596,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 CreateDiagramPage = new function()
 {
@@ -21851,10 +21634,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 DiagramListPage = new function()
 {
@@ -21881,10 +21660,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 CreateGroupPage = new function()
 {
@@ -21913,10 +21688,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 CreateTestPage = new function()
 {
@@ -21943,10 +21714,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 DeviceFirmware = new function()
 {
@@ -21973,10 +21740,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 function compare(a,b) {
   if (a.name < b.name)
@@ -22027,10 +21790,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-
 }();VariableListPage = new function()
 {
   var MAINMENU_ID = "MAINMENU_OPTIONS";
@@ -22045,10 +21804,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 DeviceConfigPage = new function()
 {
@@ -22082,10 +21837,6 @@
   {
   };
 
-  this.resize = function()
-  {
-  };
-  
 }();
 getGatewayStatus = function()
 {
@@ -22403,10 +22154,6 @@
     InterfaceMonitor.start();
   },
   
-  resize: function()
-  {
-  },
-  
   onSourceChanged: function()
   {
  /*   if (this.m_hmcfgRadioButton.checked)
