--- occu/WebUI/www/config/easymodes/etc/hmipChannelConfigDialogs.tcl.orig
+++ occu/WebUI/www/config/easymodes/etc/hmipChannelConfigDialogs.tcl
@@ -27,7 +27,7 @@
 
   puts "<script type=\"text/javascript\">load_JSFunc('/config/easymodes/MASTER_LANG/HmIP-ParamHelp.js');</script>"
 
-  if {([string equal $devType "HmIP-CCU3"] == 1) || ([string equal $devType "RPI-RF-MOD"] == 1)} {
+  if {([string equal $devType "HmIP-CCU3"] == 1) || ([string equal $devType "RPI-RF-MOD"] == 1) || ([string equal $devType "HmIP-RFUSB"] == 1)} {
     append html "[getNoParametersToSet]"
     return $html
   }
--- occu/WebUI/www/rega/esp/datapointconfigurator.fn.orig
+++ occu/WebUI/www/rega/esp/datapointconfigurator.fn
@@ -744,7 +744,7 @@
               Call("/esp/controls/hap.fn::CreateHAP()");
             }
 
-            if ((ch.Label() == "RPI-RF-MOD") || (ch.Label() =="HmIP-CCU3")) {
+            if ((ch.Label() == "RPI-RF-MOD") || (ch.Label() == "HmIP-CCU3") || (ch.Label() == "HmIP-RFUSB")) {
               WriteLine( "<script>conInfo('Create CoPro Control.');</script>" );
               sLastControlName = CN_MAINTENANCE;
               isKnownControl = true;
--- occu/WebUI/www/rega/esp/functions.fn.orig
+++ occu/WebUI/www/rega/esp/functions.fn
@@ -397,7 +397,7 @@
       !   can be MAINTENANCE_BAT_BL, MAINTENANCE_BAT_EL, MAINTENANCE_BAT_CL and so on
       ! For BidCos-RF this isn't necessary
 
-      if ((oDevHssType != "HmIP-CCU3") && (oDevHssType != "HmIP-HAP") && (oDevHssType != "HmIP-HAP-A") && (oDevHssType != "HmIP-HAP-B1") && (oDevHssType != "HmIP-HAP JS1") && (oDevHssType != "RPI-RF-MOD")) {
+      if ((oDevHssType != "HmIP-CCU3") && (oDevHssType != "HmIP-HAP") && (oDevHssType != "HmIP-HAP-A") && (oDevHssType != "HmIP-HAP-B1") && (oDevHssType != "HmIP-HAP JS1") && (oDevHssType != "RPI-RF-MOD") && (oDevHssType != "HmIP-RFUSB")) {
         bShowChannel = false;
       }
 
--- occu/WebUI/www/tools/js/model/devicetypelist.js.orig
+++ occu/WebUI/www/tools/js/model/devicetypelist.js
@@ -18,7 +18,8 @@
     "HMW-RCV-50",
     "HM-Sec-SD-Team",
     "HmIP-CCU3",
-    "RPI-RF-MOD"
+    "RPI-RF-MOD",
+    "HmIP-RFUSB"
   ],
   
   /**
@@ -110,4 +111,4 @@
   }
 
 });
- 
\ No newline at end of file
+ 
--- occu/WebUI/www/webui/js/lang/de/translate.lang.deviceDescription.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.deviceDescription.js
@@ -181,7 +181,7 @@
     "HmIP-BROLL-2": "Homematic IP Rollladenaktor f%FCr Markenschalter",
     "HmIP-BS2": "Homematic IP Wandtaster f%FCr Markenschalter - 2-fach",
     "HmIP-BSL": "Homematic IP Schaltaktor f%FCr Markenschalter - mit Signalleuchte",
-    "HmIP-CCU3": "Homematic IP Zentrale CCU3",
+    "HmIP-CCU3": "Homematic IP Zentrale",
     "HmIP-DRG-DALI": "Homematic IP DALI Gateway",
     "HmIP-DBB": "Homematic IP T%FCrklingeltaster",
     "HmIP-DLD" : "Homematic IP T%FCrschlossantrieb",
@@ -224,6 +224,7 @@
     "HmIP-PCBS2": "Homematic IP Schaltplatine - 2-fach",
     "HmIP-PMFS": "Homematic IP Netzausfall%FCberwachung",
     "HmIP-RCV-50": "Virtuelle Fernbedienung",
+    "HmIP-RFUSB": "Homematic IP Zentrale",
     "HmIP-RGBW" : "Homematic IP LED Controller - RGBW",
     "HmIP-SAM" :"Homematic IP Ersch%FCtterungs- / Beschleunigungssensor",
     "HmIP-SCI" : "Homematic IP Kontakt-Schnittstelle",
@@ -315,7 +316,7 @@
     "PSMCO": "Homematic IP Schalten/Messen",
     "RC8": "Homematic IP Fernbedienung, 8 Kanal",
     "RF-LAN-Sniffer" : "RF-LAN-Sniffer",
-    "RPI-RF-MOD" : "CO-PROCESSOR",
+    "RPI-RF-MOD" : "Homematic IP Zentrale",
     "SMI": "Homematic IP Bewegungsmelder innen",
     "SMO": "Homematic IP Bewegungsmelder au%DFen",
     "SRH": "Homematic IP Fenster-/ Drehgriffkontakt",
@@ -348,4 +349,4 @@
     "ZEL_STG_RM_WT_2": "Funk-Wandtaster 2-fach im 55er Rahmen",
     "The END": ""
   }
-});
\ No newline at end of file
+});
--- occu/WebUI/www/webui/js/lang/en/translate.lang.deviceDescription.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.deviceDescription.js
@@ -181,7 +181,7 @@
     "HmIP-BROLL-2": "Homematic IP Blind Actuator for brand switch systems, flush-mount",
     "HmIP-BS2": "Homematic IP Remote Control for brand switches - 2 channels",
     "HmIP-BSL": "Homematic IP Switch Actuator with Signal Lamp - for brand switches",
-    "HmIP-CCU3": "Homematic IP Central Control Unit CCU3",
+    "HmIP-CCU3": "Homematic IP Central Control Unit",
     "HmIP-DRG-DALI": "Homematic IP DALI Gateway",
     "HmIP-DBB": "Homematic IP Doorbell Button",
     "HmIP-DLD" : "Homematic IP Door Lock Drive",
@@ -224,6 +224,7 @@
     "HmIP-PCBS2": "Homematic IP Switch Circuit Board - 2 channels",
     "HmIP-PMFS": "Homematic IP Mains Failure Surveillance",
     "HmIP-RCV-50": "Virtual Remote Control",
+    "HmIP-RFUSB": "Homematic IP Central Control Unit",
     "HmIP-RGBW" : "Homematic IP LED Controller - RGBW",
     "HmIP-SAM": "Homematic IP Acceleration Sensor",
     "HmIP-SCTH230" : "Homematic IP CO2 Sensor, 230 V",
@@ -315,7 +316,7 @@
     "PSMCO": "Homematic IP Switch and Meter",
     "RC8": "Homematic IP Remote Contro, 8-channel",
     "RF-LAN-Sniffer" : "RF-LAN-Sniffer",
-    "RPI-RF-MOD" : "CO-PROCESSOR",
+    "RPI-RF-MOD" : "Homematic IP Central Control Unit",
     "SMI": "Homematic IP Motion Detector - indoor",
     "SMO": "Homematic IP Motion Detector, outdoor",
     "SRH": "Homematic IP Rotary Handle Sensor",
@@ -348,4 +349,4 @@
     "ZEL_STG_RM_WT_2": "Wireless Push-button 2-channel in 55mm frame",
     "The END": ""
   }
-});
\ No newline at end of file
+});
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -1678,6 +1678,12 @@
 DEV_PATHS["RPI-RF-MOD"]["50"] = "/config/img/devices/50/CCU3_thumb.png";
 DEV_PATHS["RPI-RF-MOD"]["250"] = "/config/img/devices/250/CCU3.png";
 DEV_HIGHLIGHT["RPI-RF-MOD"] = new Object();
+DEV_LIST.push('HmIP-RFUSB');
+DEV_DESCRIPTION["HmIP-RFUSB"] = "HmIP-RFUSB";
+DEV_PATHS["HmIP-RFUSB"] = new Object();
+DEV_PATHS["HmIP-RFUSB"]["50"] = "/config/img/devices/50/CCU3_thumb.png";
+DEV_PATHS["HmIP-RFUSB"]["250"] = "/config/img/devices/250/CCU3.png";
+DEV_HIGHLIGHT["HmIP-RFUSB"] = new Object();
 DEV_LIST.push('HM-OU-CF-Pl');
 DEV_DESCRIPTION["HM-OU-CF-Pl"] = "HM-OU-CF-Pl";
 DEV_PATHS["HM-OU-CF-Pl"] = new Object();
@@ -10989,7 +10995,8 @@
     "HM-Sec-SD-Team",
     "HM-Sec-SD-2-Team",
     "HmIP-CCU3",
-    "RPI-RF-MOD"
+    "RPI-RF-MOD",
+    "HmIP-RFUSB"
   ],
   
   /**
