--- occu/WebUI/www/rega/pages/tabs/admin/systemvars.htm.orig
+++ occu/WebUI/www/rega/pages/tabs/admin/systemvars.htm
@@ -233,7 +233,7 @@
         Write("<table cellspacing='0' cellpadding='0' style='width:100%;'><tr>");
         Write("<td class='CLASS05210'>");
           string sClassName = "StdButton";
-          string sOnclick = "DeleteSysVar("#sv.ID()#");";
+          string sOnclick = "DeleteSysVar("#sv.ID()#",\""#sv.Name()#"\");";
           if( sv.Unerasable() )
           {
             sClassName = sClassName#"Inactive";
@@ -427,10 +427,11 @@
     }
   }
   
-  DeleteSysVar = function(id)
+  DeleteSysVar = function(id,name)
   {
     var _id_ = id;
-    new YesNoDialog(translateKey("dialogSafetyCheck"), translateKey("dialogQuestionRemoveSysVar"), function(result) {
+	var _name_ = name;
+    new YesNoDialog(translateKey("dialogSafetyCheck"), translateKey("dialogQuestionRemoveSysVar").replace(/%s/, _name_), function(result) {
       if (result == YesNoDialog.RESULT_YES) { DeleteObject(_id_); }
     });    
   };
--- occu/WebUI/www/rega/pages/tabs/admin/userAdministration.htm.orig
+++ occu/WebUI/www/rega/pages/tabs/admin/userAdministration.htm
@@ -145,7 +145,7 @@
         Write("<td style='text-align:center; vertical-align: middle;' class='TblUserAdministration CLASS05303' >")
           Write("<div class='StdButton CLASS05304' onclick='editUser("#user.ID()#")'>${btnEdit}</div>")
           if (system.GetSessionVar('sessionUserID') != user.ID()) {
-            Write("<div class='StdButton CLASS05304' onclick='deleteUser("#user.ID()#")'>${btnRemove}</div>")
+            Write("<div class='StdButton CLASS05304' onclick='deleteUser("#user.ID()#",\""#user.Name()#"\")'>${btnRemove}</div>")
           }
         Write("</td>")
         Write("</tr>")
@@ -187,10 +187,11 @@
     CreatePopup(ID_USER_ACCOUNT_CONFIG_ADMIN, id);
   }
   
-  deleteUser = function(id)
+  deleteUser = function(id,name)
   {
     var _id_ = id;
-    new YesNoDialog(translateKey("dialogSafetyCheck"), translateKey("dialogQuestionRemoveUser"), function(result) {
+	var _name_ = name;
+    new YesNoDialog(translateKey("dialogSafetyCheck"), translateKey("dialogQuestionRemoveUser").replace(/%s/, _name_), function(result) {
       if (result == YesNoDialog.RESULT_YES)
       {  
         var url = "/esp/system.htm?sid=" + SessionId;
--- occu/WebUI/www/rega/pages/tabs/user/favOverview.htm.orig
+++ occu/WebUI/www/rega/pages/tabs/user/favOverview.htm
@@ -304,7 +304,7 @@
                   Write('<div class="CLASS06412">Bearbeiten...</div>');
                 }
                 if ((bDelAllowed) && (!oFav.Unerasable())) {
-                    Write('<div class="CLASS06409 StdButton" onclick="DeleteFavorite('#sID#');">${btnRemove}</div>');
+                    Write('<div class="CLASS06409 StdButton" onclick="DeleteFavorite('#sID#',\''#oFav.Name()#'\');">${btnRemove}</div>');
                 }
                 else {
                   Write('<div class="CLASS06412">${btnRemove}</div>');
@@ -480,10 +480,11 @@
     updateContent("/pages/tabs/user/newFav.htm");
   }
   
-  DeleteFavorite = function(id)
+  DeleteFavorite = function(id,name)
   {
     var _id_ = id;
-    new YesNoDialog(translateKey("dialogSafetyCheck"), translateKey("dialogQuestionRemoveFavorite"), function(result) {
+	var _name_ = name;
+    new YesNoDialog(translateKey("dialogSafetyCheck"), translateKey("dialogQuestionRemoveFavorite").replace(/%s/, _name_), function(result) {
       if (result == YesNoDialog.RESULT_YES) { iseFavorites.RemoveList(_id_); }
     });     
   };
--- occu/WebUI/www/webui/js/lang/de/translate.lang.extension.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.extension.js
@@ -981,7 +981,7 @@
     "channelModeHumidityLimiter" :  "Feuchtigkeitsbegrenzer",
     "channelModeTactileSwitch" :  "Taster",
     "channelModeTemperatureLimiter" : "Temperaturbegrenzer",
-    "dialogQuestionRemoveInternalLink" :  "<span class='attention'>ACHTUNG:</span><br/><br>Es handelt sich hier um eine interne Verkn%FCpfung des Ger%E4tes. Das L%F6schen dieser Verkn%FCpfung kann dazu f%FChren, dass das Ger%E4t unbedienbar wird. In diesem Fall m%FCssen Sie die Verkn%FCpfung neu erstellen.<br/><br/>M%F6chten Sie die Verkn%FCpfung wirklich l%F6schen?",
+    "dialogQuestionRemoveInternalLink" :  "<span class='attention'>ACHTUNG:</span><br/><br>Es handelt sich hier um eine interne Verkn%FCpfung des Ger%E4tes. Das L%F6schen dieser Verkn%FCpfung kann dazu f%FChren, dass das Ger%E4t unbedienbar wird. In diesem Fall m%FCssen Sie die Verkn%FCpfung neu erstellen.<br/><br/>M%F6chten Sie die Verkn%FCpfung '%s' wirklich l%F6schen?",
     "descriptionTransmitterDataTransmissionCondition0" :  "<b>Modus 1</b><br/><br/>Senden auf Pegel%E4nderung am Daten%FCbertragungspin (DU30) von HIGH- auf LOW-Pegel.",
     "descriptionTransmitterDataTransmissionCondition1" :  "<b>Modus 2</b><br/><br/>Senden auf Pegel%E4nderung am Daten%FCbertragungspin (DU30) von LOW- auf HIGH-Pegel.",
     "descriptionTransmitterDataTransmissionCondition2" :  "<b>Modus 3</b><br/><br/>Senden auf Pegel%E4nderung am Daten%FCbertragungspin (DU30) auf jede Logikpegel%E4nderung.",
--- occu/WebUI/www/webui/js/lang/de/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.js
@@ -409,10 +409,10 @@
     "dialogInfo" : "Info",
     "dialogHint" : "Hinweis",
     "dialogQuestionRemoveProgram" : "M%F6chten Sie das Programm '%s' wirklich l%F6schen?",
-    "dialogQuestionRemoveLink" : "M%F6chten Sie die Verkn%FCpfung wirklich l%F6schen?",
-    "dialogQuestionRemoveUser" : "M%F6chten Sie den Benutzer wirklich l%F6schen?",
-    "dialogQuestionRemoveSysVar" : "M%F6chten Sie die Systemvariable wirklich l%F6schen?",
-    "dialogQuestionRemoveFavorite" : "M%F6chten Sie den Favoriten wirklich l%F6schen?",
+    "dialogQuestionRemoveLink" : "M%F6chten Sie die Verkn%FCpfung '%s' wirklich l%F6schen?",
+    "dialogQuestionRemoveUser" : "M%F6chten Sie den Benutzer wirklich '%s' l%F6schen?",
+    "dialogQuestionRemoveSysVar" : "M%F6chten Sie die Systemvariable '%s' wirklich l%F6schen?",
+    "dialogQuestionRemoveFavorite" : "M%F6chten Sie den Favoriten '%s' wirklich l%F6schen?",
     "dialogQuestionRemoveCFG" : "M%F6chten Sie den " + HMIdentifier.de.HomeMaticCFGAdapter + " wirklich l%F6schen?",
     "dialogQuestionRemoveExtraSoftware" : "Wollen Sie die Zusatzsoftware wirklich entfernen?",
     "dialogQuestionRestart" : "Best%E4tigen Sie hier, um den Neustart durchzuf%FChren.",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.extension.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.extension.js
@@ -984,7 +984,7 @@
     "channelModeHumidityLimiter" :  "Humidity limiter",
     "channelModeTactileSwitch" :  "Push-button",
     "channelModeTemperatureLimiter" : "Temperature limiter",
-    "dialogQuestionRemoveInternalLink" :  "<span class='attention'>PLEASE NOTE:</span><br/><br>This is an internal connection of the device. It may occur that the device cannot be operated any more if you delete this connection. In this case, you have to create the connection again.<br/><br/>Do you really want to delete the connection?",
+    "dialogQuestionRemoveInternalLink" :  "<span class='attention'>PLEASE NOTE:</span><br/><br>This is an internal connection of the device. It may occur that the device cannot be operated any more if you delete this connection. In this case, you have to create the connection again.<br/><br/>Do you really want to delete the connection '%s'?",
     "descriptionTransmitterDataTransmissionCondition0" :  "<b>Mode 1</b><br/><br/>Transmit on level change of data transmission pin (DU30) from HIGH to LOW level.",
     "descriptionTransmitterDataTransmissionCondition1" :  "<b>Mode 2</b><br/><br/>Transmit on level change of data transmission pin (DU30) from LOW to HIGH level.",
     "descriptionTransmitterDataTransmissionCondition2" :  "<b>Mode 3</b><br/><br/>Transmit on level change of data transmission pin (DU30) on every logic level change.",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.js
@@ -408,10 +408,10 @@
     "dialogInfo" : "Info",
     "dialogHint" : "Please note",
     "dialogQuestionRemoveProgram" : "Do you really want to delete the program '%s'?",
-    "dialogQuestionRemoveLink" : "Do you really want to delete the connection?",
-    "dialogQuestionRemoveUser" : "Do you really want to delete the user?",
-    "dialogQuestionRemoveSysVar" : "Do you really want to delete the system variable?",
-    "dialogQuestionRemoveFavorite" : "Do you really want to delete the favourite?",
+    "dialogQuestionRemoveLink" : "Do you really want to delete the connection '%s'?",
+    "dialogQuestionRemoveUser" : "Do you really want to delete the user '%s'?",
+    "dialogQuestionRemoveSysVar" : "Do you really want to delete the system variable '%s'?",
+    "dialogQuestionRemoveFavorite" : "Do you really want to delete the favourite '%s'?",
     "dialogQuestionRemoveCFG" : "Do you really want to delete the " + HMIdentifier.en.HomeMaticCFGAdapter + "?",
     "dialogQuestionRemoveExtraSoftware" : "Do you really want to delete the additional software?",
     "dialogQuestionRestart" : "Please click here to perform the restart?",
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -39316,8 +39316,8 @@
 RemoveLink = function(iface, sender_address, receiver_address, sender_type, redirect_url)
 {
   var questionRemoveLink = ((iface == "HmIP-RF") && (sender_type != "HmIP-SMI55") && (sender_address.split(":")[0] == receiver_address.split(":")[0])) ? translateKey('dialogQuestionRemoveInternalLink') : translateKey('dialogQuestionRemoveLink');
-
-  new YesNoDialog(translateKey('dialogSafetyCheck'), questionRemoveLink, function(result) {
+  
+  new YesNoDialog(translateKey('dialogSafetyCheck'), questionRemoveLink.replace('%s', sender_address+' - '+receiver_address), function(result) {
     if (result == YesNoDialog.RESULT_YES)
     {
       ResetPostString();
