--- occu/WebUI/www/api/methods/sysvar/createbool.tcl.orig
+++ occu/WebUI/www/api/methods/sysvar/createbool.tcl
@@ -26,7 +26,11 @@
   sv.ValueName1("true");
   sv.State(init_val);
 
-  sv.Internal(internal);
+  if((internal == "true") || (internal == 1)) {
+    sv.Internal(true);
+  } else {
+    sv.Internal(false);
+  }
 
   if (channel) {
     sv.Channel(chnID);
@@ -35,7 +39,7 @@
 
   oSysVars.Add(sv.ID());
 
-  Write("{'name':'"#sv.Name()#"','id':'"#sv.ID()#"','value':'"#sv.Value()#"' }");
+  Write('{"name":"'#sv.Name()#'","id":"'#sv.ID()#'","value":"'#sv.Value()#'" }');
   }
 
 jsonrpc_response [hmscript $script args]
--- occu/WebUI/www/api/methods/sysvar/createenum.tcl.orig
+++ occu/WebUI/www/api/methods/sysvar/createenum.tcl
@@ -26,8 +26,11 @@
   sv.ValueList( valList );
   sv.State(0);
 
-
-  sv.Internal(internal);
+  if((internal == "true") || (internal == 1)) {
+    sv.Internal(true);
+  } else {
+    sv.Internal(false);
+  }
 
   if (channel) {
     sv.Channel(chnID);
@@ -36,7 +39,7 @@
 
   oSysVars.Add(sv.ID());
 
-  Write("{'name':'"#sv.Name()#"','id':'"#sv.ID()#"','value':'"#sv.ValueList()#"' }");
+  Write('{"name":"'#sv.Name()#'","id":"'#sv.ID()#'","value":"'#sv.ValueList()#'" }');
 }
 
 jsonrpc_response [hmscript $script args]
--- occu/WebUI/www/api/methods/sysvar/createfloat.tcl.orig
+++ occu/WebUI/www/api/methods/sysvar/createfloat.tcl
@@ -23,7 +23,12 @@
   sv.ValueMin(minValue);
   sv.ValueMax(maxValue);
   sv.State(0);
-  sv.Internal(internal);
+
+  if((internal == "true") || (internal == 1)) {
+    sv.Internal(true);
+  } else {
+    sv.Internal(false);
+  }
 
   if (channel) {
     sv.Channel(chnID);
@@ -32,7 +37,7 @@
 
   oSysVars.Add(sv.ID());
 
-  Write("{'name':'"#sv.Name()#"','id':'"#sv.ID()#"','value':'"#sv.Value()#"' }");
+  Write('{"name":"'#sv.Name()#'","id":"'#sv.ID()#'","value":"'#sv.Value()#'" }');
 }
 
 jsonrpc_response [hmscript $script args]
--- occu/WebUI/www/api/methods/sysvar/get.tcl.orig
+++ occu/WebUI/www/api/methods/sysvar/get.tcl
@@ -32,7 +32,7 @@
 ##
 
 set script {
-  var sv = dom.GetObject(id);
+  var sv = dom.GetObject(ID_SYSTEM_VARIABLES).Get(id);
   if (sv)
   {
     var sv_type = "UNKNOWN";
--- occu/WebUI/www/api/methods/sysvar/getvalue.tcl.orig
+++ occu/WebUI/www/api/methods/sysvar/getvalue.tcl
@@ -10,7 +10,7 @@
 ##
 
 set script {
-  var sv = dom.GetObject(id);
+  var sv = dom.GetObject(ID_SYSTEM_VARIABLES).Get(id);
   
   if (sv)
   {
--- occu/WebUI/www/api/methods/sysvar/getvaluebyname.tcl.orig
+++ occu/WebUI/www/api/methods/sysvar/getvaluebyname.tcl
@@ -11,11 +11,11 @@
 
 
  set script {
-    var sv = dom.GetObject(name);
+    var sv = dom.GetObject(ID_SYSTEM_VARIABLES).Get(name);
     if (sv)
     {
       Write(sv.Value());
     }
  }
 
- jsonrpc_response [json_toString [hmscript $script args]]
\ No newline at end of file
+ jsonrpc_response [json_toString [hmscript $script args]]
--- occu/WebUI/www/api/methods/sysvar/setbool.tcl.orig
+++ occu/WebUI/www/api/methods/sysvar/setbool.tcl
@@ -23,7 +23,7 @@
  }
 
  set script {
-   var sv = dom.GetObject(name);
+   var sv = dom.GetObject(ID_SYSTEM_VARIABLES).Get(name);
 
    if (sv)
    {
--- occu/WebUI/www/api/methods/sysvar/setenum.tcl.orig
+++ occu/WebUI/www/api/methods/sysvar/setenum.tcl
@@ -12,7 +12,7 @@
  ##
 
  set script {
-   var sv = dom.GetObject(name);
+   var sv = dom.GetObject(ID_SYSTEM_VARIABLES).Get(name);
 
    if (sv)
    {
@@ -21,7 +21,7 @@
  }
 
  if {[hmscript $script args] } {
-   jsonrpc_response $args(valueList)
+   jsonrpc_response \"$args(valueList)\"
  } else {
    jsonrpc_response -1
  }
--- occu/WebUI/www/api/methods/sysvar/setfloat.tcl.orig
+++ occu/WebUI/www/api/methods/sysvar/setfloat.tcl
@@ -12,7 +12,7 @@
  ##
 
  set script {
-   var sv = dom.GetObject(name);
+   var sv = dom.GetObject(ID_SYSTEM_VARIABLES).Get(name);
 
    if (sv)
    {
