--- occu/WebUI/www/api/eq3/jsonrpc.tcl.orig
+++ occu/WebUI/www/api/eq3/jsonrpc.tcl
@@ -84,7 +84,11 @@
     puts -nonewline "\"id\": $JSONRPC(ID),"
   }
   puts -nonewline "\"version\": \"1.1\","    
-  puts -nonewline "\"result\": $result," 
+  if { [string trim $result] != "" } {
+    puts -nonewline "\"result\": $result,"
+  } else {
+    puts -nonewline "\"result\": null,"
+  }
   puts -nonewline "\"error\": null"   
   puts -nonewline "\}"
 }
--- occu/WebUI/www/api/methods/ccu/restartReGa.tcl.orig
+++ occu/WebUI/www/api/methods/ccu/restartReGa.tcl
@@ -7,4 +7,6 @@
 # Rückgabewert: kein
 ##
 
-catch {exec /etc/init.d/S70ReGaHss restart}
\ No newline at end of file
+catch {exec /etc/init.d/S70ReGaHss restart}
+
+jsonrpc_response ""
--- occu/WebUI/www/api/methods/ccu/restartsshdaemon.tcl.orig
+++ occu/WebUI/www/api/methods/ccu/restartsshdaemon.tcl
@@ -7,4 +7,6 @@
 # Rückgabewert: kein
 ##
 
-catch {exec /etc/init.d/S50sshd restart}
\ No newline at end of file
+catch {exec /etc/init.d/S50sshd restart}
+
+jsonrpc_response ""
--- occu/WebUI/www/api/methods/ccu/setssh.tcl.orig
+++ occu/WebUI/www/api/methods/ccu/setssh.tcl
@@ -18,3 +18,4 @@
 
 source "/bin/setfirewall.tcl"
 
+jsonrpc_response ""
--- occu/WebUI/www/api/methods/ccu/setstickyunreach.tcl.orig
+++ occu/WebUI/www/api/methods/ccu/setstickyunreach.tcl
@@ -16,3 +16,4 @@
   catch {exec rm /etc/config/hideStickyUnreach}
 }
 
+jsonrpc_response ""
--- occu/WebUI/www/api/methods/user/setlanguage.tcl.orig
+++ occu/WebUI/www/api/methods/user/setlanguage.tcl
@@ -14,3 +14,4 @@
 set userName [string map {"../" ""} $userName]
 exec echo $args(userLang) > /etc/config/userprofiles/$userName.lang
 
+jsonrpc_response ""
--- occu/WebUI/www/api/methods/user/setregaversion.tcl.orig
+++ occu/WebUI/www/api/methods/user/setregaversion.tcl
@@ -9,3 +9,5 @@
 ##
 
 exec echo $args(ReGaVersion) > /etc/config/ReGaHssVersion
+
+jsonrpc_response ""
