--- occu/WebUI/www/config/ic_deviceparameters.cgi.orig
+++ occu/WebUI/www/config/ic_deviceparameters.cgi
@@ -40,8 +40,8 @@
 
 proc put_page {} {
 
-  global MODE redirect_url dev_descr
-
+  global MODE redirect_url dev_descr iface iface_descr
+  set HMRFIdentifier "BidCos-RF"
   #cgi_debug -on
 
   puts "<div id=\"ic_deviceparameters\">"
@@ -51,6 +51,93 @@
       puts "<div class=\"subOffsetDivPopup CLASS22000\">"
       
       puts "<script type=\"text/javascript\">"
+      if {($iface_descr($iface) == $HMRFIdentifier)} then {
+        puts "var POPUP_HEIGHT=110;"
+        puts "var POPUP_WIDTH =410;"
+        puts "this.gatewaysListbox = new UI.ListBox()"
+        puts "  .setPosition(100,10)"
+        puts "  .setWidth(POPUP_WIDTH - 130);"
+        
+        puts "this.roamingCheckbox = new UI.Checkbox()"
+        puts "  .setText(translateKey('dialogSettingsBidcosRFIFaceAssignmentLblNullify'))"
+        puts "  .setPosition(100,40)"
+        puts "  .setWidth(POPUP_WIDTH - 130)"
+        puts "  .setIsChecked(this.roaming);"
+
+        puts "homematic('Interface.listBidcosInterfaces', {'interface': '$HMRFIdentifier'}, function(interfaceStatus) {"
+        puts "  if (interfaceStatus) {"
+        puts "    for (var i = 0; i < interfaceStatus.length; i++) {"
+        puts "      if (interfaceStatus\[i\].isDefault) {"
+        puts "       var defaultGateway = {type:interfaceStatus\[i\].type, userName:'$HMRFIdentifier',serialNumber:interfaceStatus\[i\].address};"
+        puts "        this.gatewaysListbox.add({name: translateKey('dialogSettingsBidcosRFIFaceAssignmentLblStandard'), value: defaultGateway});"
+        puts "      }"
+        puts "    }"
+        puts "  }"
+        puts "});"
+
+        puts "this.gateways = \[\];"
+        puts "var bHasCustomName=false;"
+        puts "var InterfaceCellContent = document.getElementById('InterfaceCellContent');"
+        puts "var rfGateways = homematic('BidCoS_RF.getConfigurationRF');"
+        puts "if (rfGateways) {"
+        puts "  for (var i = 0, len = rfGateways.interfaces.length; i < len; i++) {"
+        puts "    var gateway = rfGateways.interfaces\[i\];"
+        puts "    var name = (gateway.userName.length > 0) ? gateway.userName: gateway.serialNumber;"
+        puts "    if (InterfaceCellContent) {"
+        puts "      if (gateway.serialNumber === InterfaceCellContent.innerHTML) {"
+        puts          "bHasCustomName=true;"
+        puts          "InterfaceCellContent.innerHTML = name;"
+        puts "      }"
+        puts "      if (this.roaming) InterfaceCellContent.innerHTML= InterfaceCellContent.innerHTML+'<br/>(ROAMING)';"
+        puts "    }"
+        puts "    this.gateways.push({name: name, value: gateway});"
+        puts "  }"
+        puts "} else {"
+        puts "  var InterfaceCell = document.getElementById('InterfaceCell');"
+        puts "  if (InterfaceCell) InterfaceCell.innerHTML = '$HMRFIdentifier';"
+        puts "}"
+        puts "if (InterfaceCellContent && bHasCustomName === false) { InterfaceCellContent.innerHTML = '$HMRFIdentifier'; }"
+        puts "if (this.gateways) {"
+        puts "  this.gateways.sort((a, b) => a.name.localeCompare(b.name, undefined, { numeric: true, sensitivity: 'base' }));"
+        puts "  for (var i = 0, len = this.gateways.length; i < len; i++) {"
+        puts "    this.gatewaysListbox.add(this.gateways\[i\]);"
+        puts "  }"
+        puts "}"
+        
+        puts "this.ok = function() { "
+        puts "  var selectedGateway = this.gatewaysListbox.getSelectedItem().value;"
+        puts "  var selectedGatewayName = (selectedGateway.userName.length > 0) ? selectedGateway.userName: selectedGateway.serialNumber;"
+        puts "  homematic('Interface.setBidcosInterface', {"
+        puts "   'interface': '$HMRFIdentifier',"
+        puts "   'deviceId': this.addr,"
+        puts "   'interfaceId': selectedGateway.serialNumber,"
+        puts "   'roaming': this.roamingCheckbox.isChecked()"
+        puts "  });"
+        puts "  if (InterfaceCellContent) {"
+        puts "    InterfaceCellContent.innerHTML = selectedGatewayName ;"
+        puts "     if (this.roamingCheckbox.isChecked()) InterfaceCellContent.innerHTML= InterfaceCellContent.innerHTML+'<br/>(ROAMING)';"
+        puts "  }"
+        puts "  DeviceList.updateInterfaceDisplayName(this.addr, selectedGatewayName);"
+        puts "  close();"      
+        puts "}"    
+
+        puts "this.close = function() { Layer.remove(gwframe.getElement()); }"   
+
+        puts "setGW = function(addr,name,roaming) {"
+        puts "  this.addr = addr;"
+        puts "  this.roaming = roaming;"
+        puts "  gwframe = new UI.Frame()"
+        puts "    .setTitle(translateKey('dialogSettingsBidcosRFIFaceAssignmentTitle')+name)"
+        puts "    .setContentSize(POPUP_WIDTH, POPUP_HEIGHT)"
+        puts "    .setPosition(parseInt((WebUI.getWidth()  - POPUP_WIDTH)  / 2), parseInt((WebUI.getHeight() - POPUP_HEIGHT) / 2))"
+        puts "    .add(new UI.Text().setPosition(10, 10).setText(translateKey('dialogSettingsBidcosRFIFaceAssignmentLblGateway')))"
+        puts "    .add(this.gatewaysListbox)"
+        puts "    .add(this.roamingCheckbox)"
+        puts "    .add(new UI.Button().setPosition(10,  POPUP_HEIGHT - 30).setText(translateKey('btnCancel')).setAction(this.close, this))"
+        puts "    .add(new UI.Button().setPosition(250, POPUP_HEIGHT - 30).setText(translateKey('btnOk'))    .setAction(this.ok,    this));"
+        puts "  Layer.add(gwframe.getElement());"
+        puts "};"
+      }
       if {$redirect_url == "IC_SETPROFILES" || $redirect_url == "IC_LINKPEERLIST"} then { 
         #puts "  setPath(\"<span onclick='WebUI.enter(LinksAndProgramsPage);'>Programme &amp; Verkn&uuml;pfungen</span> &gt; <span onclick='WebUI.enter(LinkListPage);'>Direkte Verkn&uuml;pfungen</span> &gt; Ger&auml;te- / Kanalparameter einstellen\");"
         puts "  setPath(\"<span onclick='WebUI.enter(LinksAndProgramsPage);'>\"+translateKey('menuProgramsLinksPage')+\"</span> &gt; <span onclick='WebUI.enter(LinkListPage);'>\"+translateKey('submenuDirectLinks')+\"</span> &gt; \" + translateKey('setDeviceAndChannelParams'));"
@@ -1176,13 +1263,17 @@
   set SENTRY(IMAGE)       "&nbsp;"
   set SENTRY(DESCRIPTION) $dev_descr(TYPE)
   set SENTRY(ADDRESS)     $address
-  set SENTRY(IFACE)       $iface_descr($iface)
   set SENTRY(FIRMWARE)    "&nbsp;"
 
   if { [catch { set SENTRY(NAME) $ise_CHANNELNAMES($iface;$address)} ] } then {
       set SENTRY(NAME) "$iface"
       append SENTRY(NAME) ".$address"
   }
+  if {($iface_descr($iface) == $HMRFIdentifier)} then {
+    set SENTRY(IFACE)       "<table id='InterfaceTable'><tr id='InterfaceTableRow'><td class='CLASS22012' style='border: none !important;' onclick=\"setGW('$address','$SENTRY(NAME)',$dev_descr(ROAMING));\"><div class=\"CLASS21000\" id='InterfaceCellContent'>$dev_descr(INTERFACE)</div></td></tr></table>"
+  } else {
+    set SENTRY(IFACE)       $iface_descr($iface)
+  }
   set EASYLINKMODE [get_user_easylinkmode]
   set SENTRY(IMAGE) "<div id=\"picDiv_thumb\" class=\"CLASS22005\" onmouseover=\"picDivShow(jg_250, '$dev_descr(TYPE)', 250, '-1', this);\" onmouseout=\"picDivHide(jg_250);\"></div>"
   append SENTRY(IMAGE) "<script type=\"text/javascript\">"
@@ -1328,12 +1419,12 @@
   puts "<TBODY>"
       
   puts "<tr>"
-  puts "<td>[cgi_quote_html $SENTRY(NAME)]</td>"
-  puts "<td>$SENTRY(TYPE)</td>"
+  puts "<td class='DeviceListCell'>[cgi_quote_html $SENTRY(NAME)]</td>"
+  puts "<td class='DeviceListCell'>$SENTRY(TYPE)</td>"
   puts "<td id=\"DeviceImage\">$SENTRY(IMAGE)</td>"
-  puts "<td id=\"DeviceDescription\">$SENTRY(DESCRIPTION)</td>"
-  puts "<td>$SENTRY(ADDRESS)</td>"
-  puts "<td>$SENTRY(IFACE)</td>"
+  puts "<td class='DeviceListCell' id=\"DeviceDescription\">$SENTRY(DESCRIPTION)</td>"
+  puts "<td class='DeviceListCell'>$SENTRY(ADDRESS)</td>"
+  puts "<td class='DeviceListCell' id='InterfaceCell'>$SENTRY(IFACE)</td>"
   puts "<td>$SENTRY(FIRMWARE)</td>"
   if {$EASYLINKMODE == "false"} {
     puts "<td>$SENTRY(RESTOREDEVPARAMS)</td>"
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -491,7 +491,7 @@
           <td class=\"DeviceListThumbnail\" ><div id=\"${PREFIX}${device.id}Thumbnail\" class=\"thumbnail\" onmouseover=\"picDivShow(jg_250, \'${device.deviceType.id}\', 250, \'\', this);\" onmouseout=\"picDivHide(jg_250);\">${device.getThumbnailHTML()}<\/div><\/td>\n
           <td class=\"DeviceListCell\" name=\"${device.typeDescription}\" >${device.typeDescription}<\/td>\n
           <td class=\"DeviceListCell\" >${device.address}${device.rfAddress}<\/td>\n
-          <td class=\"DeviceListCell\" >${device.interfaceName}<\/td>\n
+          <td class=\"DeviceListCell\" id=\"DeviceInterfaceDisplayName${device.id}\" >${device.interfaceDisplayName}<\/td>\n
           <td class=\"DeviceListCell j_chMode\" >{for name in device.modes}${name}<br \/>{forelse}&#160;{\/for}<\/td>\n
           <td class=\"DeviceListCell j_functions\" >{for subsection in device.subsections}${subsection.name}<br \/>{forelse}&#160;{\/for}<\/td>\n
           <td class=\"DeviceListCell j_rooms\" >{for room in device.rooms}${room.name}<br \/>{forelse}&#160;{\/for}<\/td>\n
@@ -11017,6 +11017,7 @@
       // End debugging
 
       this.interfaceName = data["interface"];
+      this.interfaceDisplayName = data["interface_displayname"];
       this.isReadyConfig = data["isReady"];
       this.enabledServiceMsg = (data["enabledServiceMsg"] == "true") ? true : false;
       this.thumbnailHTML = deviceType.getThumbnailHTML();
@@ -11232,6 +11233,13 @@
     this.rssiPeerColor = getColorCode(this.RSSI_PEER);
   },
 
+  updateInterfaceDisplayName: function(interfaceDisplayName) {
+    this.interfaceDisplayName = interfaceDisplayName;
+    var devInterfaceDisplayNameCell = document.getElementById('DeviceInterfaceDisplayName' + this.id);
+    if (devInterfaceDisplayNameCell !== null && typeof(devInterfaceDisplayNameCell) !== 'undefined') {
+      devInterfaceDisplayNameCell.innerHTML = interfaceDisplayName;
+    }
+  },
   /**
    * Legt den Namen des Geräts fest.
    **/
@@ -11597,11 +11605,50 @@
     this.channels = {};
     this.groups   = {};
 
+    
+    var BidCosGateways=[];
+    var allBidCosDevsList=[];
+    var rfGateways = homematic("BidCoS_RF.getConfigurationRF");
+    if (rfGateways) {
+      for (i = 0, len = rfGateways.interfaces.length; i < len; i++) {
+        var gateway = rfGateways.interfaces[i];
+        if (gateway.type === globalLGWTypes.HMLGW2) {
+          BidCosGateways.push(
+           new BidcosRfPage.Gateway()
+            .setGatewayClass("RF")
+            .setType(gateway.type)
+            .setUserName((gateway.userName.length > 0) ? gateway.userName : gateway.serialNumber)
+            .setAddress(gateway.serialNumber)
+            .setKey(gateway.encryptionKey)
+            .setIP(gateway.ipAddress)
+          );
+        }
+      }
+      BidCosGateways.sort((a, b) => a.m_userName.localeCompare(b.m_userName, undefined, { numeric: true, sensitivity: 'base' }));
+      for (i = 0, len = BidCosGateways.length; i < len; i++) {
+        DeviceListPage.INTERFACES.push({id:BidCosGateways[i].m_userName, name:BidCosGateways[i].m_userName});
+      }
+      allBidCosDevsList = homematic("Interface.listDevices", {"interface": "BidCos-RF"});
+    }
+    
+
     var self=this;
     homematic("Device.listAllDetail", null, function(deviceList) {
       // alert("after listAllDetail devices: " + deviceList);
       jQuery.each(deviceList, function (index, data) {
       // console.dir (data);
+        data["interface_displayname"] = data["interface"];
+        if ( allBidCosDevsList && rfGateways && data["interface"] === "BidCos-RF") {
+          var idx = allBidCosDevsList.findIndex(({ address }) => address === data["address"]);
+          if (idx > -1) {
+            var _device= allBidCosDevsList[idx];
+            var gwIdx = BidCosGateways.findIndex(({ m_address }) => m_address === _device["interface"]);
+            if ( gwIdx > -1 ) {
+              data["interface_displayname"] = (BidCosGateways[gwIdx].m_userName.length > 0) ?  BidCosGateways[gwIdx].m_userName : _device["interface"]; 
+            }
+          }
+        }
+
         if (data !== null && (data["type"] != typeCCU1))
         {
           var id = data["id"];
@@ -11665,6 +11712,16 @@
       }
     });
   },
+  
+  updateInterfaceDisplayName:function(addr,gwName) {
+    var _this_ = this;
+    for (var id in _this_.devices) {
+      var device = _this_.devices[id];
+      if (device.address === addr) {
+        device.updateInterfaceDisplayName(gwName);
+      }
+    }
+  },
 
   /**
    * Startet die Akualisierung eines Geräts.
@@ -22178,7 +22235,7 @@
             (this.TypeNameFilter.match(device.typeName))                 &&
             (this.DescriptionFilter.match(translateKey(device.typeDescription)))       &&
             (this.AddressFilter.match(device.address))                   &&
-            (this.InterfaceFilter.matchString(device.interfaceName))     &&
+            (this.InterfaceFilter.matchString(device.interfaceDisplayName))     &&
             (this.ModeFilter.matchStringArray(device.modes)) &&
             (this.RoomFilter.matchArray(device.rooms))                   &&
             (this.FuncFilter.matchArray(device.subsections)));
@@ -24410,6 +24467,7 @@
       //jQuery("."+gateway.getAddress()).attr("class","bidcosrf_tablecell " + gateway.getAddress().replace(/\s+/g,""));
       this.setGateway(gateway.getAddress());
       this.setRoaming(roaming);
+      DeviceList.updateInterfaceDisplayName(this.m_address, (gateway.getUserName()).length > 0 ? gateway.getUserName() : 'BidCos-RF');
     }
   }
 };
