--- occu/WebUI/www/api/methods/user/restarthmipserver.tcl.orig
+++ occu/WebUI/www/api/methods/user/restarthmipserver.tcl
@@ -1,6 +1,6 @@
 ##
 # User.restartHmIPServer
-# Restartet den Lighttpd Webserver
+# Restartet den HMIPServer
 #
 # Parameter:
 #   keine
@@ -8,5 +8,5 @@
 # Rückgabewert: true
 
 #exec /etc/init.d/S62HMServer start &
-exec -- /usr/bin/nohup /etc/init.d/S62HMServer restart >/dev/null &
+exec monit restart HMIPServer >/dev/null &
 jsonrpc_response true
--- occu/WebUI/www/api/methods/user/starthmipserver.tcl.orig
+++ occu/WebUI/www/api/methods/user/starthmipserver.tcl
@@ -1,11 +1,11 @@
 ##
 # User.startHmIPServer
-# Startet den Lighttpd Webserver
+# Startet den HMIPServer
 #
 # Parameter:
 #   keine
 #
 # Rückgabewert: true
 
-exec -- /usr/bin/nohup /etc/init.d/S62HMServer start >/dev/null &
+exec monit start HMIPServer >/dev/null &
 jsonrpc_response true
--- occu/WebUI/www/api/methods/user/stophmipserver.tcl.orig
+++ occu/WebUI/www/api/methods/user/stophmipserver.tcl
@@ -1,11 +1,11 @@
 ##
 # User.stopHmIPServer
-# Stoppt den Lighttpd Webserver
+# Stoppt den HMIPServer
 #
 # Parameter:
 #   keine
 #
 # Rückgabewert: true
 
-exec -- /usr/bin/nohup /etc/init.d/S62HMServer stop >/dev/null &
+exec monit stop HMIPServer >/dev/null &
 jsonrpc_response true
--- occu/WebUI/www/config/cp_maintenance.cgi.orig
+++ occu/WebUI/www/config/cp_maintenance.cgi
@@ -14,6 +14,14 @@
   5 "\${dialogSettingsCMLogLevel3}"
 }
 
+array set HMIP_LOGLEVELS {
+  "0TRACE" "\${dialogSettingsCMLogLevel5}"
+  "1DEBUG" "\${dialogSettingsCMLogLevel0}"
+  "2INFO" "\${dialogSettingsCMLogLevel1}"
+  "3WARN" "\${dialogSettingsCMLogLevel2}"
+  "4ERROR" "\${dialogSettingsCMLogLevel3}"
+}
+
 array set REGA_LOGLEVELS {
   0 "\${dialogSettingsCMLogLevel0}"
   1 "\${dialogSettingsCMLogLevel1}"
@@ -333,7 +341,7 @@
 }
 
 proc action_put_page {} {
-  global env sid REMOTE_FIRMWARE_SCRIPT LOGLEVELS REGA_LOGLEVELS RFD_URL HS485D_URL downloadOnly
+  global env sid REMOTE_FIRMWARE_SCRIPT LOGLEVELS HMIP_LOGLEVELS REGA_LOGLEVELS RFD_URL HS485D_URL downloadOnly
   http_head
 
   division {class="popupTitle j_translate"} {
@@ -553,6 +561,19 @@
                 table {class="CLASS20909"} {
                   table_row {
                     table_data {align="left"} {
+                      puts "\${dialogSettingsCMLblLogHomematicIP}"
+                    }
+                    table_data {align="right"} {
+                      cgi_select log_hmip= {id="select_log_hmip"} {
+                        foreach level [lsort [array names HMIP_LOGLEVELS]] {
+                          cgi_option $HMIP_LOGLEVELS($level) value=[string range $level 1 end] 
+                        }
+                      }
+                      get_hmip_loglevel
+                    }
+                  }
+                  table_row {
+                    table_data {align="left"} {
                       puts "\${dialogSettingsCMLblLogBidCosRF}"
                     }
                     table_data {align="right"} {
@@ -675,6 +696,7 @@
     puts {
       apply_logging = function() {
         var pb = "action=apply_logging";
+        pb += "&level_hmip="+document.getElementById("select_log_hmip").value;
         pb += "&level_rfd="+document.getElementById("select_log_rfd").value;
         pb += "&level_hs485d="+document.getElementById("select_log_hs485d").value;
         //pb += "&level_pfmd="+document.getElementById("select_log_pfmd").value;
@@ -866,6 +888,22 @@
   }
 }
 
+ proc get_hmip_loglevel {} {
+    set fd [open "/etc/config/log4j.xml" r]
+    set file_data [read $fd]
+    regsub -all {\n} $file_data { } file_data
+    close $fd
+    cgi_javascript {
+      puts "var xmlDoc = jQuery.parseXML( '$file_data' );"
+      puts "var rootElement = jQuery(xmlDoc).find('root');"
+      puts "jQuery(rootElement).children().each(function(){ if (this.nodeName === 'priority') {"
+      puts "var priorityValue = this.attributes\[0\].nodeValue; "
+      puts "jQuery(\"#select_log_hmip\").val(priorityValue).prop('selected', true); "
+      puts "} });"
+    }
+    return 2;
+  }
+
 proc get_serial { } {
   return [read_var /var/ids SerialNumber]
 }
@@ -1285,7 +1323,7 @@
 }
 
 # proc set_log_config {loghost level_rfd level_hs485d level_pfmd level_rega}
-proc set_log_config {loghost level_rfd level_hs485d level_rega} {
+proc set_log_config {loghost level_rfd level_hs485d level_rega level_hmip} {
   #global RFD_URL HS485D_URL PFMD_URL
   global RFD_URL HS485D_URL
 
@@ -1300,6 +1338,7 @@
   puts $fd "LOGLEVEL_HS485D=$level_hs485d"
   # puts $fd "LOGLEVEL_PFMFD=$level_pfmd"
   puts $fd "LOGLEVEL_REGA=$level_rega"
+  puts $fd "LOGLEVEL_HMIP=$level_hmip"
   close $fd
   
   catch { xmlrpc $RFD_URL logLevel [list int $level_rfd]}
@@ -1307,6 +1346,44 @@
   # catch { xmlrpc $PFMD_URL logLevel [list int $level_pfmd]}
   rega "system.LogLevel($level_rega)"
   
+  set log4jfilename     "/etc/config/log4j.xml"
+  
+  set appender_ref_ext  "\\<appender-ref ref=\"EXT_SYSLOG\" \\/\>"
+  set start_insert      "\<log4j\:configuration"
+  set appender_name_ext "\\<appender name=\"EXT_SYSLOG\""
+  
+  set    appender_ext_full "\\          \<appender name=\"EXT_SYSLOG\" class=\"org.apache.log4j.net.SyslogAppender\"\>\\n"
+  append appender_ext_full "\\                \<param name=\"SyslogHost\" value=\"$loghost\" \\/\>\\n"
+  append appender_ext_full "\\                \<param name=\"Facility\" value=\"USER\" \\/\>\\n"
+  append appender_ext_full "\\                \<param name=\"FacilityPrinting\" value=\"false\" \\/\>\\n"
+  append appender_ext_full "\\                \<layout class=\"org.apache.log4j.PatternLayout\"\>\\n"
+  append appender_ext_full "\\                        \<param name=\"ConversionPattern\" value=\"%c %-5p \[%t\] %m %n\"\\/\>\\n"
+  append appender_ext_full "\\               \<\\/layout\>\\n"
+  append appender_ext_full "\\        \<\\/appender\>"
+  
+  set fp [open $log4jfilename r]
+  set file_data [read $fp]
+  close $fp
+
+  regsub -all {priority value="DEBUG"}  $file_data {priority value="PLACEHOLDER"} file_data
+  regsub -all {priority value="INFO"}   $file_data {priority value="PLACEHOLDER"} file_data
+  regsub -all {priority value="WARN"}   $file_data {priority value="PLACEHOLDER"} file_data
+  regsub -all {priority value="ERROR"}  $file_data {priority value="PLACEHOLDER"} file_data
+  regsub -all PLACEHOLDER $file_data $level_hmip file_data
+
+  set fd [open $log4jfilename w]
+  puts -nonewline $fd $file_data
+  close $fd
+  
+  exec sed -i "/$appender_name_ext/,+7 d" $log4jfilename
+  exec sed -i "/$appender_ref_ext/d"      $log4jfilename 
+  if { "$loghost" != "" } {
+    exec sed -i "/appender-ref ref=\"SYSLOG\"/a\\                $appender_ref_ext" $log4jfilename
+    exec sed -i "/$start_insert/a$appender_ext_full" $log4jfilename    
+  }
+  exec unix2dos $log4jfilename
+  exec monit restart HMIPServer >/dev/null &
+  
   return 1
 }
 
@@ -1316,11 +1393,12 @@
   import level_hs485d
   # import level_pfmd
   import level_rega
+  import level_hmip
   
   http_head
   
   # if {![set_log_config $log_server $level_rfd $level_hs485d $level_pfmd $level_rega]}
-  if {![set_log_config $log_server $level_rfd $level_hs485d $level_rega]} {
+  if {![set_log_config $log_server $level_rfd $level_hs485d $level_rega $level_hmip]} {
     puts "Failure"
     return
   }
--- occu/WebUI/www/webui/js/lang/de/translate.lang.extension.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.extension.js
@@ -393,7 +393,7 @@
     "lblValue" : "Wert",
     "lblRangeOfValue" : "Wertebereich",
     "transferConfigData" : "Ger%E4te-/Kanaleinstellungen werden an Komponenten %FCbertragen...",
-    "setLoggingSuccess" : "Setzen der Logging-Einstellungen erfolgreich!",
+    "setLoggingSuccess" : "Setzen der Logging-Einstellungen erfolgreich!\nDer Homematic IP-Dienst wird im Hintergrund neu gestartet.",
     "setLoggingFailure" : "Setzen der Logging-Einstellungen fehlgeschlagen:\n",
     "comment" : "RF_REPEATER.txt",
     "maxEntriesReachedTitle" : "Achtung!",
--- occu/WebUI/www/webui/js/lang/de/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.js
@@ -3,6 +3,7 @@
     "CCUFullNameHeader" : "RaspberryMatic",
     "CCUFullNameText" : "RaspberryMatic",
     "CCUShortName" : "RaspMatic",
+    "HomematicIP" : "Homematic IP",
     "BidCosRF" : "Homematic Funk",
     "BidCosWired" : "Homematic Wired",
     "VirtualDevices" : "VirtualDevices",
@@ -491,6 +492,7 @@
     "dialogSettingsCMLblPerformSoftwareUpdateStep2" : "Schritt 2: Heruntergeladene Software ausw%E4hlen",
     "dialogSettingsCMLblPerformSoftwareUpdateStep3" : "Schritt 3: Software auf " + HMIdentifier.de.CCUShortName + " laden",
     "dialogSettingsCMLblPerformSoftwareUpdateStep4" : "Schritt 4: Update starten",
+    "dialogSettingsCMLblLogHomematicIP" : HMIdentifier.de.HomematicIP,
     "dialogSettingsCMLblLogBidCosRF" : HMIdentifier.de.BidCosRF,
     "dialogSettingsCMLblLogBidCosWired" : HMIdentifier.de.BidCosWired,
     "dialogSettingsCMLblLogCentralControl" : HMIdentifier.de.CCUShortName + "-Steuerung",
@@ -522,6 +524,7 @@
     "dialogSettingsCMLogLevel2" : "Nur Warnungen",
     "dialogSettingsCMLogLevel3" : "Nur Fehler",
     "dialogSettingsCMLogLevel4" : "Kein Logging",
+    "dialogSettingsCMLogLevel5" : "Tracing",
     "commentSettingsDialogMaintenanceEnd" : "***********************************************",
     "commentSettingsDialogSecurity" : "***********************************************",
     "dialogSettingsSecurityTitle" : HMIdentifier.de.CCUShortName + "-Sicherheit",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.extension.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.extension.js
@@ -393,7 +393,7 @@
     "lblValue" : "Value",
     "lblRangeOfValue" : "Value range",
     "transferConfigData" : "Device/channel settings will be transmitted to devices.",
-    "setLoggingSuccess" : "Saving the logging settings was successful!",
+    "setLoggingSuccess" : "Saving the logging settings was successful!\nThe Homematic IP service will be restarted in background.",
     "setLoggingFailure" : "Saving the logging settings failed:\n",
     "maxEntriesReachedTitle" : "Attention!",
     "maxEntriesReached_a" : "Maximum number of ",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.js
@@ -3,6 +3,7 @@
     "CCUFullNameHeader" : "RaspberryMatic",
     "CCUFullNameText" : "RaspberryMatic",
     "CCUShortName" : "RaspMatic",
+    "HomematicIP" : "Homematic IP",
     "BidCosRF" : "Homematic RF",
     "BidCosWired" : "BidCos-Wired",
     "BidCosRF-Filter" : "BidCos-RF",
@@ -490,6 +491,7 @@
     "dialogSettingsCMLblPerformSoftwareUpdateStep2" : "Step 2: Select downloaded software",
     "dialogSettingsCMLblPerformSoftwareUpdateStep3" : "Step 3: Upload software to " + HMIdentifier.en.CCUShortName,
     "dialogSettingsCMLblPerformSoftwareUpdateStep4" : "Step 4: Start update",
+    "dialogSettingsCMLblLogHomeMaticIP" : HMIdentifier.de.HomematicIP,
     "dialogSettingsCMLblLogBidCosRF" : HMIdentifier.en.BidCosRF,
     "dialogSettingsCMLblLogBidCosWired" : HMIdentifier.en.BidCosWired,
     "dialogSettingsCMLblLogCentralControl" : HMIdentifier.en.CCUShortName + " control",
@@ -521,6 +523,7 @@
     "dialogSettingsCMLogLevel2" : "Only alerts",
     "dialogSettingsCMLogLevel3" : "Only errors",
     "dialogSettingsCMLogLevel4" : "No logging",
+    "dialogSettingsCMLogLevel5" : "Tracing",
     "commentSettingsDialogMaintenanceEnd" : "***********************************************",
     "commentSettingsDialogSecurity" : "***********************************************",
     "dialogSettingsSecurityTitle" : HMIdentifier.en.CCUShortName + " security",
