--- occu/WebUI/www/rega/esp/controls/powermeter.fn.orig
+++ occu/WebUI/www/rega/esp/controls/powermeter.fn
@@ -339,7 +339,7 @@
                         WriteLine("<td>");
                           !WriteLine("<span id='"#chnId#"Power'>"#oPower.Value().ToString(2)#" "#oPower.ValueUnit()#"</span>");
                           WriteLine("<div name='"#chnId#"sensor' class='hidden'><span id='"#chnId#"Power'>"#oPower.Value().ToString(2)#" "#oPower.ValueUnit()#"</span></div>");
-                          WriteLine("<div name='"#chnId#"sensorGas' class='hidden'><span id='"#chnId#"PowerGas'>"#oPowerGas.Value().ToString(2)#" m<sup>3</span></div>");
+                          WriteLine("<div name='"#chnId#"sensorGas' class='hidden'><span id='"#chnId#"PowerGas'>"#oPowerGas.Value().ToString(2)#" m<sup>3</span>/h</div>");
                         WriteLine("</td>");
                       WriteLine("</tr>");
                     WriteLine("</table>");
@@ -1059,7 +1059,7 @@
                       if (oPower && oPowerGas) {
                         if (sLastControlName != "POWERMETER_IEC2.") {
                           WriteLine("<div name='"#chnId#"sensor' class='hidden'><span id='"#chnId#"Power'>"#oPower.Value().ToString(2)#" "#oPower.ValueUnit()#"</span></div>");
-                          WriteLine("<div name='"#chnId#"sensorGas' class='hidden'><span id='"#chnId#"PowerGas'>"#oPowerGas.Value().ToString(2)#" m<sup>3</span></div>");
+                          WriteLine("<div name='"#chnId#"sensorGas' class='hidden'><span id='"#chnId#"PowerGas'>"#oPowerGas.Value().ToString(2)#" m<sup>3</span>/h</div>");
                         }
                       }
                       WriteLine("<div name='"#chnId#"sensorIEC' class='hidden'><span id='"#chnId#"PowerIEC'>"#oPowerIEC.Value().ToString(decimalPlace)#" "#oPowerIEC.ValueUnit()#"</span></div>");
@@ -1105,23 +1105,31 @@
     WriteLine("opts.devFirmwareMinor = '"#devFwMinor#"';");
     WriteLine("opts.devFirmwarePatch = '"#devFwPatch#"';");
     if (sLastControlName == "POWERMETER_IEC1.") {
-      WriteLine("opts.bootFlag = "#oBoot.Value()#";");
-      WriteLine("opts.idEnergyCounter = "#oEnergyCounter.ID()#";");
-      WriteLine("opts.valEnergyCounter = "#oEnergyCounter.Value()#";");
-      WriteLine("opts.unitEnergyCounter = '"#oEnergyCounter.ValueUnit()#"';");
-      WriteLine("opts.maxValEnergyCounter = '"#oEnergyCounter.ValueMax()#"';");
-      WriteLine("opts.minValEnergyCounter = '"#oEnergyCounter.ValueMin()#"';");
-      WriteLine("opts.idPower = "#oPower.ID()#";");
-      WriteLine("opts.valPower = "#oPower.Value()#";");
-      WriteLine("opts.unitPower = '"#oPower.ValueUnit()#"';");
-      WriteLine("opts.idEnergyCounterGas = "#oEnergyCounterGas.ID()#";");
-      WriteLine("opts.valEnergyCounterGas = "#oEnergyCounterGas.Value()#";");
-      WriteLine("opts.unitEnergyCounterGas = '"#oEnergyCounterGas.ValueUnit()#"';");
-      WriteLine("opts.maxValEnergyCounterGas = '"#oEnergyCounterGas.ValueMax()#"';");
-      WriteLine("opts.minValEnergyCounterGas = '"#oEnergyCounterGas.ValueMin()#"';");
-      WriteLine("opts.idPowerGas = "#oPowerGas.ID()#";");
-      WriteLine("opts.valPowerGas = "#oPowerGas.Value()#";");
-      WriteLine("opts.unitPowerGas = '"#oPowerGas.ValueUnit()#"';");
+      if (oBoot)          { WriteLine("opts.bootFlag = "#oBoot.Value()#";"); }
+      if (oEnergyCounter) {
+        WriteLine("opts.idEnergyCounter = "#oEnergyCounter.ID()#";"); 
+        WriteLine("opts.valEnergyCounter = "#oEnergyCounter.Value()#";"); 
+        WriteLine("opts.unitEnergyCounter = '"#oEnergyCounter.ValueUnit()#"';"); 
+        WriteLine("opts.maxValEnergyCounter = '"#oEnergyCounter.ValueMax()#"';"); 
+        WriteLine("opts.minValEnergyCounter = '"#oEnergyCounter.ValueMin()#"';"); 
+      }
+      if (oPower) { 
+        WriteLine("opts.idPower = "#oPower.ID()#";"); 
+        WriteLine("opts.valPower = "#oPower.Value()#";"); 
+        WriteLine("opts.unitPower = '"#oPower.ValueUnit()#"';"); 
+      }
+      if (oEnergyCounterGas) { 
+        WriteLine("opts.idEnergyCounterGas = "#oEnergyCounterGas.ID()#";"); 
+        WriteLine("opts.valEnergyCounterGas = "#oEnergyCounterGas.Value()#";"); 
+        WriteLine("opts.unitEnergyCounterGas = '"#oEnergyCounterGas.ValueUnit()#"';"); 
+        WriteLine("opts.maxValEnergyCounterGas = '"#oEnergyCounterGas.ValueMax()#"';"); 
+        WriteLine("opts.minValEnergyCounterGas = '"#oEnergyCounterGas.ValueMin()#"';"); 
+      }
+      if (oPowerGas) { 
+        WriteLine("opts.idPowerGas = "#oPowerGas.ID()#";"); 
+        WriteLine("opts.valPowerGas = "#oPowerGas.Value()#";"); 
+        WriteLine("opts.unitPowerGas = '"#oPowerGas.ValueUnit()#"';"); 
+      }
     }
     if (oEnergyCounterIEC) {
       WriteLine("opts.idEnergyCounterIEC = "#oEnergyCounterIEC.ID()#";");
