--- occu/WebUI/www/api/methods/session/create.tcl.orig
+++ occu/WebUI/www/api/methods/session/create.tcl
@@ -11,7 +11,7 @@
 
 package require http
 
-set LOGIN_URL {127.0.0.1/login.htm}
+set LOGIN_URL {http://127.0.0.1/login.htm}
 
 ##
 # getHttpHeader
@@ -41,7 +41,7 @@
 ::http::cleanup $request
 
 if { [regexp {sid=@([^@]*)@} $location dummy _session_id_] } then {
-  ::http::cleanup [::http::geturl "127.0.0.1$location"]
+  ::http::cleanup [::http::geturl "http://127.0.0.1$location"]
   jsonrpc_response [json_toString $_session_id_]
 } else {
   jsonrpc_error 501 "could not create session $location [array get $request]"
--- occu/WebUI/www/api/methods/session/login.tcl.orig
+++ occu/WebUI/www/api/methods/session/login.tcl
@@ -12,7 +12,7 @@
 
 package require http
 
-set LOGIN_URL 127.0.0.1/login.htm
+set LOGIN_URL http://127.0.0.1/login.htm
 
 ##
 # getHttpHeader
--- occu/WebUI/www/api/methods/session/renew.tcl.orig
+++ occu/WebUI/www/api/methods/session/renew.tcl
@@ -12,7 +12,7 @@
 package require http
 
 set sid $args(_session_id_)
-set url "127.0.0.1/pages/index.htm?sid=@$sid@"
+set url "http://127.0.0.1/pages/index.htm?sid=@$sid@"
 
 ::http::cleanup [::http::geturl $url]
 
--- occu/WebUI/www/tcl/eq3/session.tcl.orig
+++ occu/WebUI/www/tcl/eq3/session.tcl
@@ -5,8 +5,8 @@
 
 package require http
 
-set LOGIN_URL 127.0.0.1/login.htm
-set RENEW_URL 127.0.0.1/pages/index.htm
+set LOGIN_URL http://127.0.0.1/login.htm
+set RENEW_URL http://127.0.0.1/pages/index.htm
 
 ##
 # session_getHttpHeader
