--- occu/WebUI/www/config/easymodes/hmip/CLIMATECONTROL_HEAT_DEMAND_BOILER_TRANSMITTER.tcl.orig
+++ occu/WebUI/www/config/easymodes/hmip/CLIMATECONTROL_HEAT_DEMAND_BOILER_TRANSMITTER.tcl
@@ -1,5 +1,6 @@
 #!/bin/tclsh
 
+source [file join $env(DOCUMENT_ROOT) config/easymodes/etc/hmipChannelConfigDialogs.tcl]
 source [file join /www/config/easymodes/em_common.tcl]
 
 proc set_htmlParams {iface address pps pps_descr special_input_id peer_type} {
