--- occu/WebUI/www/api/methods/system/saveobjectmodel.tcl.orig
+++ occu/WebUI/www/api/methods/system/saveobjectmodel.tcl
@@ -9,8 +9,9 @@
 ##
 
 
+# call system.Save() async to keep WebUI active
 set script {
-  system.Save();
+  system.Exec("echo 'load tclrega.so; rega system.Save()' | /bin/tclsh &");
   Write("true");
 }
 
--- occu/WebUI/www/webui/js/lang/de/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.js
@@ -52,6 +52,7 @@
     "alarmMsg" : "Alarmmeldungen",
     "serviceMsg" : "Servicemeldungen",
     "logout" : "Abmelden",
+    "logoutInProgress" : "Abmeldevorgang l%E4uft...",
     "regasave" : "%C4nderungen speichern",
     "noPasswd" : "Kein Kennwort gesetzt",
     "menuStartPage" : "Startseite",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.js
@@ -51,6 +51,7 @@
     "alarmMsg" : "Alarm messages",
     "serviceMsg" : "Service messages",
     "logout" : "Logout",
+    "logoutInProgress" : "Logout in progress...",
     "regasave" : "Save changes",
     "noPasswd" : "No password set",
     "menuStartPage" : "Home page",
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -32658,11 +32658,27 @@
 
 
 logout = function() {
+  // disable logout button
+  jQuery("#btnLogOut > div").css('color', 'graytext');
+  jQuery("#btnLogOut > div").attr("onclick", "").unbind("click");
+
+  // create ProgressBar
+  ProgressBar = new ProgressBarMsgBox(translateKey('logoutInProgress'), 1);
+  ProgressBar.show();
+  ProgressBar.StartKnightRiderLight();
+
+  // stop all update monitors
   regaMonitor.stop();
   InterfaceMonitor.stop();
-  location.href = "/logout.htm?lang="+getLang();
-  homematic('Session.logout', {});
-  homematic('system.saveObjectModel', {});
+
+  // logout and make sure we save the regadom
+  homematic('Session.logout', {}, function() {
+    homematic('system.saveObjectModel', {}, function() {
+      ProgressBar.hide();
+      ProgressBar.StopKnightRiderLight();
+      location.href = "/logout.htm?lang="+getLang();
+    });
+  });
 };
 
 
