--- occu/WebUI/www/webui/js/lang/de/translate.lang.help.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.help.js
@@ -2,6 +2,8 @@
   "de": {
     "genericBtnTxtHelp" : "Hilfe",
     "genericBtnTxtHelpOff" : "Hilfe aus",
+	"dialogHelpCentralLinkTitle" : "Hinweise zum Zentralen-Link",
+	"dialogHelpCentralLinkContent" : "Hier muss jetzt noch Inhalt rein",	
     "helpPowerCost" : "Hier k%F6nnen Sie den Preis pro kWh Ihres Energieanbieters eintragen.",
     "diagramHelpCustomModeTitle" : "Beschreibung der Werte f%FCr gr%F6%DFere Zeitr%E4ume",
     "diagramHelpCustomModeContent" :
--- occu/WebUI/www/webui/js/lang/en/translate.lang.help.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.help.js
@@ -2,6 +2,8 @@
   "en": {
     "genericBtnTxtHelp" : "Help",
     "genericBtnTxtHelpOff" : "Disable help",
+	"dialogHelpCentralLinkTitle" : "Note on Central-Link",
+	"dialogHelpCentralLinkContent" : "Insert content here...",	
     "helpPowerCost" : "Here you can set the price per kWh of your electricity provider.",
     "diagramHelpCustomModeTitle" : "Description of the values for longer periods",
     "diagramHelpCustomModeContent" :
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -169,6 +169,20 @@
       <\/div>\n
     <\/div>\n
     {\/if}\n
+    {if channel.channelType == \"KEY_TRANSCEIVER\"}\n
+      <div id=\"channelCentralLink\" class=\"ChannelConfigDialogSection\">\n
+      <div class=\"CLASS11003\" name=\"generalDeviceChannelConfigLblCentralLink\">Zentralen-Link&nbsp;<img src=\"/ise/img/help.png\" style=\"cursor: pointer; width:18px; height:18px; position:relative; top:2px\" onclick=\"ChannelConfigDialog.showConfigCentralLinkHelp();\"><\/div>\n
+      <hr \/>\n
+       <div>\n
+        <table border=\"0\"  class=\"ChannelConfigDialogTable\" width=\"250px\">\n
+          <tr>\n
+            <td width=\"50%\"><div id=\"ChannelConfigDialogCreateCentralLinkButton\" class=\"StdButton\" name=\"channelConfigBtnCreateCentralLink\" onclick=\"ChannelConfigDialog.createCentralLink();\">Anlegen<\/div><\/td>\n
+            <td width=\"50%\"><div id=\"ChannelConfigDialogRemoveCentralLinkButton\" class=\"StdButton\" name=\"channelConfigBtnRemoveCentralLink\" onclick=\"ChannelConfigDialog.removeCentralLink();\">Entfernen<\/div><\/td>\n
+          <\/tr>\n
+        <\/table>\n
+      </div>\n
+      </div>\n
+    {\/if}\n
   <\/div>\n
 \n
   <div id=\"ChannelConfigDialogContentMain\">\n
@@ -13627,7 +13641,42 @@
       _this_.pollTest();
     });
   },
-  
+
+  showConfigCentralLinkHelp: function() {
+    var title = translateKey("dialogHelpCentralLinkTitle"),
+    content = translateKey("dialogHelpCentralLinkContent");
+    MessageBox.show(title, content,'', 475, 450);
+  },
+
+  createCentralLink: function()
+  {
+   ResetPostString();
+   AddParam($('global_sid'));
+   poststr += "&iface=HmIP-RF";
+   poststr += "&sender_address="+this.channel.address;
+   poststr += "&receiver_address=CENTRAL_DEVICE:63";
+   poststr += "&cmd=addLink";
+   SendRequest('ic_ifacecmd.cgi');
+   ProgressBar = new ProgressBarMsgBox(translateKey("progressBarCreateLinkTitle"), 1);
+   ProgressBar.show();
+  },
+  
+  removeCentralLink: function()
+  {
+	var addr=this.channel.address;
+    new YesNoDialog(translateKey('dialogSafetyCheck'), translateKey('dialogQuestionRemoveLink').replace('%s', addr+' - ZENTRALE'), function(result) {
+      if (result == YesNoDialog.RESULT_YES) {
+        ResetPostString();
+        AddParam(document.getElementById('global_sid'));
+        poststr += "&cmd=removeLink";
+        poststr += "&iface=HmIP-RF";
+        poststr += "&sender_address="+addr;
+        poststr += "&receiver_address=CENTRAL_DEVICE:63" ;
+        SendRequest('ic_ifacecmd.cgi');
+      }
+    }, "html");
+  },
+
   /**
    * Callback. Fragt zyklisch das Ergebnis des Funktionstests ab.
    **/
