--- occu/WebUI/www/webui/js/lang/de/translate.lang.help.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.help.js
@@ -2,6 +2,10 @@
   "de": {
     "genericBtnTxtHelp" : "Hilfe",
     "genericBtnTxtHelpOff" : "Hilfe aus",
+    "dialogHelpCentralLinkTitle" : "Tasterevent an Zentrale - Hinweise",
+    "dialogHelpCentralLinkContent" :
+      "Um die Funkauslastung bzw. den DutyCycle der CCU-Zentrale oder einen eventl. Batterieverbrauch m%F6glichst gering zu halten senden gewisse Homematic-Ger%E4te mit einem Taster-/Schalterkanal nicht standardm%E4%DFig bei jeder %C4nderung des Taster-/Schalterstatus den jeweils aktuellen Wert an Ihre CCU-Zentrale%2C sondern lediglich wenn dieser Kanal in %22Programme %26 Zentralenverkn%FCpfungen%22 explizit genutzt wird.<br/><br/>" +
+      "Ist das Anlegen eines solchen Programmes nicht gew%FCnscht%2C und man möchte lediglich die Taster-/Schaltevents auch trotz potentiell h%F6herem DutyCycle/Batterverbrauch an die CCU-Zentrale ausliefern lassen, kann mit Hilfe der %22Taster-/Schaltevent an Zentrale%22 Funktion das Weitersenden der Statusevents explizit f%FCr einen Kanal aktiviert bzw. deaktiviert werden. Hierbei findet dann im jeweiligen Ger%E4t das Anlegen bzw. L%F6schen einer speziellen internen Direktverkn%FCpfung statt%2C, sodass die CCU-Zentrale %FCber s%E4mtliche Bet%E4tigungen automatisch zu Lasten eines h%F6heren DutyCycle/Batterieverbrauches mit informiert wird.",
     "helpPowerCost" : "Hier k%F6nnen Sie den Preis pro kWh Ihres Energieanbieters eintragen.",
     "diagramHelpCustomModeTitle" : "Beschreibung der Werte f%FCr gr%F6%DFere Zeitr%E4ume",
     "diagramHelpCustomModeContent" :
--- occu/WebUI/www/webui/js/lang/de/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.js
@@ -436,6 +436,7 @@
     "dialogEditScript" : "Skript bearbeiten",
     "dialogEnterValue" : "Wert eingeben",
     "dialogEnterText" : "Text eingeben",
+    "dialogQuestionAddLink" : "M%F6chten Sie die Verkn%FCpfung '%s' wirklich anlegen?",
     "dialogRangeOfValues" : "Wertebereich einstellen",
     "dialogDeleteSysVar" : "Systemvariable l%F6schen",
     "dialogCreateFavChooseProg" : "Favoritenerstellung - Programmauswahl",
--- occu/WebUI/www/webui/js/lang/en/translate.lang.help.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.help.js
@@ -2,6 +2,10 @@
   "en": {
     "genericBtnTxtHelp" : "Help",
     "genericBtnTxtHelpOff" : "Disable help",
+    "dialogHelpCentralLinkTitle" : "Pushbutton event to CCU - Info",
+    "dialogHelpCentralLinkContent" :
+      "In order to keep the radio load / duty cycle of the CCU central and potentially also the battery consumption as low as possible, Homematic devices with a push-button channel do not send the current value to the CCU central every time the status changes, but only if this push-button channel is explicitly used in 'Programs & central links'.<br/><br/>" +
+      "If the creation of such a program is not desired and you just want to make sure to receive all status events, forwarding of these status events can be explicitly activated or deactivated for a button channel with the help of the 'Pushbutton event to CCU' function. In this case, the creation or deletion of a special internal direct link takes place in the respective device, so that the CCU central is automatically informed about all button actuations thought this might increase the duty cycle and battery consumption to some extend.",
     "helpPowerCost" : "Here you can set the price per kWh of your electricity provider.",
     "diagramHelpCustomModeTitle" : "Description of the values for longer periods",
     "diagramHelpCustomModeContent" :
--- occu/WebUI/www/webui/js/lang/en/translate.lang.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.js
@@ -435,6 +435,7 @@
     "dialogEditScript" : "Edit script",
     "dialogEnterValue" : "Enter value",
     "dialogEnterText" : "Enter text",
+    "dialogQuestionAddLink" : "Do you really want to add the connection '%s'?",
     "dialogRangeOfValues" : "Set value range",
     "dialogDeleteSysVar" : "Delete system variable",
     "dialogCreateFavChooseProg" : "Creating favourite - program selection",
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -195,6 +195,20 @@
       <\/div>\n
     <\/div>\n
     {\/if}\n
+    {if channel.channelType == \"KEY_TRANSCEIVER\" || channel.channelType == \"KEY\" || channel.channelType == \"MULTI_MODE_INPUT_TRANSMITTER\"}\n
+      <div id=\"channelCentralLink\" class=\"ChannelConfigDialogSection\">\n
+      <div class=\"CLASS11003\" name=\"generalDeviceChannelConfigLblCentralLink\">Taster-/Schaltevent an Zentrale&nbsp;<img src=\"/ise/img/help.png\" style=\"cursor: pointer; width:18px; height:18px; position:relative; top:2px\" onclick=\"ChannelConfigDialog.showConfigCentralLinkHelp();\"><\/div>\n
+      <hr \/>\n
+       <div>\n
+        <table border=\"0\" class=\"ChannelConfigDialogTable\" width=\"250px\">\n
+          <tr>\n
+            <td width=\"50%\"><div id=\"ChannelConfigDialogCreateCentralLinkButton\" class=\"StdButton\" name=\"channelConfigBtnCreateCentralLink\" onclick=\"ChannelConfigDialog.createCentralLink('${channel.channelType}');\">Aktivieren<\/div><\/td>\n
+            <td width=\"50%\"><div id=\"ChannelConfigDialogRemoveCentralLinkButton\" class=\"{if !channel.hasProgramIds()}StdButton{else}StdButtonInactive{\/if}\" name=\"channelConfigBtnRemoveCentralLink\" onclick=\"{if !channel.hasProgramIds()}ChannelConfigDialog.removeCentralLink('${channel.channelType}');{\/if}\">Deaktivieren<\/div><\/td>\n
+          <\/tr>\n
+        <\/table>\n
+      </div>\n
+      </div>\n
+    {\/if}\n
   <\/div>\n
 \n
   <div id=\"ChannelConfigDialogContentMain\">\n
@@ -14879,7 +14893,43 @@
       _this_.pollTest();
     });
   },
+
+  showConfigCentralLinkHelp: function() {
+    var title = translateKey("dialogHelpCentralLinkTitle"),
+    content = translateKey("dialogHelpCentralLinkContent");
+    MessageBox.show(title, content,'', 475, 450);
+  },
+
+  createCentralLink: function(channeltype)
+  {
+    var addr=this.channel.address;
+    new YesNoDialog(translateKey('dialogSafetyCheck'), translateKey('dialogQuestionAddLink').replace('%s', addr+' <> CCU'), function(result) {
+      if (result == YesNoDialog.RESULT_YES) {
+        if (channeltype === "KEY_TRANSCEIVER" || channeltype === "MULTI_MODE_INPUT_TRANSMITTER") {
+          homematic("Interface.reportValueUsage", {"interface": "HmIP-RF", "address": addr, "valueId": "PRESS_SHORT", "refCounter":"1"});
+        } else if (channeltype === "KEY") {
+          homematic("Interface.reportValueUsage", {"interface": "BidCos-RF", "address": addr, "valueId": "PRESS_SHORT", "refCounter":"1"});
+        }
+      }
+    }, "html");
+  },
   
+  removeCentralLink: function(channeltype)
+  {
+    var addr=this.channel.address;
+    new YesNoDialog(translateKey('dialogSafetyCheck'), translateKey('dialogQuestionRemoveLink').replace('%s', addr+' <> CCU'), function(result) {
+      if (result == YesNoDialog.RESULT_YES) {
+        if (channeltype === "KEY_TRANSCEIVER" || channeltype === "MULTI_MODE_INPUT_TRANSMITTER") {
+          homematic("Interface.reportValueUsage", {"interface": "HmIP-RF", "address": addr, "valueId": "PRESS_SHORT", "refCounter":"0"});
+          homematic("Interface.reportValueUsage", {"interface": "HmIP-RF", "address": addr, "valueId": "PRESS_LONG", "refCounter":"0"});
+        } else if (channeltype === "KEY") {
+          homematic("Interface.reportValueUsage", {"interface": "BidCos-RF", "address": addr, "valueId": "PRESS_SHORT", "refCounter":"0"});
+          homematic("Interface.reportValueUsage", {"interface": "BidCos-RF", "address": addr, "valueId": "PRESS_LONG", "refCounter":"0"});
+        }
+      }
+    }, "html");
+  },
+
   /**
    * Callback. Fragt zyklisch das Ergebnis des Funktionstests ab.
    **/
