--- occu/WebUI/www/webui/js/lang/de/translate.lang.help.js.orig
+++ occu/WebUI/www/webui/js/lang/de/translate.lang.help.js
@@ -2,6 +2,10 @@
   "de": {
     "genericBtnTxtHelp" : "Hilfe",
     "genericBtnTxtHelpOff" : "Hilfe aus",
+    "dialogHelpCentralLinkTitle" : "Tasterevent an Zentrale - Hinweise",
+    "dialogHelpCentralLinkContent" :
+      "Um die Funkauslastung bzw. den DutyCycle der CCU-Zentrale m%F6glichst gering zu halten senden Homematic-Ger%E4te mit einem Tasterkanal nicht standardm%E4%DFig bei jeder %C4nderung des Tasterstatus den jeweils aktuellen Wert an Ihre CCU-Zentrale%2C sondern lediglich wenn dieser Tasterkanal in %22Programme %26 Zentralenverkn%FCpfungen%22 explizit genutzt wird.<br/><br/>" +
+      "Ist das Anlegen eines solchen Programmes nicht gew%FCnscht%2C kann mit Hilfe der %22Tasterevent an Zentrale%22 Funktion das Weitersenden der Statusevents explizit f%FCr einen Tasterkanal aktiviert bzw. deaktiviert werden. Hierbei findet dann im jeweiligen Ger%E4t das Anlegen bzw. L%F6schen einer speziellen internen Direktverkn%FCpfung statt%2C sodass die CCU-Zentrale %FCber s%E4mtliche Tasterbet%E4tigungen automatisch mit informiert wird.",
     "helpPowerCost" : "Hier k%F6nnen Sie den Preis pro kWh Ihres Energieanbieters eintragen.",
     "diagramHelpCustomModeTitle" : "Beschreibung der Werte f%FCr gr%F6%DFere Zeitr%E4ume",
     "diagramHelpCustomModeContent" :
--- occu/WebUI/www/webui/js/lang/en/translate.lang.help.js.orig
+++ occu/WebUI/www/webui/js/lang/en/translate.lang.help.js
@@ -2,6 +2,10 @@
   "en": {
     "genericBtnTxtHelp" : "Help",
     "genericBtnTxtHelpOff" : "Disable help",
+    "dialogHelpCentralLinkTitle" : "Pushbutton event to CCU - Info",
+    "dialogHelpCentralLinkContent" :
+      "In order to keep the radio load / duty cycle of the CCU central as low as possible, Homematic devices with a push-button channel do not send the current value to your CCU central every time the push-button status is changed, but only if this push-button channel is explicitly used in 'Programs & central links'.<br/><br/>" +
+      "If the creation of such a program is not desired, the forwarding of the status events can be activated or deactivated explicitly for a button channel with the help of the 'Pushbutton event to CCU' function. In this case, the creation or deletion of a special internal direct link takes place in the respective device, so that the CCU central is automatically informed about all button actuations.",
     "helpPowerCost" : "Here you can set the price per kWh of your electricity provider.",
     "diagramHelpCustomModeTitle" : "Description of the values for longer periods",
     "diagramHelpCustomModeContent" :
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -169,6 +169,20 @@
       <\/div>\n
     <\/div>\n
     {\/if}\n
+    {if channel.channelType == \"KEY_TRANSCEIVER\" || channel.channelType == \"KEY\"}\n
+      <div id=\"channelCentralLink\" class=\"ChannelConfigDialogSection\">\n
+      <div class=\"CLASS11003\" name=\"generalDeviceChannelConfigLblCentralLink\">Tasterevent an Zentrale&nbsp;<img src=\"/ise/img/help.png\" style=\"cursor: pointer; width:18px; height:18px; position:relative; top:2px\" onclick=\"ChannelConfigDialog.showConfigCentralLinkHelp();\"><\/div>\n
+      <hr \/>\n
+       <div>\n
+        <table border=\"0\" class=\"ChannelConfigDialogTable\" width=\"250px\">\n
+          <tr>\n
+            <td width=\"50%\"><div id=\"ChannelConfigDialogCreateCentralLinkButton\" class=\"StdButton\" name=\"channelConfigBtnCreateCentralLink\" onclick=\"ChannelConfigDialog.createCentralLink('${channel.channelType}');\">Aktivieren<\/div><\/td>\n
+            <td width=\"50%\"><div id=\"ChannelConfigDialogRemoveCentralLinkButton\" class=\"{if !channel.hasProgramIds()}StdButton{else}StdButtonInactive{\/if}\" name=\"channelConfigBtnRemoveCentralLink\" onclick=\"{if !channel.hasProgramIds()}ChannelConfigDialog.removeCentralLink('${channel.channelType}');{\/if}\">Deaktivieren<\/div><\/td>\n
+          <\/tr>\n
+        <\/table>\n
+      </div>\n
+      </div>\n
+    {\/if}\n
   <\/div>\n
 \n
   <div id=\"ChannelConfigDialogContentMain\">\n
@@ -13627,7 +13641,43 @@
       _this_.pollTest();
     });
   },
+
+  showConfigCentralLinkHelp: function() {
+    var title = translateKey("dialogHelpCentralLinkTitle"),
+    content = translateKey("dialogHelpCentralLinkContent");
+    MessageBox.show(title, content,'', 475, 450);
+  },
+
+  createCentralLink: function(channeltype)
+  {
+    if (channeltype === "KEY_TRANSCEIVER") {
+      homematic("Interface.reportValueUsage", {"interface": "HmIP-RF", "address": this.channel.address, "valueId": "PRESS_SHORT", "refCounter":"1"});  
+    } else if (channeltype === "KEY") {
+      homematic("Interface.reportValueUsage", {"interface": "BidCos-RF", "address": this.channel.address, "valueId": "PRESS_SHORT", "refCounter":"1"});  
+    }
+  },
   
+  removeCentralLink: function(channeltype)
+  {
+    var addr=this.channel.address;
+    new YesNoDialog(translateKey('dialogSafetyCheck'), translateKey('dialogQuestionRemoveLink').replace('%s', addr+' - ZENTRALE'), function(result) {
+      if (result == YesNoDialog.RESULT_YES) {
+        if (channeltype === "KEY_TRANSCEIVER") {
+          ResetPostString();
+          AddParam(document.getElementById('global_sid'));
+          poststr += "&cmd=removeLink";
+          poststr += "&iface=HmIP-RF";
+          poststr += "&sender_address="+addr;
+          poststr += "&receiver_address=CENTRAL_DEVICE:63" ;
+          SendPOSTRequest('ic_ifacecmd.cgi');
+        } else if (channeltype === "KEY") {
+          homematic("Interface.reportValueUsage", {"interface": "BidCos-RF", "address": addr, "valueId": "PRESS_SHORT", "refCounter":"0"});  
+          homematic("Interface.reportValueUsage", {"interface": "BidCos-RF", "address": addr, "valueId": "PRESS_LONG", "refCounter":"0"});  
+        }
+      }
+    }, "html");
+  },
+
   /**
    * Callback. Fragt zyklisch das Ergebnis des Funktionstests ab.
    **/
