--- occu/WebUI/www/config/ic_common.tcl.orig
+++ occu/WebUI/www/config/ic_common.tcl
@@ -583,11 +583,10 @@
 
 #cgi_debug -on
 
-  global env
+  global env _cgi _cgi_uservar
   upvar $pPROFILE PROFILE_TMP
 
-  set params [split [cgi_unquote_input $env(QUERY_STRING)] &]
-
+  set params $_cgi(uservars)
 
   #Parametername eines evtl. Subsets auf den URL-Parametern generieren. z.B. subset_4_1
   set i 1
@@ -595,7 +594,10 @@
 
   foreach p $params {
 
-    if { [regexp {^(.*)=(.*)$} $p dummy name value] } then {
+    set name $p
+    set value $_cgi_uservar($name)
+
+    if { $name != "" } then {
 
       if { [lsearch -exact $IGNORE_PARAMS $name] >= 0} then {
 
@@ -648,7 +650,7 @@
 
   global HTMLTITLE env
 
-  set IGNORE_PARAMS {address cmd iface paramid peer pnr ps_id ps_type sid SUBSET_OPTION_VALUE NAME}
+  set IGNORE_PARAMS {address cmd iface paramid peer pnr ps_id ps_type sid SUBSET_OPTION_VALUE NAME _}
 
   if { $profile != "" } then {
     
@@ -1128,7 +1130,7 @@
     set idval "separate_${pnr}_$j"
 
     if { ! ($operations & 3) } then { continue }
-    if {    $operations & 1  } then { catch {set value $ps($param_id)} }
+    if {    $operations & 1  } then { catch {set value [cgi_quote_html $ps($param_id)]} }
     if {    $operations & 2  } then { set access "" } else { set access "disabled=\"disabled\"" }
 
     append s "<tr>"
--- occu/WebUI/www/config/ic_ifacecmd.cgi.orig
+++ occu/WebUI/www/config/ic_ifacecmd.cgi
@@ -639,6 +639,7 @@
   set ps_type ""
   set paramid ""
   set pnr     ""
+  set new_profilepath ""
   set HmIPIdentifier "HmIP-RF"
   set HmIPWiredIdentifier "HmIP-Wired"
 
--- occu/WebUI/www/config/ic_neweasymode.cgi.orig
+++ occu/WebUI/www/config/ic_neweasymode.cgi
@@ -13,7 +13,7 @@
 
   global env base_pnr
 
-  set IGNORE_PARAMS {EasyModeName base_pnr cmd ps_id ps_ids sensor actor}
+  set IGNORE_PARAMS {EasyModeName base_pnr cmd ps_id ps_ids sensor actor _}
 
   upvar $pBASEPROFILE BASEPROFILE
   upvar $pNEWPROFILE  PROFILE_TMP
--- occu/WebUI/www/webui/webui.js.orig
+++ occu/WebUI/www/webui/webui.js
@@ -31461,12 +31461,13 @@
     
     /* <!-- ELV */
     var opts;
+    var url = file + "?sid=" + SessionId;
     if (file.substring(0, UI_PATH.length) == UI_PATH)
     {
-      //method: 'get' ist für die ELV-CGIs notwendig.
       opts = {
         evalScripts: true,
-        method: 'get',
+        method: 'post',
+        parameters:argsForUrl,
         onComplete:function(){
           iseRefr(true);
           bUpdateContentRunning=false;
@@ -31489,12 +31490,10 @@
           }, 100);
         }
       };
+      if (argsForUrl){ url += argsForUrl; }
     }
     /* ELV --> */
     
-    var url = file + "?sid=" + SessionId;
-    if (argsForUrl){ url += argsForUrl; }
-    
     //console.info( "updateContent with "+pb );
     
     new Ajax.Updater("content", url, opts);
@@ -42949,6 +42948,27 @@
   new Ajax.Updater(id, scriptname, opt);
 };
 
+SendPOSTRequest = function(scriptname, html_container_id, callback)
+{
+  scriptname = Get_ReGa_Path(scriptname);
+  var params = poststr;
+
+  if ((html_container_id) && (html_container_id !== "")) { id = html_container_id; }
+  else                                                   { id = Get_UI_CONTENTBOX_ID(); }
+
+  var _callback_ = callback; 
+  
+  var opt =
+  {
+    method:'post',
+    parameters:params,
+    evalScripts:true,
+    onComplete: function() { if (_callback_) { _callback_(); } }
+  };
+
+  new Ajax.Updater(id, scriptname + '?sid=' + document.getElementById('global_sid').value, opt);
+};
+
 ShowLoadingBox = function(msg)
 {
   
@@ -43243,7 +43263,7 @@
   poststr += "&devname=" + devname;
   poststr += "&devtype=" + devtype;
 
-  SendRequest('ic_ifacecmd.cgi');
+  SendPOSTRequest('ic_ifacecmd.cgi');
 
   ProgressBar = new ProgressBarMsgBox("Verknüpfungen und Programme werden abgefragt...", 1);
   ProgressBar.show();
@@ -43494,7 +43514,7 @@
       if ((redirect_url) && (redirect_url !== "")) { poststr += "&redirect_url=" + redirect_url; }
       else                                         { poststr += "&redirect_url=IC_LINKPEERLIST"; }
 
-      SendRequest('ic_ifacecmd.cgi');
+      SendPOSTRequest('ic_ifacecmd.cgi');
     }
   }, "html");
 };
@@ -43563,7 +43583,7 @@
   poststr += "&receiver_address=" +receiver_address;
   poststr += "&cmd=ShowConfigPendingMsg";
 
-  SendRequest('ic_ifacecmd.cgi');
+  SendPOSTRequest('ic_ifacecmd.cgi');
 };
 
 
@@ -43756,7 +43776,7 @@
   poststr += "&param_id="          + param_id;
   poststr += "&html_inputelem_id=" + html_inputelem_id;
 
-  SendRequest('ic_ifacecmd.cgi');
+  SendPOSTRequest('ic_ifacecmd.cgi');
 
   //ProgressBar = new ProgressBarMsgBox("Parameter wird festgesetzt...", 1);
   ProgressBar = new ProgressBarMsgBox(translateKey("dialogDetermineParameterTitle"), 1);
@@ -45286,7 +45306,7 @@
   ProgressBar.show();
     ProgressBar.StartKnightRiderLight();
 
-  SendRequest('ic_ifacecmd.cgi');
+  SendPOSTRequest('ic_ifacecmd.cgi');
 };
 
 ShowNewLinkSummary = function(iface, sender_address, receiver_address, name, description, group_name, group_description)
@@ -45628,7 +45648,7 @@
   poststr += "&receiver_address=" + receiver_address;
   poststr += "&cmd=activateLinkParamset";
 
-  SendRequest('ic_ifacecmd.cgi');
+  SendPOSTRequest('ic_ifacecmd.cgi');
 };
 
 SetLinkInfo = function (iface, sender_address, receiver_address, name, description) {
@@ -45643,7 +45663,7 @@
   poststr += "&description=" + elv_toQueryString(description);
   poststr += "&cmd=setLinkInfo";
 
-  SendRequest('ic_ifacecmd.cgi');
+  SendPOSTRequest('ic_ifacecmd.cgi');
 };
 
 
@@ -45679,7 +45699,7 @@
     }
     poststr += "&new_profilepath=" + is_newProfile(paramid).split('-')[0] + fileExtension;
   }
-  SendRequest('ic_ifacecmd.cgi');
+  SendPOSTRequest('ic_ifacecmd.cgi');
 };
 
 CollectData_SaveProfileSettings = function (reload) {
@@ -46122,7 +46142,7 @@
     poststr += "&iface=" + $F('global_iface');
     poststr += "&address=" + $F('global_address');
     poststr += "&cmd=firmware_update";
-    SendRequest('ic_ifacecmd.cgi');
+    SendPOSTRequest('ic_ifacecmd.cgi');
     ProgressBar = new ProgressBarMsgBox(translateKey("performingFirmwareUpdate"), 1);
     ProgressBar.show();
     ProgressBar.StartKnightRiderLight();
@@ -46462,7 +46482,7 @@
     });
   }
 
-  SendRequest('ic_ifacecmd.cgi', null, function() {
+  SendPOSTRequest('ic_ifacecmd.cgi', null, function() {
     if (device)
     {
       DeviceList.beginUpdateDevice(device.id);
@@ -46481,7 +46501,7 @@
   poststr += "&receiver=" + receiver;
   poststr += "&longKeyPress=" + simLongKeyPress;
   poststr += "&cmd=SendInternalKeyPress";
-  SendRequest('ic_ifacecmd.cgi');
+  SendPOSTRequest('ic_ifacecmd.cgi');
 };
 
 SetInternalKey = function(iface, address, special_input_id, peerAddress)
@@ -46510,7 +46530,7 @@
   AddSeparateSettings('subset_'   + special_input_id + '_', pnr);
   
   //alert(poststr);
-  SendRequest('ic_ifacecmd.cgi');
+  SendPOSTRequest('ic_ifacecmd.cgi');
 };
 
 ShowInternalKeyProfile = function(selectelem, channel, counter)
