--- occu/WebUI/www/config/cp_maintenance.cgi.orig
+++ occu/WebUI/www/config/cp_maintenance.cgi
@@ -1428,15 +1428,16 @@
   puts "Content-Type:application/x-download"
   puts "Content-Disposition:attachment;filename=[set HOSTNAME]-$year-$month-$day.log\n"
   
-  cd /var/log
-  foreach f {messages.1 messages.0 messages hmserver.log.1 hmserver.log} {
+  foreach f [glob -directory /var/log *] {
     catch {
-      set fd [open $f r]
-      puts -nonewline "\r\n***** $f *****\r\n"
-      while { ! [eof $fd]} {
-        puts -nonewline "[gets $fd]\r\n"
+      if {[file extension $f] != ".gz" && [file type $f] == "file"} {
+        set fd [open $f r]
+        puts -nonewline "\r\n***** $f *****\r\n"
+        while { ! [eof $fd]} {
+          puts -nonewline "[gets $fd]\r\n"
+        }
+        close $fd
       }
-      close $fd
     }
   }
 }
