--- occu/WebUI/www/rega/esp/programs.fn.orig
+++ occu/WebUI/www/rega/esp/programs.fn
@@ -84,11 +84,11 @@
                     {
                       if( iRV1 == 0 )
                       {
-                        sRet = sRet#', ${ruleConditionLblWhen} '#oLV.ValueName0()#', ';
+                        sRet = sRet#' '#oLV.ValueName0()#' ';
                       }
                       if( iRV1 == 1 )
                       {
-                        sRet = sRet#', ${ruleConditionLblWhen} '#oLV.ValueName1()#', ';
+                        sRet = sRet#' '#oLV.ValueName1()#' ';
                       }
                     }
                   }
