--- occu/WebUI/www/rega/esp/controls/dimmer.fn.orig
+++ occu/WebUI/www/rega/esp/controls/dimmer.fn
@@ -65,8 +65,10 @@
 
         WriteLine("<td class='CLASS02529'>");
           WriteLine("<table width='100%' height='100%' cellspacing='10'>");
-            WriteLine("<tr><td><div id='"#chnId#"On'  class='ControlBtnOn CLASS02545 j_translate'>${actionStatusControlLblOn}</div></td></tr>");
-            WriteLine("<tr><td><div id='"#chnId#"Off' class='ControlBtnOff CLASS02545 j_translate'>${actionStatusControlLblOff}</div></td></tr>");
+            string BtnOnState="On"; string BtnOffState="Off";                                                                                                                             
+            if (iState == 0) { BtnOnState="Off"; BtnOffState="On"; }
+            WriteLine("<tr><td><div id='"#chnId#"On'  class='ControlBtn"#BtnOnState#" CLASS02545 j_translate'>${actionStatusControlLblOn}</div></td></tr>");
+            WriteLine("<tr><td><div id='"#chnId#"Off' class='ControlBtn"#BtnOffState#" CLASS02545 j_translate'>${actionStatusControlLblOff}</div></td></tr>");
           WriteLine("</table>");
         WriteLine("</td>");
 
@@ -215,8 +217,10 @@
 
         WriteLine("<td class='CLASS02529'>");
           WriteLine("<table width='100%' height='100%' cellspacing='10'>");
-            WriteLine("<tr><td><div id='"#chnId#"On'  class='ControlBtnOn CLASS02545 j_translate'>${actionStatusControlLblOn}</div></td></tr>");
-            WriteLine("<tr><td><div id='"#chnId#"Off' class='ControlBtnOff CLASS02545 j_translate'>${actionStatusControlLblOff}</div></td></tr>");
+            string BtnOnState="On"; string BtnOffState="Off";                                                                                                                             
+            if (iState == 0) { BtnOnState="Off"; BtnOffState="On"; }
+            WriteLine("<tr><td><div id='"#chnId#"On'  class='ControlBtn"#BtnOnState#" CLASS02545 j_translate'>${actionStatusControlLblOn}</div></td></tr>");
+            WriteLine("<tr><td><div id='"#chnId#"Off' class='ControlBtn"#BtnOffState#" CLASS02545 j_translate'>${actionStatusControlLblOff}</div></td></tr>");
           WriteLine("</table>");
         WriteLine("</td>");
 
@@ -297,8 +301,10 @@
 
         WriteLine("<td class='CLASS02529'>");
           WriteLine("<table width='100%' height='100%' cellspacing='10'>");
-            WriteLine("<tr><td><div id='"#chnId#"On'  class='ControlBtnOn CLASS02545 j_translate'>${actionStatusControlLblOn}</div></td></tr>");
-            WriteLine("<tr><td><div id='"#chnId#"Off' class='ControlBtnOff CLASS02545 j_translate'>${actionStatusControlLblOff}</div></td></tr>");
+            string BtnOnState="On"; string BtnOffState="Off";                                                                                                                             
+            if (iState == 0) { BtnOnState="Off"; BtnOffState="On"; }
+            WriteLine("<tr><td><div id='"#chnId#"On'  class='ControlBtn"#BtnOnState#" CLASS02545 j_translate'>${actionStatusControlLblOn}</div></td></tr>");
+            WriteLine("<tr><td><div id='"#chnId#"Off' class='ControlBtn"#BtnOffState#" CLASS02545 j_translate'>${actionStatusControlLblOff}</div></td></tr>");
           WriteLine("</table>");
         WriteLine("</td>");
 
