--- occu/WebUI/www/config/easymodes/etc/hmip_helper.tcl.orig
+++ occu/WebUI/www/config/easymodes/etc/hmip_helper.tcl
@@ -2685,6 +2685,8 @@
         hmip-fsm  {return true}
         hmip-fsm16  {return true}
         hmip-psm  {return true}
+        hmip-psm-2  {return true}
+        "hmip-psm-2 qhj"  {return true}
         hmip-usbsm  {return true}
       default {return false}
     }
--- occu/WebUI/www/config/easymodes/etc/uiElements.tcl.orig
+++ occu/WebUI/www/config/easymodes/etc/uiElements.tcl
@@ -33,6 +33,8 @@
     switch [string tolower $devType] {
         hmip-psm -
         hmip-fsm16 {return 3680.0}
+        hmip-psm-2 -
+        "hmip-psm-2 qhj" {return 3000.0}
         hmip-bsm -
         hmip-fsm  {return 1150.0}
         hmip-usbsm  {return 60.0}
@@ -44,6 +46,8 @@
     switch [string tolower $devType] {
         hmip-psm -
         hmip-fsm16 {return "3680.00"}
+        hmip-psm-2 -
+        "hmip-psm-2 qhj" {return "3000.00"}
         hmip-bsm -
         hmip-fsm  {return "1150.00"}
         hmip-usbsm  {return "60.00"}
@@ -182,6 +186,8 @@
 proc getUserDefinedCondTXThresholdUnitMinMaxDescr {devType param} {
    switch [string tolower $devType] {
       hmip-psm -
+      hmip-psm-2 -
+      "hmip-psm-2 qhj" -
       hmip-fsm16 -
       hmip-bsm -
       hmip-fsm -
