--- occu/WebUI/www/config/cp_software.cgi.orig
+++ occu/WebUI/www/config/cp_software.cgi
@@ -452,14 +452,14 @@
   if { $loop > -1 } {
     translatePage $loop
   } else {
-    puts "<script type=\"text/javascript\">translatePage('#messagebox')</script>"
+    puts "<script type=\"text/javascript\">translatePage('#messagebox');dlgPopup.readaptSize();</script>"
   }
 }
 
 proc translatePage {loop} {
 
 
-  puts "<script type=\"text/javascript\">translatePage('#messagebox')</script>"
+  puts "<script type=\"text/javascript\">translatePage('#messagebox');dlgPopup.readaptSize();</script>"
   global swVersion swUpdate
   cgi_javascript {
     puts "function getVersion(url, callback) {"
--- occu/WebUI/www/config/cp_time.cgi.orig
+++ occu/WebUI/www/config/cp_time.cgi
@@ -636,6 +636,7 @@
       });
 
       translatePage('#messagebox');
+      dlgPopup.readaptSize();
     }
   }
   puts "</div><script type=\"text/javascript\">new Draggable(\"dragTime\");</script>" 
