--- occu/WebUI/www/config/easymodes/etc/uiElements.tcl.orig
+++ occu/WebUI/www/config/easymodes/etc/uiElements.tcl
@@ -32,9 +32,9 @@
   if {[string equal $extraparam "TX_THRESHOLD_POWER"] == 1} {
     switch [string tolower $devType] {
       hmip-psm -
-      hmip-psm-2 -
-      "hmip-psm-2 qhj" -
       hmip-fsm16 {return 3680.00}
+      hmip-psm-2 -
+      "hmip-psm-2 qhj" {return 3000.00}
       hmip-bsm -
       hmip-fsm  {return 1150.00}
       hmip-usbsm  {return 60.00}
@@ -45,9 +45,9 @@
   if {([string equal $extraparam "COND_TX_THRESHOLD_LO"] == 1) || ([string equal $extraparam "COND_TX_THRESHOLD_HI"] == 1)} {
     switch [string tolower $devType] {
       hmip-psm -
-      hmip-psm-2 -
-      "hmip-psm-2 qhj" -
       hmip-fsm16 {return "3680.00"}
+      hmip-psm-2 -
+      "hmip-psm-2 qhj" {return "3000.00"}
       hmip-bsm -
       hmip-fsm  {return "1150.00"}
       hmip-usbsm  {return "60.00"}
