--- occu/WebUI/www/rega/esp/datapointconfigurator.fn.orig
+++ occu/WebUI/www/rega/esp/datapointconfigurator.fn
@@ -591,7 +591,7 @@
                 if (ch.MetaData("channelMode") == 3) {
                   Call("/esp/controls/door_window_contact.fn::CreateDoorWindowContact()");
                 } else {
-                  Call("/esp/datapointconfigurator.fn::CreateNoFunction()");
+                  Call("/esp/controls/key_switch_sc.fn::CreateContent()");
                 }
               }
             }
