--- occu/WebUI/www/config/easymodes/hmip/SWITCH_VIRTUAL_RECEIVER.tcl.orig
+++ occu/WebUI/www/config/easymodes/hmip/SWITCH_VIRTUAL_RECEIVER.tcl
@@ -66,7 +66,7 @@
   set wgtFirstVirtCh false
   set wgtChnMode "--"
 
-  if {$chn == 4} {set wgtFirstVirtCh true}
+  if {($isWGT != -1) && ($chn == 4)} {set wgtFirstVirtCh true}
 
   # HmIP-WGT(-A) Get the selected mode (modeBWTH or modeSWITCH) of the first virtual switch actor (wtc chn. 4)
   # The next virtual switch actor channels are using the global variable wgtChnMode to determine their mode.
