--- a/drivers/nvmem/rockchip-efuse.c.orig	2018-04-24 11:38:54.605779600 +0200
+++ b/drivers/nvmem/rockchip-efuse.c	2018-04-24 11:15:57.953005980 +0200
@@ -504,7 +504,7 @@
 		return PTR_ERR(nvmem);
 
 	platform_set_drvdata(pdev, nvmem);
-	ret = rockchip_efuse_read(efuse, 0, efuse_buf, 32);
+	ret = rockchip_rk3288_efuse_read(efuse, 0, efuse_buf, 32);
 	if (ret < 0) {
 		printk(" failed to rockchip_efuse_read\n");
 	}
