--- a/arch/arm/boot/dts/rk3288.dtsi.orig	2022-04-08 00:04:12.060918630 +0200
+++ b/arch/arm/boot/dts/rk3288.dtsi	2022-04-08 00:06:00.578306333 +0200
@@ -456,13 +456,6 @@
 	};
 
 	thermal-zones {
-		reserve_thermal: reserve-thermal {
-			polling-delay-passive = <1000>; /* milliseconds */
-			polling-delay = <5000>; /* milliseconds */
-
-			thermal-sensors = <&tsadc 0>;
-		};
-
 		cpu_thermal: cpu-thermal {
 			polling-delay-passive = <100>; /* milliseconds */
 			polling-delay = <5000>; /* milliseconds */
@@ -534,6 +527,13 @@
 				};
 			};
 		};
+
+		reserve_thermal: reserve-thermal {
+			polling-delay-passive = <1000>; /* milliseconds */
+			polling-delay = <5000>; /* milliseconds */
+
+			thermal-sensors = <&tsadc 0>;
+		};
 	};
 
 	tsadc: tsadc@ff280000 {
