--- homematic/WebUI/www/config/cp_security.cgi.orig	2016-12-14 20:34:28.446089427 +0100
+++ homematic/WebUI/www/config/cp_security.cgi	2017-03-22 16:10:46.686956003 +0100
@@ -192,7 +192,7 @@
         }
     }
     catch {
-        exec run-parts /etc/config/rc.d stop
+        exec run-parts -a stop /etc/config/rc.d
     }
     if { [catch {
         exec crypttool -r
@@ -202,8 +202,8 @@
         # exec /usr/sbin/ubiattach -p /dev/mtd6
         # exec /usr/sbin/ubimkvol /dev/ubi1 -N user -m
         # exec mount /usr/local
-        exec touch /var/doFactoryReset
-        exec kill -SIGQUIT 1
+        exec touch /usr/local/.doFactoryReset
+        exec /sbin/reboot
     }]} {
 
       # TWIST-22
@@ -278,13 +278,13 @@
 
 proc action_backup_restore_check {} {
     global env
-    cd /tmp/
+    cd /usr/local/tmp/
     
     http_head
     set i 0
     if { [catch {
-        exec tar xf new_config.tar
-        file delete -force /tmp/new_config.tar
+        exec tar xf /usr/local/tmp/new_config.tar 2>/dev/null
+        file delete -force /usr/local/tmp/new_config.tar
     
     set config_version [read_version "firmware_version"]
     set ccu1_backup false
@@ -414,7 +414,7 @@
 
 proc action_backup_restore_go {} {
     global env
-    cd /tmp/
+    cd /usr/local/tmp/
     http_head
     
     set system_version [read_version "/boot/VERSION"]
@@ -476,14 +476,15 @@
     }
     
   if { "false" == $ccu1_backup } {  # backup for version >= 2
-    exec umount /usr/local
-          exec /usr/sbin/ubidetach -p /dev/mtd6
-          exec /usr/sbin/ubiformat /dev/mtd6 -y
-          exec /usr/sbin/ubiattach -p /dev/mtd6
-          exec /usr/sbin/ubimkvol /dev/ubi1 -N user -m
-          exec mount /usr/local
+    #exec umount /usr/local
+          #exec /usr/sbin/ubidetach -p /dev/mtd6
+          #exec /usr/sbin/ubiformat /dev/mtd6 -y
+          #exec /usr/sbin/ubiattach -p /dev/mtd6
+          #exec /usr/sbin/ubimkvol /dev/ubi1 -N user -m
+          #exec mount /usr/local
+          exec rm -rf /usr/local/*
 
-    if { [catch {exec tar xzf /tmp/usr_local.tar.gz} errorMessage] } {
+    if { [catch {exec tar xzf /usr/local/tmp/usr_local.tar.gz} errorMessage] } {
       # set msg "Beim Einspielen des Systembackups ist ein Fehler aufgetreten. Bitte versuchen Sie es erneut. "
       # append msg "Falls dieser Fehler wiederholt Auftritt, wenden Sie sich bitte mit der folgenden Fehlermeldung an den Kundenservice:\n<br>"
       # append msg $errorMessage
@@ -494,22 +495,22 @@
     }    
   } else {  # backup for version < 2      
     #delete existing files
-    file delete -force /tmp/backup
+    file delete -force /usr/local/tmp/backup
     file delete -force /etc/config/hs485d /etc/config/hs485types /etc/config/rfd /etc/config/userprofiles 
     file delete -force /etc/config/homematic.regadom /etc/config/homematic.regadom.bak
     file delete -force /etc/config/ids /etc/config/ntpclient /etc/config/rega.conf /etc/config/syslog 
     file delete -force /etc/config/tweaks /etc/config/TZ /etc/config/server.pem /etc/config/keys
     file delete -force /etc/config/time.conf
     
-    file mkdir /tmp/backup
-    cd /tmp/backup
-    if { [catch {exec tar xzf /tmp/usr_local.tar.gz} errorMessage] } {
+    file mkdir /usr/local/tmp/backup
+    cd /usr/local/tmp/backup
+    if { [catch {exec tar xzf /usr/local/tmp/usr_local.tar.gz} errorMessage] } {
       put_message "\${dialogSettingsSecurityMessageSysBackupErrorTitle}" "\${dialogSettingsSecurityMessageSysBackupErrorContent} $errorMessage"
       set backuperror true
     } else {
       #copy old files
-      if { [file exists /tmp/backup/usr/local/etc/config] } {
-        cd /tmp/backup/usr/local/etc/config
+      if { [file exists /usr/local/tmp/backup/usr/local/etc/config] } {
+        cd /usr/local/tmp/backup/usr/local/etc/config
       }
       
       if { [file exists hs485d] } { file copy hs485d /etc/config/hs485d }        
@@ -609,7 +610,7 @@
       puts "translatePage('#messagebox');"
     }
 
-    file delete -force /tmp/new_config.tar /tmp/firmware_version /tmp/signature /tmp/usr_local.tar.gz /tmp/backup
+    file delete -force /usr/local/tmp/new_config.tar /usr/local/tmp/firmware_version /usr/local/tmp/signature /usr/local/tmp/usr_local.tar.gz /usr/local/tmp/backup
 }
 
 proc put_message {title msg args} {
@@ -1051,39 +1052,36 @@
 
 proc action_create_backup {} {
     set HOSTNAME [exec hostname]
+    set system_version [read_version "/boot/VERSION"]
     set iso8601_date [exec date -Iseconds]
+    set tmpdir [exec mktemp -d -p /usr/local/tmp]
     regexp {^(\d+)-(\d+)-(\d+)T(\d+):(\d+):(\d+)([+-]\d+)$} $iso8601_date dummy year month day hour minute second zone
+    set backupfile [set HOSTNAME]-$system_version-$year-$month-$day-$hour$minute.sbk
     #save DOM
     rega system.Save()
     cd /
-    exec tar czf /tmp/usr_local.tar.gz usr/local
-    cd /tmp/
+    exec tar --exclude=usr/local/tmp --exclude=usr/local/lost+found -czf $tmpdir/usr_local.tar.gz usr/local
+    cd $tmpdir/
     #sign the configuration with the current key
     exec crypttool -s -t 1 <usr_local.tar.gz >signature
     #store the current key index
     exec crypttool -g -t 1 >key_index
     file copy -force /boot/VERSION firmware_version
-    set fd [open "|tar c usr_local.tar.gz signature firmware_version key_index"]
-    catch {fconfigure $fd -translation binary}
-    catch {fconfigure $fd -encoding binary}
-    puts "Content-Type:application/x-download"
-    puts "Content-Disposition:attachment;filename=[set HOSTNAME]-$year-$month-$day.sbk\n"
-    catch {fconfigure stdout -translation binary}
-    catch {fconfigure stdout -encoding binary}
-    while { ! [eof $fd]} {
-        puts -nonewline [read $fd 65536]
-    }
-    close $fd
-    file delete -force /tmp/usr_local.tar.gz /tmp/firmware_version /tmp/signature
+    exec tar -cf /usr/local/tmp/last_backup.sbk usr_local.tar.gz signature firmware_version key_index
+    cd /
+    exec rm -rf $tmpdir
+    puts "X-Sendfile: /usr/local/tmp/last_backup.sbk"
+    puts "Content-Type: application/octet-stream"
+    puts "Content-Disposition: attachment; filename=\"$backupfile\"\n"
 }
 
 proc action_backup_upload {} {
     global env sid
-    cd /tmp/
+    cd /usr/local/tmp/
     
     http_head
     import_file -client backup_file
-    file rename -force -- [lindex $backup_file 0] "/tmp/new_config.tar"
+    file rename -force -- [lindex $backup_file 0] "/usr/local/tmp/new_config.tar"
     cgi_javascript {
         puts "var url = \"$env(SCRIPT_NAME)?sid=$sid\";"
         puts {
