--- homematic/arm-gnueabihf/packages-eQ-3/RFD/etc/config_templates/rfd.conf.orig	2016-12-05 11:00:17.737254728 +0100
+++ homematic/arm-gnueabihf/packages-eQ-3/RFD/etc/config_templates/rfd.conf	2016-12-05 11:00:53.812376089 +0100
@@ -22,6 +22,6 @@
 [Interface 0]
 Type = CCU2
 ComPortFile = /dev/mmd_bidcos
-#AccessFile = /dev/null
-#ResetFile = /dev/ccu2-ic200
+AccessFile = /dev/null
+ResetFile = /dev/null
 
