--- homematic/arm-gnueabihf/packages-eQ-3/HS485D/etc/init.d/S49hs485d.orig	2016-12-13 09:57:04.258978176 +0100
+++ homematic/arm-gnueabihf/packages-eQ-3/HS485D/etc/init.d/S49hs485d	2016-12-13 10:07:01.972664917 +0100
@@ -25,6 +25,19 @@
 	else
 		cmp -s $CFG_TEMPLATE_DIR/InterfacesList.xml /etc/config/InterfacesList.xml || cp $CFG_TEMPLATE_DIR/InterfacesList.xml /etc/config
 	fi
+
+  # if there is no HmIP support enabled we have to strip
+  # the HmIP-RF part in InterfacesList.xml
+  source /var/ids 2>/dev/null
+  if [[ ${HM_MODE} != "HmIP" ]]; then
+    line_start=$(cat /etc/config/InterfacesList.xml | grep -n '>HmIP-RF<' | cut -d: -f1 | head -1)
+    line_end=$(cat /etc/config/InterfacesList.xml | grep -n '>HmIP-RF<' | cut -d: -f1 | tail -1)
+    if [[ -n ${line_start} && -n ${line_end} ]]; then
+      line_start=$((${line_start}-1))
+      line_end=$((${line_end}+1))
+      sed -i "${line_start},${line_end}d" /etc/config/InterfacesList.xml
+    fi
+  fi
 }
 
 start() {
