--- homematic/arm-gnueabihf/packages-eQ-3/RFD/etc/init.d/S61rfd.orig	2016-12-12 22:04:53.864706566 +0100
+++ homematic/arm-gnueabihf/packages-eQ-3/RFD/etc/init.d/S61rfd	2016-12-12 22:15:43.947535829 +0100
@@ -16,14 +16,20 @@
 	#Migration for existing rfd.conf in user space
 	if [ ! -e /etc/config/rfd.conf ] ; then
 		cp $CFG_TEMPLATE_DIR/rfd.conf /etc/config
-	else
-		sed -i 's/\/dev\/ttyAPP0/\/dev\/mmd_bidcos/' /etc/config/rfd.conf || true
-		sed -i 's/^AccessFile/#AccessFile/' /etc/config/rfd.conf || true
-		sed -i 's/^ResetFile/#ResetFile/' /etc/config/rfd.conf || true
-                if ! grep -q "Improved Coprocessor Initialization" /etc/config/rfd.conf ; then
-                        sed -i 's/\[Interface 0\]/Improved\ Coprocessor\ Initialization\ =\ true\n\n&/' /etc/config/rfd.conf || true
-                fi
-	fi
+  fi
+
+  source /var/ids 2>/dev/null
+  if [[ ${HM_MODE} == "HmIP" ]]; then
+		sed -i 's/^ComPortFile = \/dev\/.*$/ComPortFile = \/dev\/mmd_bidcos/' /etc/config/rfd.conf
+  else
+		sed -i 's/^ComPortFile = \/dev\/.*$/ComPortFile = \/dev\/ttyAMA0/' /etc/config/rfd.conf
+  fi
+
+	sed -i 's/^#AccessFile = \/dev\/.*$/AccessFile = \/dev\/null/' /etc/config/rfd.conf
+	sed -i 's/^#ResetFile = \/dev\/.*$/ResetFile = \/dev\/null/' /etc/config/rfd.conf
+  if ! grep -q "Improved Coprocessor Initialization" /etc/config/rfd.conf ; then
+    sed -i 's/\[Interface 0\]/Improved\ Coprocessor\ Initialization\ =\ true\n\n&/' /etc/config/rfd.conf
+  fi
 
         # Bug fix
         grep "/etc/config/rfd/keys" /etc/config/rfd.conf
@@ -43,14 +49,14 @@
 
 waitStartupComplete() {
     echo -n "Waiting for rfd to get ready"
-	STEPS=60
+	STEPS=20
     for i in $(seq 1 $STEPS)
     do
 		sleep 2
 		echo -n "."
-        RFDSTATUSPID=`cat /var/status/rfd.status 2>/dev/null`
+        RFDSTATUSPID=`cat /var/status/rfd.status 2>&1`
         RFDPID=`pidof rfd`
-        if [ $RFDSTATUSPID = $RFDPID ] 
+        if [ "$RFDSTATUSPID" = "$RFDPID" ] 
 		then
             echo "rfd is ready now."
             break
