--- homematic/arm-gnueabihf/packages-eQ-3/RFD/etc/init.d/S61rfd.orig	2016-12-01 16:45:56.689404401 +0100
+++ homematic/arm-gnueabihf/packages-eQ-3/RFD/etc/init.d/S61rfd	2016-12-01 16:47:33.859368573 +0100
@@ -7,22 +7,32 @@
 CFG_TEMPLATE_DIR=/etc/config_templates
 PIDFILE=/var/run/rfd.pid
 
+update() {        
+	echo "0" > /sys/class/gpio/gpio18/value
+	sleep 1
+	echo "1" > /sys/class/gpio/gpio18/value
+	if [ ! -e /etc/config/no-coprocessor-update ] ; then
+		echo "checking if firmware update is needed..."
+		/bin/eq3configcmd update-coprocessor -p "/dev/bcm2835-raw-uart" -u
+		if ! [ $? -eq 0 ] ; then
+			echo "error while updating coprocessor, force..."
+			/bin/eq3configcmd update-coprocessor -p "/dev/bcm2835-raw-uart" -u -f
+		else
+			echo "done"
+		fi
+	else
+		echo "firmware update disabled"
+	fi
+}
+
 init() {
 	export TZ=`cat /etc/config/TZ`
 
 	if [ ! -d /etc/config/rfd ] ; then
   		mkdir /etc/config/rfd
 	fi
-	#Migration for existing rfd.conf in user space
 	if [ ! -e /etc/config/rfd.conf ] ; then
 		cp $CFG_TEMPLATE_DIR/rfd.conf /etc/config
-	else
-		sed -i 's/\/dev\/ttyAPP0/\/dev\/mmd_bidcos/' /etc/config/rfd.conf || true
-		sed -i 's/^AccessFile/#AccessFile/' /etc/config/rfd.conf || true
-		sed -i 's/^ResetFile/#ResetFile/' /etc/config/rfd.conf || true
-                if ! grep -q "Improved Coprocessor Initialization" /etc/config/rfd.conf ; then
-                        sed -i 's/\[Interface 0\]/Improved\ Coprocessor\ Initialization\ =\ true\n\n&/' /etc/config/rfd.conf || true
-                fi
 	fi
 
         # Bug fix
@@ -43,14 +53,14 @@
 
 waitStartupComplete() {
     echo -n "Waiting for rfd to get ready"
-	STEPS=60
+	STEPS=20
     for i in $(seq 1 $STEPS)
     do
 		sleep 2
 		echo -n "."
-        RFDSTATUSPID=`cat /var/status/rfd.status 2>/dev/null`
+        RFDSTATUSPID=`cat /var/status/rfd.status 2>&1`
         RFDPID=`pidof rfd`
-        if [ $RFDSTATUSPID = $RFDPID ] 
+        if [ "$RFDSTATUSPID" = "$RFDPID" ] 
 		then
             echo "rfd is ready now."
             break
@@ -65,6 +75,7 @@
 start() {
 	echo "Starting rfd: "
 	init
+	update
 	start-stop-daemon -S -q -b -m -p $PIDFILE --exec /bin/rfd -- -f /etc/config/rfd.conf -l $LOGLEVEL_RFD
 	waitStartupComplete
 }
