--- homematic/WebUI/www/webui/js/lang/tr/translate.lang.js.orig	2016-11-25 11:20:29.675646850 +0100
+++ homematic/WebUI/www/webui/js/lang/tr/translate.lang.js	2016-11-25 11:28:51.000012075 +0100
@@ -1,8 +1,8 @@
 jQuery.extend(true,HMIdentifier , {
   "tr" : {
-    "CCUFullNameHeader" : "HomeMatic CCU2",
-    "CCUFullNameText" : "Homematic CCU",
-    "CCUShortName" : "CCU",
+    "CCUFullNameHeader" : "RaspberryMatic",
+    "CCUFullNameText" : "RaspberryMatic",
+    "CCUShortName" : "RaspMatic",
     "BidCosRF" : "BidCos-RF",
     "BidCosWired" : "BidCos-Wired",
     "VirtualDevices" : "VirtualDevices",
--- homematic/WebUI/www/webui/js/lang/de/translate.lang.js.orig	2016-11-25 11:30:28.821752867 +0100
+++ homematic/WebUI/www/webui/js/lang/de/translate.lang.js	2016-11-25 11:30:58.557066805 +0100
@@ -1,8 +1,8 @@
 jQuery.extend(true,HMIdentifier , {
   "de" : {
-    "CCUFullNameHeader" : "HomeMatic CCU2",
-    "CCUFullNameText" : "Homematic CCU2",
-    "CCUShortName" : "CCU2",
+    "CCUFullNameHeader" : "RaspberryMatic",
+    "CCUFullNameText" : "RaspberryMatic",
+    "CCUShortName" : "RaspMatic",
     "BidCosRF" : "BidCos-RF",
     "BidCosWired" : "BidCos-Wired",
     "VirtualDevices" : "VirtualDevices",
--- homematic/WebUI/www/webui/js/lang/en/translate.lang.js.orig	2016-11-25 11:31:56.819723455 +0100
+++ homematic/WebUI/www/webui/js/lang/en/translate.lang.js	2016-11-25 11:32:56.974337722 +0100
@@ -1,8 +1,8 @@
 jQuery.extend(true,HMIdentifier , {
   "en" : {
-    "CCUFullNameHeader" : "HomeMatic CCU2",
-    "CCUFullNameText" : "Homematic CCU",
-    "CCUShortName" : "CCU",
+    "CCUFullNameHeader" : "RaspberryMatic",
+    "CCUFullNameText" : "RaspberryMatic",
+    "CCUShortName" : "RaspMatic",
     "BidCosRF" : "BidCos-RF",
     "BidCosWired" : "BidCos-Wired",
     "VirtualDevices" : "VirtualDevices",
