--- homematic/WebUI/www/webui/webui.js.orig	2017-03-22 10:29:02.525052893 +0100
+++ homematic/WebUI/www/webui/webui.js	2017-03-22 10:25:40.462016779 +0100
@@ -11285,8 +11285,9 @@
     var isOk = true;
     for (var i = 0, len = ips.length; i < len; i++) {
       var ip = ips[i];
-      if (!ip.match(/^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+(\/[0-9]+)?$/)) {
+      if (!ip.match(/^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+(\/[0-9]+)?$/) && !ip.match(/^[a-f0-9]+:/)) {
         isOk = false;
+        break;
       }
     }
 
@@ -18124,7 +18125,7 @@
     var script = document.createElement("script");
     script.id = "homematic_com_script";
     script.type = "text/javascript";
-    script.src = "http://update.homematic.com/firmware/download?cmd=js_check_version&version="+WEBUI_VERSION+"&product=HM-CCU2&serial=" + serial;
+    script.src = "https://gitcdn.xyz/repo/jens-maus/RaspberryMatic/master/release/LATEST-VERSION.js"
     $("body").appendChild(script);
   },
 
@@ -18220,7 +18221,7 @@
    **/
   setLatestVersion: function(latestVersion, product)
   {
-    if (product == "HM-CCU2") {
+    if (product == "HM-RASPBERRYMATIC") {
       homematic.com.m_latestVersion = latestVersion;
       homematic.com.m_isUpdateAvailable = (WEBUI_VERSION != latestVersion);
     } else {
