--- homematic/WebUI/www/config/cp_maintenance.cgi.orig	2017-03-21 13:52:40.253500448 +0100
+++ homematic/WebUI/www/config/cp_maintenance.cgi	2017-03-21 13:53:12.704702906 +0100
@@ -377,7 +377,7 @@
                                         table_row {
                                             table_data {
                                                 division {class="CLASS20908" style="display: none"} {id="btnFwDownload"} {} "onClick=\"window.location.href='$REMOTE_FIRMWARE_SCRIPT?cmd=download&version=$cur_version&serial=$serial&lang=de&product=HM-CCU2';\"" {}
-                                                division {class="CLASS20908"}  "onClick=\"showCCULicense();\"" {puts "\${dialogSettingsCMBtnPerformSoftwareUpdateDownload}"}
+                                                division {class="CLASS20908"}  "onClick=\"window.open('https://github.com/jens-maus/RaspberryMatic/releases','_blank');\"" {puts "\${dialogSettingsCMBtnPerformSoftwareUpdateDownload}"}
                                             }
                                         }
                                     }
@@ -387,66 +387,68 @@
                         table_row {
                             td {width="20"} {}
                             table_data {align="left"} {colspan="2"} {
-                                puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep2}"
-                            }
-                        }
-                        table_row {
-                            td {width="20"} {}
-                            table_data {colspan="2"} {
-                                form "$env(SCRIPT_NAME)?sid=$sid" name=firmware_form {target=firmware_upload_iframe} enctype=multipart/form-data method=post {
-                                    export action=firmware_upload
-                                    export downloadOnly=$downloadOnly
-                                    file_button firmware_file size=30 maxlength=1000000
-                                }
-                                puts {<iframe name="firmware_upload_iframe" style="display: none;"></iframe>}
-                            }
-                        }
-                        table_row {
-                            td {width="20"} {}
-                            table_data {align="left"} {
-                                puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep3}"
-                            }
-                            table_data {
-                                division {class="popupControls CLASS20905"} {
-                                    table {
-                                        table_row {
-                                            table_data {
-                                                division {class="CLASS20919"} {onClick="document.firmware_form.submit();showUserHint();"} {
-                                                  puts "\${dialogSettingsCMBtnPerformSoftwareUpdateUpload}"
-                                                }
-                                            }
-                                        }
-                                    }
-                                }
-                            }
-                        }
-                        table_row {
-                            td {width="20"} {}
-                            table_data {align="left"} {colspan="2"} {class="CLASS20920"} {
-                                puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep4}"
+                                #puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep2}"
+                                puts "Step 2: Read <a href='https://github.com/jens-maus/RaspberryMatic' target='_blank'>documentation</a> and <a href='https://homematic-forum.de/forum/viewtopic.php?f=65&t=34497' target='_blank'>update instructions</a>"
                             }
                         }
+                        #table_row {
+                        #    td {width="20"} {}
+                        #    table_data {colspan="2"} {
+                        #        form "$env(SCRIPT_NAME)?sid=$sid" name=firmware_form {target=firmware_upload_iframe} enctype=multipart/form-data method=post {
+                        #            export action=firmware_upload
+                        #            export downloadOnly=$downloadOnly
+                        #            file_button firmware_file size=30 maxlength=1000000
+                        #        }
+                        #        puts {<iframe name="firmware_upload_iframe" style="display: none;"></iframe>}
+                        #    }
+                        #}
+                        #table_row {
+                        #    td {width="20"} {}
+                        #    table_data {align="left"} {
+                        #        puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep3}"
+                        #    }
+                        #    table_data {
+                        #        division {class="popupControls CLASS20905"} {
+                        #            table {
+                        #                table_row {
+                        #                    table_data {
+                        #                        division {class="CLASS20919"} {onClick="document.firmware_form.submit();showUserHint();"} {
+                        #                          puts "\${dialogSettingsCMBtnPerformSoftwareUpdateUpload}"
+                        #                        }
+                        #                    }
+                        #                }
+                        #            }
+                        #        }
+                        #    }
+                        #}
+                        #table_row {
+                        #    td {width="20"} {}
+                        #    table_data {align="left"} {colspan="2"} {class="CLASS20920"} {
+                        #        puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep4}"
+                        #    }
+                        #}
                     }
                 }
                 table_data {align="left"} {class="CLASS20921"} {
-                    puts "\${dialogSettingsCMHintSoftwareUpdate1}"
-                    number_list {class="j_noForcedUpdate"} {
-                        li {
-                          ${dialogSettingsCMHintSoftwareUpdate2}                        }
-                        li {
-                             ${dialogSettingsCMHintSoftwareUpdate3}
-                        }
-                        set bat_level [get_bat_level]
-                        if {$bat_level < 50} {
-                            set msg " \${dialogSettingsCMHintSoftwareUpdate4a} $bat_level%. "
-                            append msg  \${dialogSettingsCMHintSoftwareUpdate4b}
-                            li $msg
-                        }
-                    }
+                    puts "RaspberryMatic is a community driven project. If you want to support it, please consider a <a href='https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL' target='_blank'>PayPal donation</a> to support further development.<br><br><a href='https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL' target='_blank'><img alt='PayPal' border=0 src='https://www.paypalobjects.com/en_US/i/btn/btn_donateCC_LG.gif'></a>"
+                    #puts "\${dialogSettingsCMHintSoftwareUpdate1}"
+                    #number_list {class="j_noForcedUpdate"} {
+                    #    li {
+                    #      ${dialogSettingsCMHintSoftwareUpdate2}                        }
+                    #    li {
+                    #         ${dialogSettingsCMHintSoftwareUpdate3}
+                    #    }
+                    #    set bat_level [get_bat_level]
+                    #    if {$bat_level < 50} {
+                    #        set msg " \${dialogSettingsCMHintSoftwareUpdate4a} $bat_level%. "
+                    #        append msg  \${dialogSettingsCMHintSoftwareUpdate4b}
+                    #        li $msg
+                    #    }
+                    #}
 
-                    division {class="j_forcedUpdate" style="padding:10px;"} {
-                      puts "<br/>\${dialogSettingsCMHintSoftwareUpdate3}"
-                    }
+                    #division {class="j_forcedUpdate" style="padding:10px;"} {
+                    #  puts "<br/>\${dialogSettingsCMHintSoftwareUpdate3}"
+                    #}
                 }
             }
             table_row {class="CLASS20902 j_noForcedUpdate j_fwUpdateOnly"} {
