--- homematic/WebUI/www/config/cp_maintenance.cgi.orig	2016-11-25 21:36:55.773395730 +0100
+++ homematic/WebUI/www/config/cp_maintenance.cgi	2016-11-25 22:10:04.059356309 +0100
@@ -377,7 +377,7 @@
                                         table_row {
                                             table_data {
                                                 division {class="CLASS20908" style="display: none"} {id="btnFwDownload"} {} "onClick=\"window.location.href='$REMOTE_FIRMWARE_SCRIPT?cmd=download&version=$cur_version&serial=$serial&lang=de&product=HM-CCU2';\"" {}
-                                                division {class="CLASS20908"}  "onClick=\"showCCULicense();\"" {puts "\${dialogSettingsCMBtnPerformSoftwareUpdateDownload}"}
+                                                division {class="CLASS20908"}  "onClick=\"window.open('https://github.com/jens-maus/RaspberryMatic/releases','_blank');\"" {puts "\${dialogSettingsCMBtnPerformSoftwareUpdateDownload}"}
                                             }
                                         }
                                     }
