--- homematic/arm-gnueabihf/packages-eQ-3/RFD/etc/init.d/S60multimacd.orig	2016-12-12 22:00:43.139457618 +0100
+++ homematic/arm-gnueabihf/packages-eQ-3/RFD/etc/init.d/S60multimacd	2016-12-12 22:41:11.351135142 +0100
@@ -7,39 +7,8 @@
 CFG_TEMPLATE_DIR=/etc/config_templates
 PIDFILE=/var/run/multimacd.pid
 
-update() {        
-	if [ ! -e /etc/config/no-coprocessor-update ] ; then
-		echo "checking if firmware update is needed..."
-		/bin/eq3configcmd update-coprocessor -p "/dev/mxs_auart_raw.0" -u
-		if ! [ $? -eq 0 ] ; then
-			echo "error while updating coprocessor, recovering..."
-			avrprog --bo
-			/bin/eq3configcmd update-coprocessor -p "/dev/mxs_auart_raw.0" -u -f
-			sleep 1
-		else
-			echo "done"
-			sleep 1
-		fi
-	else
-		echo "firmware update disabled"
-	fi
-}
-check_bo(){
-	avrprog --devinfo | grep -q ec,fd,da,ef
-	if ! [ 0 -eq $? ];
-	then
-	avrprog --bo
-	fi
-}
-check_rf_address(){
-        sed 1q /sys/module/plat_eq3ccu2/parameters/radio_mac > /var/log/rf_address
-        /bin/eq3configcmd read-default-rf-address -l 0 -v -h -f /dev/mxs_auart_raw.0 | grep -q -F -f /var/log/rf_address
-        if ! [ 0 -eq $? ];
-        then
-        avrprog --bo
-        fi
-        rm /var/log/rf_address
-}
+source /var/ids 2>/dev/null
+
 init() {
 	export TZ=`cat /etc/config/TZ`
 	
@@ -63,9 +32,9 @@
     do
 		sleep 2
 		echo -n "."
-        MMDSTATUSPID=`cat /var/status/multimacd.status 2>/dev/null`
+        MMDSTATUSPID=`cat /var/status/multimacd.status 2>&1`
         MMDPID=`pidof multimacd`
-        if [ $MMDSTATUSPID = $MMDPID ] 
+        if [ "$MMDSTATUSPID" = "$MMDPID" ] 
 		then
             echo "multimacd is ready now."
             break
@@ -78,23 +47,21 @@
 }
 
 start() {
-	echo "Starting multimacd: "
-	init
-	if [ ! -e $PIDFILE ] ; then
-		check_bo
-                check_rf_address
-		update
-	else
-		echo "Skipping coprocessor update because multimacd is running."
-	fi
-	init
-	start-stop-daemon -S -q -b -m -p $PIDFILE --exec /bin/multimacd -- -f /etc/config/multimacd.conf -l $LOGLEVEL_RFD
-	waitStartupComplete
+	echo -n "Starting multimacd: "
+  if [[ ${HM_MODE} == "HmIP" ]]; then
+	  init
+	  start-stop-daemon -S -N -10 -q -b -m -p $PIDFILE --exec /bin/multimacd -- -f /etc/config/multimacd.conf -l $LOGLEVEL_RFD
+	  waitStartupComplete
+  else
+    echo "not required"
+  fi
 }
 stop() {
 	echo -n "Stopping multimacd: "
-	start-stop-daemon -K -q -p $PIDFILE
-	rm -f $PIDFILE
+  if [[ ${HM_MODE} == "HmIP" ]]; then
+	  start-stop-daemon -K -q -p $PIDFILE
+	  rm -f $PIDFILE
+  fi
 	echo "OK"
 }
 restart() {
