--- homematic/WebUI/www/api/homematic.cgi	2016-12-13 13:45:20.253603436 +0100
+++ homematic/WebUI/www/api/homematic.cgi	2016-11-22 12:56:53.000000000 +0100
@@ -22,15 +22,13 @@
 # um eine Notlösung. 
 ##
 
-set central true
+set central true 
 
 # Zentrale oder Konfigtool (/www/ = Zentrale)
 if {$env(DOCUMENT_ROOT) != "\/www\/"} then {
   set central false
 }
 
-set central true
-
 #####################
 # Verwendete Module #
 #####################
