--- homematic/arm-gnueabihf/packages/lighttpd/etc/lighttpd/conf.d/cgi.conf.orig	2016-12-16 13:46:39.807222464 +0100
+++ homematic/arm-gnueabihf/packages/lighttpd/etc/lighttpd/conf.d/cgi.conf	2017-03-20 15:09:56.907743421 +0100
@@ -1,5 +1,7 @@
 #### CGI module
 cgi.assign = ( ".pl"  => "/usr/bin/perl",
-                ".cgi" => "/opt/hm/bin/tclsh",
+                ".cgi" => "/bin/tclsh",
                 ".ccc" => ""
              )
+cgi.x-sendfile = "enable"
+cgi.x-sendfile-docroot = ( "/usr/local/tmp" )
