--- homematic/arm-gnueabihf/packages/lighttpd/etc/lighttpd/lighttpd.conf.orig	2017-02-26 20:09:08.397575287 +0100
+++ homematic/arm-gnueabihf/packages/lighttpd/etc/lighttpd/lighttpd.conf	2017-02-27 15:55:38.840081180 +0100
@@ -14,10 +14,10 @@
 ## chroot example aswell.
 ##
 var.log_root    = "/var/log"
-var.server_root = "/opt/hm/www"
+var.server_root = "/www/"
 var.state_dir   = "/var/run"
 var.home_dir    = "/var/lib/lighttpd"
-var.conf_dir    = "/opt/hm/etc/lighttpd"
+var.conf_dir    = "/etc/lighttpd"
 
 ## 
 ## run the server chrooted.
@@ -87,8 +87,8 @@
 ##
 server.port = 80
 
-$SERVER["socket"]==":8181" {
-        server.document-root="/opt/hm/www"
+$SERVER["socket"] == "[::]:8181" {
+        server.document-root="/www/"
 }
 
 ##
@@ -99,7 +99,7 @@
 ##
 ## bind to a specific IP
 ##
-#server.bind = "localhost"
+server.bind = "0.0.0.0"
 
 ##
 ## Run as a different username/groupname.
