--- homematic/arm-gnueabihf/packages/lighttpd/etc/lighttpd/lighttpd.conf.orig	2017-02-26 00:34:20.519204768 +0100
+++ homematic/arm-gnueabihf/packages/lighttpd/etc/lighttpd/lighttpd.conf	2017-02-26 00:32:24.649889528 +0100
@@ -14,10 +14,10 @@
 ## chroot example aswell.
 ##
 var.log_root    = "/var/log"
-var.server_root = "/opt/hm/www"
+var.server_root = "/www/"
 var.state_dir   = "/var/run"
 var.home_dir    = "/var/lib/lighttpd"
-var.conf_dir    = "/opt/hm/etc/lighttpd"
+var.conf_dir    = "/etc/lighttpd"
 
 ## 
 ## run the server chrooted.
@@ -90,7 +90,7 @@
 ##
 ## Use IPv6?
 ##
-# server.use-ipv6 = "enable"
+server.use-ipv6 = "enable"
 
 ##
 ## bind to a specific IP
