--- homematic/arm-gnueabihf/packages/lighttpd/etc/lighttpd/lighttpd_ssl.conf.orig	2017-02-26 00:39:56.971410070 +0100
+++ homematic/arm-gnueabihf/packages/lighttpd/etc/lighttpd/lighttpd_ssl.conf	2017-02-26 00:40:14.303008584 +0100
@@ -14,7 +14,7 @@
 ## chroot example aswell.
 ##
 var.log_root    = "/var/log"
-var.server_root = "/www"
+var.server_root = "/www/"
 var.state_dir   = "/var/run"
 var.home_dir    = "/var/lib/lighttpd"
 var.conf_dir    = "/etc/lighttpd"
@@ -90,7 +90,7 @@
 ##
 ## Use IPv6?
 ##
-# server.use-ipv6 = "enable"
+server.use-ipv6 = "enable"
 
 ##
 ## bind to a specific IP
--- homematic/arm-gnueabihf/packages-eQ-3/WebUI/etc/lighttpd/lighttpd_ssl.conf.orig	2017-02-26 00:39:56.971410070 +0100
+++ homematic/arm-gnueabihf/packages-eQ-3/WebUI/etc/lighttpd/lighttpd_ssl.conf	2017-02-26 00:40:14.303008584 +0100
@@ -14,7 +14,7 @@
 ## chroot example aswell.
 ##
 var.log_root    = "/var/log"
-var.server_root = "/www"
+var.server_root = "/www/"
 var.state_dir   = "/var/run"
 var.home_dir    = "/var/lib/lighttpd"
 var.conf_dir    = "/etc/lighttpd"
@@ -90,7 +90,7 @@
 ##
 ## Use IPv6?
 ##
-# server.use-ipv6 = "enable"
+server.use-ipv6 = "enable"
 
 ##
 ## bind to a specific IP
