--- homematic/arm-gnueabihf/packages-eQ-3/HS485D/etc/init.d/S60hs485d.orig	2017-03-06 19:37:20.030960088 +0100
+++ homematic/arm-gnueabihf/packages-eQ-3/HS485D/etc/init.d/S60hs485d	2017-03-08 17:21:40.559001309 +0100
@@ -7,9 +7,12 @@
 
 CFG_TEMPLATE_DIR=/etc/config_templates
 
-init() {
-	export TZ=`cat /etc/config/TZ`
+source /var/hm_mode 2>/dev/null
+
+# skip this startup in LAN Gateway mode
+[[ ${HM_MODE} == "HMLGW" ]] && exit 0
 
+init() {
 	if [ ! -e /etc/config/syslog ] ; then
 		cp $CFG_TEMPLATE_DIR/syslog /etc/config
 	fi
