--- homematic/WebUI/www/rega/esp/datapointconfigurator.fn.orig
+++ homematic/WebUI/www/rega/esp/datapointconfigurator.fn
@@ -34,11 +34,6 @@
     Call("/esp/datapointconfigurator.fn::StringCut()");
   
     WriteLine('<tr>');
-      WriteLine('<td class="CLASS02506"></td>');
-      WriteLine('<td class="CLASS02506"></td>');
-    WriteLine('</tr>');
-  
-    WriteLine('<tr>');
       WriteLine('<td class="CLASS02507">');
       WriteLine('<div class="CLASS02509">'#sName#'</div></td>');
       WriteLine('<td id="dpc'#cId#'" class="CLASS02508">');
@@ -46,10 +41,6 @@
       WriteLine('</td>');
     WriteLine('</tr>');
   
-    WriteLine('<tr>');
-      WriteLine('<td class="CLASS02506"></td>');
-      WriteLine('<td class="CLASS02506"></td>');
-    WriteLine('</tr>');
   }
   WriteLine('</table>');
 }
@@ -88,7 +79,7 @@
     if( bSysVar )
     {
       WriteLine( '<table id="ctrl'#chnId#'tbl" cellspacing="0" class="ctrlTbl CLASS02510" border="0">' );
-      WriteLine( '<tr align="center">' );
+      WriteLine( '<tr>' );
       Call( "/esp/datapointconfigurator.fn::CreateSysVar()" );
       WriteLine( '</tr>' );
       WriteLine( '</table>' );
@@ -1039,8 +1030,8 @@
           {
             WriteLine( '<td>' );
 
-              WriteLine( '<div class="ControlBtnInfo CLASS02532" style="height:2.5em;" >' );
-                WriteLine( '<span class="_CLASS02533"><span class="CLASS02534 j_translate" style="float: none;">'#sTitle#'</span>' );
+              WriteLine( '<div class="ControlBtnInfo CLASS02532">' );
+                WriteLine( '<span class="_CLASS02533"><span class="CLASS02534 j_translate">'#sTitle#':  </span>' );
                 boolean tbRead =  ( ( oDP.Operations() & OPERATION_READ ) != 0 );
                 boolean tbEvent = ( ( oDP.Operations() & OPERATION_EVENT ) != 0 );
                 boolean tbWrite = ( ( oDP.Operations() & OPERATION_WRITE ) != 0 );
@@ -1189,7 +1180,6 @@
   WriteLine( '<td class="CLASS02537">' );
     WriteLine( '<div class="ControlBtnInfo CLASS02538" >' );
       sName = cObj.Name();
-      WriteLine( '<span class="CLASS02534">'#sName#':  </span>' );
       
       string sSysVarVal = '';
       object cObjTmp = cObj;
@@ -1218,7 +1208,7 @@
 !        if( !(i%2) ) { WriteLine( '<tr>' ); }
         WriteLine( '<td class="CLASS02540">' );
           WriteLine( '<div class="ControlBtnInfo CLASS02538" >' );
-            WriteLine( '<span class="CLASS02534">'#oSV.Name()#':  </span>' );
+            WriteLine( '<span class="CLASS02534">'#oSV.Name()#':</span>' );
             
             string sSysVarVal = "";
             object cObjTmp = oSV;
--- homematic/WebUI/www/rega/pages/tabs/startpage.htm.orig
+++ homematic/WebUI/www/rega/pages/tabs/startpage.htm
@@ -115,9 +115,6 @@
                   Write('<td class="CLASS01807">'#dom.GetObject(sid).Name()#'</td>');
                   Write('</tr>');
                   Write('<tr>');
-                  Write('<td class="CLASS01808"></td>');
-                  Write('</tr>');
-                  Write('<tr>');
                   Write('<td id="dpc'#oEntry.ID()#'" class="CLASS01812">');
                   integer cId = oEntry.ID();
                   integer iStatusOnly = 0;
@@ -140,9 +137,6 @@
                   Call("/esp/datapointconfigurator.fn::dcCreate()");
                   Write('</td>');
                   Write('</tr>');
-                  Write('<tr>');
-                  Write('<td class="CLASS01808"></td>');
-                  Write('</tr>');
                   Write('</table>');
                   Write('</br>');
                   Write('</td>');
--- homematic/WebUI/www/webui/style.css.orig
+++ homematic/WebUI/www/webui/style.css
@@ -4787,7 +4787,7 @@
 .CLASS01812 {
 	background-color:$_(activeBackground);
 	width:580px;
-	height:120px;
+	height:auto;
 	border:solid 1px $_(black); /* black */
 	color: $_(textColorB); /* black */
 }
@@ -4971,7 +4971,7 @@
 	background-color:$_(activeBackground);
 	width:520px;
 	max-width:502px;
-	height:120px;
+	height:auto;
 	border:solid 1px $_(black); /* black */
 	color: $_(textColorB); /* black */
 	overflow:auto;
@@ -5129,7 +5129,9 @@
 .CLASS02532 {
 	font-size: 10px;
 	width:220px;
-	height:25px;
+	height:auto;
+	clear: both;
+	display: inline-block;
   /* line-height:25px; problem with controls with a long parameter name */
 }
 
@@ -5158,6 +5160,11 @@
 
 .CLASS02535 {
 	float: right;
+	text-align: right;
+	width: 98%;
+	word-wrap: break-word;
+	max-width: 480px;
+	display: inline-block;
 }
 
 .CLASS02536 {
@@ -5165,13 +5172,18 @@
 }
 
 .CLASS02537 {
+	text-align: center;
 	padding: 5px;
+	width: 100%;
 	background-color: $_(activeBackground) ! important;
 }
 
 .CLASS02538 {
-	width:220px;
-	height:25px;
+	width:98%;
+	height:100%;
+	max-width: 480px;
+	clear: both;
+	display: inline-block;
 }
 
 .CLASS02539 {
@@ -5180,7 +5192,7 @@
 
 .CLASS02540 {
 	text-align:center;
-	width:100%;
+	width:220px;
 	border:0px;
 }
 
