--- homematic/WebUI/www/config/ic_common.tcl	2016-12-13 13:45:20.317601828 +0100
+++ homematic/WebUI/www/config/ic_common.tcl	2016-11-22 12:56:53.000000000 +0100
@@ -23,10 +23,10 @@
 set GL_FLAG_RECEIVER_UNKNOWN     0x8
 
 set HTMLTITLE "HomeMatic Interne Konfiguration"
-set INTERFACES_FILE "/opt/hm/etc/config/InterfacesList.xml"
+set INTERFACES_FILE "/etc/config/InterfacesList.xml"
 
 if { ![info exists env(CONFIG_ROOT)] } {
-    set env(CONFIG_ROOT) "/opt/hm/etc/config"
+    set env(CONFIG_ROOT) "/etc/config"
 }
 
 set USERPROFILESPATH "$env(CONFIG_ROOT)/userprofiles"
@@ -1067,9 +1067,14 @@
   #Evtl. irgendwann umbauen.
   global iface_url
   set channel_type ""
+  set chn ""
+
+  set hmDisEPIdentifier "HM-Dis-EP-WM55"
+  set hmDisWM55Identifier "HM-Dis-WM55"
 
   if { ! [catch { array set ch_descr [xmlrpc $iface_url($iface) getDeviceDescription [list string $address]] } ] } then {
     set channel_type $ch_descr(TYPE)
+    catch {set chn $ch_descr(INDEX)}
     set parent_type ""
     catch {set parent_type $ch_descr(PARENT_TYPE)}
   }
@@ -1106,6 +1111,10 @@
 
     if {[info exists unit] == 0} {
      set unit ""
+    } else {
+      if {($unit == "??C") || ($unit == "Â°C")} {
+        set unit "&#176;C"
+      }
     }
 
     # omit internal and invisible parameters
@@ -1123,7 +1132,14 @@
     if {$ps_type == "MASTER" && $parent_type == "" } then {
       append s "<td><span class=\"stringtable_value\">${param_id}</span></td>"
     } elseif {$ps_type == "MASTER" || $ps_type == "VALUES"} then {
-      append s "<td><span class=\"stringtable_value\">$channel_type|${param_id}</span></td>"
+      # We have to rename the translation of the parameters for the channels >=4 of the HM-Dis-EP-WM55 (Text Zeile x > Text Block x)
+      if {($parent_type != $hmDisEPIdentifier) || ($chn < 4)} {
+        # original translation
+        append s "<td><span class=\"stringtable_value\">$channel_type|${param_id}</span></td>"
+      } else {
+        # new translation
+        append s "<td><span>\${lblTextBlock}</span></td>"
+      }
     } else {
 
       # Nötig, zum übersetzen der Parameter auf der Senderseite (wie PEER_NEEDS_BURST, oder EXPECT_AES)
@@ -1149,17 +1165,40 @@
         } else {
           puts "<script type=\"text/javascript\">load_JSFunc('/config/easymodes/MASTER_LANG/KEY_4Dis.js');</script>"
           set helpText [getStatusDisplayHelp]
-          array set dev_descr [xmlrpc $url getDeviceDescription [list string $address]]
-          set chn $dev_descr(INDEX)
-          if {$param_id == "TEXTLINE_1"} {
-            # Fortlaufende Nummerierung der Textblöcke hinzufügen.
-            # Berechnung:
-            # 1. Parameter = Kanalnummer * 2 - 1
-            append s "<td>[expr $chn * 2 - 1]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" onblur=\"encodeStringStatusDisplay('$idval', true);\" value=\"$value\" $id $access /></td>"
+
+          # The parameter numbering of the channels 1 and 2 are the same as from the HM-Dis-WM55
+          if {($parent_type != $hmDisEPIdentifier) || ($chn < 4)} {
+            if {$param_id == "TEXTLINE_1"} {
+              # Fortlaufende Nummerierung der Textblöcke hinzufügen.
+              # Berechnung:
+              # 1. Parameter = Kanalnummer * 2 - 1
+              if {($parent_type != $hmDisEPIdentifier) && ($parent_type != $hmDisWM55Identifier)} {
+                append s "<td>[expr $chn * 2 - 1]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true);\" value=\"$value\" $id $access /></td>"
+              } else {
+                append s "<td>[expr $chn * 2 - 1]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true, '_');\" value=\"$value\" $id $access /></td>"
+              }
+            } else {
+              # 2. Parameter = Kanalnummer * 2
+              if {($parent_type != $hmDisEPIdentifier) && ($parent_type != $hmDisWM55Identifier)} {
+                append s "<td>[expr $chn * 2]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true);\" value=\"$value\" $id $access /></td>"
+              } else {
+                append s "<td>[expr $chn * 2]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true, '_');\" value=\"$value\" $id $access /></td>"
+              }
+
+              append s "<td><img src=\"/ise/img/help.png\"/ size=\"24\" width=\"24\" onclick=\"MessageBox.show(translateKey('dialogHelpTitle'), '$helpText', '', 450, 375) ;\"></td>"
+            }
           } else {
-            # 2. Parameter = Kanalnummer * 2
-            append s "<td>[expr $chn * 2]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" value=\"$value\" $id $access /></td>"
-            append s "<td><img src=\"/ise/img/help.png\"/ size=\"24\" width=\"24\" onclick=\"MessageBox.show(translateKey('dialogHelpTitle'), '$helpText', '', 450, 320) ;\"></td>"
+            # Here we set the parameter numbering for the channels 4 - 8 of the HM-Dis-EP-WM55
+            if {$param_id == "TEXTLINE_1"} {
+              # Fortlaufende Nummerierung der Textblöcke hinzufügen.
+              # Berechnung:
+              # 1. Parameter = Kanalnummer * 2 - 7
+              append s "<td>[expr $chn * 2 - 7]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true, '_');\" value=\"$value\" $id $access /></td>"
+            } else {
+              # 2. Parameter = Kanalnummer * 2 -6
+              append s "<td>[expr $chn * 2 - 6]&nbsp;&nbsp;<input type=\"text\" name=\"$param_id\" maxlength=\"12\" onblur=\"encodeStringStatusDisplay('$idval', true, '_');\" value=\"$value\" $id $access /></td>"
+              append s "<td><img src=\"/ise/img/help.png\"/ size=\"24\" width=\"24\" onclick=\"MessageBox.show(translateKey('dialogHelpTitle'), '$helpText', '', 450, 375) ;\"></td>"
+            }
           }
         }
         append s "<td>$unit</td>"
@@ -1249,7 +1288,7 @@
           append s "<input type=\"hidden\" name=\"$param_id\"   value=\"$value_orig\" $id                 $access style=\"visibility:hidden;display:none;\" />"
           append s "<input type=\"text\"   name=\"__$param_id\" value=\"$value\"       id=\"$input_idval\" $access $hidden"
           append s "  onkeyup=\"ProofAndSetValue('$input_idval', '${idval}', parseInt($min), parseInt($max), parseFloat([expr 1 / $factor]));\" /></td>"
-          append s "<td><div id=\"${input_idval}_unit\" $hidden class=\"stringtable_value\">$unit ($min-$max)</div></td>"
+          append s "<td><div id=\"${input_idval}_unit\" $hidden class=\"_stringtable_value\">$unit ($min-$max)</div></td>"
       }
       "FLOAT" {
         set input_idval   ${idval}_temp
@@ -1691,7 +1730,9 @@
   upvar $ps_arr ps
   set pname $name 
   set s ""
-  set value $ps($pname)
+  # change because of HmIP
+  # set value $ps($pname)
+  set value [lindex $ps($pname) 0]
 
   if {[string first "." $value] >= 0} then {
     set value [format "%.1f" $value]
@@ -1866,7 +1907,7 @@
   if {$paramids != ""} then {
     foreach filename $paramids {
 
-      if { [file exists easymodes/$filename.tcl] } then {
+      if { [file exists easymodes/$filename.tcl] || [file exists easymodes/hmip/$filename.tcl] } then {
         set paramid $filename
         break
       }
