--- homematic/firmware/HM-MOD-UART/fwmap.orig	2017-01-30 14:51:52.401460699 +0100
+++ homematic/firmware/HM-MOD-UART/fwmap	2017-01-30 14:52:01.985264726 +0100
@@ -1,7 +1,7 @@
 #Type           Filename                Version
 
 #Coprozessor
-CCU2                    coprocessor_update.eq3                          1.4.1          #CoProzessor CCU2  (HM-MOD-UART)
+#CCU2                    coprocessor_update.eq3                          1.4.1          #CoProzessor CCU2  (HM-MOD-UART)
 
-#CCU2					dualcopro_si1002_update_blhm.eq3				2.7.0			#Dual CoProzessor CCU2
+CCU2					dualcopro_si1002_update_blhm.eq3				2.7.0			#Dual CoProzessor CCU2
 
