--- homematic/firmware/HM-MOD-UART/fwmap.orig	2017-03-02 17:06:02.819834881 +0100
+++ homematic/firmware/HM-MOD-UART/fwmap	2017-03-02 17:06:48.050832634 +0100
@@ -1,7 +1,7 @@
 #Type           Filename                Version
 
 #Coprozessor
-CCU2                    coprocessor_update.eq3                          1.4.1          #CoProzessor CCU2  (HM-MOD-UART)
+#CCU2                    coprocessor_update.eq3                          1.4.1          #CoProzessor CCU2  (HM-MOD-UART)
 
-#CCU2					dualcopro_si1002_update_blhm.eq3				2.8.4		   #Dual CoProzessor CCU2
+CCU2					dualcopro_si1002_update_blhm.eq3				2.8.4		   #Dual CoProzessor CCU2
 
