--- homematic-b6c9bbd1eac8169de1e933deec2a277895a88bfe/WebUI/www/api/methods/ccu/setsshpassword.tcl.orig	2017-02-06 21:40:20.964294568 +0100
+++ homematic-b6c9bbd1eac8169de1e933deec2a277895a88bfe/WebUI/www/api/methods/ccu/setsshpassword.tcl	2017-02-06 21:41:56.734049233 +0100
@@ -9,9 +9,7 @@
 ##
 
 
-  package require md5crypt
-
-  set digest [md5crypt::md5crypt $args(passwd) \[md5crypt::salt\]]
+  set digest [exec echo "$args(passwd)" | mkpasswd -m sha512]
   set command "s%^root:\\(\[^:\]*:\\)%root:$digest:%"
   set rc [catch {exec sed -i "$command" /etc/config/shadow} msg]
 
