--- homematic/WebUI/www/config/cp_software.cgi.orig	2017-03-22 14:31:29.135159837 +0100
+++ homematic/WebUI/www/config/cp_software.cgi	2017-03-22 14:35:37.869129058 +0100
@@ -488,11 +488,11 @@
 
 proc action_image_upload {} {
     global env sid
-    cd /tmp/
+    cd /usr/local/tmp/
     
     http_head
     import_file -client firmware_file
-    file rename -force -- [lindex $firmware_file 0] "/var/new_firmware.tar.gz"
+    file rename -force -- [lindex $firmware_file 0] "/usr/local/tmp/new_firmware.tar.gz"
     cgi_javascript {
         puts "var url = \"$env(SCRIPT_NAME)?sid=$sid\";"
         puts {
