--- homematic/WebUI/www/config/control_panel.cgi.orig	2017-04-12 22:09:27.110113255 +0200
+++ homematic/WebUI/www/config/control_panel.cgi	2017-04-12 22:11:56.354767638 +0200
@@ -182,10 +182,9 @@
 set i 1
 
 if { "[read_var /etc/config/tweaks CP_DEVCONFIG]" != "" } {
-  puts "<!-- devconfig -->"
-  puts "<td>"
+  puts "<td><div class=\"cpButton\">"
   puts "<div class=\"StdTableBtn CLASS21701\" onclick=\"window.open('/tools/devconfig.cgi?sid=$sid');\">devconfig</div>"
-  puts "</td><td class=\"StdTableBtnHelp\"></td>"
+  puts "<div class=\"StdTableBtnHelp\"></div></td>"
   incr i
 }
 
