--- homematic/WebUI/www/tools/devconfig.cgi.orig	2017-04-12 22:18:15.234306179 +0200
+++ homematic/WebUI/www/tools/devconfig.cgi	2017-04-12 22:21:34.657850260 +0200
@@ -139,6 +139,7 @@
         h3 [url "Link List" $env(SCRIPT_NAME)?cmd=list_links&$urlsid]
         h3 [url "Service Messages" $env(SCRIPT_NAME)?cmd=service_messages&$urlsid]
         h3 [url "RSSI" $env(SCRIPT_NAME)?cmd=show_rssi&$urlsid]
+        h3 [url "Device Types" /tools/devicetypes.html]
     }
 }
 
