--- linux/net/rfkill/rfkill-bt.c.orig	2017-10-12 14:34:37.382473540 +0200
+++ linux/net/rfkill/rfkill-bt.c	2017-10-12 14:34:47.770656733 +0200
@@ -700,8 +700,9 @@
     if (gpio_is_valid(rfkill->pdata->reset_gpio.io))
         gpio_free(rfkill->pdata->reset_gpio.io);
     
-    if (gpio_is_valid(rfkill->pdata->poweron_gpio.io))
+    if (gpio_is_valid(rfkill->pdata->poweron_gpio.io)) {
         gpio_free(rfkill->pdata->poweron_gpio.io);
+    }
 	clk_disable_unprepare(rfkill->pdata->ext_clk);
     g_rfkill = NULL;
 
