--- /www/rega/esp/system.fn.orig
+++ /www/rega/esp/system.fn
@@ -859,6 +859,51 @@
     svValue="";
   }
 
+  if( system.GetVar("createNew") == 0 )
+  {
+    string SingleCond; object SingleCondObject; integer CntValSC; integer CntValSV;
+    foreach(SingleCond,dom.GetObject(ID_SCONDITIONS).EnumIDs() )
+    {
+      SingleCondObject = dom.GetObject(SingleCond);
+      if( SingleCondObject)
+      {
+        if( ((SingleCondObject.LeftVal() == sv.ID()) && (SingleCondObject.LeftValType()== ivtSystemId )))
+        {
+          if( ist == istEnum ) {CntValSC = SingleCondObject.RightVal1(); CntValSV = (web.webGetValueListCount(sv.ValueList()))-1;}
+          if( ((ist == istAlarm) || (ist == istBool)) && (SingleCondObject.RightVal1ValType() != ivtBinary))
+          {
+            SingleCondObject.ConditionType(1);
+            SingleCondObject.RightVal1ValType(ivtBinary); SingleCondObject.RightVal1(false);
+            SingleCondObject.RightVal2ValType(ivtBinary); SingleCondObject.RightVal2(false);
+          }
+          elseif( (ist == istGeneric) && (SingleCondObject.RightVal1ValType() != ivtFloat))
+          {
+            SingleCondObject.ConditionType(8);
+            SingleCondObject.RightVal1ValType(ivtFloat); SingleCondObject.RightVal1(system.GetVar("iMinVal"));
+            SingleCondObject.RightVal2ValType(ivtFloat); SingleCondObject.RightVal2(0.0);
+          }
+          elseif( (ist == istEnum) && (SingleCondObject.RightVal1ValType() == ivtInteger) && (CntValSC > CntValSV))
+          {
+            SingleCondObject.RightVal1(CntValSV);
+            SingleCondObject.RightVal2(0);
+          }
+          elseif( (ist == istEnum) && (SingleCondObject.RightVal1ValType() != ivtInteger))
+          {
+            SingleCondObject.ConditionType(1);
+            SingleCondObject.RightVal1ValType(ivtInteger); SingleCondObject.RightVal1(0);
+            SingleCondObject.RightVal2ValType(ivtInteger); SingleCondObject.RightVal2(0);
+          }
+          elseif( (ist == istChar8859) && (SingleCondObject.RightVal1ValType() != ivtString))
+          {
+            SingleCondObject.ConditionType(1);
+            SingleCondObject.RightVal1ValType(ivtString); SingleCondObject.RightVal1("");
+            SingleCondObject.RightVal2ValType(ivtString); SingleCondObject.RightVal2("");
+          }
+        }
+      }
+    }
+  }
+
   if (oChn) 
   {
     oChn.DPs().Add( sv.ID() );
