--- uboot/include/configs/rpi.h.orig	2020-01-06 21:56:31.000000000 +0100
+++ uboot/include/configs/rpi.h	2020-06-04 12:05:40.714314603 +0200
@@ -79,6 +79,8 @@
 
 /* Environment */
 #define CONFIG_SYS_LOAD_ADDR		0x1000000
+#define CONFIG_BOOT_RETRY_TIME	20
+#define CONFIG_RESET_TO_RETRY
 
 /* Shell */
 
@@ -89,9 +91,9 @@
 
 /* Environment */
 #define ENV_DEVICE_SETTINGS \
-	"stdin=serial,usbkbd\0" \
-	"stdout=serial,vidconsole\0" \
-	"stderr=serial,vidconsole\0"
+	"stdin=usbkbd\0" \
+	"stdout=vidconsole\0" \
+	"stderr=vidconsole\0"
 
 #ifdef CONFIG_ARM64
 #define FDT_HIGH "ffffffffffffffff"
