diff -ru node-v16.16.0.orig/node.gyp node-v16.16.0/node.gyp
--- node-v16.16.0.orig/node.gyp	2022-07-07 13:54:38.000000000 +0200
+++ node-v16.16.0/node.gyp	2022-09-20 14:52:41.271586048 +0200
@@ -295,7 +295,7 @@
               'action_name': 'run_mkcodecache',
               'process_outputs_as_sources': 1,
               'inputs': [
-                '<(mkcodecache_exec)',
+                @MAYBE_WRAPPER@ '<(mkcodecache_exec)',
               ],
               'outputs': [
                 '<(SHARED_INTERMEDIATE_DIR)/node_code_cache.cc',
@@ -320,7 +320,7 @@
               'action_name': 'node_mksnapshot',
               'process_outputs_as_sources': 1,
               'inputs': [
-                '<(node_mksnapshot_exec)',
+                @MAYBE_WRAPPER@ '<(node_mksnapshot_exec)',
               ],
               'outputs': [
                 '<(SHARED_INTERMEDIATE_DIR)/node_snapshot.cc',
Only in node-v16.16.0: node.gyp.orig
diff -ru node-v16.16.0.orig/tools/v8_gypfiles/v8.gyp node-v16.16.0/tools/v8_gypfiles/v8.gyp
--- node-v16.16.0.orig/tools/v8_gypfiles/v8.gyp	2022-07-07 13:54:39.000000000 +0200
+++ node-v16.16.0/tools/v8_gypfiles/v8.gyp	2022-09-20 14:53:18.412082273 +0200
@@ -68,7 +68,7 @@
         {
           'action_name': 'run_torque_action',
           'inputs': [  # Order matters.
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)torque<(EXECUTABLE_SUFFIX)',
+            @MAYBE_WRAPPER@ '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)torque<(EXECUTABLE_SUFFIX)',
             '<@(torque_files)',
           ],
           'outputs': [
@@ -99,7 +99,7 @@
             '<@(torque_outputs_inc)',
           ],
           'action': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)torque<(EXECUTABLE_SUFFIX)',
+            @MAYBE_WRAPPER@ '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)torque<(EXECUTABLE_SUFFIX)',
             '-o', '<(SHARED_INTERMEDIATE_DIR)/torque-generated',
             '-v8-root', '<(V8_ROOT)',
             '<@(torque_files_without_v8_root)',
@@ -211,7 +211,7 @@
         {
           'action_name': 'generate_bytecode_builtins_list_action',
           'inputs': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)bytecode_builtins_list_generator<(EXECUTABLE_SUFFIX)',
+            @MAYBE_WRAPPER@ '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)bytecode_builtins_list_generator<(EXECUTABLE_SUFFIX)',
           ],
           'outputs': [
             '<(generate_bytecode_builtins_list_output)',
@@ -395,7 +395,7 @@
             ],
           },
           'inputs': [
-            '<(mksnapshot_exec)',
+            @MAYBE_WRAPPER@ '<(mksnapshot_exec)',
           ],
           'outputs': [
             '<(INTERMEDIATE_DIR)/snapshot.cc',
@@ -1503,7 +1503,7 @@
         {
           'action_name': 'run_gen-regexp-special-case_action',
           'inputs': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)gen-regexp-special-case<(EXECUTABLE_SUFFIX)',
+            @MAYBE_WRAPPER@ '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)gen-regexp-special-case<(EXECUTABLE_SUFFIX)',
           ],
           'outputs': [
             '<(SHARED_INTERMEDIATE_DIR)/src/regexp/special-case.cc',
Only in node-v16.16.0/tools/v8_gypfiles: v8.gyp.orig
