--- node-v16.16.0/Makefile.orig	2022-09-20 08:48:42.172733518 +0200
+++ node-v16.16.0/Makefile	2022-09-20 08:49:03.556963313 +0200
@@ -116,12 +116,12 @@
 else
 ifeq ($(BUILD_WITH), ninja)
 ifeq ($(V),1)
-	NINJA_ARGS := $(NINJA_ARGS) -v
+NINJA_ARGS := $(NINJA_ARGS) -v
 endif
 ifdef JOBS
-	NINJA_ARGS := $(NINJA_ARGS) -j$(JOBS)
+NINJA_ARGS := $(NINJA_ARGS) -j$(JOBS)
 else
-	NINJA_ARGS := $(NINJA_ARGS) $(filter -j%,$(MAKEFLAGS))
+NINJA_ARGS := $(NINJA_ARGS) $(filter -j%,$(MAKEFLAGS))
 endif
 $(NODE_EXE): config.gypi out/Release/build.ninja
 	ninja -C out/Release $(NINJA_ARGS)
