--- a/startup.sh
+++ b/startup.sh
@@ -17,15 +17,32 @@ if [ ! -e $USER_DIR ] ; then
   
   $BASE_DIR/install.tcl
 
-fi
+else
+
+  # make sure cloudmatic button is always present
+  if ! grep -q "ID mh CONFIG_NAME CloudMatic" /etc/config/hm_addons.cfg 2>/dev/null; then
+    $BASE_DIR/install.tcl
+  fi
 
-mkdir -p /dev/net
-mknod /dev/net/tun c 10 200
-insmod $BASE_DIR/tun.ko
+fi
 
+if [ ! -d /dev/net ]; then
+  mkdir -p /dev/net
+  if [ ! -e /dev/net/tun ]; then
+    mknod /dev/net/tun c 10 200
+    insmod $BASE_DIR/tun.ko
+  fi
+fi
 
-l1=`ifconfig | grep 'eth0' | tr -s ' ' | cut -d ' ' -f5 | tr ':' '-'`
-l2=`ifconfig | grep 'usb0' | tr -s ' ' | cut -d ' ' -f5 | tr ':' '-'` 
+if [[ -e /sys/devices/platform/ccu2-ic200 ]]; then
+  # this is for a CCU2
+  l1=`ifconfig | grep 'eth0' | tr -s ' ' | cut -d ' ' -f5 | tr ':' '-'`
+  l2=`ifconfig | grep 'usb0' | tr -s ' ' | cut -d ' ' -f5 | tr ':' '-'`
+else
+  # this is for RaspberryMatic
+  l1=`grep Serial /proc/cpuinfo | sed 's|Serial||' | sed 's|:||' | sed 's/^[ \t]*//'`
+  l2=`ifconfig | grep 'eth0' | tr -s ' ' | cut -d ' ' -f5 | tr ':' '-'`
+fi
 echo SerialNumber=$l1 > $USER_DIR/ids
 echo BidCoS-Address=$l2 >> $USER_DIR/ids
 
@@ -54,4 +71,4 @@ if [ $dienst -ge 1 ] ; then
 	(crontab -l 2>/dev/null; echo "0 */6 * * * /bin/sh /usr/local/etc/config/addons/mh/cloudmaticcheck.sh >> /dev/null") | crontab -
   # autostart vpn service
   dohammer=`/bin/sh $USER_DIR/loophammer.sh`
-fi
\ No newline at end of file
+fi
--- a/user/cleanup.sh
+++ b/user/cleanup.sh
@@ -37,7 +37,14 @@ ln -s $BASE_DIR/www $WWW_DIR
   
 $BASE_DIR/install.tcl
 
-l1=`ifconfig | grep 'eth0' | tr -s ' ' | cut -d ' ' -f5 | tr ':' '-'`
-l2=`ifconfig | grep 'usb0' | tr -s ' ' | cut -d ' ' -f5 | tr ':' '-'` 
+if [[ -e /sys/devices/platform/ccu2-ic200 ]]; then
+  # this is for a CCU2
+  l1=`ifconfig | grep 'eth0' | tr -s ' ' | cut -d ' ' -f5 | tr ':' '-'`
+  l2=`ifconfig | grep 'usb0' | tr -s ' ' | cut -d ' ' -f5 | tr ':' '-'`
+else
+  # this is for RaspberryMatic
+  l1=`grep Serial /proc/cpuinfo | sed 's|Serial||' | sed 's|:||' | sed 's/^[ \t]*//'`
+  l2=`ifconfig | grep 'eth0' | tr -s ' ' | cut -d ' ' -f5 | tr ':' '-'`
+fi
 echo SerialNumber=$l1 > /etc/config/addons/mh/ids
 echo BidCoS-Address=$l2 >> /etc/config/addons/mh/ids
