--- uboot/include/configs/rpi.h.orig
+++ uboot/include/configs/rpi.h
@@ -77,6 +77,8 @@
 #define CONFIG_ENV_SIZE			SZ_16K
 #define CONFIG_SYS_LOAD_ADDR		0x1000000
 #define CONFIG_PREBOOT			"usb start"
+#define CONFIG_BOOT_RETRY_TIME 20
+#define CONFIG_RESET_TO_RETRY
 
 /* Shell */
 
@@ -88,9 +89,9 @@
 /* Environment */
 #define CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG
 #define ENV_DEVICE_SETTINGS \
-	"stdin=serial,usbkbd\0" \
-	"stdout=serial,vidconsole\0" \
-	"stderr=serial,vidconsole\0"
+	"stdin=usbkbd\0" \
+	"stdout=vidconsole\0" \
+	"stderr=vidconsole\0"
 
 /*
  * Memory layout for where various images get loaded by boot scripts:
